<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>DNA Mechanisms for storing and processing information. Part I</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="A lot of people use the term DNA. But there are almost no articles describing how it works (not understandable to biologists). I have already describe...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>DNA Mechanisms for storing and processing information. Part I</h1><div class="post__text post__text-html js-mediator-article"><div style="text-align:center;"><img src="https://habrastorage.org/webt/vs/9k/wg/vs9kwg8cww3sonbpyvtu-uipt1a.jpeg"></div><br>  A lot of people use the term DNA.  But there are almost no articles describing how it works (not understandable to biologists).  I have already described in general terms the <a href="https://habr.com/post/424041/">structure of the cell</a> and the very basics of its <a href="https://habr.com/post/424275/">energy processes</a> .  We now turn to DNA. <br>  DNA stores information.  Everybody knows that.  But how does she do it? <br><br>  To begin with, where it is stored in a cage.  Approximately 98% is stored in the kernel.  The rest is in mitochondria and chloroplasts (photosynthesis takes place in these guys).  DNA is a huge polymer composed of monomeric units.  It looks like this. <br><br><img src="https://habrastorage.org/webt/v2/x-/hq/v2x-hqku3k1kgvpfy50kcqckbes.jpeg">
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      What do we see here?  First, DNA is a double-stranded molecule.  Why it is so important - a little later.  Next we see the blue pentagons.  These are <b>deoxy</b> ribose molecules (such sugar is slightly less glucose. It differs from ribose by the absence of one OH group, which gives stability to the DNA molecule, unlike RNA, which uses ribose. Next, for simplicity, I‚Äôll omit the deoxy prefix and just say ribose, yes forgive us scrupulous comrades).  Small circles - residues of phosphoric acid.  Well, actually there are nitrogen bases.  There are 5 of them altogether, but in DNA there are mostly 4. They are Adenine, Guanin, Timine and Cytosine.  That is, there is a ribose with which the nitrogenous base is associated.  Together they form the so-called nucleosides, which bind to each other with the help of phosphoric acid residues.  Thus we get a long chain consisting of monomers.  Now look at the enlarged left chain.  You see C and G are connected by three dashed lines, and T and A are two.  What does it mean?  Yes, DNA consists of two chains, but what keeps them together?  There is such a thing as a hydrogen bond.  It looks like this.  A partial negative charge forms on the oxygen (O) and nitrogen (N) atoms, and a positive charge on the hydrogen (H).  This leads to the formation of weak links. <br><a name="habracut"></a><br><img src="https://habrastorage.org/webt/xf/y9/wo/xfy9wou-abaim1xamjpu29nbcr4.jpeg"><br><br>  The connections are really very weak.  Their energy can be 200 times lower than the energy of covalent bonds (formed due to the overlapping of a pair of electron clouds, for example, a bond in a CO2 molecule).  However, there are many such connections.  In each of our cells, the DNA chains are connected by almost 16 billion weak links, not a few, do you agree? <br><br>  But back to the number of links between the bases.  Cytosine and Guanine are connected by three connections, and Adenin and Tymin - by two.  This leads to the fact that G and C are connected much stronger than A and T. Some organisms need special stability of DNA bonds, for example, living at high temperatures.  When heated, DNA containing more HZ pairs is more stable.  So you want to live in a geyser - have a lot of HZ pairs.  Although recent studies say that there is no clear connection between the GC composition (% GC pairs from all pairs) and the temperature of the habitat.  It should be said that it varies greatly.  So, Candidatus Carsonella ruddii PV (intracellular endosymbiont) is about 16%, we have almost 41%, and Anaeromyxobacter K (the bacterium is quite average size) reaches 75%. <br><br>  Here you can see the connection of the GC composition with the size of the bacterial genome.  Mb is a million nucleotide pairs.  The indicator is quite variable.  By the way, it is often used as a feature when teaching various kinds of classifiers.  He himself recently wrote a classifier for recognizing pathogens based on raw sequencing data and it turned out that the GC composition can be used even for one reader. <br><br><img src="https://habrastorage.org/webt/0f/v4/9g/0fv49gdsgkm27xfzxz1ue-pxv-w.jpeg"><br><br>  Until I forgot.  Why is it important that DNA is double stranded?  On the basis of one circuit, you can restore another.  If in one chain there is a damaged piece opposite to the sequence Adenine-Adenine-Cytosine, then we know for sure that before the damage there was Timing-Timin-Guanin.  Thus, the presence of the second circuit allows you to store information more reliably. <br><cut></cut><br>  Cool!  Now back to the DNA molecule itself.  This is a chain of 4 types of links.  But how long?  Candidatus Carsonella ruddii PV, already mentioned above, has a total of 160,000 nucleotides.  We have 3.2 billion with you (in a haploid cell, that is, with one set of chromosomes. Most of our cells have two).  It seems a lot, right?  Not really.  In the unicellular amoeba (Amoeba dubia), it is approximately 670 billion base pairs.  It seems that this is an infinitely long chain, so let's convert the size to our favorite meters.  If all of our chromosomes (there are 46 of them, we don‚Äôt forget; 23 two copies each) turn and stretch into one line, we‚Äôll get about a 2 meter chain.  The DNA of one amoeba is enough to surround a football stadium.  But what am I leading to?  The nucleus in which the DNA is stored is not very large.  We have an average diameter of 6 microns.  Not very much, if you want to roll a 2 meter string, even if it is very thin.  And you need not just push the thread into the core.  It is necessary to turn in such a way that at any time it was possible to provide access to any of its parts.  The task is complicated.  And specialized proteins successfully cope with it.  They create a series of spirals and loops that provide increasingly high levels of packaging and do not allow for DNA tangling into the Gordian knot.  Let's talk about how it is packed. <br><br>  I must say, it is packaged in very different ways.  But if you throw away the exotic, then there are two ways.  The first is characteristic of bacteria, the second for eukaryotes (or nuclear). <br><br><h3>  DNA packaging in bacteria </h3><br>  Let's start with our smaller brothers.  Bacteria themselves do not have a very large genome, an average of 1 to 5 million base pairs.  Their most characteristic difference from us is that they do not have a nucleus and the DNA is floating in the cell.  Not quite floating, it is partially attached to the cell membrane and also collapsed, but not as much as ours. <br><br>  The second.  Bacterial DNA is often circular.  So it is easier to copy (there are no ends that can be lost when copying and there is no need to invent mechanisms for saving ends).  Usually such a ring is one, but some bacteria can have 2 or 3. There are also smaller rings (from a couple of thousands to a couple of hundreds of thousands of residues). They have a plasmid, and this is a different story altogether. <br><br><img src="https://habrastorage.org/webt/ku/dj/04/kudj04kiud4jslmf2itabm6ch40.jpeg"><br><br>  Let's go back to the DNA packaging.  DNA is packed with histone proteins (there are also histone-like proteins).  DNA is deoxyribonucleic acid.  <b>Acid</b> .  This means that it is negatively charged (due to residues of phosphoric acid).  Therefore, the proteins that bind it are positively charged.  In this way, they can bind to DNA.  The DNA of bacteria together with its packaging proteins form a nucleoid, while DNA accounts for 80% of its mass.  It looks like this.  That is, the circular DNA is divided into domains of 40 thousand base pairs.  Then twisting occurs.  Twisting occurs within the domains as well, but its degree differs in different domains.  On average, the degree of bacterial DNA packaging varies from a hundred to a thousand times. <br><br><img src="https://habrastorage.org/webt/tg/3y/3k/tg3y3kntc4ekjfu7rpl1fkuh82o.jpeg"><br><br>  There is still a <a href="https://www.youtube.com/watch%3Fv%3D30B0wGAID4o">cool video</a> . <br><br><h3>  DNA packaging in eukaryotes </h3><br>  It's all much more interesting.  Our DNA is well packed and hidden inside the nucleus.  And it is much more efficiently packed than in bacteria.  During mitosis (cell division), the size of chromosome 22 is 2 Œºm.  If you unravel it and pull it out, it will already be 1.5 cm. That corresponds to a packing degree of 10,000 times.  This is about the maximum degree of packing our DNA.  During division, DNA needs to be packed as much as possible in order to effectively divide it between the daughter cells.  In everyday life, the degree of compaction is about 500 times.  Too packed DNA is difficult to read. <br><br><h3>  There are several levels of eukaryotic DNA packaging </h3><br>  The first is the nucleosome level.  8 histone proteins form a particle on which DNA is wound.  Then another protein fixes it.  It looks like this. <br><br><img src="https://habrastorage.org/webt/lv/di/x4/lvdix4swi6vjjf2uspsrr8bcap8.gif"><br><img src="https://habrastorage.org/webt/kh/v9/yv/khv9yvyfcgdueqtqjyasrjb_a14.jpeg"><br><br>  It turns out a kind of beads.  Due to this, the packing density increases by 7-10 times.  Next, nucleosomes are packaged into fibrils.  A bit like a salt pans.  Here the total degree of packing can reach 60 times. <br><br><img src="https://habrastorage.org/webt/6o/z_/dl/6oz_dlp8df99ehl5sqfqkkfqzb0.png"><br><br>  The next stage of DNA compaction is associated with the formation of loop-like structures called chromomeres.  Fibril is divided into sections of 10‚Äì80 thousand pairs of nitrogenous bases.  Globules of nonhistone proteins are located in the breakdown sites.  DNA binding proteins recognize globules of non-histone proteins and bring them together.  Formed the mouth of the loop.  The average loop length includes about 50 thousand bases.  This structure is called interphase chromonema.  And that is where our DNA is located most of the time.  The level of packaging here reaches 500-1500 times. <br><br>  If necessary, the cell can further compact the genetic material.  The formation of larger loops of chromomeric fibrils.  These loops in turn form new loops (loops in loops ... and this is not knitting).  Which ultimately form the chromosome. <br><br>  In general, the packaging process can be described something like this. <br><br><img src="https://habrastorage.org/webt/ge/am/ue/geamuezfmjskc9dnige0s18pn5g.jpeg"><br><br>  As a result, from the DNA strands we get, when dividing, supercoiled structures that can be seen under a microscope.  We call them chromosomes. <br><br><img src="https://habrastorage.org/webt/zw/b5/us/zwb5ustuacp9x-chnbf-sukuer8.png"><br><br>  Actually the substance of chromosomes is called chromatin.  And the degree of its packaging differs depending on the chromosome region.  There is euchromatin and heterochromatin.  Euchromatin is a rather unraveled region of chromatin, in which DNA is located at a chromomeric level (packing 500‚Äì1000 times).  Here is the active reading of information.  For example, if the cell is now actively synthesizing protein A, then the DNA region encoding it will be in the state of euchromatin, so that the enzymes that ‚Äúread‚Äù the DNA could reach it.  Heterochromatin also contains that part of the DNA that the cell does not really need right now.  That is, the DNA is packed as tightly as possible so as not to get under your feet.  Depending on the needs of the cell, some areas of chromatin can be partially untied, while others can be intertwined.  Thus, regulation is also carried out (a very rough approximation), because the twisted area cannot be reached, and therefore it cannot be read. <br><br>  Actually, that's all for now.  We discussed how the storage medium is stored.  Let's make a short pause and in a couple of days we will talk about the coding of the information itself. </div><p>Source: <a href="https://habr.com/ru/post/424809/">https://habr.com/ru/post/424809/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../424799/index.html">Two-way analytics of partner iframe widget using Google Tag Manager</a></li>
<li><a href="../424801/index.html">Zeppelin OS - another step towards secure smart contracts</a></li>
<li><a href="../424803/index.html">Taiwan hacker was going to crack Mark Zuckerberg's Facebook page online, but later abandoned his threats</a></li>
<li><a href="../424805/index.html">As I graduated from LaTeX with GitHub, Docker and TravisCI</a></li>
<li><a href="../424807/index.html">EOIP between DD-WRT and Keenetic Giga 2 to merge two apartments into one network or one HDD into two apartments</a></li>
<li><a href="../424813/index.html">The machine on the Arduino, controlled by an Android device via Bluetooth, is the application code and mic (part 2)</a></li>
<li><a href="../424815/index.html">Facebook hacked: up to 90 million user accounts were attacked due to an error in the code, the company apologizes</a></li>
<li><a href="../424817/index.html">Design of digital products. Purpose, role, method</a></li>
<li><a href="../424819/index.html">Data Verification - Java & Spring Validation</a></li>
<li><a href="../424821/index.html">About demons and teleportation: two technologies that bring the ‚Äúquantum future‚Äù closer</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>