<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Practical bioinformatics p.5. DNA sequencing</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In an effort to talk about the most difficult, as quickly as possible, obviously, you forget about the simplest. And, in my case, not only about a sim...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Practical bioinformatics p.5. DNA sequencing</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/storage2/b2d/3a8/e29/b2d3a8e290b637c3ecbcb423237059ac.jpg" align="left">  In an effort to talk about the most difficult, as quickly as possible, obviously, you forget about the simplest.  And, in my case, not only about a simple, but also about an important link.  Causation slightly disrupted.  In my previous articles ( <a href="http://habrahabr.ru/blogs/bioinformatics/137069/">1</a> , <a href="http://habrahabr.ru/blogs/bioinformatics/137082/">2</a> , <a href="http://habrahabr.ru/blogs/bioinformatics/137267/">3</a> , <a href="http://habrahabr.ru/blogs/bioinformatics/137453/">4</a> ) the mathematical aspect and programming are described, but there is practically no biology in them.  Therefore, this article is about exactly what part of molecular biology they are trying to uncover, predict, see and solve the programs and algorithms that I describe. <br><a name="habracut"></a><br clear="all"><br><br>  The drawing with the image of an egg cell and sperm cells on the surface symbolizes the stage I missed when everything is just being born.  An interesting fact is that the union of two cells gives rise to about 10 trillion cells of the human body. <br><br>  Let's start with a little excursion into molecular biology.  I will try to describe in simple words complex things, omitting the details.  DNA is in the cell.  We consider exactly the DNA and the processes that take place on it, we are not very interested in exactly where it is located in the cell.  It is necessary to clarify that examples will be given for <a href="http://en.wikipedia.org/wiki/Eukaryote">eukaryotic</a> cells with which I work, but most likely many things are suitable for <a href="http://en.wikipedia.org/wiki/Prokaryote">prokaryotic</a> . 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      All processes in the cell begin with DNA, it is on the DNA that the <a href="http://en.wikipedia.org/wiki/Nucleotide">nucleotide</a> sequences are located, the copying of which is later responsible for the reactions and transformations in the cells.  We are interested in the copying process, called <a href="http://en.wikipedia.org/wiki/Transcription_(genetics)">transcription</a> .  The process of copying transcripts and the number of final products obtained is called <a href="http://en.wikipedia.org/wiki/Gene_expression">gene expression.</a>  But we call it gene expression and we measure exactly the number of transcript copies (regardless of the number of final products obtained). <br><br>  Below is an illustration of a textbook on biology, called ‚Äúthe four main genetic processes occurring in a cell‚Äù (see Fig. 1).  At the moment I am engaged in the processes depicted in this figure between the numbers 1 and 2. The technologies that help to study them are called DNA-seq and RNA-seq.  These technologies can be used both individually and jointly.  In this article I will focus in more detail on the mechanisms of DNA-seq. <br><br><img src="https://habrastorage.org/storage2/b61/b51/a8b/b61b51a8b5e332a86e84cc0261ede874.png"><br><br>  Fig.  one <br><br><img src="https://habrastorage.org/storage2/9c2/ffd/43d/9c2ffd43d1c0b77467d43a4af1df3944.png" align="left">  As a special case of DNA-seq, consider the ‚ÄúDNA-seq Pol II‚Äù (also see ‚Äú <a href="http://habrahabr.ru/blogs/bioinformatics/137453/">Preparing to work with ZINBA</a> ‚Äù).  The basis of the understanding of the result obtained with its help lies in the process of transcription.  An important role in the process of transcription is played by the RNA polymerase protein ( <a href="http://en.wikipedia.org/wiki/RNA_polymerase">RNA Polymerase</a> ).  Polymerase is fixed in a special section of DNA (see Fig. 2.1), which is called a promoter.  While the preparatory operations of the transcription process are underway (the initiation stage), the polymerase remains fixed on the promoter.  Polymerase begins to create a copy of the upper helix from the start of the site (the beginning of the replicated chain, indicated by a blue dot in the figure).  Then, moving at a speed of about 1000 bases per minute (at 37C), it copies the DNA segment to the stop site (the end of the replicated chain, indicated in the figure by a red dot).  Once again, the polymerase promoter is in the position of the promoter most of the time, and less in each section of the copied DNA.  For further study, scientists are interested in the promoter and start site, which are a necessary condition for the start of transcription.  Also in the promoter laid information on the number of copies, which is also an important object of study. <br><br>  The DNA-seq Pol II process with precipitation takes place in several stages: 1. fix the current position of the polymerase chemically or at a temperature (fixation of the polymerase means that the processes have stopped, the polymerase does not move more along the DNA);  2. in an arbitrary way (ultrasound, <a href="http://en.wikipedia.org/wiki/Micrococcal_nuclease">MNase</a> , etc.) we cut the DNA (in most cases, the sections of DNA <a href="http://en.wikipedia.org/wiki/Micrococcal_nuclease">protected by</a> polymerase are cut);  3. using immunoprecipitation of chromatin (an antibody that is immune to all other proteins, except Pol II), we select those regions that contain polymerase.  As a result, we obtain fragments of about 150 bases in length, which we will later send to the sequencer.  It should be emphasized that the cutting of the polymerase does not occur strictly along the edges.  Thus, the process turns out to be the following: we prepared millions of the same DNA from different cells, fixed polymerases, most of which are on the promoter, and some are randomly distributed throughout the gene.  The result of the digitization of the described process is observed in the next figure, shown in one of the articles (see Fig.3).  The green lines are circled, defining the intended promoter (high-density reads), followed by low-density reads, corresponding to polymerase in motion. <br><br><img src="https://habrastorage.org/storage2/a20/4c2/66b/a204c266b79679219700a145fd53e93b.png"><br><br><br>  Fig.  3 <br>  Next, I will briefly talk about some other proteins.  At the top of Figure 4, you can see how the DNA is packaged (see Fig. 4.1.), And at the bottom of the figure shows in more detail the transcription process described above (see Fig. 4.2.).  As can be seen in Figure 4.1., The DNA packaging is very dense, the double helix is ‚Äã‚Äãwound on the nucleosome, the length of the DNA coil is approximately 146 bases.  The nucleosome consists of histones.  Histones can have various modifications; modification of histones is the subject of a huge separate article.  Antibodies are developed for a specific modification of the histone, and it is the DNA segments that are wrapped around this modification of the histone that will be precipitated (filtered).  Let us give an example of the designation of the modification: H3K4Me3.  It will be read as trimethylation of lysine in the fourth position of the histone H3.  Such modifications of histones can be numerous, and all of them are scattered along DNA, therefore, in the article ‚Äú <a href="http://habrahabr.ru/blogs/bioinformatics/137453/">Preparing to work with ZINBA</a> ‚Äù in Figure 2, which depicts the H3K27Me3 landscape (trimethylation of lysine 27 of histone 3), such frequent accumulation of peaks is observed. <br><br>  There are DNA-seq technologies without precipitation, when DNA is simply cut, for example, with the help of DNase enzyme, and we get a huge amount of fragments.  In most cases, the fragments will correspond to the turns that are wrapped around the nucleosomes, since the DNase cuts between them.  The resulting fragments are sent to the sequencer.  It is believed that with deep sequencing (a huge number of fragments, about 100 million reads, goes to the sequencer, while it is expensive) by the resulting landscape you can recognize exactly where the protein is.  Moreover, the landscape will look like a volcano, a small depression at the top must correspond to a squirrel.  Most often, the DNase method is used to find DNA regions sensitive to this enzyme, i.e.  those plots that are best cut. <br><br><img src="https://habrastorage.org/storage2/6d7/d5e/2b2/6d7d5e2b247f0903dd316c9c4684d1eb.png"><br><br>  Fig.  four <br>  At the bottom of the figure, we see RNA polymerase, as well as multi-colored clouds and ovals placed in it and on DNA.  These additional elements are called transcription factors; they play an important role in the regulation of gene expression.  Transcription factors can either increase or decrease or simply block the possibility of adding other transcription factors, thereby implicitly leading to the regulation of expression.  They are also proteins.  As with all other proteins, antibodies are being developed for them.  As an example of a transcription factor, take the <a href="http://en.wikipedia.org/wiki/CTCF">CTCF</a> protein; one of its roles is to block the operation of other transcription factors.  DNA-seq experiments with it are described in the article ‚Äú <a href="http://habrahabr.ru/blogs/bioinformatics/137453/">Preparing to work with ZINBA</a> ‚Äù, the corresponding CTCF chromatin immunoprecipitation landscape can be seen in the following <a href="">figure</a> .  As can be seen, the regions protected by this protein are small, therefore, the spread in the neighborhood is not high, only 150-200bp. <br><br>  An exemplary diagram of the types of DNA-seq experiments (sequencing methods): <br><img src="https://habrastorage.org/storage2/a71/73e/434/a7173e434742f764f5b6d452db8b36a1.jpg"><br><br>  Fig.  five <br>  I have broken the scheme into four conditional levels.  The first level is fixation, it can be carried out with or without formaldehyde.  The second level - the method of cutting DNA.  The third level is filtering by size, in addition to this filtering can be carried out using chromatin immunoprecipitation.  The fourth level is the sequencing method. <br><br>  Thus, each type of experiment corresponds to the method of sequencing.  Depending on the sequencing method, different kinds of fragments are created.  Consider those experiments, the results of which are fragments containing protein.  Protein can be placed in the center of the piece or on one of the two sides of the piece.  The sequencer does not decode the entire segment, but only a small part of it from the 5 'end of the helix of each fragment.  Decoded pieces are called reads.  In Figure 6, the red dots mark the 5 'ends of each helix.  Blue and red arrows marked the beginning of the reads.  These beginnings are randomly distributed from the 5 'end of each helix to the protein border. <br><img src="https://habrastorage.org/storage2/5bd/bb7/220/5bdbb7220bed1b236c6085a9158fa0db.png"><br><br>  Fig.  6 <br><br><br>  The experiment involves a lot of the original molecules, their DNA is cut in some neighborhood of the protein, when many reads are mapped onto the coordinate axis (of one of the proteins), a picture similar to this is obtained: <br><img src="https://habrastorage.org/storage2/146/a7e/801/146a7e8011019dfd33f5fbca99dc446f.png"><br><br>  Fig.  7 <br>  Reeds from the upper helix (‚Äú+‚Äù strand) are displayed above the coordinate axis (dark blue), and reads from the lower helix (‚Äú-‚Äù strand) are displayed below the axis (light blue).  To say exactly where the protein was impossible, it is assumed that the protein binding site is located between these two peaks. <br><br>  The resulting picture resembles a signal, and therefore some algorithms were applied to the data for analyzing signals, one of which is described in this article <a href="http://habrahabr.ru/blogs/algorithm/135281/">habrahabr.ru/blogs/algorithm/135281</a> - ‚ÄúWhat are hidden Markov models‚Äù. <br><br>  I hope that after this article the subject has become more understandable, and further study will not cause a little horror.  If someone is interested in a more detailed presentation of the material, there is a wonderful book Molecular Cell Biology edited by Lodish, in which the basics of molecular biology are very informative and accessible, some of the illustrations are borrowed from it. <br>  Review by Andrey Kartashov, Cincinnati, OH, porter@porter.st. </div><p>Source: <a href="https://habr.com/ru/post/137626/">https://habr.com/ru/post/137626/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137620/index.html">Beautiful buttons for Twitter Bootstrap</a></li>
<li><a href="../137622/index.html">Smartphones recently and now</a></li>
<li><a href="../137623/index.html">Smart-IP.net 2.0 - total service update</a></li>
<li><a href="../137624/index.html">Algorithm ImageZero (IZ): fast compression without loss of quality</a></li>
<li><a href="../137625/index.html">Prospects for the development of civilization in the context of the problem of contactless information exchange</a></li>
<li><a href="../137627/index.html">Apple Demeet - sharing experience in Moscow</a></li>
<li><a href="../137630/index.html">The history of the blocked application</a></li>
<li><a href="../137632/index.html">Top science fiction novels, or I will make my ‚ÄúIMDB for books‚Äù, with preference and librarians</a></li>
<li><a href="../137633/index.html">Specialists from the Ruhr University managed to crack the encryption system satellite telephone</a></li>
<li><a href="../137634/index.html">Are you afraid that they will close ex.ua again? Not worth it - everything can be downloaded on the server of the Ministry of Internal Affairs of Ukraine</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>