<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>DNA Mechanisms for storing and processing information. Part II</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Hi Habr! Today we will continue the last story about DNA. In it, we talked about how much it is, how DNA is stored and why it is so important . Today ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>DNA Mechanisms for storing and processing information. Part II</h1><div class="post__text post__text-html js-mediator-article"><div style="text-align:center;"><img src="https://habrastorage.org/webt/xp/cm/1w/xpcm1wk37w14fseyl0cyupnpo_w.jpeg"></div><br>  Hi Habr!  Today we will continue the last story about DNA.  In it, we talked about how <a href="https://habr.com/post/424809/">much it is, how DNA is stored and why it is so important</a> .  Today we begin with a historical background and finish with the basics of coding information in DNA. <br><br><h3>  Story </h3><br>  By itself, DNA was isolated back in 1869 by Johann Friedrich Miescher from leukocytes, which he received from pus.  Leukocytes are white blood cells that perform a protective function.  There are a lot of pus in them, because they tend to damaged tissues where they are ‚Äúeaten‚Äù by bacterial cells.  He identified a substance that contains nitrogen and phosphorus.  At first it got the name nuclein, however, when acidic properties were found on him, the name was changed to nucleic acid.  The biological function of the newly discovered substance was unclear, and for a long time it was considered that phosphorus was stored in it.  Even at the beginning of the 20th century, many biologists believed that DNA had nothing to do with information transfer, since the structure of the molecule, as it seemed, was too monotonous and could not encode so much information. <br><a name="habracut"></a><br>  By 1901, Albrecht Kossel identified and described five nitrogenous bases that make up DNA and RNA.  A little later, Peter Leven found that the carbohydrate component of nucleic acids are deoxyribose and ribose.  Nucleic acids, which include ribose, became known as ribonucleic acids or, in short, RNA, and those that contained deoxyribose, deoxyribonucleic acids, or DNA. <br><br>  Now, the question arose of how the individual links are interconnected.  To do this, the DNA chain had to be broken down and look at what would happen after the destruction.  For this, the DNA polymer was hydrolyzed.  However, Leven changed the method of hydrolysis.  Now instead of boiling for hours in an acidic medium, he used enzymes.  This time, it was possible to isolate not only individual adenine, guanine, thymine, cytosine, deoxyribose, and phosphoric acid from hydrolysates, but also larger fragments, for example, compounds of nitrogenous bases with carbohydrate or carbohydrate with phosphoric acid.  At the same time <u>, compounds consisting of two nitrogenous bases, or compounds of the base type ‚Äî phosphoric acid, were not found in nucleic acid hydrolysates</u> .  That is, it became clear that phosphoric acid combines with sugar, and he, in turn, with a nitrogenous base.  The compounds of nitrogenous bases with carbohydrate were proposed to be called nucleosides, and the phosphoric esters of nucleosides were called nucleotides. <br><cut></cut><br>  As a result of these works, Leven concluded that nucleic acids are polymers.  Nucleotides are used as monomers.  The content of each of the four nucleotides in DNA, or RNA, according to chemical analysis of that time, seemed to Leven equal.  Therefore, Leven proposed the following theory of the structure of nucleic acids: they are polymers whose monomers are blocks of four nucleotides connected in series. <br>  The theory of the tetranucleotide structure at that time looked quite reasonable, having entered all the textbooks before the war.  However, the question of DNA function remained unclear.  To clarify this issue took almost half a century. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      A period began during which biologists accumulated information on the distribution of nucleic acids in various types of animal and plant tissues, in bacteria and viruses, in some unicellular organisms. <br><cut></cut><br>  At that time, the scientific community seriously believed that it was proteins that were responsible for storing genetic information.  The traditional understanding of the primary role of proteins in the life process did not even allow thinking that such an important substance as the substance of heredity could be anything other than protein.  Proteins were extremely diverse in their structure, which then could not be said about nucleic acids.  The famous Soviet geneticist-cytologist N. K. Koltsov calculated that by varying the sequence of 20 amino acids that make up a protein molecule, trillions of proteins unlike each other can be created. <blockquote>  If we wanted to print in the most simplified form, as printed logarithmic tables, this trillion molecules and provided all existing printing presses of the world for this plan, releasing 50,000 volumes of 100 printed pages per year, then so much time had passed how many have passed since the Archean period d of our days. </blockquote><cut></cut><br>  Indeed a lot ... 20 to 20th ... But the sequences are much longer than 20 amino acids. <br><br>  But as R. R. Kizel writes about this, one of the most erudite biochemists of that time. <br><blockquote>  From the views on the role of nucleic acid just cited ... it follows that it does not belong to the structure of genes and it follows that the genes are composed of some other material.  We still do not know for certain this material, despite the fact that in most cases it is directly called protein. </blockquote>  The first success came from microbiology.  In 1944, the results of the experiments of Avery and his colleagues (USA) on the transformation of bacteria were published.  A few words about transformation. <br><br>  The transformation itself was discovered in 1928 by microbiologist Griffiths. <br><br>  Griffith worked with pneumococcal cultures (Streptococcus pneumoniae) of the causative agent of a form of pneumonia.  Some strains of this bacterium are virulent, causing inflammation of the lungs.  Their cells are covered with a polysaccharide capsule that protects the bacteria from the action of the immune system.  In culture, such bacteria form large smooth colonies of regular spherical shape.  Because of this, they were called S ‚Äì strains (from the English smooth - smooth). <br><br>  There are various virulent pneumococcal strains, they differ in antibodies that are produced in the body when bacteria enter it.  They are called IS, IIS, IIIS, etc. From time to time, some cells of the virulent S strains mutate, losing the ability to synthesize the polysaccharide membrane, and become avirulent.  In culture, they form small, rough colonies of irregular shape, because of this they are called R ‚Äì strains (from English rough - rough).  Sometimes there are reverse mutations that restore the ability to synthesize a polysaccharide membrane, but only in the groups of the corresponding strains: <br><br><h3>  IIS - IIR <br>  IIIS - IIIR </h3><br>  This suggests that avirulent R-strains always correspond to the parent virulent S-strain. <br><br><img src="https://habrastorage.org/webt/l2/oz/w4/l2ozw4tlbtokfcuvkpqk40ycrgc.jpeg"><br><br>  Griffith introduced to different groups of laboratory mice a virulent and avirulent strain of pneumococcus.  In the first control group, injection of a virulent IIIS strain resulted in the death of animals.  Animals of the second control group after injection of the avirulent IIR strain remained alive.  After that, Griffith heated a solution with a culture of a virulent strain IIIS at a temperature of 60 ¬∞ C, which led to the death of bacteria.  Killed by heating bacteria, he introduced the third group of experimental mice.  The animals remained alive, which in principle was expected.  However, this is not all.  He introduced parts of the surviving mice bacterium of an avirulent IIR strain. <br><br>  It seemed that this could not lead to any terrible consequences for mice.  However, contrary to expectations, the animals died.  When bacteria were isolated from their bodies and cultured, it turned out that they belong to the virulent strain IIIS. <br><br><img src="https://habrastorage.org/webt/m8/lv/hz/m8lvhzkp9q5slcfkx_zfyj3qlmc.png"><br><br>  The fact that the death-causing cells of the mice synthesized a type III rather than II polysaccharide membrane showed that they could not result from the reverse mutation of IIR - IIS.  From this Griffith made a very important conclusion.  Avirulent bacteria of the IIR strain can be transformed into virulence bacteria, somehow interacting with the heat-killed bacteria of the IIIS strain that still remained in the body of the mice.  In other words, the avirulent bacteria of the IIR strain receive from the dead bacteria of the IIIS strain some factor that turns them into virulence.  However, what is this factor, Griffith did not know. <br><br>  Actually this phenomenon was called bacterial transformation.  It is a unidirectional transfer of inherited traits from one bacterial cell to another. <br><br>  Now back to the experiments of Avery.  The scheme of their experiments is somewhat similar to the experiments of Griffiths.  Avery and employees set themselves the task of finding out the chemical nature of the transforming agent.  They destroyed the suspension of pneumococci and removed proteins, capsular polysaccharide and RNA from the extract, however, the transforming activity of the extract remained.  The transforming activity of the drug was not lost when it was treated with crystalline trypsin or chymotrypsin (destroying proteins), ribonuclease (destroying RNA).  It was clear that the drug was neither protein nor RNA.  However, the transforming activity of the drug was completely lost when it was treated with deoxyribonuclease (destructive DNA), and insignificant amounts of the enzyme caused complete inactivation of the drug.  Thus, it was found that the transforming factor in bacteria is pure DNA.  This conclusion was a significant discovery, and Avery perfectly aware of this.  He wrote that this is exactly what genetics had long dreamed of, namely the substance of the gene.  Seems here it is proof.  But belief in protein, as a substance of heredity, was too strong.  Some believed that the insignificant impurities of protein that remained in the preparation could also cause transformation. <br><br>  New evidence of the direct genetic role of DNA was the experiments of virologists Hershey and Chase.  They worked with bacteriophage T2 (Bacteriophages - bacteria viruses) that infect <i>Escherichia coli</i> (E. coli). <br><br><img src="https://habrastorage.org/webt/yc/df/sm/ycdfsmv9arpposhsvk8l-5svspo.png"><br><br>  Actually what they did.  They included radioactive phosphorus (P32) into the DNA composition of some bacteriophages, and the sulfur isotope (S35) into the proteins of others.  To do this, some bacteria were grown on the medium with the addition of radioactive phosphorus in the composition of phosphate ion, others - on the medium with the addition of radioactive sulfur in the composition of sulfate ion.  Then bacteriophage T2 was added to these bacteria, which, multiplying in bacterial cells, included a radioactive label in its DNA (P is in DNA, but not in proteins), or proteins (S is in proteins, but not in DNA). <br><br><img src="https://habrastorage.org/webt/fi/za/xr/fizaxr5brmqt_mtswgcxlihgj68.png"><br><br>  After isolation of the radioactively labeled bacteriophages, they were added to the culture of fresh (isotope-free) bacteria.  What led to the infection of these bacteria.  The bacteriophage joins the bacterial cell and ‚Äúinjects‚Äù its DNA.  After that, the medium with bacteria was subjected to vigorous shaking in a special mixer (it was shown that in this case the phage membranes were separated from the surface of bacterial cells), and then infected bacteria were separated from the medium.  When in the first experiment labeled with phosphorus-32 bacteriophages were added to the bacteria, it turned out that the radioactive label was in the bacterial cells.  When, in the second experiment, bacteriophages labeled with sulfur-35 were added to the bacteria, the label was found in a fraction of the medium with protein shells, but it was not in the bacterial cells.  This confirmed that the material with which the bacteria were infected is DNA.  Since full viral particles containing viral proteins are formed inside infected bacteria, this experience was one of the decisive evidence that genetic information (information about the structure of proteins) is contained in DNA. <br><br>  These discoveries greatly influenced many biologists of the time.  Especially on the famous Chargaff.  He believed that Avery had essentially opened up a 'new language', or at least showed where to look for it. <br><br>  Chargaff began to look for a difference in the nucleotide composition and arrangement of nucleotides in DNA preparations obtained from various sources.  And, since there were no methods that could accurately give a chemical characterization of DNA at the time ... he had to invent them.  They were shown that the old tetranucleotide theory of the structure of nucleic acids is incorrect.  DNA in different organisms in composition and structure are very different.  At the same time, new facts were discovered that were not previously established for other natural polymers, namely, regularities in the ratio of individual bases in the composition of all the tested DNA.  Now even schoolchildren know them as Chargaff rules. <br><br><ol><li>  The amount of adenine is equal to the amount of thymine, and guanine is equal to cytosine: A = T, T = C. </li><li>  The number of purines is equal to the number of pyrimidines: A + G = T + C. </li><li>  It follows from the first and second.  The number of bases with amino groups in position 6 is equal to the number of bases with keto groups in position 6: A + C = T + G. </li></ol><br>  We touched <a href="https://habr.com/post/424809/">on the</a> mechanism <a href="https://habr.com/post/424809/">in the last article</a> , so here I will not dwell on it. <br><br>  Slowly we approached two legendary people who discovered the structure of DNA.  Francis Creek and James Watson met for the first time in 1951.  Watson then decided to take up the structure of DNA.  As a biologist, he understood that when choosing a specific DNA structure, one must take into account the existence of some simple principle of doubling the DNA molecule embedded in its structure.  Indeed, one of the most important properties of genes is the transmission of hereditary information. <br>  A cry was created the theory of x-ray diffraction on spirals, allowing to determine whether the structure under study is in spiral conformation or not.  DNA radiographs already existed at that time.  They were received in London by Maurice Wilkins and Rosalind Franklin. <br><br>  By the nature of the X-ray patterns of DNA, Watson and Crick realized that the structure under study was in a helical conformation.  They also knew that the DNA molecule is a long linear polymer chain consisting of monomers and nucleotides.  The phosphodeoxyribose backbone of this polymer is continuous, and nitrogenous bases are attached to the deoxyribose residues on the side.  To build the models, it remained to solve the problem of how many chains of a linear polymer are packed in a compact structure. <br><br>  Based on the X-ray diffraction pattern of the DNA, Watson and Crick suggested that the DNA molecule consists of two linear polynucleotide chains with a phosphodeoxyribose skeleton on the outside of the molecule and nitrogenous bases inside it.  What later confirmed.  It only remained to decide on the order of the location of the nitrogenous bases of the two chains inside the bipirale. <br><br>  Considering possible combinations of nitrogenous base pairs, Watson found that adenine ‚Äì thymine and guanine ‚Äì cytosine pairs are the same size and are stabilized by hydrogen bonds.  The rules of Chargaff were immediately explained: if a DNA adenine of one strand always binds to the thymine of another strand, and a guanine always comes together with cytosine, then the adenine in DNA should always be as much as thymine, and guanine - the same how much cytosine.  It was also clear how the doubling of the DNA molecule should occur.  Each strand is complementary to the other, and during the DNA replication process, the bispiral strand must disperse and a strand complementary to it must be completed on each polynucleotide strand.  Here, too, there were several theories, but about them in a week, in the next article. <br><br><h3>  Coding Information </h3><br>  So, we know that DNA is a carrier of information, we know what it consists of.  But how the information is encoded is still not clear. <br><br>  Let's go from the task.  DNA encodes 20 amino acids (we can say that 21, but so far we have not touched selenocysteine).  There are 4 variants of nucleotides.  That is, one nucleotide can encode 4 options, 2 - 16, 3 -64.  It is logical to assume that the code is triplet (that is, the three bases encode one amino acid).  You can read about experimental confirmation <a href="http://biologylib.ru/books/item/f00/s00/z0000012/st180.shtml">here</a> .  I'm afraid there's a lot of history here ... <br><br>  Actually, we have 64 variants and 20 amino acids.  Amino acids can be encoded by different codons.  There are also start and stop codons from which reading begins. <br>  Do not forget that DNA is first read into RNA, which is already being read into protein. <br>  The table below is the correspondence of codons to RNA amino acids.  Remember that there is no thymine in RNA, uracil goes instead of it. <br><br><img src="https://habrastorage.org/webt/uy/jq/bu/uyjqbuvi328dddekivo_hw1xqk4.gif"><br><br>  If you did not find the start codon in the table - look for AUG.  It encodes methionine and at the same time is the starting one.  When broadcasting prokaryotic, plastid and mitochondrial genes, the starting amino acid is N-formylmethionine (this is just for reference)). <br><br>  If we paint all the way from DNA to protein, we get something like that. <br><br><img src="https://habrastorage.org/webt/vu/qf/rx/vuqfrxh3x1ssxogz2ovhjqecxpk.jpeg"><br><br>  In this figure, the synthesis comes from the red chain.  As a consequence, the RNA will coincide with the blue chain (do not forget about the replacement of T with Y) <br><br>  As I said, each amino acid can be encoded by several codons.  At first glance, this does not seem to be a particularly necessary side effect of redundancy in the number of codons.  But he, in fact, quite an important role. <br><br>  Here we will touch on a little mutation.  They come in different types.  From chromosomal, when whole pieces of chromosomes are removed from the genome, they are swapped, duplicated, to the point ones, when one nitrogenous base is replaced by another.  Focus on point mutations. <br><br><h3>  What can point mutations lead to? </h3><br>  A codon can begin to encode another amino acid, which is not always scary.  Such mutations are called missense mutations (that is, with a change of meaning).  This can affect the structure of the protein.  For example, if a positively charged amino acid is replaced by a negatively charged one, this can make the protein unstable, or lead to the fact that it coagulates into another conformation (yes, the linear amino acid sequence usually collapses into a certain form) and cannot perform its functions (or better, it already smacks of evolution). <br><br>  Specifically, hemoglobin S has a single nucleotide substitution (A for T) in the coding gene.  As a result, the GAG ‚Äã‚Äãtriplet, coding for glutamate, is replaced by TG, which encodes valine.  Hemoglobin S can also transport oxygen, but makes it worse than regular hemoglobin. <br><br>  In the Hikari hemoglobin molecule, asparagine is substituted for lysine, but it is still good to carry oxygen. <br><br>  As an example with loss of function, consider hemoglobin M. Another point mutation in the hemoglobin gene leads to a complete loss of function (histidine is changed to tyrosine in the active center). <br><br>  By the way, folding protein looks like this, if you omit all the nuances. <br><br><img src="https://habrastorage.org/webt/ox/zh/su/oxzhsuj9zhktj4hbwblwxeh1fqg.gif"><br><br>  What else can happen? <br><br>  Replacing a single nitrogenous base can also lead to the appearance of a stop codon in the center of the sequence, or vice versa, the stop codon at the end will disappear.  The output will be either an incomplete circuit or an extremely long chain, which in any case will not be able to function normally.  Such mutations are called nonsense. <br><br><img src="https://habrastorage.org/webt/qf/8d/pb/qf8dpb8f4fm1tgg72ewjnx0zar8.jpeg"><br><br>  There is a third type of mutation - silence mutation.  In fact, there is a change of codon to another, encoding the same amino acid.  Protein properties do not change. <br><br>  Let's summarize the general scheme. <br><br><img src="https://habrastorage.org/webt/n6/qw/4c/n6qw4cpq1n4ss6jzgrydtnwypmi.png"><br><br>  In conclusion, I would like to tell you about one interesting feature.  One amino acid can encode several codons.  This we know.  But what does it mean?  The body uses all codons at once for coding.  But some more often, some less. <br><br>  Compare a person and ... E. coli ( <i>Escherichia coli</i> ) in terms of frequency of use of cysteine ‚Äã‚Äãencoding codons. <br><br>  It is encoded by two codons UGU and UGC. <br><br>  <b>Person</b> <br>  UGU 10.6 <br>  UGC 12.6 <br><br>  <b>E. coli</b> (strain O127: H6) <br>  UGU 19.1 <br>  UGC 0.0 <br><br>  The numbers are the occurrence of a triplet per thousand.  It can be seen that we use both codons with approximately the same frequency, while <i>E. coli</i> almost does not use the UGC codon. <br><br>  This feature must be remembered, especially when you are engaged in genoengineering and you want to develop the product of a gene of one organism in another.  If the human gene, with frequent occurrence of the UGC codon, try to insert into the E. coli of this strain - you will be disappointed.  In the cell, amino acids are associated with transport RNA, each of which corresponds to its own codon.  So, the tRNA corresponding to the UGC codon will be extremely small, which will greatly slow down the synthesis. <br><br>  If interested, <a href="http://www.kazusa.or.jp/codon/">here</a> you can see the differences in the codon composition of different organisms. <br><br>  The codon composition can be very different in both organisms of different species and different strains.  So in <i>Escherichia coli</i> O157: H7 EDL933 is more or less equally in terms of UGC and UGU.  Or another example.  The strains of the tubercle bacillus <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5964552/">isolated in different countries also differ in code composition</a> . <br><br><h3>  Summarize </h3><br>  This time there was a lot of history and relatively little biology.  More of this will not happen.  We talked about how it became clear that DNA is the carrier of information, as it is stored in the DNA itself.  We talked about the redundancy of the gene code and what this leads to.  Mutated by the mutations and the difference in the frequency of use of certain codons. <br><br>  Next time let's talk about DNA replication. <br><br>  PS: there will also be a story, but much less.  I will try not to make such pauses in writing. </div><p>Source: <a href="https://habr.com/ru/post/427153/">https://habr.com/ru/post/427153/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../427143/index.html">The second retrocomputer badge from Hackaday</a></li>
<li><a href="../427145/index.html">The best way to start studying modern genetics, molecular biology, genetic engineering and genomics</a></li>
<li><a href="../427147/index.html">‚ÄúBIAN Reference Model‚Äù for the banking industry or how to stop reinventing the wheel</a></li>
<li><a href="../427149/index.html">RADIOTEHNIKA S-30 speakers from old to new</a></li>
<li><a href="../427151/index.html">Educational process in IT: Olympiads, scholarships, support programs and ITMO University community</a></li>
<li><a href="../427155/index.html">Bus filter from Southampton - the first step to cleaner cities</a></li>
<li><a href="../427159/index.html">We collect Mikrotik firewall logs into the database</a></li>
<li><a href="../427163/index.html">New material will help make thermal solar power plants more efficient.</a></li>
<li><a href="../427165/index.html">How to develop integration tests for Atlassian Jira Server (jira-func-test-plugin)</a></li>
<li><a href="../427167/index.html">What can DSS - or old songs about the main thing</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>