<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Support vector machine for finding polymorphisms in the genome</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="The 2013 article ‚ÄúA support vector for single-nucleotide polymorphisms from next-generation sequencing data‚Äù (O'Fallon, Wooderchak-Donahue, Crockett) ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Support vector machine for finding polymorphisms in the genome</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/storage2/bb0/4f5/c4a/bb04f5c4a716848bfb821507f8ec2e3a.jpg" align="right">  The 2013 article <a href="http://bioinformatics.oxfordjournals.org/content/29/11/1361">‚ÄúA support vector for single-nucleotide polymorphisms from next-generation sequencing data‚Äù</a> (O'Fallon, Wooderchak-Donahue, Crockett) proposes a new method for determining polymorphisms in the genome using the <a href="http://bioinformatics.oxfordjournals.org/content/29/11/1361">support vector machine</a> (SVM ).  Although earlier in the 2011 article, <a href="http://www.nature.com/ng/journal/v43/n5/full/ng.806.html">‚ÄúThe framework for variation discovery and genotyping using next-generation DNA sequencing data,‚Äù</a> has already described the use of machine learning methods for determining <a href="http://ru.wikipedia.org/wiki/%25D0%259E%25D0%25B4%25D0%25BD%25D0%25BE%25D0%25BD%25D1%2583%25D0%25BA%25D0%25BB%25D0%25B5%25D0%25BE%25D1%2582%25D0%25B8%25D0%25B4%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25BF%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25BE%25D1%2580%25D1%2584%25D0%25B8%25D0%25B7%25D0%25BC">single-nucleotide polymorphisms</a> ( <a href="https://en.wikipedia.org/wiki/Single-nucleotide_polymorphism">SNPs</a> , snaps), an approach based on the use of SVM, is described for the first time in this article. <br><br>  Determining polymorphisms in the genome is important (for example, for a <a href="http://ru.wikipedia.org/wiki/%25D0%259F%25D0%25BE%25D0%25BB%25D0%25BD%25D0%25BE%25D0%25B3%25D0%25B5%25D0%25BD%25D0%25BE%25D0%25BC%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25BF%25D0%25BE%25D0%25B8%25D1%2581%25D0%25BA_%25D0%25B0%25D1%2581%25D1%2581%25D0%25BE%25D1%2586%25D0%25B8%25D0%25B0%25D1%2586%25D0%25B8%25D0%25B9">full genomic search for</a> aka <a href="http://en.wikipedia.org/wiki/Genome-wide_association_study">GWAS associations</a> ), but not a trivial task.  It is necessary to take into account that many organisms are heterozygous, and also that the data may contain erroneous information. <br><a name="habracut"></a><br>  The standard approach to the determination of SNP is based on the alignment of sequencing data (reads, genome fragments) relative to the reference (reference) genome.  However, reads may contain errors and may be incorrectly aligned to the reference.  Since it is believed that the occurrence of an error in this position is a rarer event than the appearance of polymorphism, the simplest approach in this case would be to filter out the options with a small frequency.  However, in the 2010 article, <a href="http://www.nature.com/nrg/journal/v11/n6/full/nrg2779.html">‚ÄúUncovering the whole genome sequencing‚Äù</a> , it was shown that rare polymorphisms that occur with a frequency of &lt;1% have the greatest effect on many diseases.  Frequency-based filtering will entail a large number of false negatives for rare SNPs, so there is a need to create more sensitive methods for determining genomic variations. <br><br>  As stated in the article, the advantage of methods based on machine learning is that they allow you to combine various factors affecting the plausibility of the occurrence of polymorphisms in a given position in the genome, which, among other things, increases the sensitivity of the method to rarer polymorphisms. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Unlike other machine learning methods (for example, the reinforcement learning approach described in the <a href="http://www.nature.com/nrg/journal/v11/n6/full/nrg2779.html">above article</a> ), SVM requires significantly less data to train. <br><br>  This article does not describe the detailed method for implementing the algorithm, but provides links to other articles containing technical details.  <a href="http://www.csie.ntu.edu.tw/~cjlin/libsvm/">LIBSVM is</a> used as a platform providing an implementation of the SVM algorithm.  The first publication about LIBSVM refers to 2011, and the latest release on the site to 2013. It should be noted that <a href="http://samtools.sourceforge.net/">SAM or BAM</a> files are accepted as input data, which seems to be a convenient and natural approach to bioinformatics.  As a result of the algorithm, a <a href="http://en.wikipedia.org/wiki/Variant_Call_Format">VCF</a> file is generated. <br><br>  Two types of data are used to train the model: <br><ol><li>  alignments corresponding to true polymorphisms; </li><li>  nucleotide mismatches in certain positions not related to the presence of polymorphisms. </li></ol><br>  I wonder how you can collect data corresponding to the second type.  The authors cite as an example two possible strategies: <br><ol><li>  Identify polymorphisms using <a href="http://www.broadinstitute.org/gatk/guide/topic%3Fname%3Dbest-practices">GATK's best practices guidelines</a> , including <a href="http://www.broadinstitute.org/gatk/gatkdocs/org_broadinstitute_sting_gatk_walkers_genotyper_UnifiedGenotyper.html">UnifiedGenotyper</a> .  The authors used snaps, which were determined only in one case (while excluding snipes, which are described in the data of the <a href="http://www.1000genomes.org/">1000 Genomes</a> project), from the remaining ones, snaps were chosen with a very low leveling quality value. </li><li>  The authors chose polymorphisms that occur at least 8 times in the local database of 57 <a href="http://en.wikipedia.org/wiki/Exome">exomes</a> (which also do not correspond to data from 1000 Genomes). </li></ol><br>  It should be noted that polymorphisms can also occur in non-transcribed regions of the genome; therefore, it is possible that the use of information obtained solely from the analysis of exomes may entail some inaccuracies. <br><br>  On the received data training is performed.  The article states that the training includes the determination of the numerical parameters of the hyperplane that best separates the two classes of data on which the training is performed. <br><br>  It is also interesting, what signs were chosen by the authors for training the model.  The authors taught the model, starting with a set consisting of three signs.  This gave a bad result, so many of the signs expanded.  This process is described in the article not very detailed, and it is not entirely clear in which order the authors added new signs, because it is obvious that they did not go through all the existing subsets of features.  In total, the authors considered 16 features, among them the probability of an error in the reads, the average quality of a given nucleotide in the reads, the sum of the qualities of the nucleotides in a given position (the last two did not give any improvement when added to the set of features), the quality of the read alignment, the depth of sequencing also did not improve the work of the model), the balance of alleles, the quality of nearby nucleotides. <br><br>  According to the graph provided in the article, the described method is superior to other methods for determining polymorphisms in sensitivity and specificity on a particular NA12878 <a href="">dataset</a> .  The method identified 95.6% of all true positives, while other methods: UnifiedGenotyper, UnifiedGenotyper + <a href="http://gatkforums.broadinstitute.org/discussion/39/variant-quality-score-recalibration-vqsr">VQSR</a> , <a href="http://samtools.sourceforge.net/">SAMtools</a> and <a href="https://github.com/ekg/freebayes">FreeBayes</a> identified only 90.6%, 94.1%, 89.5% and 88.7%, respectively. <br><br><img src="https://habrastorage.org/storage2/01c/a8d/262/01ca8d2624d178c8616ee6108b88fe52.png"><br><br>  Among the shortcomings of the described approach, it can be noted that the SVM model can give poor results on data containing different error profiles.  In addition, despite the multithreading, the implementation was 10-20% slower than the GATK UnifiedGenotyper method. <br><br>  In general, the chosen model fits well with the specifics of the task and corresponds to the general trend of using machine learning methods for analyzing genomic sequences. <br><br>  The article was written as part <a href="http://bioinformaticsinstitute.ru/courses/bioinf/2013/spring">of a bioinformatics workshop</a> . </div><p>Source: <a href="https://habr.com/ru/post/183330/">https://habr.com/ru/post/183330/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../183320/index.html">Software package for computing clusters Fujitsu HPC Cluster Suite</a></li>
<li><a href="../183322/index.html">Proper use of exceptions in Java</a></li>
<li><a href="../183324/index.html">How difficult is it to change a business model after 2 years? Startup restart history</a></li>
<li><a href="../183326/index.html">Is it possible to create a business card site in 40 minutes? My experience</a></li>
<li><a href="../183328/index.html">Optimizing the speed of loading a site on Wordpress using the tips of Page Speed</a></li>
<li><a href="../183332/index.html">Switch for two parameters in C ++</a></li>
<li><a href="../183334/index.html">Node.js vs Java + Rhino + Jetty + FreeMarker</a></li>
<li><a href="../183338/index.html">What to do if the program suddenly stopped working</a></li>
<li><a href="../183340/index.html">Backstreet Access To Intersystems Cache '</a></li>
<li><a href="../183342/index.html">Superhydrophobicity in life or available nanotechnology</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>