<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Algorithms in Bioinformatics Part 1</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In previous articles ( 1 , 2 ) we got acquainted with how the data may look depending on the biological experiment performed. Based on these visualize...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Algorithms in Bioinformatics Part 1</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/storage2/b34/e64/eb8/b34e64eb8396c1d0527876ef846d34f3.png" alt="bioinformatic" align="left">  In previous articles ( <a href="http://habrahabr.ru/blogs/bioinformatics/137453/">1</a> , <a href="http://habrahabr.ru/blogs/bioinformatics/137626/">2</a> ) we got acquainted with how the data may look depending on the biological experiment performed.  Based on these visualized data, assumptions were made about what happens inside the cell.  Now we will focus on how to mathematically and algorithmically analyze the data so that the machines for us can perform the routine work.  Unfortunately, after reading a lot of articles on data analysis, I got the impression that there is no unambiguous or most universal solution.  There are algorithms that show themselves well on a certain set of data, and in other cases no longer respond to the tasks set. <br><a name="habracut"></a><br clear="all">  First I want to remind you and explain a little the basic concepts used in the article.  Every time I talk about the reads, and how they are produced, I come across surprising and misunderstanding.  And then by chance, throwing a pack of paper in a shredder, I understood how to try to explain what reads are.  Suppose you have a book, the book itself is a cell, and each chapter of the book is DNA.  You already have several million copies of this book.  You fed the books to the shredder, which cut them into strips.  Here you take a container with nakromsannymi paper strips, mix them well and take out what the first one got.  Next, carefully lay out the strips, from both sides cut into small pieces and put the cut pieces in a separate box.  These little pieces are reads, the strips are fragments.  Further, with the help of a certain program, we determine in which part of the book the cut piece was located.  Sometimes we get lucky and we can pinpoint the place in the book where the piece was cut from.  Sometimes the program identifies several possible locations for this piece, for example, due to the fact that some chapters began with a repetition of the previous one or as a result of all operations some letters were erased and they were not found in the book.  The number of variations on the topic that could happen with pieces of strips is very large, if you can discuss them all, you can not get to the algorithms.  I hope, now it has become clearer what is what. <br><br>  Describing the algorithm, I will talk about it, using the data on which the developers tested it.  But this does not mean that it cannot be used on data of another type.  I will also try to identify the shortcomings of the algorithms that I encountered.  I do not attempt to systematize the algorithms; I will describe them in groups, in the order in which they were used in the general analysis of the sequenced data.  In this article I will talk about a simple analysis of ‚Äúpollution by the adapter‚Äù and about the preliminary preparation of the landscape and, probably, in the next article or through one - about assessing the quality of the data obtained. <br><br>
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <br><h1>  Contamination adapter </h1><br>  Companies producing equipment for sequencing also produce related kits: slides, tubes, reagents.  One type of reagent is the polynucleotide chain necessary for sequencing, which is called an adapter. <br><br>  There are various types of contamination of the prepared material.  One of them is pollution by the adapter.  The analysis of such pollution is simple.  It is necessary to calculate the frequency of nucleotides on the first, second, etc.  positions of the received FASTA / FASTQ file.  Based on the data obtained, a table is compiled.  Consider the analysis process in more detail, using the following example, below are the first 9 columns of the table (see Table 1): <br><br>  Table 1 <br><table><tbody><tr><td>  Pos / Nucl </td><td>  0 </td><td>  one </td><td>  2 </td><td>  3 </td><td>  four </td><td>  five </td><td>  6 </td><td>  7 </td></tr><tr><td>  A </td><td>  446500 </td><td>  2869138 </td><td>  599400 </td><td>  941816 </td><td>  1145756 </td><td>  1133404 </td><td>  3599581 </td><td>  3291736 </td></tr><tr><td>  C </td><td>  1922795 </td><td>  1367021 </td><td>  1761361 </td><td>  3723679 </td><td>  1030494 </td><td>  1173573 </td><td>  963827 </td><td>  1284534 </td></tr><tr><td>  T </td><td>  485441 </td><td>  1337742 </td><td>  3273475 </td><td>  918001 </td><td>  1016981 </td><td>  1006168 </td><td>  1494250 </td><td>  1291715 </td></tr><tr><td>  G </td><td>  4130093 </td><td>  1425977 </td><td>  1365764 </td><td>  1416504 </td><td>  3806769 </td><td>  3686855 </td><td>  942342 </td><td>  1132015 </td></tr><tr><td>  N </td><td>  15171 </td><td>  122 </td><td></td><td></td><td></td><td></td><td></td><td></td></tr></tbody></table><br><br>  Using the data table, you can build a graph (Fig. 1).  The peaks of the graph correspond to the letter (nucleotide) most frequently found at the current position.  In particular, the letter ‚ÄúG‚Äù in the first, the letter ‚ÄúA‚Äù in the second, ‚ÄúT‚Äù in the third position. <br><br><img src="https://habrastorage.org/storage2/a2a/f72/586/a2af72586bf6055fea2d9a7f34c4da04.png"><br>  Fig.  one <br><br><br>  If you make up the entire chain of letters corresponding to the peaks, you get a sequence that is most often found in the original data.  If this sequence coincided with the sequence from the list of adapters (which are supplied by manufacturers of equipment for sequencing), as for example, in the considered example, one of the Illumina adapter sequences was obtained, then the adapter was contaminated. <br><br>  To calculate the percentage of pollution adapter and the source material, perform the following simple steps.  1. We count the total number of letters in one of the positions in the table, in the given example, they are exactly 7000000. 2. If there was no contamination, then the number of letters in each position should be approximately the same.  Excluding from consideration the letter corresponding to the peak, we consider the rest.  As follows from the graph, there are about 1,200,000 such letters in each position. 3. Since there must be the same number of letters in each position, we get: 1,200,000 * 4 = 4,800,000. 4. Thus, the adapter's pollution is: 4,800,000 / 7,000,000 * 100% = 32%.  Those.  when preparing the material the next time you need to reduce the number of added adapters. <br><br><br><br><br><h1>  Preparing a profile for further analysis </h1> <br>  Most of the algorithms known to me perform the initial data processing - they prepare a profile.  That is, group the data, and then evaluate their level of significance.  The grouped data will determine the location of the likely location of the protein on the DNA that we are trying to find using algorithms. <br><br>  Consider the data resulting from the use of chromatin immunoprecipitation (for more details, ‚Äú <a href="http://habrahabr.ru/blogs/bioinformatics/137626/">DNA Sequencing</a> ‚Äù).  Let me remind you about the features of this method: fragments are selected using antibodies corresponding to the proteins of interest, as a result, most fragments contain the sequence to which the protein was attached.  Also, when cutting, fragments of different size are obtained, they are additionally filtered in length (no more than 150-300 nucleotides).  Most of the reads are ultimately located on both sides of the protein.  This feature we will use in the future to determine the likely location of the protein. <br><br><img src="https://habrastorage.org/storage2/c08/724/341/c08724341c3bd71ef8d6fbe349ba7e5c.png"><br>  Fig.  2 <br><br><br>  In Figure 2, the yellow arrows show the possible locations of the DNA cut, this is where fragments can begin or end.  Red lines depict reads obtained from a positive helix (indicated with a ‚Äú+‚Äù sign), and blue lines show a negative helix (indicated by a ‚Äú-‚Äù sign).  In some cases, if a more expensive experiment is performed (‚Äú <a href="">pair-end sequencing</a> ‚Äù), the sequencing equipment is able to determine which two reads were obtained from a single fragment.  It is also stated that the use of such a method several times increases the quality of determining the proposed position of a protein [1].  The pair-end method is still not used frequently, so in most cases we are dealing with an unrelated set of reads. <br><br>  In the ideal case, the landscape of the location of the reads around the protein should be as in Figure 3. Given that the possible number of random components of the experiment is large, this is rare. <br><br><img src="https://habrastorage.org/storage2/1f9/369/284/1f9369284435af3e2f0b8ea1388bd7c4.png"><br>  Fig.  3 <br><br><br>  Below are examples of landscapes from experimental data (Fig. 4).  The width of the column (window) corresponds to 20 nucleotides, the maximum height of the columns is only 5-8 reads.  If the width of the window were equal to one nucleotide, then the image would become even more discrete and noisy.  But, if the data are processed using the profile preparation algorithms proposed below, the landscape will be more like the ideal one (Fig. 3). <br><br><img src="https://habrastorage.org/storage2/00c/ba2/d9a/00cba2d9a2cf0487b7d17a56dd438377.png"><br>  Fig.  four <br><br><br><br><br><br><h2>  Manual profile preparation </h2><br>  How to bring the data to a more readable form for the machine?  In some cases, we know in advance how long fragments are expected.  For example, in the case of H3K4Me3 (MNase method), they are about 148bp [2].  Shift the reads by half the fragment length: 148/2 = 74. The reads found on the positive spiral are shifted to the right by 74, found on the negative spiral, shifted to the left by 74. Add the absolute values ‚Äã‚Äãand display on the graph (Fig. 5) [3, four]. <br><br><img src="https://habrastorage.org/storage2/70d/64d/155/70d64d155585138e9d5fb42e11c4f0ba.png"><br>  Fig.  five <br><br>  In the figure (Fig. 5), the peak in purple is displayed as a result of adding the number of reads from the positive and negative sides of the spiral.  The top of the peak should correspond with some accuracy to the center of the location of the protein on the DNA.  With such a simple transformation, we amplified the signal, and now it should become more noticeable. <br><br>  There are various ways of plotting the graphs above.  One of the ways: along the ‚ÄúY‚Äù axis, we postpone the number of reads originating at the ‚ÄúX‚Äù point.  Another way: plotting using segments, along the ‚ÄúY‚Äù axis, we postpone the number of read crossings at a given ‚ÄúX‚Äù point.  Sometimes a graph is built, setting off along the ‚ÄúY‚Äù axis the intersection not of reads, but of fragments.  The meaning of such graphs is the density of DNA coverage with reads / fragments. <br><br>  This is good when you can estimate the fragment size in advance and transfer its value to the program.  But what to do if the fragment length is not known, how can we calculate the value by which the data must be shifted? <br><br><br><br><br><h2>  Cross-correlation training profile </h2><br>  One way to calculate the magnitude of the shift is to cross-correlate read data from a positive and negative helix.  For each chromosome ‚Äúc‚Äù we define the function <img src="https://habrastorage.org/getpro/habr/post_images/13b/1d4/8ea/13b1d48ea3831fbfb315988152e74347.png" alt="image">  , which takes at each point ‚Äúx‚Äù a value equal to the number of beginnings on the side of the helix ‚Äús‚Äù.  Define for shear <img src="https://habrastorage.org/getpro/habr/post_images/d79/be1/1bb/d79be11bb8ad0f90d921780c96f3ea50.png" alt="image">  function <img src="https://habrastorage.org/getpro/habr/post_images/0db/56d/17f/0db56d17fc8704a6558ac81379a1f2f2.png" alt="image">  where: <br><ul><li><img src="https://habrastorage.org/getpro/habr/post_images/d53/7bb/ae6/d537bbae62926f40976dfcaaa69ba200.png" alt="image">  - linear Pearson coefficient between vectors ‚Äúa‚Äù and ‚Äúb‚Äù; </li><li>  C - a set of all chromosomes; </li><li>  N <sub>c</sub> is the number of reads on the ‚Äúc‚Äù chromosome; </li><li>  N is the total number of reads. </li></ul><br><br>  Choose a range in which there can be a shift value, for example, from -100 to 300 and construct a graph (Fig. 6) [4]. <br><br><img src="https://habrastorage.org/storage2/00c/ebe/1bd/00cebe1bde29bd2be5667fb90cfcae68.png"><br>  Fig.  6 <br><br><br>  The peak of this graph along the ‚Äúx‚Äù axis will correspond to the shift with the highest correlation.  In the case shown in Figure 6, the shift is 100/2 = 50 nucleotides.  The correlation of data with all chromosomes may be less accurate with respect to the correlation of the area of ‚Äã‚Äãinterest.  So get their <img src="https://habrastorage.org/getpro/habr/post_images/d79/be1/1bb/d79be11bb8ad0f90d921780c96f3ea50.png" alt="image">  shear for each enriched segment. <br><br><br><br><br><br><h2>  Kernel density profile </h2><br><br>  Consider the algorithm using the kernel density functions [4,5,6].  Redefine the coverage density for each position i in accordance with the formula <img src="https://habrastorage.org/getpro/habr/post_images/04d/958/5c7/04d9585c7b63d4f61f270290550ae8fd.png" alt="image">  where: <br><ul><li>  h - kernel density bandwidth, in article [5] 30bp was used, and in article [4] they took 0.45 of the width of the peak in the middle, shown in Figure 6; </li><li><img src="https://habrastorage.org/getpro/habr/post_images/c39/f81/11b/c39f8111b3901aeee12e5e22c24f5ff5.png" alt="image">  - Gaussian kernel density function; </li><li><img src="https://habrastorage.org/getpro/habr/post_images/276/164/bb2/276164bb26a25dac73a7c5971a16ecaa.png" alt="image">  - returns the number of reads that ended at point j on the positive and negative spiral, respectively. </li></ul><br>  Thus, having recalculated the density for each helix and amplifying the signal, for well-enriched areas we can determine the peaks on each side of the helix and take half the difference between them for the shift value (Fig. 7, ‚ÄúPeak shift‚Äù). <br><br><img src="https://habrastorage.org/storage2/365/26d/9ae/36526d9aef70a8237f8ef768460fa1af.jpg"><br>  Fig.  7 <br><br><br><br><br><h2>  Geometric profile building method </h2><br>  This algorithm uses simple mathematical operations.  For a window wide in w and for position i, we define: p1 (i), as the number of tags in the window on the positive spiral to the left of i;  p2 (i), as the number of tags to the right of position i.  We define n1 (i) and n2 (i) according to the definition of p, only for the negative spiral (Fig. 8.1).  Calculate the value at point i using the following formula: <img src="https://habrastorage.org/getpro/habr/post_images/132/564/4fa/1325644faa9a424482db5fba5a050f19.png" alt="image">  within the specified window we find the maximum.  The maximum found is the estimated sticking position of the protein.  This method is called WTD from Windows Tag Density. <br>  In article [4] he was also offered to modify <img src="https://habrastorage.org/getpro/habr/post_images/c4e/ddf/6de/c4eddf6de871c2e71938e200d73d0a0b.png" alt="image">  where P [a, b] is the linear Pearson correlation coefficient between vectors a and b, such that, vector ‚Äúa‚Äù is the number of reads on the positive helix from position i to i + k, and vector ‚Äúb‚Äù is the number Reads on the negative helix from position i to ik, <img src="https://habrastorage.org/getpro/habr/post_images/dc4/571/edf/dc4571edf0d26469d673f2f932e12242.png" alt="image">  (Figure 8.2).  Called MTC from Mirrow Tag Correlation. <br><br><br><br><img src="https://habrastorage.org/storage2/ee6/f00/744/ee6f00744f7ab4e5ae4330c203b0b742.jpg"><br>  Fig.  eight <br><br><br>  There are other algorithms used in the construction of the profile [7].  I tried to highlight the most frequently used.  All algorithms already have an implementation in any software package, links are given in articles, here are some of them: <br>  <a href="http://sissrs.rajajothi.com/">sissrs.rajajothi.com</a> <br>  <a href="http://home.gwu.edu/~wpeng/Software.htm">http://home.gwu.edu/~wpeng/Software.htm</a> <br>  <a href="http://mendel.stanford.edu/sidowlab/downloads/quest/">mendel.stanford.edu/sidowlab/downloads/quest</a> <br>  <a href="http://compbio.med.harvard.edu/software.html">http://compbio.med.harvard.edu/software.html</a> . <br><br>  Proper construction of a profile with the help of algorithms is the most important part of all programs, because the more precisely it is built, the more likely it is to determine where the protein is located.  Another part of the algorithms is aimed at mathematically estimating how significant the found sections are.  But, unfortunately, all algorithms are not able to replace a person, and can only help in the routine processing of data, hinting at the place where you should search.  So while the whole system of interaction of biology, chemistry, mathematics and programming is far from perfect, and there is a huge field for searching, implementing and improving ideas. <br><br>  1. Wang, C., et al., An effective approach for protein in vivo protein-DNA binding sites from ChIP-Seq data.  BMC Bioinformatics, 2010. 11: p.  81. <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2831849/">www.ncbi.nlm.nih.gov/pmc/articles/PMC2831849</a> <br><br>  2. Barski, A., et al., High-resolution profiling of the human genome.  Cell, 2007. 129 (4): p.  823-37.  <a href="http://www.ncbi.nlm.nih.gov/pubmed/17512414">www.ncbi.nlm.nih.gov/pubmed/17512414</a> <br><br>  3. Pepke, S., B. Wold, and A. Mortazavi, Computation for ChIP-seq and RNA-seq studies.  Nat Methods, 2009. 6 (11 Suppl): p.  S22-32.  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19844228">www.ncbi.nlm.nih.gov/pubmed/19844228</a> <br><br>  4. Kharchenko, PV, MY Tolstorukov, and PJ Park, ChIP-seq experiments for DNA-binding proteins.  Nat Biotechnol, 2008. 26 (12): p.  1351-9.  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19029915">www.ncbi.nlm.nih.gov/pubmed/19029915</a> <br><br>  5. Valouev, A., et al., Genome-wide analysis of transcription factor binding sites based on ChIP-Seq data.  Nat Methods, 2008. 5 (9): p.  829-34.  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19160518">www.ncbi.nlm.nih.gov/pubmed/19160518</a> <br><br>  6. Boyle, AP, et al., F-Seq: a feature density estimator for high-throughput sequence tags.  Bioinformatics, 2008. 24 (21): p.  2537-8.  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC2732284/">www.ncbi.nlm.nih.gov/pmc/articles/PMC2732284</a> <br><br>  7. Zang, C., et al., A clustering approach for identification of enriched domains from histone modification ChIP-Seq data.  Bioinformatics, 2009. 25 (15): p.  1952-8. <br><br>  Some of the drawings were taken from nature.com and printed with their permission: <br>  License Date: Feb 16, 2012, <br>  License Numbers: 2850830942235, 2850831421638 <br><br>  Review by Andrey Kartashov, Covington, KY, porter@porter.st </div><p>Source: <a href="https://habr.com/ru/post/138486/">https://habr.com/ru/post/138486/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../138474/index.html">Kiev GTUG - hackathon on Android February 18-19, 2012</a></li>
<li><a href="../138475/index.html">Student convicted of hacker attack on Facebook</a></li>
<li><a href="../138481/index.html">Google has released a version of Chromium with support for Dart "Dartium"</a></li>
<li><a href="../138483/index.html">Maintain strong ties. Community or social network 4</a></li>
<li><a href="../138485/index.html">ITSM, knowledge is power</a></li>
<li><a href="../138487/index.html">Inter for 30 minutes scored 5k likes on Facebook</a></li>
<li><a href="../138489/index.html">Usability testing KidsReview.ru site - we are looking for volunteers</a></li>
<li><a href="../138490/index.html">Practical application of DNSSEC</a></li>
<li><a href="../138492/index.html">Dynamic translation of the page into another language</a></li>
<li><a href="../138494/index.html">VLC reaches 2.0</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>