<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Tale of how SQL saves time</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="There is a company providing a platform for working with big data. This platform allows you to store genetic data and effectively manage them. For the...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Tale of how SQL saves time</h1><div class="post__text post__text-html js-mediator-article">  There is a company providing a platform for working with big data.  This platform allows you to store genetic data and effectively manage them.  For the full operation of the platform, you need the ability to process dynamic requests in the runtime environment in no more than two seconds.  But how to overcome this barrier?  To transform the existing system, it was decided to use SQL data warehouse.  Look under the cat for details! <br><br><img src="https://habrastorage.org/webt/ik/-m/wp/ik-mwpsigumh5jppshwksyy8_aw.jpeg"><a name="habracut"></a><br><br><h2>  Situation </h2><br>  Since approximately 99.5% of the human genome is common, sequence sequencing typically compares sequence patterns with selected known reference genomes.  Comparison results are saved to a raw data file with a .vcf extension.  It contains only differences between genome samples and the reference genome.  It contains approximately one million lines, each of which represents a gene mutation, that is, a gene index and a value that differs from the value of the reference genome for a given index. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      The resulting VCF file goes through a normalization procedure, which prepares it to use the various scripts available on the platform.  It is a sequence of steps for adding metadata and annotations from external gene databases.  The normalization process creates a huge file containing about 1 million lines and approximately 50 columns. <br><br><h2>  Problem </h2><br>  Doctors and researchers need the ability to process dynamic queries in a runtime environment, using a dashboard to review patient sequencing results.  The basic query supports filtering samples based on the contents of several columns.  Extended queries can associate a sample with a reference database (with 60 million rows).  A more detailed scenario implies a series of join operations within a single query, which allows genetic analysis of the pedigree and contacting the repository of the external reference database. <br><br>  Since requests are generated using the dashboard, it was necessary to get the result in less than two seconds.  If it took more time to get the result, the operation was considered completed unsuccessfully.  Processing such requests at runtime (for example, if each sample contains a large number of rows and columns) requires a significant amount of computational resources and takes a lot of time. <br><br><h2>  Interaction </h2><br>  The goal of our cooperation with this promising company was to measure the performance of query processing in order to further evaluate the effectiveness of a number of scalable storage system architectures: <br><br><h4>  Running Impala on <a href="http://www.cloudera.com/partners/solutions/microsoft.html">Cloudera</a> Cluster </h4><br>  <a href="http://impala.apache.org/">Impala is</a> recommended for processing SQL queries within the framework of interactive research analytics of large data sets, and the <a href="https://hive.apache.org/">Hive</a> and <a href="https://en.wikipedia.org/wiki/MapReduce">MapReduce</a> modules for performing batch tasks with a long processing time. <br><br>  Impala also uses a column storage system that handles requests of the type described above much more efficiently.  This is due to the way physical data is stored on disk.  When performing SELECT queries and projecting several columns, only the corresponding blocks of the selected columns are read from the disk, since the data in different rows of the same column is located next to each other.  For more information about column storage, see <a href="http://the-paper-trail.org/blog/columnar-storage/">here</a> . <br><br><h4>  Running Spark in an <a href="https://azure.microsoft.com/en-us/services/hdinsight/">HDInsight</a> Cluster </h4><br>  <a href="https://databricks.com/spark/about">Spark</a> is the technology that is recommended for processing big data in HDFS first.  Spark will successfully prepare the system to use other technologies (for example, Impala).  For more information about Spark technology, see <a href="http://spark.apache.org/">here</a> . <br><br><h4>  SQL Data Warehouse </h4><br>  <a href="https://azure.microsoft.com/en-us/services/sql-data-warehouse/">SQL Data Warehouse</a> is a distributed database management system for mass-parallel processing (MPP).  By posting data and processing across sites, SQL Data Warehouse provides flexible scaling options.  For more information about SQL Data Warehouse, see <a href="https://azure.microsoft.com/en-us/documentation/articles/sql-data-warehouse-overview-what-is/">here</a> . <br><br><h2>  Requests </h2><br>  We agreed on the number of queries that support generic scripts.  We used the same sample files for different storage systems and checked how long it took to process requests for each of these files.  For clarity, the following queries are written "in the image and likeness of SQL."  They are processed by each storage system in accordance with its own syntax. <br><br>  The reference database table contains 60 million lines within a similar VCF file structure, as well as additional metadata for each gene.  In the normalization process, this data is used to annotate the lines in the sample VCF file. <br><br>  The following table provides a description of the queries and the results of performance measurements: <br><table><tbody><tr><th>  . </th><th>  Technology </th><th>  Impala </th><th>  Spark in HD Insight </th><th>  Spark in Cloudera </th><th>  SQL Data Warehouse </th><th>  SQL Data Warehouse </th></tr><tr><td align="center" width="77">  . </td><td align="center" width="150">  Cluster Type </td><td align="center" width="76">  3 x D13 </td><td align="center" width="78">  3 x D12 </td><td align="center" width="95">  3 x DS13 (5 P30 disks each) </td><td align="center">  400 DWU (Data Warehouse) </td><td align="center">  1000 DWU (data storage units) </td></tr><tr><td align="center" width="79">  Index </td><td align="center" width="150">  Scenario </td><td align="center" width="76">  with </td><td align="center" width="78">  with </td><td align="center" width="95">  with </td><td align="center">  with </td><td align="center">  with </td></tr><tr><td align="center" width="77">  one </td><td width="150">  Choose from samples, sort by gene position, take the first 100 entries </td><td align="center" width="76">  2 </td><td align="center" width="78">  five </td><td align="center" width="95">  2 </td><td align="center">  one </td><td align="center">  one </td></tr><tr><td align="center" width="77">  2 </td><td width="150">  Select from the table of the reference database, filter by one field, sort by another field, take the first 100 records </td><td align="center" width="76">  eleven </td><td align="center" width="78">  96 </td><td align="center" width="95">  38 </td><td align="center">  2 </td><td align="center">  6 </td></tr><tr><td align="center" width="77">  3 </td><td width="150">  Associate the sample with the reference database table by gene identifier, sort by gene position, take the first 100 entries </td><td align="center" width="76">  775 </td><td align="center" width="78">  347 </td><td align="center" width="95">  168 </td><td align="center">  15 </td><td align="center">  6 </td></tr><tr><td align="center" width="77">  four </td><td width="150">  Associate the sample with the reference database table by gene identifier, take the first 100 entries </td><td align="center" width="76">  211 </td><td align="center" width="78">  275 </td><td align="center" width="95">  121 </td><td align="center">  one </td><td align="center">  one </td></tr><tr><td align="center" width="77">  five </td><td width="150">  Associate the sample with the reference database table by gene identifier, filter by one field, sort by gene position, take the first 100 records </td><td align="center" width="76">  13 </td><td align="center" width="78">  111 </td><td align="center" width="95">  61 </td><td align="center">  one </td><td align="center">  one </td></tr><tr><td align="center" width="77">  6 </td><td width="150">  Associate the sample with the reference database table by gene identifier, group by one field, sort by another field, take the first 100 records </td><td align="center" width="76">  37 </td><td align="center" width="78">  41 </td><td align="center" width="95">  23 </td><td align="center">  five </td><td align="center">  2 </td></tr><tr><td align="center" width="77">  7 </td><td width="150">  Select from the table of the reference database, group by one field, sort by another field, take the first 100 records </td><td align="center" width="76">  12 </td><td align="center" width="78">  25 </td><td align="center" width="95">  20 </td><td align="center">  15 </td><td align="center">  7 </td></tr><tr><td align="center" width="77">  eight </td><td width="150">  Select from the sample row file that does not exist in the reference database table, take the first 100 entries </td><td align="center" width="76">  37 </td><td align="center" width="78">  367 </td><td align="center" width="95">  432 </td><td align="center">  five </td><td align="center">  2 </td></tr><tr><td align="center" width="77">  9 </td><td width="150">  Link four samples by gene identifier, filter by one field, sort by another field, take the first 100 entries </td><td align="center" width="76">  7 </td><td align="center" width="78">  sixteen </td><td align="center" width="95">  20 </td><td align="center">  3 </td><td align="center">  2 </td></tr><tr><td align="center" width="77">  ten </td><td width="150">  Add 1 million records to a table containing 60 million records </td><td align="center" width="76">  7 </td><td align="center" width="78">  35 </td><td align="center" width="95">  33 </td><td align="center">  four </td><td align="center">  3 </td></tr><tr><td align="center" width="77">  eleven </td><td width="150">  Add 1 million entries to a new empty table </td><td align="center" width="76">  4.5 </td><td align="center" width="78">  35 </td><td align="center" width="95">  33 </td><td align="center">  6 </td><td align="center">  21 </td></tr><tr><td align="center" width="77">  12 </td><td width="150">  Add 60 million entries to a new empty table </td><td align="center" width="76">  140 </td><td align="center" width="78">  500 </td><td align="center" width="95">  280 </td><td align="center">  214 </td><td align="center">  124 </td></tr></tbody></table><br><h2>  Conclusion </h2><br>  Considering the results obtained, the above alternatives do not satisfy the requirement for the duration of treatment (2 s). <br><br>  We recommend the following steps. <br><br><ol><li>  In the short term: use a relational DBMS (for example, SQL Server, MySql, Postgres) to process most real-time queries sent from a dashboard that need to be processed in less than two seconds.  Make sure that the execution plan does not provide for a full table scan. </li><li>  Deploy a small Hadoop cluster (including Spark, Impala, HDFS and Parquet).  Use it to download individual complex web requests sent from a dashboard that do not require a response from a relational database to Impala in less than a second.  After a while, increase the number of queries processed outside the relational DBMS, using Impala to perform the most complex and resource-intensive tasks. </li><li>  Use additional ad hoc queries and optimize data preparation with Impala and Spark.  Pre-process the raw data using Spark and Impala, and then load the appropriate data slices into a relational DBMS.  Use Impala technology to build new special requests in the web application and add the appropriate changes to the interface (for example, "Wait ...", "Loading ...") so that the increased response time does not affect the user experience. </li></ol></div><p>Source: <a href="https://habr.com/ru/post/344060/">https://habr.com/ru/post/344060/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../344050/index.html">What's new in WebStorm 2017.3</a></li>
<li><a href="../344052/index.html">Short shoulder match</a></li>
<li><a href="../344054/index.html">Practical use of multiple bounds generic in Java</a></li>
<li><a href="../344056/index.html">Recover Huffman Tables in Intel ME 11.x</a></li>
<li><a href="../344058/index.html">Mac OS High Sierra, is there a remote ‚Äúpasswordless root‚Äù? (Yes, but unlikely)</a></li>
<li><a href="../344062/index.html">The evolution of solid-state drives: from the first models of the 70s to the present day</a></li>
<li><a href="../344064/index.html">GDPR - new rules for the processing of personal data in Europe for the international IT market</a></li>
<li><a href="../344066/index.html">[Translation of the article] 7 basic rules for protection against phishing</a></li>
<li><a href="../344068/index.html">Instructions for use sysadmin in small business</a></li>
<li><a href="../344070/index.html">As in banks: the state will insure exchange investments of citizens in the amount of up to 1.4 million rubles</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>