<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Genetic engineering of bacteria: how to get the protein we need from bacteria</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="So, in the two previous articles on the genetic engineering of bacteria ( one and two ), we figured out how to collect the genes we need, in what form...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Genetic engineering of bacteria: how to get the protein we need from bacteria</h1><div class="post__text post__text-html js-mediator-article">  So, in the two previous articles on the genetic engineering of bacteria ( <a href="https://geektimes.ru/post/286520/">one</a> and <a href="https://geektimes.ru/post/286574/">two</a> ), we figured out how to collect the genes we need, in what form they can be introduced into a bacterium, and how exactly they can be inserted there.  Suppose we did all these manipulations in order to make a biofactory for the production of protein.  Then now it‚Äôs easy to get the bacteria out of our bacteria in the purest form possible. <br>  There are many methods for solving this problem, most of them are related to chromatography.  How these methods work read under a cat. <br><div style="text-align:center;"><img src="https://habrastorage.org/files/52e/50a/cdd/52e50acdd90e4e46935d3578a4f85229.jpg"></div><br>  <i>The stationary phase "catches" molecules passing by for their (-OH) groups.</i> <br><a name="habracut"></a><br>  So, we got the strain we needed and increased the biomass.  Now we need to somehow remove our protein from it. <br><br>  First of all, we take our culture and pour it into bottles for centrifugation.  Bacteria are rather large and heavy objects, so for their precipitation no monstrous overload values ‚Äã‚Äãare necessary when centrifuging: 10,000g will be enough for more than 20 minutes.  As a result, we obtain a precipitate at the bottom of the vials and liquid (it is called the supernatant).  We pour out the liquid, because our protein is in the bacteria that are in the sediment.  Then you can act on the circumstances: either immediately work with the obtained sediment, or freeze and store until it is needed.  It can be stored for a very long time.  The further procedure of working with sediment does not depend on whether we kept it in the freezer or not. <br><br>  Let's say it's time to work with sediment.  What's next? 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      First of all, we need to destroy the cells so that the protein can get out.  To do this, use the following methods: <br><br><ul><li>  <b>Sequential freeze-defrost.</b>  Not the most effective way to destroy the cell wall, but it is safe for protein, so they often start with several such cycles and then move on to more efficient methods; <br></li><li>  <b>Ultrasound treatment (sonication).</b>  The method is widely used, it is simple and effective.  Usually, ultrasound is fed directly into the volume of water using an elongated "speaker", which is inserted into the fluid with the cells so that the end of the "speaker" almost reaches to the bottom.  This is a rather ‚Äúhard‚Äù method.  There is also a "softer" alternative - ultrasound is fed into a water bath in which the tubes float. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/436/349/76c/43634976cab34495b9ce1c03928ef0ed.jpg"></div><br>  <i>MIT graduate warns - watch your sonicists carefully.</i> <br><br>  When carrying out ultrasonic destruction of cells, it is important to monitor the temperature: water heats up quickly under the influence of ultrasound, and we do not need denaturation of our protein at all, and all the more we would not want to boil it.  Therefore, in the first case (with an ‚Äúelongated column‚Äù), a tube with cells is inserted into a cup with ice and water, and there should be plenty of ice.  Water in this case acts as a thermal interface between the ice and the tube cooled by it.  In the case of the ‚Äúsoft alternative‚Äù, ice is added directly to the water bath. <br><br><iframe width="560" height="315" src="https://www.youtube.com/embed/jacYVhl2P7k" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe><br>  <i>Demonstration of "hard" ultrasonic processing.</i>  <i>Pay attention to the creepy noise standing around the installation, it is especially well heard at 2:06 of this video.</i>  <i>Someone who can save himself is trying to: someone tries to go out of the room for the time of sonication, someone wears special noise-reducing headphones.</i> <br><br><iframe width="560" height="315" src="https://www.youtube.com/embed/YR5PgwEDFzs" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe><br>  <i>Demonstration of "soft" ultrasonic treatment.</i> <br></li><li>  <b>Adding lysing agents to the medium.</b>  These are mainly detergents, that is, ‚Äúsoap‚Äù (in general, the word ‚Äúdetergent‚Äù in English means ‚Äúdetergent‚Äù).  Detergents are essentially related to lipids in cell membranes.  Once in the medium with the cells, the detergents begin to be embedded in the membrane, which leads to their destruction.  And we just need this.  Sometimes lysozyme is also added - an enzyme that destroys the cell wall of bacteria. <br></li><li>  <b>French press.</b>  The cell suspension is sucked through the inlet valve into the working cylinder, from where it is squeezed by a press through a narrow slit.  The transverse forces that occur when pressure drops to atmospheric pressure when cells pass through the gap, lead to the destruction of the cell wall and membrane. </li></ul><br>  From my own experience I can say that sonication in combination with detergents is usually more than enough. <br><br>  So, we got a cell lysate - ‚Äúsoup‚Äù, consisting of the protein we need and a huge amount of ‚Äúgarbage‚Äù (chromosomal and plasmid DNA, pieces of the cell wall, various proteins, etc.).  How to get rid of all this? <br><br>  The first question that needs to be answered is: ‚ÄúIn what form did our protein exist in the cell?‚Äù.  There are two main options: <br><ul><li>  The protein was soluble before lysis and with a very high degree of probability remained so after it, which means that we cannot sediment it by centrifugation.  But on the other hand, we can precipitate large ‚Äútrash‚Äù remaining after lysis: this way we will get rid of pieces of the cell wall, chromosomal DNA and other large fragments.  The precipitate is discarded, and the supernatant is sent to chromatography; <br></li><li> Protein in the conditions of bacterial cytoplasm was insoluble and it formed the so-called.  "Inclusion bodies", preserved after lysis.  Inclusion bodies are protein balls that are formed if the protein does not have time to fold (that is, to take its normal soluble form) during the synthesis, before it collides with another similar unfolded protein.  Non-folded proteins almost always have hydrophobic regions sticking out, with which they will stick to each other.  After the two squirrels have "stuck together", they no longer have enough degrees of freedom to complete their folding.  Then the third, fourth, fifth ... swim to them.  The result is a protein granule.  This usually happens when we synthesize eukaryotic proteins in bacteria (as I wrote earlier, the bacterium does not know how to produce them).  As a result, we have a mass of protein with a ‚Äúpoor‚Äù spatial structure.  But there is a blessing in disguise, because as a result, granules are formed that consist almost entirely of the protein we need! <br><br>  Since, in this case, our protein is insoluble, it is in the sediment when centrifuged along with the ‚Äúgarbage‚Äù.  We discard the supernatant, and then the bodies in the sediment are washed off from all unnecessary things as follows: <br><ol><li>  Pouring wash buffer No. 1 was poured in which the ‚Äúgarbage component of sediment No. 1‚Äù is dissolved; </li><li>  Stir the precipitate until homogeneous and set on a shaker (a device that shakes); </li><li>  Centrifuged and poured the supernatant.  In the end, what was dissolved in the wash buffer No. 1 was separated from the sediment; </li><li>  Pouring wash buffer No. 2 was poured in which the ‚Äúgarbage component of sediment No. 2‚Äù is dissolved; </li><li>  Well, and so on.  In total, about 5 washes are usually needed. </li></ol><br>  <b>Let me</b> remind you once again that if we are dealing with inclusion bodies, then we are dealing with an <b>unfolded</b> protein, which means that it is not able to fulfill its functions.  Therefore, before subsequent purification, it is refolded, that is, it is brought into active form.  The most common refolding method consists of the following steps: <br><ol><li>  <b>Dissolve the calf in "non-physiological" conditions.</b>  Usually, the calves dissolve well at high pH values ‚Äã‚Äã(about 12) in the presence of two-molar urea.  The idea of ‚Äã‚Äãthis method is that at high pH there are very few protons in the medium and everyone who could get rid of them can easily do it.  A proton is positively charged, which means that all proton donors receive a negative charge, and then everything that makes Coulomb's law does ‚Äî proteins that have a large negative charge start to repel each other; <br></li><li>  <b>Drop by drop add the entire volume of the ‚Äúprotein solution‚Äù from the first item to a much larger actively mixed volume of buffer with protein-comfortable conditions.</b>  The idea of ‚Äã‚Äãthis stage is that at a low protein concentration, he is much more likely to fold before he meets with his unfolded counterpart.  Usually, the drop-by-step transfer of the protein solution is carried out in buffer volumes exceeding the volume of the solution itself at least 10 times. </li></ol></li></ul><br>  So, we have ready some volumes of protein solutions in the folded state.  The further procedure is the same for any "initial conditions" - we filter our solutions through filters with a pore size of 100-450 nm (in order not to clog the chromatographic columns and the chromatograph itself), and then perform chromatography. <br><br><h3>  Types of chromatography </h3><br>  The principle of any chromatography is that the various components of the solution move through the column at different speeds; as a result, the components ‚Äúentering the column‚Äù together end up at the output at different times. <br><br>  But first, we recall another method - protein electrophoresis in a gel. <br><br>  This method allows the separation of proteins according to their masses.  For this purpose, a buffer containing <a href="https://ru.wikipedia.org/wiki/%25D0%259B%25D0%25B0%25D1%2583%25D1%2580%25D0%25B8%25D0%25BB%25D1%2581%25D1%2583%25D0%25BB%25D1%258C%25D1%2584%25D0%25B0%25D1%2582_%25D0%25BD%25D0%25B0%25D1%2582%25D1%2580%25D0%25B8%25D1%258F">SDS</a> and a dye (usually bromophenol blue) is added to the sample.  At the same time, the density of the buffer is higher than the density of water; this is done to make the mixture of the sample and buffer easier to apply to the wells on the gel.  The dye is added to control the procedure for introducing the sample into the well, and it also serves as an indicator that it is time to finish the Forez: the ink moves faster than the protein, so when it reaches the end of the gel, the Forex stops.  This is very convenient, since we do not see the movement of the protein along the gel. <br><br><iframe width="560" height="315" src="https://www.youtube.com/embed/nfxXYG2JMQ8" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe><br>  <i>SDS-PAG electrophoresis.</i>  <i>It is better to read and see once than read a hundred times.</i>  <i>The molecular weight marker was applied to the first well on the forez in the video, and the samples under study were applied to the remaining wells.</i> <br><br>  SDS is a key component: this negatively charged detergent (surfactant) denatures proteins, after which it ‚Äúsurrounds‚Äù them, and it turns out that the number of SDS molecules attached to a protein is directly proportional to its mass.  Then it follows from Coulomb's law that in the absence of any resistance to movement, all proteins surrounded by SDS molecules would have the same acceleration in an electric field (the force grows linearly with the charge, but linearly with the charge the mass grows). <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/c3e/f57/8e4/c3ef578e40b04968bfc040e2dbffb163.gif"></div><br>  <i>Diagram of the effect of negative charged detergent SDS on protein.</i>  <i>First, he denatures him, and then ‚Äúenvelops‚Äù him.</i>  <i>As a result, the protein consists of a negatively charged membrane consisting of SDS, the charge of which is directly proportional to the mass of the protein.</i> <br><br>  And here we come to the aid of PAGE (PAG - polyacrylamide gel).  During polymerization, it forms a network of channels of approximately the same diameter, through which, all other things being equal, small objects move faster than large ones.  It is obvious that the heavy protein is larger than the lung, so light proteins move faster. <br><br>  After the end of the electrophoresis procedure, the gel is stained.  More precisely, the places where the protein is in the gel are stained - blue stripes are obtained (or black, depending on what to paint).  In order to understand what mass corresponds to one or another strip, you need to compare with something, therefore, so-called is applied to each gel.  "Molecular weight marker" - a mixture of components with known masses. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/c0c/d4c/147/c0cd4c1473c24a7a9c1a01326a642746.jpg"></div><br>  <i>An example of molecular weight marker strips.</i>  <i>This marker is also convenient due to the fact that its two bands are painted in an individual color: red (70 kilodaltons) and green (10 kilodaltons).</i>  <i>This greatly helps the experimenter to navigate the marker itself.</i>  <i>1 dalton, also known as atomic mass unit, is defined as 1‚Äì12 mass of a free carbon 12 atom at rest, which is in the ground state.</i> <br><br>  It would seem a miracle, not a method - the strips do not overlap, which means that the proteins are ideally divided!  But in fact, scientists have not learned how to conduct protein electrophoresis on an industrial scale, and therefore we still use column chromatography. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/0bd/cd8/6da/0bdcd86da8fb43339267dc5138f02d27.jpg"></div><br>  <i>An approximate scheme of any chromatography.</i>  <i>A column is something filled with a cylinder.</i>  <i>This ‚Äúsomething‚Äù is called a ‚Äústationary phase,‚Äù and the ‚Äúmobile phase‚Äù is a solution, the components of which we need to separate.</i>  <i>Components move at different speeds and therefore leave the column separately.</i> <br><br>  Most chromatographic techniques are based on the fact that passing by proteins cling to the fixed filler of the column "with different enthusiasm": those who "cling" often and strongly move the slowest of all, and those who do not "cling" to anything that quickly leave the column.  As a result, a substantial part of the mixture often leaves the column without ever ‚Äúcatching on‚Äù for it - they go together at the very beginning of chromatography (this fraction is called ‚Äúovershoot‚Äù, as they slipped through the column without ‚Äúcatching on‚Äù for it).  But what to do with those who "hooked"? <br><br>  To begin with, let us deal with what is happening in the column with those who ‚Äúcling‚Äù.  The ‚Äúhook‚Äù itself occurs because there is something on the surface of the stationary phase that is capable of ‚Äúcapturing‚Äù squirrels passing by for a while (how it happens - just below).  That is, the ‚Äúclinging‚Äù protein spends some time in a state associated with a solid phase, then it detaches from it (due to its own thermal motion, collisions, etc.), swims further, again binds to the solid phase, again detaches and so on until he leaves the column.  It is clear that the more often and the stronger it is associated with the stationary phase, the slower it moves along the column. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/2fd/097/406/2fd0974064f3490e84919bd751c63569.png"></div><br>  <i>The principle of chromatography.</i>  <i>That which does not bind to the column (overshoot) moves at the speed of the mobile phase (adjusted for size).</i>  <i>The binding components have different speeds, which leads to their separation at the output.</i> <br><br>  Generally speaking, only in the case of great bad luck there will be different components in the mixture that move with the same speed, so we can just wait for them to come out of the column.  But it is often ‚Äújust to wait‚Äù for a very long time, as the components move slowly.  In this case, they need to be ‚Äúcustomized‚Äù, and the way they are ‚Äúcustomized‚Äù depends on the type of the column. <br><br>  Finally, consider the chromatographic methods themselves, which are suitable for the industrial production of belov. <br><ol><li>  <b>Ion exchange chromatography.</b>  The idea of ‚Äã‚Äãthe method is that the stationary phase has a charge under the conditions of chromatography.  If the charge is positive, then they say about "anion-exchange chromatography", if negative, then about "cation-exchange". <br><br>  The logic of the name is simple, consider it on the example of an anion exchanger: negatively charged components of the mixture ‚Äústick‚Äù to the positively charged solid phase of the column.  Moreover, if we follow a specific positively charged group on the surface of a solid phase, then one or another negatively charged component will ‚Äústick‚Äù to it one by one.  Thus, there is a kind of anion exchange between the stationary and mobile phases, hence the name ‚Äúanion-exchange chromatography‚Äù. <br><br>  Obviously, if we want to use anion-exchange chromatography, then we must carry out the separation of the mixture in such conditions under which the solid phase is positively charged.  At the same time, we have a choice of two options for further development of events depending on what kind of charge a protein has under these conditions: if ‚Äú-‚Äù, then it ‚Äúsits‚Äù on the column, and if ‚Äú+‚Äù, then it will be in overshoot. <br><br>  How to speed up the output of related components, that is, how to carry out the "elution"?  You can act in two ways: <br><ul><li>  You can change the charge of the stationary phase and the bound components by changing the pH of the mobile phase.  It is important to make the process of pH change gradually, then the components will accelerate their movement, but they will not ‚Äúfall out‚Äù at the same time all at once; <br></li><li>  Add other anions to the column.  Indeed, the column has a finite number of places for which anions can catch, and if we take all these places with something, the components of the mixture will move faster.  For anion exchange chromatography, the addition of chlorine anions is typical.  The addition of competing anions should also be carried out gradually. </li></ul><br>  The procedure for cation exchange chromatography is generally the same, only the opposite is true. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/2fd/0b4/bcf/2fd0b4bcf1d34cc8bd63710f2147302f.JPG"></div><br>  <i>Cation-exchange chromatography with elution due to changes in the pH of the mobile phase.</i> <i><br></i>  <i>1 - at pH = 2, there are a lot of protons in the mobile phase, so all the components of the mixture picked them up from the liquid, while receiving a positive charge;</i> <i><br></i>  <i>2 - at pH = 5, the number of protons in the mobile phase has decreased and the ‚Äúpurple‚Äù component already acts not as an acceptor, but as a proton donor, therefore, it has a negative charge at a given pH value and it leaves the column.</i>  <i>The ‚Äúgreen‚Äù component still acts as a proton acceptor and it retains a positive charge at a given pH value, which means it is still holding on to the column;</i> <i><br></i>  <i>3 - at pH = 9, there are very few protons in the mobile phase, the ‚Äúgreen‚Äù component also becomes a proton donor, acquires a negative charge and leaves the column last.</i> <br><br>  In general, the choice of the type of ion exchanger and the conditions of separation depends on what we want to isolate and what else is in our mix. <br></li><li>  <b>Another separation method is based on the hydrophobicity (~ non-polarity) and hydrophilicity (~ polarity) of certain substances.</b>  The idea here is exactly the same as in the case of the ion exchanger: the components, according to their physical properties, are ‚Äútorn‚Äù between hydrophobic and hydrophilic environments, some linger longer, some less. <br><br>  First of all, let me remind you that the non-polar dissolves in the non-polar, and the polar in the polar.  Often, ‚Äúhydrophobic‚Äù and ‚Äúnon-polar‚Äù are the same as ‚Äúhydrophilic‚Äù and ‚Äúpolar‚Äù, although in general this is not the case.  An example of a non-polar solvent is ethanol, and an example of a polar one is water. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/419/fa0/76e/419fa076eb474b6d811ca6df576143f9.jpg"></div><br>  <i>The polar bear laments the fact that it dissolves in water, which is a polar solvent.</i>  <i>"Non-polar" brown bear is not threatened.</i> <br><br>  Now consider an example.  Suppose the stationary phase is hydrophobic (non-polar).  Then the mobile should be generally hydrophilic (polar).  All hydrophobic components will ‚Äúsit down‚Äù on the column, since it is ‚Äúscary‚Äù to swim in the hydrophilic mobile phase.  Hydrophilic components will be in the overshoot, as they prefer the hydrophilic mobile phase.  Then, the hydrophobic components ‚Äúsat down‚Äù on the column should be eluted by gradually increasing the hydrophobicity of the mobile phase.  The regulation of the hydrophobicity of the mobile phase is carried out by a pump that simultaneously pumps the polar and non-polar solvent from two different tanks through the column.  The total polarity of the mobile phase depends on the ratio between them. <br></li><li>  <b>Metal chelate chromatography with his-tag (His-tag).</b>  The method is based on the fact that areas of six histidines (histidine is one of 20 proteinogenic amino acids) bind well to the nickel ions on the column through their imidazole groups.  The method is convenient because there is little in which natural protein there is a histidine tag, which means with a high degree of probability all proteins except ours will be in the breakthrough. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/d0e/606/d02/d0e606d025074af5adf02932a9967450.png"></div><br>  <i>The structure of the molecules of imidazole and histidine.</i> <br><br>  The disadvantage of the method is that it is necessary to modify the protein, and this may affect its properties.  Addition of 6 histidine tags to the protein occurs at the stage of gene creation, and to minimize the effect of the tag on the protein, these amino acids are added at the beginning or end of the amino acid sequence.  Naturally, the method will not work if the protein is such that its beginning and end are hidden inside: the nickel ions and the tag simply cannot meet. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/0ea/da6/f47/0eada6f47fbd41cdb8a774436e572638.jpg"></div><br>  <i>Chromatography on a nickel column using his-tag tag.</i>  <i>All proteins, except for the ‚Äúpurple hexagon‚Äù protein, do not have His-tag and are in the gap.</i>  <i>The purple hexagonal proteins are eluted by imidazole coating on the column.</i> <br><br>  "Remove" protein from the column can be a gradual addition of imidazole.  The essence of the elution method is the same as that of ion exchangers - imidazole competes with the His-tag protein for binding sites on the stationary phase (only on the ion exchanger binding sites were abstract ions, and on the metal chelate column, the binding sites are nickel ions).  As a result, it occupies most of the available binding sites, and the squirrel has no choice but to leave the column. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/63c/3f4/e72/63c3f4e72e434ac3a0db8a1b8fe59ef3.png"></div><br>  <i>Electrophoresis according to the results of nickel metal chelate chromatography.</i> <br>  <i>1 - the original "dirty" material for application to the column;</i>  <i>2 - overshoot;</i>  <i>3 and 4 - washing without imidazole;</i>  <i>M - molecular weight marker;</i>  <i>5-10 sequential elution with imidazole.</i>  <i>The protein we need is labeled with an arrow to the left.</i>  <i>As we see, in the breakthrough there is no our protein, but in it there is a lot of "garbage".</i>  <i>The eluate is a purified target protein.</i>  <i>On the right are the masses of the respective bands of the molecular weight marker.</i> <br><br>  The method could potentially be convenient also by the fact that there is a dependence of the binding force on the amount of histidine tags in the protein composition.  Thus, it would be possible to attach to one protein tag only at the beginning, and to the other from both ends and synthesize them in one bacterium.  But in reality it is not very practical: the difference in the concentrations of imidazole that elute such proteins is small (approximately 75 mM and 100 mM imidazole for one and two tags, respectively), so there is a possibility of difficulties with their separation from one mixture.  And making two separate strains is no more difficult than making one with two genes. <br><br>  Another possible difficulty in working with the His-tag is the need to obtain a protein without any modifications, which exists in some experiments.  In principle, it is possible to cut off the protein from the enzymes, but some of the protein will remain with the tag (that is, ‚ÄúEFFICIENCY &lt;1‚Äù), and this will also lead to the fact that we will need to carry out an additional stage of purification from these enzymes.  In addition, each stage of purification is associated with losses of the target product. <br></li><li>  <b>Gel filtration</b>  I‚Äôll just say that this method can be called suitable for protein extraction only conditionally: if we need to get a small amount of protein for research, then using gel filtration makes sense, but this technique is not suitable for working with truly industrial volumes, and here's why. <br><br>  In all previous methods, the speed of movement of the components associated with a column in the absence of ‚Äúfitting factors‚Äù is so small that it can generally be considered close to zero.  With gel filtration, everything is different: all components, even the largest ones, move at quite noticeable speeds.   ,  -   -  ,        (   ,         ),          .       :      -     1-2%   . ,          ,       ,     10       400 : ,            .  -        . <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/b94/ca3/a5f/b94ca3a5f0b5444a94437fd270c99988.gif"></div><br> <i>  -:             ,      .</i> <br><br>  - ?     ,   ,     .   ,      ,             ,    .   -     ,  . <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/a3b/d31/e43/a3bd31e43ec440b9b2a6dcb4b7b49044.png"></div><br> <i>   -.       . ,        .  ‚Äî .        .</i> <br><br>       -,     .      ,      ,         ,      . - ‚Äî   ,           . </li></ol><br>  Well then.                ,           ,      . </div><p>Source: <a href="https://habr.com/ru/post/402685/">https://habr.com/ru/post/402685/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../402675/index.html">Harvard CS50 course in Russian: lecture twenty-one</a></li>
<li><a href="../402677/index.html">Course Overview - Building robots and other devices on the Arduino (week 1)</a></li>
<li><a href="../402679/index.html">Universe and heavy metal</a></li>
<li><a href="../402681/index.html">DIY: Lightsaber with sound</a></li>
<li><a href="../402683/index.html">Some additions for 3d printer Anet A8</a></li>
<li><a href="../402687/index.html">Divoom TimeBox: desktop bluetooth speaker with pixel display</a></li>
<li><a href="../402689/index.html">Dangerous products for drivers. Test breathalyzer DrinkMate</a></li>
<li><a href="../402691/index.html">Falcon 9 check completed for SES 10 mission</a></li>
<li><a href="../402695/index.html">Treat gus with beer! Or about the license Beerware</a></li>
<li><a href="../402697/index.html">A vulnerability has been discovered in the Miele dishwasher's web server.</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>