<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Practical bioinformatics</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Found a tough shortage of information on bioinformatics in the Russian segment. I don‚Äôt know, it‚Äôs true or not, but I want to provide the reader with ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Practical bioinformatics</h1><div class="post__text post__text-html js-mediator-article">  Found a tough shortage of information on bioinformatics in the Russian segment.  I don‚Äôt know, it‚Äôs true or not, but I want to provide the reader with an introduction that can be called practical bioinformatics, which I really didn‚Äôt have enough to familiarize myself with the subject.  In this chapter, I want to describe the path that I had to go up to the present moment, when I no longer shy away from phrases: here's a FASTQ file for you and build me a bed graph for the genome browser.  In order to continue the conversation about the interesting, I want to walk diagonally through the definitions and programs of primary data processing, without which it is difficult to speak the same language. <br><a name="habracut"></a><br>  First, some definitions.  We assume that chromosomes are one-dimensional coordinate axes, from one to about 10e8.  The length of the axis depends on the length of the chromosome.  Each point of the axis is an integer.  Biologists and chemists conducted a large number of experiments and thanks to them they were able to describe parts of the chromosomes with great accuracy (about 90%).  These descriptions are called annotations.  The abstract contains information about the length of chromosomes, the coordinates of individual sections of chromosomes, the most famous of them are the gene, <a href="http://en.wikipedia.org/wiki/Intron">intron</a> , <a href="http://en.wikipedia.org/wiki/Exon">exon</a> .  There is a huge number of these sections, but their main property is that these segments or a set of segments are located on the coordinate axis.  Some segments may include others or intersect in some way.  Here is a set of sites where you can see annotations of human genomes, mice, and the like. <br><ul><li>  <a href="http://useast.ensembl.org/Homo_sapiens/Info/">useast.ensembl.org/Homo_sapiens/Info</a> </li><li>  <a href="http://genome.ucsc.edu/cgi-bin/hgGateway">genome.ucsc.edu/cgi-bin/hgGateway</a> </li><li>  <a href="http://www.genenames.org/">www.genenames.org</a> </li></ul><br>  Biologists with chemists conduct their experiments and as a result of their operations on cells they get a solution containing relatively small pieces of DNA or RNA (I don‚Äôt really want to go into details of difference or sameness, just a sequence of <a href="http://en.wikipedia.org/wiki/Nucleotide">nucleotides</a> ).  This solution is passed through sequencing equipment, the output of which is small lines.  These strings are the ends of pieces of DNA or RNA that were in solution.  The length of the strings obtained from the equipment is only 36-50 bases (the length of the string in nucleotides) is sometimes longer, but at the current moment it seems to be no more than 200. These segments, obtained from sequencing equipment and defined by a sequence of nucleotides, are called reads (from English reads - ‚Äú reading").  It should be noted that reads are characterized only by the sequence of nucleotides, and not by their location on the genome.  Sometimes these sequences are supplemented with a string of probabilities that matches the position of a nucleotide to its probability of being in that position.  A FASTA file is a file without probabilities, FASTQ is a file with probabilities. <br><br>  Further, depending on what was the result of the experiment - pieces of DNA or RNA, one of the two sequencing methods <a href="http://en.wikipedia.org/wiki/Chip-sequencing">ChIP-seq</a> or <a href="http://en.wikipedia.org/wiki/RNA_sequencing">RNA-seq</a> , respectively, is performed.  More details about them are described here <a href="http://en.wikipedia.org/wiki/DNA_sequencing">http://en.wikipedia.org/wiki/DNA_sequencing</a> . <br><br>  After the expensive sequencing machines work out and produce the result in the FAST A / Q file, you need to find the resulting sequences in the genome.  For ChIP-seq, the <a href="http://bowtie-bio.sourceforge.net/">bowtie</a> program does a very quick search at home, <a href="http://bowtie-bio.sourceforge.net/">finding</a> millions of reads in just 5 minutes.  Those.  it is looking for the entry of a string of 36‚Äì50 characters consisting of at least four alphabetic alphabets in strings with a total length of 10‚Äì9.  Why at least a turnover is used: in addition to the standard use of the A / TG ‚Äã‚Äã/ C alphabet, the symbol N is often added, replacing any possible letter ( <a href="http://genome.ucsc.edu/FAQ/FAQdownloads.html">http://genome.ucsc.edu/FAQ/FAQdownloads.html#download5 for</a> more details).  The program has many parameters.  For example, you can resolve an error in a line (read) or two (up to a maximum of 3 errors).  She can search not only one entry of the read into the genome, but many.  It can slyly sort the data, for example, if the read with one error was unequivocally found in the genome, and with two or three errors was found in many parts of the genome, then only the first result should be output.  This process of finding the reads in the genome is called the mapping from the English mapping (also called alignment).  The algorithm of such a quick search is very interesting, but you can devote a separate article to this or find a <a href="http://genomebiology.com/2009/10/3/R25">link</a> to an English article from the developer‚Äôs website, which tells how the bzip2 algorithm prompted them to such a solution.  There are a lot of mapping programs and sites where they make it online;  For the keywords blast, eland + genome in Google you can find additional information. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      For RNA-seq, the procedure is a bit more complicated, ChIP-seq mapping is performed for it, and then those reads that are not found during ChIP-seq are processed.  A good program that actively uses the bowtie for their work is called <a href="http://tophat.cbcb.umd.edu/">tophat</a> .  As a result of <a href="http://en.wikipedia.org/wiki/RNA_splicing">splicing</a> and the <a href="http://en.wikipedia.org/wiki/Protein_isoform">isoformism</a> generated by it <a href="http://en.wikipedia.org/wiki/Protein_isoform">,</a> parts of the <a href="http://en.wikipedia.org/wiki/Protein_isoform">reed</a> may be located in different places of the genome.  For example, the first 15 characters may fall on one region of the genome, and the other 11 on another.  This division of a reed into parts, at the end of one exon and at the beginning of another, is called splice-junctions.  Tophat allows them to be found, and also identifies new possible isoforms of existing genes. <br><br>  The result of these programs is the <a href="http://samtools.sourceforge.net/">sam / bam</a> file, which contains information from the FAST A / Q file plus information about the coordinates on the corresponding spirals of the chromosome.  The process from the library to the sam / bam file is often referred to as the pipeline procedure and in many laboratories is on the stream, so you should ask what parameters and software versions are installed by default.  In general, this is what the intro part ends with, then comes the line of research.  It may be noted that from this moment we have data that are with some certainty in the same coordinates: the coordinate axis, annotations, reads with the corresponding coordinates. <br><br>  You can proceed to the analysis of data from a simple elucidation of uniformity and continuity, complexity, ending with complex statistical calculations that help to divide the reads into certain groups: noise, zero level and enrichment.  Data analysis is necessary in order to be able to reasonably discard certain data in the future. <br><br>  If you are interested in the introductory part, then at each stage I can elaborate.  Unfortunately, a simple introductory part already occupies several pages of dry text without pictures, therefore, I did not dare to describe the written programs and mechanisms in this chapter.  I myself am most attracted to the last paragraph, where I briefly mentioned statistics.  I would like to consecrate existing libraries and mechanisms for working with such data.  Here you can attach the methods included in datamining (different types of clustering), which are described in Habr√©.  How to apply a Poisson distribution to analyze data without control, how to apply a complex chain of Poisson, f-test, to find areas enriched in Reed on the genome (Diarac delta function)?  Are ready-made libraries useful for working with intervals boost.intervals, boost.icl? <br><br>  And, of course, if this topic is interesting, maybe someone will tell you how and where to dig, in what matters, and add.  And maybe he will write his own.  For the solution of biological problems without mathematics and programming at this stage is definitely impossible.  There are English-speaking resources where similar issues are discussed <a href="http://www.seqanswers.com/">www.seqanswers.com</a> .  But I would like in the future to get away from the description of finished products and engage in a discussion of the legitimacy of applying mathematical and statistical methods in these programs and the possibility of applying new methods. <br><br>  At the current stage of our work, we tried to find parameters by which you can filter reads, both for interesting for research and for noise.  The task is rather nontrivial in the absence of control.  In the future, it was decided to add a control to the library for sequencing, which will allow measuring the level of error, but the statistics will not let go anyway. </div><p>Source: <a href="https://habr.com/ru/post/137069/">https://habr.com/ru/post/137069/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137063/index.html">Two rules for a good iPhone app icon</a></li>
<li><a href="../137065/index.html">Multithreading in practice</a></li>
<li><a href="../137066/index.html">Memory capture / release speed in C #</a></li>
<li><a href="../137067/index.html">116 sites for promoting iPhone / iPad applications</a></li>
<li><a href="../137068/index.html">If there is no money left for marketing, or is there life in the valley of death?</a></li>
<li><a href="../137070/index.html">Novice Project Manager - Design</a></li>
<li><a href="../137071/index.html">Malware JavaScript decryption</a></li>
<li><a href="../137076/index.html">Historical essay on the protection of information</a></li>
<li><a href="../137077/index.html">Postindustrial or Production 2.0</a></li>
<li><a href="../137078/index.html">Video review of Android applications and games - kedDroid</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>