<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>‚ÄúLive graphs‚Äù - growing graphs on cellular automata with Silverlight examples</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Introduction 

 Perhaps nothing for so long, for centuries, did not interest scientists, as questions about the origin of life and mind. How did natur...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>‚ÄúLive graphs‚Äù - growing graphs on cellular automata with Silverlight examples</h1><div class="post__text post__text-html js-mediator-article"><h5>  Introduction </h5><br><br>  Perhaps nothing for so long, for centuries, did not interest scientists, as questions about the origin of life and mind.  How did nature guess to create the human brain?  What determines the structure of the neural network in our head and how does the autoassembly of a multicellular organism from a single cell work?  Why is it that when a human embryo develops at a certain stage, something similar to fish gills can be observed? <br><br>  Yes, and just a curious man in the street, not burdened with the details of organic chemistry, such questions are not avoided. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      That would have been a toy designer, with which you can collect simple growing organisms.  Then, having built an extremely simplified model that demonstrates many of the phenomena of the living, one could get closer to the answers to the questions of the device of life, or at least to an understanding of where to look for these answers. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/2c0/71f/cc7/2c071fcc7ca9970a236af957a3cd2281.png" alt="live count"><br><br>  Such an extremely simplified and visual model can be ‚ÄúLive graphs‚Äù - finite automata on a graph, each node of which contains some executing device (automaton) with a finite number of states and with a set of primitive rules governing the creation or modification of new connections between nodes. <br><br><a name="habracut"></a><br><br>  <i>‚ÄúJust see how I grew cabbage!‚Äù</i> (C) Emperor and Downshifter Diocletian <br><br>  The article will describe one of the embodiments of the ‚ÄúLive graph‚Äù - the cellular automaton for the deployment of the graph GUCA (Graph Unfolding Cellular Automata).  You can ‚Äúfeel‚Äù the demonstration of GUSA right in the browser with the <b>Silverlight</b> plug-in connected.  The web application visualizes the short life of some of my first copies of growing graphs ‚Äî both personally designed products of ‚Äúgenetic engineering‚Äù, and ‚Äúnaturally selected‚Äù genetic forms. <br><br>  Even if the initial graph consists of only one node, and all generating nodes of the graph will have the same copy of the rule set with each new iteration of the rules triggering, it will be possible to observe the development of a wide variety of structures.  It is easy to see that the initial node is analogous to the embryo of a multicellular organism, the other nodes of the graph are similar to cells, the sets of rules are the chromosome, and the rules themselves are genes. <br><br><br><h5>  Origins </h5><br><br><br>  <b>And it all started with a passion for artificial neural networks</b> (ANNs), the study of which gave rise to the idea of ‚Äã‚ÄãGUCA. <br><br>  A small reminder.  Artificial neural networks that have found widespread use in artificial control systems can be thought of as interconnected ‚Äúneurons‚Äù that themselves perform fairly simple operations, for example, summing the incoming signals and transforming the resulting value.  But being connected according to some rules into a single network, the ‚Äúneurons‚Äù form a system, the capabilities of which significantly exceed the capabilities of its constituent elements - the same adder.  The system as a whole already has functions of pattern recognition, classification, filtering of signals - all that is necessary for building a complex ‚Äúintelligent‚Äù control system. <br><br>  At the dawn of the use of neural networks, the network topology was set by the researcher, and the connection parameters were selected in advance by a predetermined algorithm using the accumulated data on the problem being solved.  If for sufficiently simple types of neurons, the problem of building a network topology was sufficiently deterministic and solvable (which does not mean that the problem of the domain itself is solvable), then using cascades of neural nonlinear dynamic networks with feedback, this task turned out to be non-trivial and the idea arose to entrust the construction topology of the neural network computer.  Came to the aid of methods, peeped in nature itself. <br><br>  <b>We are talking about neuroevolutionary methods,</b> in which genetic algorithms are used to build the topologies of neural networks. <br><br><h5>  Neuro-evolutionary methods </h5><br><br>  The review of such methods was recently published on Habr√©: [1] <a href="http://habrahabr.ru/blogs/artificial_intelligence/84015/">habrahabr.ru/blogs/artificial_intelligence/84015</a> , <br><br>  I will not retell the already known methods for implementing such algorithms, I will only recall the main provisions of indirect coding of networks / graphs that also underlie the GUCA method under consideration, the implementation of which differs only in locality, contextual choice of the rules applied and ‚Äúprogrammed death‚Äù. <br><br>  In indirect methods of coding neural networks, in the ‚Äúgenome‚Äù of an instance of a network, the topology is not specified by an explicit listing of nodes and connections, but indirectly through the rules for constructing a network from its small fragment.  After all, it is in this way that the organism of an animal develops from a small embryo, and indeed the human brain. <br><br>  I also recall that the genetic algorithm provides: random generation or modification of codes, evaluation of the ‚Äúresult‚Äù of decoding the ‚Äúgenome‚Äù (that is, evaluation of the resulting neural network using a fitness function) and selection of the best representatives. <br><br>  For neural systems that perform the role of object management systems (robots, manipulators), the fitness function may be the performance of the control object in a certain environment (labyrinth, handling objects, in flight, in battle, etc.) <br><br>  At the same time, indirect encoding methods provide the following properties: <br><br>  1) Modularity.  Recursive construction of fractal structures according to simple rules and repetition of certain building blocks in the body structure. <br><br>  2) Completeness.  For any given network (graph), you can pick up the code. <br><br>  3) Redundancy.  The code may contain redundant data that does not affect the decoding result. <br><br>  4) (Subject to use in context encoding rules) <br><br>  a.  Generation of topology depending on the signals passing in the neural network <br><br>  b.  Regeneration - recoverable with local damage <br><br>  Review [1] shows that the main efforts of researchers are aimed at inventing more or less successful methods of coding network topology, rather than designing the principles of a neuron / network and coding its parameters, which is not surprising - it is the topology that determines the work of the entire system. neural network". <br><br><h5>  Topology Cultivation </h5><br><br>  Indeed, it would be useful to single out the problem of ‚Äúgrowing‚Äù the network topology using the same genetic algorithm and indirect coding (in fact, a directed graph, and abstracting even more from the NA ‚Äî an undirected graph).  A fitness function (determining the success of an instance's survival) will no longer be the ability of a neural network to cope with managed objects, but, for example, the geometric properties of a topology.  If the graph is expanded on a plane or in space, then the properties of the resulting image, including  raster. <br><br>  <b>The concentration of efforts on a narrower task</b> (building topologies, abstracting from the neural network) <b>will allow</b> : <br><br>  ‚Ä¢ Find a wider <b>application in areas other than ANN</b> (auto-assembly mechanisms, the configuration of distributed computing in a computer network or viruses, the interaction of combat robots). <br><br>  ‚Ä¢ <b>Visualize the result</b> (not only final, but also intermediate) for the researcher, which can help in the development of algorithms to better understand cause-effect relationships.  In addition, a visual representation can suggest ideas / solutions / dead ends. <br><br>  ‚Ä¢ It is more objective to <b>compare different topology coding solutions</b> or combine successful topology coding solutions with various ‚Äúneuron‚Äù designs.  Objectively - that is,  regardless of the specific application domain. <br><br>  ‚Ä¢ <b>It is easier</b> (faster and cheaper) to <b>calculate fitness functions</b> (as compared to assessing the suitability of a neural network as a control system) - i.e.  it is to investigate topological algorithms, without spending additional resources, including computational ones, on derived tasks. <br><br>  At the same time, such curious properties as the above-mentioned ‚Äúmodularity‚Äù, ‚Äúregeneration‚Äù - will remain available for research. <br><br>  It can be added that the evolution of the brain of animals also originates from the result achieved by the evolution of the form of a multicellular organism (axisymmetry - cartilage - spine - spinal cord - asymmetry - brain brain seal). <br><br>  One of such methods is the cellular automatic deployment of the graph GUCA, the principles of which will be described in the next section. <br><br><h5>  Cellular automatic deployment of the graph GUCA (Graph Unfolding Cellular Automata) </h5><br><br>  The ‚Äúliving graph‚Äù of GUCA is a finite <a href="http://ru.wikipedia.org/wiki/%25D0%259A%25D0%25BB%25D0%25B5%25D1%2582%25D0%25BE%25D1%2587%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25B0%25D0%25B2%25D1%2582%25D0%25BE%25D0%25BC%25D0%25B0%25D1%2582">cellular automaton</a> in which each node is in one of the states (the number of states of course) and can move to another state according to certain rules, depending both on the state of the node itself and on its environment.  These rules are the same for all nodes, they are constant in time and are processed synchronously for all nodes simultaneously. <br><br>  The classic representative of the cellular automaton is the <a href="http://ru.wikipedia.org/wiki/%25D0%2596%25D0%25B8%25D0%25B7%25D0%25BD%25D1%258C_%2528%25D0%25B8%25D0%25B3%25D1%2580%25D0%25B0%2529">‚ÄúLife‚Äù game</a> .  The main difference between ‚ÄúLive graphs‚Äù and the classic game ‚ÄúLife‚Äù and its variations is that the ‚Äúintracellular‚Äù automata are not located on a rectangular plane, but are located at the nodes of the graph with a variable number of connected neighbors. <br><br>  At the same time, unlike most of the coding methods of the network topology described in the review [1], the ‚ÄúLiving Graph‚Äù GUCA professes the following principles: <br><br>  ‚Ä¢ Contextual selection of applicable rules <br><br>  ‚Ä¢ Only local and one-step changes. <br><br>  ‚Ä¢ Not only birth, but death <br><br>  Strangely enough, all these features are inherently inherent in the ‚ÄúLife‚Äù cellular automaton, although they were chosen as a result of a long search for sets of rules and operations that solve the problem <br>  graph deployment is the most simple and complete. <br><br>  However, the removal operation was offered in one of the lecture exercises [2] <br><br><h5>  Principles </h5><br><br>  So, in the ‚ÄúLive graph‚Äù GUCA node is in one of the <b>states</b> .  And there are <b>rules</b> according to which under certain <b>conditions of</b> operation operations are performed on the graph. <br><br>  An example of a set of rules that satisfy the above-described ‚Äúmeta-rules‚Äù for some instance of a ‚Äúlive graph‚Äù is the following list: <br><br>  1. "If the node is in state A, then create a linked node in state B" <br><br>  2. "If the node is in state B, and the number of links is less than two, then create a linked node in state B" <br><br>  3. "If the node is in state B and the previous state was B, then go to state C" <br><br>  four.‚Ä¶ <br><br>  <b>A state</b> is one of the elements of a finite set.  You can denote the number or letter of the alphabet - A, B, C, D, .... <br><br>  <b>Operations</b>  It is clear that it is possible to carry out a variety of operations on graphs - add / divide nodes, replace nodes with a related group, combine nodes.  I stopped at four primitive operations, which seem to be quite enough to reproduce any arbitrarily complex structure. <br><br>  ‚Ä¢ Transition of the node to the state X (thus, the state X is the operand of the operation) <br><br>  ‚Ä¢ The birth of a new node associated with the current.  The status of the new node X <br><br>  ‚Ä¢ A node is connected to the nearest un-connected node in state X <br><br>  ‚Ä¢ Disconnecting a node in state X <br><br>  Now about the conditions that dictate the operation in the rule.  I was not limited to the dependence on the current state of the node and the number of links between this node and other nodes.  To these conditions added dependencies on the previous state of the node and dependence on the number of parents / divisions of the node (after all, each node, according to the list of operations, is generated by one and only one node-parent). <br><br>  <b>Conditions:</b> <br><br>  ‚Ä¢ The node is in state X <br><br>  ‚Ä¢ Previous node state is Y <br><br>  ‚Ä¢ Number of node connections - from C1 to C2 <br><br>  ‚Ä¢ The number of ancestors of the node (from which it was born) - from P1 to P2. <br><br>  The last condition and the counter of the number of node divisions (an analogue of <a href="http://ru.wikipedia.org/wiki/%25D0%25A2%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0%25D0%25B7%25D0%25B0">telomerase</a> ) are introduced to ensure the development of the whole organism and / or its individual modules stop. <br><br>  I also considered the operations ‚Äúconnecting a node with all nodes in state X‚Äù, ‚Äúnode death‚Äù - but I considered them redundant.  I am also thinking of introducing the operation ‚Äúresetting the counter of divisions‚Äù.  I note that all operations are local and ‚Äúrotate‚Äù around a node (they change only the node or its immediate environment). <br><br>  However, perhaps the reader will be able to find a more coherent system of rules. <br><br><h5>  Grammar rules </h5><br><br>  GUCA rules can also be written in a short form, denoting the current state as a letter, the previous one with a letter in brackets, the number of neighbors with the ‚Äúc‚Äù symbol (from connections), the number of parents - ‚Äúp‚Äù (from parents).  If the previous state is ignored in the condition, then we will indicate a dash in brackets.  Let us divide the rule of operation from conditions by a colon symbol, and we denote possible operations as follows: <br><br>  ‚Ä¢ The operation of transition to another state X is denoted by X <br><br>  ‚Ä¢ The operation of the birth of a connected node is denoted by ++ X <br><br>  ‚Ä¢ The operation of connecting to the node X is denoted by + X <br><br>  ‚Ä¢ The detachment operation from node X is denoted by ‚ÄìX <br><br>  Then, for some instance of a ‚Äúlive graph,‚Äù a set of rules will be written using the grammar: <br><br>  1. A (A), p = 0: ++ B <br>  2. B (-), c &lt;2: ++ B <br>  3. B (B): C <br>  4. A (A): G <br>  5. C (B), c = 1: G <br>  6. G (G), c &lt;5: H <br><br>  If the initial node of the graph is in the state ‚ÄúA‚Äù, then after 12 consecutive iterations of the application of the rules, we get the ‚ÄúFinger dumbbell‚Äù: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/276/606/7ad/2766067ade2bac3c19dac35d88e14f41.png" alt="Earl 'Finger dumbbell' and the conditional mapping of her gene map."><br><br>  <i>Figure Graph "Finger dumbbell" and the conditional display of its gene map.</i> <br><br>  This is the very first graph that I designed several years ago to debug a finite state machine on graphs.  It is so simple and clear that I mention it only in order to pay tribute to the pioneer.  The rules laid down in its ‚Äúgenome‚Äù are simple: from the node with the initial state A, a short thread is born that ends with nodes (in the G state) for which the rule is triggered: while the number of neighbors is less than or equal to five, create a new neighbor. <br><br>  Hereinafter, when displaying a graph on a plane and on a conditional display of a gene map, red indicates the state A, pink - B, green - C, crimson - G. <br><br>  Turning off the gene number 5 of the "finger dumbbells" you can create a "freak" - "ordinary hand": <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ccc/15e/ffc/ccc15effcf03b005e059a486aa157200.png" alt="Earl hand"><br><br>  If you turn the gene back, the second "hand" will grow back. <br><br>  Thus, a set of simple rules can define the topology of a graph.  Generally speaking, the topology of a graph determines not only the set of rules and the initial state of its node (s), but also some global constraints ‚Äî for example, the maximum number of iterations or the maximum possible number of connections of nodes. <br><br><h5>  Idea check </h5><br><br>  One way to make sure that the system of rules described above is sufficient for constructing graphs of any topology is to solve ‚Äúcontrol problems‚Äù for construction.  Having thought of this or that figure, try to come up with a ‚Äúgene‚Äù, from which it will be born.  And if the conceived figure cannot be constructed in any way, then the rule system is unlikely to be suitable not only for growing graphs, but also for constructing a complex artificial neural network. <br><br>  On the one hand, ‚Äúcontrol tasks‚Äù should be extremely simple, on the other hand, ‚Äúindicative‚Äù (please do not confuse with ‚Äúspectacular‚Äù) to demonstrate the qualitative properties and effects of a complex living organism - modularity, hybridization, self-healing ... whimsical, finally.  Such examples can be - the simplest fractals, the figures growing from within, the grids with a large number of cells and simply ‚Äúwild‚Äù plants. <br><br>  Some of these tasks were extremely difficult (long) and unreliable (the answer often did not converge) to solve in mind or on paper, and in order to carry out these first experiments, I had to arm myself with an ‚Äúexperimental setup‚Äù. <br><br><h5>  Experimental installation (Silverlight application) </h5><br><br>  <b>This experimental setup</b> was a software application that allows you to debug the "genetic code" and the playback machine.  In addition, with its help you can: <br><br>  ‚Ä¢ visualize the process of graph growth, <br><br>  ‚Ä¢ visually display the color "map" of the chromosome and the activity of genes in the process of graph growth <br><br>  ‚Ä¢ look at what the shutdown of one or another gene or cutting the ‚Äúlive graph‚Äù will lead to. <br><br>  The demonstration Silverlight version of the ‚Äúexperimental setup‚Äù is available at <a href="http://roma.goodok.ru/guca/GUCA_DemoSL.html">http://roma.goodok.ru/guca/GUCA_DemoSL.html</a> <br><br><img src="https://habrastorage.org/getpro/habr/post_images/804/4da/049/8044da049c1130bd4c63c29716798e72.png" alt="Experimental setup for Live Counts"><br><br>  <i>The Silverlight version of the Experimental Setup for growing a graph makes it possible to observe the development of a graph, the activity of genes on a chromosome map, and to cut individual branches with a knife.</i> <br><br>  To <b>start</b> the graph expansion, you need to select an example (Select sample) from the drop-down list on the right ‚Äúcontrol panel‚Äù and click on the start button (‚ÄúStart‚Äù). <br><br>  In the central area with a black background, the graph will be displayed during its growth from a single germ node.  Different states of nodes are displayed in the corresponding color  You can change the scale and appearance of the display or slow down the growth of the graph to view the details of the process. <br><br>  You can not only observe the development and life of the graph, <b>but also intervene</b> in this process by incising with a scalpel (Knife switch) or turning off one or another gene, bringing to light ‚Äúfreaks‚Äù, achieving ‚Äúfatal outcome‚Äù or ‚Äúcancerous swelling‚Äù <br><br>  At the bottom right <b>, a chromosome ‚Äúmap‚Äù</b> (‚ÄúGenome‚Äù) is conventionally <b>displayed</b> - each rectangle on it corresponds to a ‚Äúgene‚Äù (paragraph of the rule).  Again, the states of the nodes in the rule condition, the rule itself, and the node state in the rule operand are color coded.  In the process of growth of the graph, green genes mark those genes that were active during the next iteration of the rules execution. <br><br>  Hovering the mouse on the rectangle of a separate "gene", you can see the decoding of the rule encoded by the gene in the form of text, and by pressing the mouse button, turn off (or turn back) the selected gene. <br><br>  In this article I will not dwell on the description of the ‚Äúlaboratory setup‚Äù device, although by itself its ‚Äúassembly‚Äù was a very interesting occupation - besides graph theory and discrete mathematics, knowledge and experience in the numerical solution of equations of mathematical physics, geometry, etc. were useful. besides, its genetic ‚Äúsource‚Äù code itself is a result of evolution and contains the heritage of distant ancestors - after all, the first version and design solutions were implemented on Delphi 7.0.  You could even say this is my first ‚ÄúHellow World!‚Äù On the Silverlight | WPF platform.  I will only note the use of the QuickGraph libraries (graphs and algorithms) and AForge (the genetic algorithm). <br><br><h5>  The simplest examples </h5><br><br>  So, let's look at some of the simplest examples of ‚Äúlive graphs‚Äù that demonstrate living properties.  Some of the specimens (for example, the aforementioned "finger dumbbell") were designed only as a divine beginning for the world of living graphs (i.e., Me), others are the result of evolution, and still others are genetically modified products. <br><br><h6>  "Hairy Circle" </h6><br><br><img src="https://habrastorage.org/getpro/habr/post_images/6d5/de2/6a9/6d5de26a9db3d210ba684872fda26a4d.png" alt="Earl 'Hairy Circle'"><br><br>  This figure is a genetically modified product!  First, using the genetic algorithm, an infinitely growing inside circle was obtained (the fitness function depended only on the topological characteristics of the graph ‚Äî the obligatory presence of two faces).  Then another ‚Äúgene‚Äù was added - the ‚Äúhair gene‚Äù (rightmost on the gene map) <br><br>  By launching an ‚Äúorganism‚Äù growing from within, one can observe how individual segments of a circle appear as a result of cyclic repetition of insertion operations of new edges. <br><br><h6>  "Finger circle" (hybrid) </h6><br><br>  If we take the ‚Äúdumbbell‚Äù and ‚Äúhairy circle‚Äù genes and combine them in one chromosome of the ‚Äúembryo‚Äù, then a hybrid will arise from it, inheriting both the properties of the ‚Äúdumbbell‚Äù and the ‚Äúround‚Äù properties: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/793/c6d/d9d/793c6dd9dbd961169d6e7ecb212fd8cb.png" alt="Hybrid of 'toes and dumbbells' and 'hairy circumference'"><br><br>  <i>Figure: Hybrid "finger dumbbells" and "hairy circumference" has common properties</i> <br><br>  Please note that some of the ‚Äúdumbbell‚Äù dummy genes remain inactive during the entire process of graph deployment (on the gene map, gene activity is indicated by a green light). <br><br><h6>  "Bushes" (Fractals) </h6><br><br>  The simplest genetic code of two "genes" creates an arbitrarily large graph. "  If you cut off any of its branches, the "bush" almost instantly recover in the same size and shape. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ef2/773/ffb/ef2773ffb554a35d97d9caa72a8df1ac.png" alt="Primitive fractal"><br><br>  <i>Figure Primitive fractal is given only a pair of genes</i> <br><br><h6>  "Kokoshnik (rectangular mesh)" </h6><br><br>  This and the next two strange figures are the result of my attempts to construct a square grid in the form of a square. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/341/771/9a9/3417719a932a785c87906684fa8cf3b3.png" alt="'Kokoshnik' - unrolling a rectangular grid"><br><br>  <i>Figure "Kokoshnik" - expanding a rectangular grid</i> <br><br>  By changing a pair of genes, one can grow an arbitrarily large (and small) mesh.  A knife can make holes.  Try to turn off the gene ‚Ññ13 "O (O): + L" and turn it on after deployment. <br><br><h6>  "Strange Figure 1" and "Strange Figure 2" </h6><br><br>  <b>‚ÄúStrange Figure 1‚Äù</b> is my mistake.  In fact, the goal was to create an equilateral square grid.  The figure is depicted at the very first figure in the article (Figure 1).  If, however, cutting its processes directly on a live graph, then, having released new processes instead of cutting, the figure will take a more extended form: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/616/eac/463/616eac463667e384626bc81c14fb9718.png" alt="Strange Figure 1"><br><br>  <i>Figure "Strange figure" after circumcision</i> <br><br>  <b>Strange figure 2</b> .  The chromosome of this figure differs from the chromosome of ‚ÄúStrange figure 2‚Äù by just one gene, more precisely, even a part of the gene (close relatives), but the form of the ‚Äúdecoding‚Äù result is a little like the form of the previous instance. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ad1/18e/6c8/ad118e6c869e7b63fe63ebf7b060378e.png" alt="Strange Figure 2"><br><br>  <i>"Strange figure 2" - a close relative of "Strange figure 1" - differs in only one gene</i> <br><br><h6>  "Hexagon with spikes" (the result of evolution) </h6><br><br>  It took thousands of iterations of the genetic algorithm to get 10 genes of this figure.  The fitness function in Russian sounded something like this: ‚Äúthe closer the number of faces to two, and the number of nodes of degree 1 to six and the number of nodes of degree 3 to six and the number of genes to zero - the more chances a graph will survive‚Äù (the degree of a node is the number connections) <br><br>  The evolution of this species has experienced several jumps: a triangle, a square with processes, a six square with irregular processes, and, finally, a regular one.  Slowing the process of unfolding, you can see what solution evolution has found to build a shape. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/049/223/5d1/0492235d1f280ff3bb1313fe00a84711.png" alt="Hexagon with leaves"><br><br>  <i>Hexagon drawing with "leaves" - The result of the genetic algorithm</i> <br><br><h5>  Conclusion </h5><br><br>  So, we have seen how using such abstract objects as finite automata and graphs, we can create systems that are amusingly similar to multicellular living organisms, which demonstrates the common principles underlying the "living graphs" and in the laws of formation of geometric forms of the living. <br><br>  However, this does not mean that we have revealed the secrets of life.  It simply means that you can move on: explore the patterns of the evolution of "live graphs".  Let me remind you that in the demonstration application we observed ‚Äúontogenesis‚Äù and the result of evolution, and not evolution itself. <br><br>  Issues of further research can be both the determination of the boundaries of the ‚Äúlive graphs‚Äù model, and the search for quantitative and qualitative patterns in an unencumbered evolutionary model with excessive details (dependence of gene lengths on the complexity of forms, statistics of genetic information on the position of a species in the evolution tree, the rate of evolution on the set of valid operations, etc.). <br><br>  According to the results of these studies, determining the most effective evolutionary algorithms, you can try to apply "live graphs" in the source area: in artificial neural networks or in the study of the evolution of robots, "animats", artificial control systems.<font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Of particular interest is the combination of dynamic nonlinear neural networks with live graphs, in which the state of excitement of the neuron is one of the states of the finite automaton "inside" the live graph. It will be possible to observe something like a ‚Äúgenetic memory.‚Äù </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Also, it would be curious to create a material embodiment of living graphs in the form of a self-organizing colony of primitive robots, the firmware of which sews the genetic code of the community. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">All these areas are difficult to reach for a solitary researcher, and I hope some of the readers will still make their own small and big discoveries in this field of research. Quite possibly, models like Live Counts will be the starting point for your own research. True, this will require other "experimental installations".</font></font><br><br><h5>  Literature </h5><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">[1] ‚ÄúReview of Neural Network Evolution Methods‚Äù </font></font><a href="http://habrahabr.ru/blogs/artificial_intelligence/84015/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">http://habrahabr.ru/blogs/artificial_intelligence/84015/</font></font></a> <br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> [2] ‚ÄúEvolutionary Computation for Modeling and Optimization‚Äù Daniel Ashlock, January 12, 2004 </font></font><a href="http://orion.math.iastate.edu/danwell/ma378/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">http: //orion.math. iastate.edu/danwell/ma378/ </font></font></a> <br><br> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">UPDATE</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Added the ability to load the ‚Äúgenome‚Äù from xml. </font><font style="vertical-align: inherit;">Example file: </font></font><a href=""><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">the ‚Äústrange figure 1‚Äù genome</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font><font style="vertical-align: inherit;">Now you can try to grow your graphs, check your ideas ... finally find mistakes. </font></font><br> <b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">UPDATE 2</font></font></b><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> Laid out the </font></font><a href=""><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">source (1.8Mb)</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> with release build.</font></font></div><p>Source: <a href="https://habr.com/ru/post/107387/">https://habr.com/ru/post/107387/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../107378/index.html">Comparison of processor speeds for dedicated servers</a></li>
<li><a href="../107379/index.html">Now there will be ... iWear</a></li>
<li><a href="../107380/index.html">How to start work on a startup?</a></li>
<li><a href="../107383/index.html">Video from HighLoad ++ in Mail.Ru: what are you doing?</a></li>
<li><a href="../107386/index.html">The helicopter flew 12 hours on remote charging</a></li>
<li><a href="../107388/index.html">Taiwan scientists have created a promising flexible display</a></li>
<li><a href="../107389/index.html">The difference in the approaches of code analysis by the compiler and the selected tool</a></li>
<li><a href="../107390/index.html">Democamp 2010 Report</a></li>
<li><a href="../107393/index.html">1-bit symphony of Tristan Peric</a></li>
<li><a href="../107394/index.html">Social networks in business. Employee Conduct Policies</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>