<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Genomes of sequenced organisms - errors in the bases</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="The most well-known base containing the genomes of sequenced organisms - NCBI , contains a large number of systematic errors. Because of this, it is p...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Genomes of sequenced organisms - errors in the bases</h1><div class="post__text post__text-html js-mediator-article">  The most well-known base containing the genomes of sequenced organisms - <a href="http://ru.wikipedia.org/wiki/%25D0%259D%25D0%25B0%25D1%2586%25D0%25B8%25D0%25BE%25D0%25BD%25D0%25B0%25D0%25BB%25D1%258C%25D0%25BD%25D1%258B%25D0%25B9_%25D1%2586%25D0%25B5%25D0%25BD%25D1%2582%25D1%2580_%25D0%25B1%25D0%25B8%25D0%25BE%25D1%2582%25D0%25B5%25D1%2585%25D0%25BD%25D0%25BE%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B8%25D1%2587%25D0%25B5%25D1%2581%25D0%25BA%25D0%25BE%25D0%25B9_%25D0%25B8%25D0%25BD%25D1%2584%25D0%25BE%25D1%2580%25D0%25BC%25D0%25B0%25D1%2586%25D0%25B8%25D0%25B8">NCBI</a> , contains a large number of systematic errors.  Because of this, it is practically impossible to use these data, and all the more it is impossible to study the mechanism of mutations (and, consequently, evolution), since in this case human errors during sequencing are investigated, and not natural mutations.  Therefore, before using this data it is necessary to clarify this database. <br><br>  And this is a time-consuming task; it cannot be solved for a separate, necessary organism.  Therefore, I would like to find those who would like to create their own Russian-language source similar to the NCBI, but with updated information. <br><br>  The article shows how massive the errors of the genomes found in NCBI are and how to make sure of this, and some ways to correct it. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <a name="habracut"></a><br><br>  <b>Where are the genomes</b> <br><br>  All the sequenced genomes are located on ftp ftp.ncbi.nih.gov/genomes/.  And then ftp.ncbi.nih.gov/genomes/Bacteria/ bacterial genomes - with them, it is worth starting. <br><br>  We need the file all.fna.tar.gz - it contains genomes of about 2000 bacteria.  What is a genome?  This is a chain of DNA - letters A, T, C, G. Download, unpack - we get a bunch of directories with species names in Latin.  Inside there are usually several NC files _ ###### - in each file there is a separate so-called.  locus - DNA strand (chromosome or plasmid). <br><br>  For simplicity, we will deal with RNA, since proteins are a bit more complicated to process.  For this we need two more files: <br>  1. all.rnt.tar.gz - contains a list and location (beginning, end, direction) of all RNAs at a specific locus <br>  2. all.frn.tar.gz - contains a list of actually cut pieces of RNA from DNA (don‚Äôt be surprised even if it is RNA - there will not be replacements of T with U here, because this is the DNA code from which RNA will be created) <br><br>  <b>How to perform preprocessing</b> <br><br>  These files are not very convenient for processing.  The .fna files contain a comment in the first line, and then the DNA code in each line of 70 characters, then a line break.  Naturally, this is not where it is suitable for search, and you need to glue it into one line without hyphenation, and delete the comment.  The file processed in this way will be given the extension .fna.txt <br><br>  In addition, there is another nuance: RNA can be transcribed from right to left or vice versa, and since DNA is a double helix, then right to left means transcription from one strand of DNA, and from left to right with another complementary . <br><br>  This means that in order to find, for example, RNA for which a negative direction is indicated in the file all.rnt.tar.gz - we need to search not in the received .fna.txt file - so we will not find anything.  We need to create a reverse file (give it the extension .fna_.txt).  The point is that we take the .fna.txt file - we read it letter by letter from the end and make complementary replacements: <br><br>  T = A;  G = C;  A = T;  C = G;  M = K;  R = Y;  W = W;  S = S;  Y = R;  K = M;  V = B;  H = D;  D = H;  B = V <br><br>  The first 4 are clear and known.  The rest are rather unexpected :), although they are rare, but they are.  The point is that if during sequencing it is impossible to distinguish G from A, R is put, etc. <br><br>  <b>Find errors or cross-analysis</b> <br><br>  I gave this process the name cross-analysis.  The essence of such a file of interest .frn take the code of one RNA.  And we are looking for a match across the entire set of .fna.txt and .fna_.txt files. <br><br>  How many matches do you think you will get?  Enough.  And it may turn out that in the .rnt file there may not be a corresponding entry.  And most often it turns out that the beginning and end in the file will be not so, but shifted by 1 or 3 positions.  There may even be another direction.  I also met with more significant errors when it is indicated that it is Ile RNA, and in fact it is Met RNA. <br><br>  By the number of these errors almost 50%.  How can I work with such erroneous data I do not know.  Why people from NCBI do not conduct such a simple cross-analysis - I don‚Äôt know either. <br><br>  But think how many erroneous conclusions then do biologists trust in this data? <br><br>  At the same time, when errors are corrected - the same method allows you to make an experiment of the type described here. <a href="http://habrahabr.ru/post/146331/">Interesting results about the evolutionary systematics of prokaryotes or ‚Äúmany-species origin‚Äù</a> , we note separately how much it is an elementary method, but giving exact facts that can speak about a lot. <br><br>  <b>One simplest example</b> <br><br>  Open the sequenced organism <a href="http://www.ncbi.nlm.nih.gov/nuccore/NC_005043">Chlamydophila pneumoniae TW-183</a> .  Searching for the ‚ÄúCpBt08‚Äù tag, the complement is indicated there (266485..266557) - this is the beginning and the end, respectively.  There is also a link to <a href="http://www.ncbi.nlm.nih.gov/sites/entrez%3Fdb%3Dgene%26cmd%3DRetrieve%26dopt%3Dfull_report%26list_uids%3D3284349">GeneID: 3284349</a> .  Then there is the link <a href="">FASTA</a> - there is a sequence <br><br>  CGGGGACTTAGCTTAGTTGGTAGAGCGTCTGATTTGCATTCAGAAGGTCAGGAGTTCGAATCTCCTAGTCTCC <br><br>  she's not right - really should be <br><br>  GGGGACTTAGCTTAGTTGGTAGAGCGTCTGATTTGCATTCAGAAGGTCAGGAGTTCGAATCTCCTAGTCTCCA <br><br>  (and it really is in the full DNA sequence, it is only incorrectly identified) <br><br>  put them next <br><br><pre> CGGGGACTTAGCTTAGTTGGTAGAGCGTCTGATTTGCATTCAGAAGGTCAGGAGTTCGAATCTCCTAGTCTCC
   GGGGACTTAGCTTAGTTGGTAGAGCGTCTGATTTGCATTCAGAAGGTCAGGAGTTCGAATCTCCTAGTCTCCA
</pre><br><br>  see that the difference is in the shift. <br><br>  Why? <br><br>  Now go to another organism <a href="http://www.ncbi.nlm.nih.gov/nuccore/NC_000922">Chlamydophila pneumoniae CWL029</a> , look for the tag: CPnt08.  And similarly <a href="">we find the gene</a> <br><br>  GGGGACTTAGCTTAGTTGGTAGAGCGTCTGATTTGCATTCAGAAGGTCAGGAGTTCGAATCTCCTAGTCTCCA <br><br>  Do you think this is just another sequence?  An, no - this is the same but shifted.  Of course there is, the question is which one is correct.  And the worst thing here is to solve this automatically.  You have to decide based on the error rate and some knowledge of what RNA sequences can be.  However, for tRNA there is a more specific criterion to check the compliance of the anticodon in positions 34-36 and the presence of the end of SSA (which all tRNA should have). <br><br>  PS Who will be interested in it so much to try to look for mistakes, correct or even make an experiment similar to mine, but for other data - please send a personal message. </div><p>Source: <a href="https://habr.com/ru/post/146814/">https://habr.com/ru/post/146814/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../146805/index.html">The digest of interesting news and materials from the world of ayti for the last week ‚Ññ12 (June 23 - 29, 2012)</a></li>
<li><a href="../146807/index.html">Australian provider Telstra scared users</a></li>
<li><a href="../146808/index.html">YouTube users can now change their nicknames to Google+ profiles</a></li>
<li><a href="../146811/index.html">Foundation 3</a></li>
<li><a href="../146812/index.html">Selling photos from Instagram</a></li>
<li><a href="../146816/index.html">A small study on the mechanics of self-organization of online communities</a></li>
<li><a href="../146818/index.html">Apple bans US Galaxy Nexus sales</a></li>
<li><a href="../146819/index.html">Plugin that converts Photoshop styles to css code</a></li>
<li><a href="../146820/index.html">ksoap2-android and arrays in the query</a></li>
<li><a href="../146821/index.html">Publish module on cpan</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>