<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Not for Selfies: Digital ELISA with a new chip embedded in a smartphone</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="From which office in any hospital do the cries of children, and sometimes adults, come from time to time? What did the parents, brazenly deceiving us ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Not for Selfies: Digital ELISA with a new chip embedded in a smartphone</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/webt/lp/pw/mn/lppwmn0c0a6_lniksk5bdmj8sns.jpeg"><br><br>  From which office in any hospital do the cries of children, and sometimes adults, come from time to time?  What did the parents, brazenly deceiving us in childhood, compare with the mosquito bite?  I think you already guessed that this is a blood test.  Now this procedure has become faster and less painful.  One thing has not changed - its importance.  Diagnostics in medicine plays the most important role in the first stages of the struggle with any disease.  After all, to defeat the disease, it must first be detected.  Having survived the procedure of blood sampling for analysis, you calmly go to wait for its results.  At this time in the laboratories, people with the help of complex, bulky and very expensive devices analyze your blood, finding out what it is and in what quantity.  And it's good when such a laboratory is in your local hospital, but this is not always the case.  What if there was a pocket lab, small and inexpensive, but analyzing samples with the same accuracy and efficiency as the usual one?  Sounds like science fiction, right?  The phrase ‚Äúpocket analyzer of blood sugar‚Äù sounded similarly futuristic at the time.  Today we will get acquainted with the research and implementation of the technology of quantitative analysis of proteins and amino acids through a compact device of a new type.  What does this miracle consist of, how does it work and how effective is it?  These and other questions will be answered in the scientists' report.  Go. <a name="habracut"></a><br><br><h3>  The basis of the study </h3><br>  We live in an era of digital technologies that are successfully implemented in various areas of our life.  Laboratory tests (diagnostic) are no exception.  Scientists note that digital drip analysis is 1000 times more accurate than traditional, and allows you to spend millions of analyzes in parallel within one drop of a sample with a volume measured in femtoliter (fl, 1 fl = 10 <sup>-15</sup> l). 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      The use of digital analysis is extremely useful for detecting nucleic acids and proteins, analyzing individual cells and even actosomes. <blockquote>  <b>Exosomes *</b> - extracellular vesicles (diameter: 30-100 nm), which are secreted by cells into the extracellular space.  Exosomes are involved in the work of immunity, secretion of proteins, etc. </blockquote>  Currently, the most well-known digital assay methods are dELISA (digital enzyme-linked immunosorbent assay / cIF) and qPCR (digital polymerase chain reaction).  These techniques allow you to work with individual cells, while obtaining very accurate results that do not require correction.  So recently, using these methods, successful quantitative analysis of protein and mRNA in a single cell was carried out simultaneously. <br><br>  Such techniques and the demonstration of their talents once again show that the implementation of parallel analysis within an extremely small environment (sample) is quite possible.  However, like any other technology, these methods also have disadvantages.  They are quite trivial - the size, price and complexity of manufacturing.  Researchers remind us that one installation for QuIFTER (Quanterix's Simoa) costs about 100,000 dollars.  And not every private clinic can afford such a sum, I‚Äôm not talking about public ones. <br><br>  Of course, this "beast" of Quanterix's Simoa is very powerful if you exaggerate.  It uses microcell plates with 200,000 cells of 40 fl each. <br><br><img src="https://habrastorage.org/webt/t1/1e/9z/t11e9zsxhoulvukjofepetcxzxu.jpeg"><br>  <i>Quanterix's Simoa</i> <br><br>  At the same time, this device can process in parallel up to 4 ELISA plates of 96 cells each.  Thus, one device is capable of producing results of 66 samples within an hour, each of which can be subjected to 10-plex analysis (i.e. 1 sample is analyzed for 10 indicators simultaneously).  The numbers are truly incredible.  But again, the question arises of the price and dimensions of such a wonderful machine. <br><br><img src="https://habrastorage.org/webt/ij/kd/vb/ijkdvbahwndkhzsos3fjdhbnooa.jpeg"><br>  <i>Image number 1</i> <br><br>  And here scientists propose to turn their eyes towards microfluidic drip systems.  Classic systems of this kind cannot boast of Quanterix's Simoa colossal indicators, but they can serve as the basis for a new device.  The method of microfluidic drops with a continuous flow in theory can analyze up to 1 million cells.  However, in practice, such indicators have not yet been achieved for several reasons.  First, the throughput (less than 104 drops per second), when the drops are generated sequentially (not in parallel) and are monodisperse.  Secondly, the detection of the fluorescence of each drop is carried out by passing them one through the laser spot.  In other words, all in the queue, one by one.  This process is shown in image <b>1a</b> (separation, incubation and determination; 3 hours for analysis of 10 <sup>7</sup> drops). <br><br>  The main problems of converting this method to a compact format are the complexity of parallelizing optics for multi-color fluorescence detection, the complexity of integrating the sample preparation process, and the need for certain tools to generate strictly controlled droplet flows.  However, scientists are not accustomed to give up in the face of difficulties, no matter how impressive they are. <br><br>  Micro droplet Megascale Detector (MD, macro detector) is a creation of our today's heroes.  This device can not only be implemented in any mobile (handheld) device, but also meets the standards for the quantitative analysis of conventional full-size laboratories.  The process is shown in image <b>1b</b> (separation, incubation and determination; 10 minutes per analysis of 10 <sup>7</sup> drops). <br><br>  To achieve this, according to the researchers, three main tasks were implemented: <br><br><ol><li>  Instead of generating 1 drop each, parallel generation of microfluidic drops was used, working 100 times faster.  And the achievements of fellow scientists in the production of monodisperse drops (a <a href="https://pubs.rsc.org/en/Content/ArticleLanding/2016/LC/C6LC01075J">link to this study</a> ) allowed us to get rid of the dependence of the monodispersity of the drops on the flow velocity.  This allows the use of very inexpensive peristaltic pumps that can be embedded in a mobile (pocket) device. </li><li>  Rapid reading of fluorescence drops at a speed of more than 105 per second (remember the limit of 104, which I mentioned above) was achieved thanks to the visualization on the base of a mobile phone, which is 100 times faster than the usual reading (when the drops are read in turn).  In this case, there is no need for expensive optics, and implementation in handheld mobile devices is obvious.  The main distinguishing feature of this innovation is the ability to overcome the limitations of the low frame rate of a digital image and provide multi-color fluorescence detection by modulating several sources of excitation of LEDs or laser diodes of different colors with unique non-periodic signals.  Videos can be decoded to capture droplet fluorescence data, overcoming camera frame rate limitations.  Thus it is possible to achieve the very (about which we spoke earlier) 1 million drops per second. </li><li>  And finally, the integration of the processing unit microgranules (or microbasin, microscopic spherical objects), a generator of droplets, signal delay lines for incubation of droplets and a fluorescence detector.  Collectively, this provides an inexpensive, compact and efficient device for inputting untreated serum (sample) and outputting molecular data (result). </li></ol><br>  As a demonstration of their invention, scientists have implemented a multiplexed SIF using microgranules of different colors, obtained using fluorescent dyes.  Each color is a color ‚Äúcode‚Äù of the protein that the microgranule antibody ( <b>1c</b> ) is aimed at. <br><br>  A multiplex analysis of serum GM-CSF and IL6 cytokines was performed using ultraviolet and green fluorescent granules when the droplets contained microgranules with a fluorescent red immunocomplex.  Bovine serum was used as the medium for quantitative analysis, and the detection limit was 0.004 pg / ml (picogram per milliliter, 1 pg = 10 <sup>‚àí12</sup> g).  It is 1000 times more accurate than the standard ELISA, and corresponds to the level of accuracy of the digital ELISA. <br><br>  It takes only 10 minutes to process 10 million drops.  The process itself includes the generation and incubation of droplets, as well as the detection of fluorescent droplets for each sample. <br><br><h3>  Device structure and analysis process </h3><br><img src="https://habrastorage.org/webt/wz/xj/a_/wzxja_za2unuiyvs0jlgts3eotq.jpeg"><br>  <i>Image # 2: MD device structure.</i> <br><br>  A little more about the image above: <b>2a</b> - chip layout, top and bottom view;  <b>2b</b> - photo of an MD chip on which all opto-fluid channels are visible;  <b>2c</b> is a micrograph of the process of encapsulating microgranules in droplets with a diameter of 40 Œºm;  <b>2d</b> is a fluorescence micrograph of droplets after the delay line;  <b>2e</b> - a schematic representation of the MD platform (mobile phone, 3 light sources and the MD chip itself). <br><br>  The main components of MD can be called a microgranule processor, where the latter capture target proteins from serum.  After that, the granules are labeled with immunocomplexes for subsequent amplification inside the droplets.  Between each such process, an iterative (several times) cleaning takes place.  There is also a droplet generator, where the microgranules mix with the enzyme substrate and encapsulate in water-oil droplets. <br><br>  Next is the microfluidic channel through which the drops pass within 3.2 minutes.  This channel is necessary as a delay / deceleration of the process, which allows the enzyme to enhance the fluorescent signal.  The final part is a detector (or scanner) based on a mobile phone (camera), where the fluorescence of the droplets is detected. <br><br>  The microgranule processor consists of a semi-permeable membrane for immobilization of granules.  Several reagents and wash buffers are delivered to the immobilized granules.  After that, the granules are released for further analysis. <br><br>  The membrane itself is made of polycarbonate.  On the membrane etched track area of ‚Äã‚Äã300 mm <sup>2</sup> with pores with a diameter of 3 microns. <br><br>  In this experiment, there were two groups of microgranules: (d = 5.4 Œºm, ex / em = 470/490 nm, CFH-5052-2), functionalized with GM-CSF antibody (MAB2172) and (d = 4.5 ¬µm, ex / em = 370/410 nm, CFP-4041-2), functionalized with anti-IL6 antibody (MAB206). <br><br>  First of all, the microgranules undergo the process of incubation with the sample for 1 hour, and only after that they are captured on the membrane, which was mentioned above. <br><br>  At this stage (within the membrane), the granules are washed with 1 ml of T20 buffer at a flow rate of 10 ml / h, incubated with 0.1 ml of 0.7 nM detecting antibody in T20 buffer for 0.5 hour, and again washed with 1 ml of T20 buffer at 10 ml / h, and after that it is released from the membrane by changing the flow rate to 6 ml / h. <br><br>  After that, the released microgranules are mixed with the ELISA substrate and encapsulated in droplets with a diameter of 40 Œºm.  To ensure accurate mixing of the granules and the substrate and to minimize the background signal from the enzymes that generate the fluorescent signal, a special channel 14 mm long is used. <br><br>  The droplet generator is made so that the droplet diameter does not depend on the flow rate.  In this device there are only 100 such generators, which at the output give a capacity of 100,000 drops per second. <br><br>  Each drop is encapsulated with either 1 pellet, or remains intact.  At the same time, a certain concentration is achieved - 10 drops more than microgranules (for example, 20 drops - 10 with granules and 10 without).  This reduces the likelihood that in one drop there will be two granules up to 0.5%. <br><br>  Following the droplet generators, there is a delay line similar to a spiral with a channel width of 1.8 mm and a height of 1.5 mm.  The delay line should be long enough, but it is impossible to increase the device in size.  Therefore, 4 spirals were made one by one, which would take 3.2 minutes to fully pass at a flow rate of 67 ml / h. <br><br>  In order to integrate such a device into a mobile platform, it was necessary to solve some tasks related to the camera phone.  The use of the usual constant-time light excitation leads to the fact that droplets moving in the field of vision of the camera are visualized as stripes.  The length of this band sets the minimum distance between the drops and, thus, seriously limits the throughput. <br><br>  If we use light excitation with a pseudo-random sequence (in time), this will allow us to ‚Äúsee‚Äù individual drops.  Light modulation speed is 10 times faster than camera exposure time.  Due to this difference, droplets form bands, the distance between them (three droplet diameters) is sufficient for their individual determination.  You can skip 120 parallel drip channels in front of the camera. <br><br>  Another important point in detection and scanning is fluorescence.  In order to conduct multiplex ELISA, you need several different fluorescent signals. To do this, 3 light sources were used at once, each of which has a wavelength necessary to excite a specific fluorescent dye.  This ternary system consists of two diode lasers (blue, green) and one LED (UV). <br><br><img src="https://habrastorage.org/webt/fr/nz/wh/frnzwhdztskralgnjjfca5pkuci.jpeg"><br>  <i>Image number 3: the process of "sample-result" (decoding data from the camera phone).</i> <br><br>  To accurately decode the video from the phone‚Äôs camera, it was necessary to conduct correlation detection for the three expected modulation patterns ( <i>m</i> ), which correspond to each of the three light sources. <br><br>  The result is a correlation vector ( <b>3a</b> ), where: <i>k</i> - frames;  <i>n</i> = 1: 120 channels in the device;  <i>R</i> , <i>G</i> , <i>B</i> - color channels of the digital camera;  <i>r</i> , <i>g</i> , <i>b</i> - color excitation. <br><br>  The droplet pattern was created by a sequence of maximum length (MLS) with | m |  = 63 bits.  In addition, each bit is 10 pixels in a digital image, that is, a total of 63 bits is 630 pixels (1/3 of the frame width in 1920). <br><br>  Fluorescence scanning is necessary in order to determine whether a drop contains a microgranule, and if so, determine the color (UV or green ‚Äî protein, red ‚Äî target molecule).  After receiving these data, they must be retrieved.  To do this, the video frame is divided into red, green and blue components in accordance with the camera sensors ( <b>3d</b> ). <br><br>  This device uses cloud technology.  This was done to reduce the load on the iron (that is, on the phone itself).  Instead of controlling the speed of the drop or phase, cloud computing was performed to determine drops with an unknown phase or speed ( <b>3c</b> ).  After determining the optimal phases and velocity of the drop, we can accurately determine the peaks in the correlation space Œ® <sup>r, g, b</sup> <sub>k, n</sub> (x, œÖ <sub>c</sub> , Œ∏ <sub>c</sub> ) ( <b>3f</b> and <b>3g</b> ). <br><br>  The collected data is uploaded to a special application (currently only on Android OS), which sends them to the cloud for processing using MATLAB on a remote server.  After that, the already processed data is returned to the smartphone and displayed on the screen. <br><br>  After conducting all the preparatory and testing work, the scientists decided to conduct a ‚Äúsparring‚Äù with the participation of their creation and the already existing commercial full-sized Simoa device. <br><br>  In the test duel, three working medium variants were used: PBS ‚Äî sodium phosphate buffer, FBS ‚Äî fetal bovine serum and human serum.  The most important indicator was the detection limit (LOD), that is, the minimum content of the analyte in the sample. <br><br><img src="https://habrastorage.org/webt/lc/gu/hn/lcguhnuzormjrmayi8ydpxps6n0.jpeg"><br>  MD chip test results. <br><br>  Several single-plex measurements of M-CSF (image <b>A</b> above) and IL6 (image <b>B</b> ) were carried out in PBS medium by measuring serial dilutions from 104 to 102 pg / ml.  In this test, very good detection limits were obtained: LOD = 0.0045 pg / ml for GM-CSF and LOD = 0.0070 pg / ml for IL6. <br><br>  Similar measurements were also carried out in FBS (1: 4) solution.  At this stage, the sample for analysis was divided in half between the device under investigation and the commercial heavyweight Simoa.  As a result, the work of scientists showed excellent results, which were practically not inferior to Simoa's indicators (R2 = 0.95, image <b>C</b> above). <br><br>  But it was a one-plex analysis, that is, an analysis of one indicator.  Now it was necessary to check how the MD chip will cope with the parallel analysis of several proteins, that is, with the duplex analysis of GM-CSF and IL6 simultaneously.  To begin with, a certain amount of GM-CSF was added to FBS, and the IL6 concentration was zero (images <b>F</b> and <b>G</b> ).  Then it was done the other way around: zero concentration of GM-CSF and some IL6. <br><br>  In both cases, the detection limit values ‚Äã‚Äãdid not differ much from the results of a single-sided analysis conducted earlier (p&gt; 0.88 for GM-CSF and p&gt; 0.90 for IL6). <br><br>  After that, a certain amount of both GM-CSF and IL6 (image h) was added to the sample.  The detection accuracy was excellent - R2&gt; 0:99 for GM-CSF and R2&gt; 0:99 for IL6. <br><br>  The most significant test was the analysis of human serum.  Blood samples were taken from 14 subjects.  The researchers conducted a quantitative analysis of the GM-CSF and IL6 of these samples using an MD and Simoa chip. <br><br><img src="https://habrastorage.org/webt/w8/yu/bv/w8yubvgsql0hc6khbsjxig7ontw.jpeg"><br>  <i>The results of the quantitative analysis of GM-CSF and IL6 human serum using MD and Simoa.</i> <br><br>  The analysis results by means of the MD chip turned out to be very close to the results of Simoa (R2 = 0: 96), which is currently the most accurate analyzer. <br><br><iframe width="560" height="315" src="https://www.youtube.com/embed/AFBVKukh1Ak" frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope; picture-in-picture" allowfullscreen=""></iframe><br>  <i>Demonstration of the device.</i> <br><br>  For more detailed acquaintance with the nuances and details of the study I recommend to look into the <a href="https://www.pnas.org/content/pnas/early/2019/02/13/1814110116.full.pdf">report of the research group</a> and <a href="https://www.pnas.org/content/pnas/suppl/2019/02/14/1814110116.DCSupplemental/pnas.1814110116.sapp.pdf">additional materials</a> to it. <br><br><h3>  Epilogue </h3><br>  Speed ‚Äã‚Äãplays a huge role in medicine.  The faster the accurate diagnosis is carried out, the faster the treatment can begin.  Sometimes it is not even about days, but minutes that cannot be wasted.  However, time is not always the main factor when you simply do not have the tools to diagnose.  Dimensions, complexity of production and the price of some accurate devices for analysis, as well as the equipment of laboratories with all the necessary is not available everywhere and not all. <br><br>  Creating devices like the MD chip is not just a good idea, it is a brilliant and incredibly important idea.  Inexpensive analyzer, which at the same time demonstrates excellent accuracy and speed of work can greatly affect medicine throughout the world, particularly in those regions where the use of standard tools is impossible.  According to the scientists themselves, the cost of a prototype of such a device is about $ 500.  With mass production, the price for the consumer will be only 5. <br><br>  Everyone has the right to treatment, but for many reasons this right is not always and not everywhere comparable with reality.  Similar studies and similar devices will help change this. <br><br>  Thank you for your attention, remain curious and have a good working week, guys. <br><br>  Thank you for staying with us.  Do you like our articles?  Want to see more interesting materials?  Support us by placing an order or recommending to friends, <b>30% discount for Habr users on a unique analogue of the entry-level servers that we invented for you:</b> <a href="https://habr.com/company/ua-hosting/blog/347386/">The whole truth about VPS (KVM) E5-2650 v4 (6 Cores) 10GB DDR4 240GB SSD 1Gbps from $ 20 or how to share the server?</a>  (Options are available with RAID1 and RAID10, up to 24 cores and up to 40GB DDR4). <br><br>  <b>VPS (KVM) E5-2650 v4 (6 Cores) 10GB DDR4 240GB SSD 1Gbps before summer for free</b> if you pay for a period of six months, you can order <a href="https://ua-hosting.company/vpsnl">here</a> . <br><br>  <b>Dell R730xd 2 times cheaper?</b>  Only we have <b><a href="https://ua-hosting.company/serversnl">2 x Intel Dodeca-Core Xeon E5-2650v4 128GB DDR4 6x480GB SSD 1Gbps 100 TV from $ 249</a> in the Netherlands and the USA!</b>  Read about <a href="https://habr.com/company/ua-hosting/blog/329618/">How to build an infrastructure building.</a>  <a href="https://habr.com/company/ua-hosting/blog/329618/">class c using servers Dell R730xd E5-2650 v4 worth 9000 euros for a penny?</a> </div><p>Source: <a href="https://habr.com/ru/post/442618/">https://habr.com/ru/post/442618/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../442608/index.html">The QuadrigaCX cryptobirds cold wallets, the founder of which died, were empty</a></li>
<li><a href="../442610/index.html">Telegram-bot + Google Analytics</a></li>
<li><a href="../442612/index.html">Cardboard engine for an electrical board game. As we brought it closer to reality</a></li>
<li><a href="../442614/index.html">CI / CD using Jenkins at Kubernetes</a></li>
<li><a href="../442616/index.html">Overclock event handling up to 1.6 million per second</a></li>
<li><a href="../442620/index.html">Machine learning in IT monitoring</a></li>
<li><a href="../442622/index.html">How to make Korutin in Unity a little more convenient</a></li>
<li><a href="../442624/index.html">The book "The perfect algorithm. Basics</a></li>
<li><a href="../442626/index.html">Habrarating: building a cloud of Russian-language words on the example of Habr's headlines</a></li>
<li><a href="../442628/index.html">Methods for effective transformation of information based on the modified Bailey-Borwein-Plaff formula</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>