<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Distant Joining: choose a representative set of genes</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Modern methods of bioinformatics allow fairly accurate restoration of evolutionary histories based on sequences of genes or proteins of living organis...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Distant Joining: choose a representative set of genes</h1><div class="post__text post__text-html js-mediator-article">  Modern methods of bioinformatics allow fairly accurate restoration of evolutionary histories based on sequences of genes or proteins of living organisms.  And thanks to the <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B5%25D1%2582%25D0%25BE%25D0%25B4%25D1%258B_%25D1%2581%25D0%25B5%25D0%25BA%25D0%25B2%25D0%25B5%25D0%25BD%25D0%25B8%25D1%2580%25D0%25BE%25D0%25B2%25D0%25B0%25D0%25BD%25D0%25B8%25D1%258F_%25D0%25BD%25D0%25BE%25D0%25B2%25D0%25BE%25D0%25B3%25D0%25BE_%25D0%25BF%25D0%25BE%25D0%25BA%25D0%25BE%25D0%25BB%25D0%25B5%25D0%25BD%25D0%25B8%25D1%258F">sequencing</a> technologies <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B5%25D1%2582%25D0%25BE%25D0%25B4%25D1%258B_%25D1%2581%25D0%25B5%25D0%25BA%25D0%25B2%25D0%25B5%25D0%25BD%25D0%25B8%25D1%2580%25D0%25BE%25D0%25B2%25D0%25B0%25D0%25BD%25D0%25B8%25D1%258F_%25D0%25BD%25D0%25BE%25D0%25B2%25D0%25BE%25D0%25B3%25D0%25BE_%25D0%25BF%25D0%25BE%25D0%25BA%25D0%25BE%25D0%25BB%25D0%25B5%25D0%25BD%25D0%25B8%25D1%258F">of the new generation, the</a> sequences are produced faster than they are analyzed.  Here is just an evolutionary reconstruction - it is computationally expensive and it would be nice to be able to get representative samples of a size suitable for analysis.  How to do this and what is generally ‚Äúrepresentative‚Äù in this case - under the cut. <br><img src="https://habrastorage.org/webt/hs/wi/b3/hswib3bhhrkoqhrr9fq6nz1vnhy.jpeg" alt="The tree of life by Hug et al. 2016"><br><a name="habracut"></a><br><div class="spoiler">  <b class="spoiler_title">What is KDPV?</b> <div class="spoiler_text">  <a href="https://www.nature.com/articles/nmicrobiol201648">The most complete tree of life that currently exists</a> .  All multicellular life fits into a green spot in the lower right corner and makes up a rather small share of it, and all animals and mushrooms (together called Opistokonta) - in the rightmost ledge of this spot.  This is approximately the diversity of life on the planet and such is the amount of data that is produced in the process of studying it. <br>  Red circles, by the way, isolated groups of unicellular organisms that no one has ever seen.  Culture does not grow in any laboratory in Petri dishes or clever bioreactors.  Their existence has become known only in recent years from metagenomic studies, when all the DNA that is from the medium is isolated from the medium and analyzed all the time.  But it is, by the way.  The picture is included in the article solely as the most amazing tree in recent years and is not important for understanding the article. <br></div></div><br><h3>  What should be done </h3><br>  First of all, let us state clearly what is happening and what we are trying to achieve.  We want to know where it came from and how a particular gene, say, a man, evolved.  Or potatoes, or unicellular algae, the methods for any organism will be about the same.  In biological databases one can find related sequences of other organisms, shove them into <a href="http://www.iqtree.org/">some kind of</a> <a href="https://sco.h-its.org/exelixis/software.html">magic</a> <a href="http://mrbayes.sourceforge.net/">program</a> and get a phylogenetic tree at the output.  The problem is that there will probably be thousands of related sequences.  And the construction of phylogenetic trees with <a href="https://en.wikipedia.org/wiki/Computational_phylogenetics">maximum likelihood</a> methods or <a href="https://en.wikipedia.org/wiki/Bayesian_inference_in_phylogeny">Bayesian phylogenetics</a> is an NP-complete problem and the solution consists, roughly speaking, in the search for different trees and parameters of the evolutionary model.  I usually have from a couple of days to a week on 2 16-core AMD Opteron 6276 processors with 64 Gb of RAM for several hundred genes of medium length.  There are also polynomial methods of phylogenetic analysis, but they are less accurate and publishing an article based on them in a decent journal will be unrealistic. <br><br>  So, you need to take from this set the correct sample: select a relatively small set of sequences of a given size and draw conclusions from it about the entire set.  Further, some phylogenetic terminology will be used, explanations - under the spoiler. <br><br><div class="spoiler">  <b class="spoiler_title">Basic concepts and terms</b> <div class="spoiler_text">  For a start, what is a phylogenetic tree?  This is a binary tree describing the emergence of many modern DNA sequences or proteins from a single ancestor by successive divergences.  The genes (or proteins) united by such an ancestor are <i>homologous</i> to each other.  Even if it existed billions of years ago, there must have been a single ancestor, otherwise the entire analysis does not make sense. <br>  For example (picture below): once there was an ancestral gene A, which had descendants B and C. B left descendants D and E, and C - F and G. There are only four sequences D, E, F and G;  ancestors from A to C existed once upon a time and have disappeared to our days.  Accordingly, we can reason about them exclusively on the basis of modern sequences and mainly in terms of the type ‚Äúcommon ancestor of such and such and such and such‚Äù.  A part of a tree united by a common ancestor is called a <i>treasure</i> .  This term can be applied both to the actual part of the graph and to the sequences at its ends.  In the tree in the picture, for example, there are clades (D, E) and (F, G).  Although nominally a treasure in a tree is the same as internal nodes, usually the researcher is only interested in a few treasures that are significant to his work. <br><img src="https://habrastorage.org/webt/-s/sx/m5/-ssxm5gshcl-0djtwzp27w84sxq.png" alt="The simplest tree of four genes"><br>  One might ask: why am I talking about the evolution of <i>sequences</i> , although everyone is used to thinking about the evolution of <i>species</i> .  In the end, whole organisms feed, multiply and die out, not small pieces of DNA.  This is generally true.  The ‚ÄúTree of Life‚Äù was originally invented for species, and phylogenetic trees perfectly describe their evolution.  But there are two reasons for discussing trees of sequences: first, phylogenetics as a field of bioinformatics most often deals with sequences.  The evolution of organisms with a set of features is an intricate and difficult to formalize process, moreover, it took place millions of years ago under the influence of unknown factors and in general it is not very clear with whom.  Disputes about what exactly the first eukaryote was, for example, are unlikely to end before the invention of the time machine.  On the other hand, replacing, inserting and deleting letters in a string is quite amenable to mathematical modeling. <br>  Secondly, the evolution of an individual gene may differ from the evolution of the species in which it is represented.  The simplest scenario is described on the tree above - the common ancestor of the species studied had one copy of the gene under study, and during evolution it was simply inherited, gradually accumulating small mutations.  Such genes are attributed to each other by <i>orthologs</i> .  By definition, orthologs are genes of different species, going back to the single gene of their common ancestor. <br>  But suppose that ancestor C had a <i>duplication of</i> the gene under study.  There was one gene, and it became two identical, which began to be inherited and mutate independently of each other.  It is likely that one of the copies of the gene continued to do the same as before, and the second (once its old function and without it has someone to perform) mutates and brings some new advantage to the mind.  But the species E or one of his recent ancestors lost this gene altogether.  This happens if the gene is not so vital.  Although the relationships of the species have not changed, the tree will take the following form: <br><img src="https://habrastorage.org/webt/bw/43/ki/bw43ki5c1re1unvp20dhbh4yig4.png" alt="Tree complicated by one duplication and one deletion"><br>  Based on this tree, it is already quite difficult to talk about the evolution of species.  E is not at all, and F and G occur twice each.  But it makes it possible to understand when the genes * 1 and * 2 appeared and how far they diverged.  The F1 and G1 genes are parallels to each other - genes of the same species, dating back to the single gene of its ancestor. <br>  In addition to duplications and deletions, there are other events: sometimes genes are acquired by organisms not from parents, but from other non-related organisms (called horizontal gene transfer or HGT), sometimes a piece from some other gene is added to one gene or, vice versa , a piece is cut off, sometimes several genes are combined into one.  A lot of interesting things happen.  So it is not known in advance what the result will be if you throw all homologous sequences into the analysis. <br></div></div><br>  Let us define the ‚Äúcorrect sample‚Äù as follows: let there be a certain number of treasure in the tree that includes all the sequences from the set.  What is their biological meaning, it does not matter;  the main thing is that they are mostly monophyletic (that is, they go back to a single ancestor each) and are basically separated by a more or less long branch from the rest of the tree.  Each clade corresponds to a subset of its sequences.  We shall assume that the correct sample is such that each of these subsets is represented in it by at least one sequence. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <h3>  What we will not do </h3><br>  The most obvious idea is to take sequences at random and hope that the sample will be more or less correct.  But an accident - it is an accident;  may be correct, and maybe not.  In practice, most likely it will not, because the biosphere is studied unevenly.  A person and a number of model objects (the front sight of Drosophila, Baker's yeast, Escherichia coli E. <i>coli</i> , etc.) are studied in great detail and everything that can be sequenced is repeatedly sequenced for them.  Economically or medically significant objects (for example, rice and HIV) and close relatives of significant objects (for example, primates as human relatives) are studied a little worse, but also very well.  But single-celled plankton, no matter how diverse it may be, is of little interest to anyone and is sequenced relatively rarely.  Therefore, there may well be 30 sequences of animals, 30 sequences of plants, 20 sequences of fungi, and one or two sequences of all other eukaryotic groups in a set of 100 homologous genes. <br><br>  For a random sample, the ratio of the number of groups as a whole will remain.  But we do not need to save it at all; it still reflects more the number of scientists involved in one or another taxon and their financing than some kind of biological reality.  We need to cover as many groups as possible, including small ones. <br><br>  Okay, the next idea is that since we already have taxonomic meta information (which organism has which sequence), why not use it?  Take one human gene, more or less evenly take a dozen of the genes of other animals, a dozen of the genes of fungi, and so on.  The idea is not bad, <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4044049/">people do it</a> , but not universal.  If all the genes in the original set are each other orthologs, that is, their evolution fully coincides with the evolution of their owners, then it is best to do so.  But this is not always the case.  Take a look, for example, at the picture: <br><br><div class="spoiler">  <b class="spoiler_title">Eukaryotic chitin synthase tree</b> <div class="spoiler_text"><img src="https://habrastorage.org/webt/2c/f1/ap/2cf1apv53d3s0lkxi2se2wiwn6k.png" alt="Chitin synthase tree, very confusing"><br></div></div><br>  This is the evolution of chitin synthases (proteins responsible for the synthesis of chitin) from <a href="https://academic.oup.com/glycob/article/26/6/635/2363177">our previous article</a> .  As can be seen, the fungi proteins (Fungal class *) do not form a single treasure;  instead, they are scattered around two different ends of the tree.  The same thing happens with other organisms: for example, most oomycete species have one gene from each of the four oomycete treasures.  Even if we take for simplicity that the treasure is only two, they still need to be taken into account. <br><br>  The conditional ‚Äúten genes of fungi‚Äù should not be a dozen of the genes of one of the treasure, it should be several genes from one clade and several from another.  The same is true for the other groups. <br>  Chitin synthases have a rather complicated evolution, and for most genes the tree will be much simpler.  But the complexity of the tree cannot be estimated before it is built, and therefore the universal sampling method must be able to work with such data. <br><br>  Finally, the third idea.  Somewhere at the beginning casually mentioned were quick and less reliable methods of phylogenetic analysis.  Why not build some tree with their help, select treasures in it and not take a sample of them already?  Yes, the tree will not be very accurate, but most likely it will be more or less similar to the true one.  You can make a selection on the basis of such a tree, and already calculate it properly.  In this form, the idea is not very practical, because ‚Äúisolating treasures‚Äù is not a trivial task.  I mentioned above that the treasure is actually the same as the number of nodes in the tree, and it is not known in advance which ones interest the user.  But you can build on this idea. <br><br><h3>  What do we do </h3><br>  One of the most common ways to quickly get a tree is to build a distance matrix for sequences (and there are a lot of different distance metrics for biological sequences) and on its basis build a tree with some kind of <a href="https://en.wikipedia.org/wiki/Computational_phylogenetics">NJ or UPGMA</a> .  Of course, the construction of the distance matrix is ‚Äã‚Äãquadratic in time and memory of the number of sequences, but unlike ML or Bayes phylogenetics, it is at least polynomial. <br><br>  Even without building a tree, on the basis of the distance matrix, you can get a sample that will include as many different sequences as possible.  Since the sequences in any clade are by definition more similar to each other than to other sequences, this method should cover most of the treasure.  The algorithm is quite simple: we include in the sample an arbitrary sequence, for each of the remaining ones we remember the distance to it.  Then we include in the sample the remaining one, for which this distance is maximal, for the others we remember the distance and to it too.  Then, from the rest, choose the one with the largest minimum distance to one of the sequences in the sample, add it to the sample and memorize the distance.  Repeat until the sample reaches the desired size.  For similarities to the Neighbor Joining algorithm, we called this method Distant Joining.  In <a href="https://github.com/SynedraAcus/sampler/blob/53637266eee2070d49f3b5242e4b140800bf3923/reductor.py">code,</a> it looks like this: <br><br><pre><code class="python hljs">leader = not_sampled[<span class="hljs-number"><span class="hljs-number">0</span></span>] reduced_list.append(leader) minima = {x: dist(leader, x) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> x <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> not_sampled[<span class="hljs-number"><span class="hljs-number">1</span></span>:]} <span class="hljs-keyword"><span class="hljs-keyword">while</span></span> len(reduced_list) &lt; final_count: leader = max(minima.items(), key=<span class="hljs-keyword"><span class="hljs-keyword">lambda</span></span> a: a[<span class="hljs-number"><span class="hljs-number">1</span></span>]) reduced_list.append(leader[<span class="hljs-number"><span class="hljs-number">0</span></span>]) <span class="hljs-keyword"><span class="hljs-keyword">del</span></span> minima[leader[<span class="hljs-number"><span class="hljs-number">0</span></span>]] not_sampled.remove(leader[<span class="hljs-number"><span class="hljs-number">0</span></span>]) <span class="hljs-keyword"><span class="hljs-keyword">for</span></span> i <span class="hljs-keyword"><span class="hljs-keyword">in</span></span> not_sampled: d = self[(leader[<span class="hljs-number"><span class="hljs-number">0</span></span>], i)] <span class="hljs-keyword"><span class="hljs-keyword">if</span></span> d &lt; minima[i]: minima[i] = d</code> </pre> <br><br>  Or picture: <br><img src="https://habrastorage.org/webt/_h/zg/zk/_hzgzkf4dnkwyyawysdqcscwvoq.png" alt="The sequence of steps in Distant Joining"><br><br>  Red - added at this iteration, black - already added. <br><br>  As it turned out, the described method works quite well.  Here are the results on a thousand simulated sequence sets (blue - Distant Joining, black - RS, random, red - SS, the choice of how many sequences closest to a given one): <br><br><img src="https://habrastorage.org/webt/cu/ka/ng/cukang5oangbv0_sopjdtvpaam0.png" alt="Different sampling methods on simulated data"><br><br>  For each simulation, a tree of about a dozen or so treasure (a normal distribution with an average of 15 and a standard deviation of 5) of different sizes, from units to hundreds of sequences each, was created.  According to this tree, random nucleotide sequences were generated, and then from the generated set, samples were taken from 10% to 90% of the sequences.  Efficiency was estimated as the fraction of the treasure represented in the reduced sample in at least one sequence.  As you can see, choosing a small number of DJ sequences is much more efficient than the other two methods. <br><br>  By the way, what is this monstrously inefficient and obviously meaningless choice of n sequences closest to any one?  This, of course, is not a method of reducing samples; it is a possible artifact.  Roughly speaking, a request to the biological database can be formulated either as ‚Äúgive me 100 sequences most similar to this one‚Äù, or as ‚Äúgive me all the sequences that look like this at least that much‚Äù.  The first formulation is useful for other tasks (and most often it is by default), but in the case of phylogenetic analysis it can lead to serious distortions in the data set. <br><br>  On real data, testing can also include the taxonomy-based <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4044049/">AST</a> method.  The results on two data sets are shown below: on the left a large set of ribosomal RNAs.  This is a classic marker gene, inherited almost without any duplication or deletion and therefore widely used to study the evolution of species.  On the right is a set of wood chitin synthases above. <br><br><img src="https://habrastorage.org/webt/d6/tq/7y/d6tq7ye-mo7ne0o2gza50ofs9ns.png" alt="Different sampling methods on real data"><br><br>  The result is fairly predictable: in the case of rRNA, taxonomic metadata ideally describes the evolution and therefore AST bypasses other methods.  But with chitin synthases he can not cope for the reasons described above.  The horizontal straight line at the top is not a frame around the graph, this DJ covers all the clades even if the data set is reduced five times. <br><br><h3>  Conclusion </h3><br>  Strictly speaking, in the described algorithm there is nothing specifically biological.  Yes, in this case, I used the sequence and biological metrics of the distance between them.  But in the same way, it is possible to take samples from the aggregates of any objects for which there is a matrix of distances and the variety of which needs to be sampled fairly evenly. <br><br>  It is only necessary to keep in mind that Distant Joining does not struggle with emissions (which it perceives as a significant diversity and is one of the first in the sample) and does not preserve the ratio of cluster sizes. <br><br>  A prototype in Python 3 <a href="https://github.com/SynedraAcus/sampler">is available on github</a> .  If he suddenly comes in handy in scientific work, please quote <a href="http://journal-biogen.org/article/view/66">this article</a> . </div><p>Source: <a href="https://habr.com/ru/post/351324/">https://habr.com/ru/post/351324/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../351312/index.html">3. Check Point for maximum. Content Awareness</a></li>
<li><a href="../351316/index.html">Introvert management introverts or experience managing technical teams</a></li>
<li><a href="../351318/index.html">John Carmack: Weekly vacation spent programming</a></li>
<li><a href="../351320/index.html">Profiling: measurement and analysis</a></li>
<li><a href="../351322/index.html">Evil by Design: interfaces from Mephistopheles (part one)</a></li>
<li><a href="../351326/index.html">Information security of bank non-cash payments. Part 4 - Overview of Threat Modeling Standards</a></li>
<li><a href="../351328/index.html">Uncompetitive wrestling: how crossover tournaments have changed in their first year</a></li>
<li><a href="../351332/index.html">What is PAT? Lab at Packet Tracer</a></li>
<li><a href="../351334/index.html">What drives IT in business: jumps and obstacles</a></li>
<li><a href="../351336/index.html">About the ‚Äúshelf life‚Äù of the employee</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>