<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Again on bioinformatics: the assembly of bacterial genomes</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Since I published the first review article on bioinformatics , almost a year has passed. I would like to write more often, but just bioinformatics is ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Again on bioinformatics: the assembly of bacterial genomes</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/getpro/habr/post_images/101/85d/f2e/10185df2e6e6b00c796aaaec4d14ce92.jpg" align="right">  Since I published the <a href="http://habrahabr.ru/post/143115/">first review article on bioinformatics</a> , almost a year has passed.  I would like to write more often, but just bioinformatics is distracted and does not give.  In the previous article, the task of assembling the genome was mentioned, as well as the Laboratory of Algorithmic Biology ( <abbr title="St. Petersburg Academic University of the Russian Academy of Sciences">SPbAU RAS</abbr> ), in which it is engaged.  As already mentioned, this task has been worked in many universities of the world for a long time and quite successfully.  However, to assemble a genome, biologists get a huge amount of different types of data, each of which has its own characteristics.  About five years ago, <abbr title="Multiple Displacement Amplification">MDA</abbr> technology appeared, which opened up great opportunities in the field of studying bacteria.  As a result, a data type emerged that required a new genomic collector.  A few years later it was developed, and not at Stanford or Massachusetts, but in St. Petersburg, at the young Academic University.  But first things first. <br><a name="habracut"></a><br><h4>  Again about the genome assembly </h4><br>  Briefly recall basic concepts.  The genome for biologists is a long <abbr title="Deoxyribonucleic acid">DNA</abbr> molecule, which is a double chain of four types of nucleotides (adenine, cytosine, guanine and thymine).  For example, the length of a DNA chain in a bacterium is measured in millions of nucleotides, while the length of human DNA is approximately 3 billion.  For bioinformatists, the genome is just a big line above the alphabet of four characters {A, C, G, T}.  ‚ÄúRead‚Äù the entire DNA molecule is completely impossible.  Modern technology allows you to read only small pieces of a length of several hundred nucleotides from random places, and then with errors.  These pieces are called reads or reads, and the reading process is called sequencing.  In one experiment, several tens or even hundreds of millions of readings are performed.  Next comes the bioinformatics that develop genomic assemblers - programs that attempt to restore the original sequence from these readings.  As a rule, assemblers do not manage to restore the entire genome as a whole, but from millions of short readings they allow to get dozens of sequences hundreds of thousands of nucleotides in length.  Such sequences can already be further studied by biologists. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/5d9/25a/5a7/5d925a5a747f7cf179842e05355668bb.png" align="left">  Who may need to collect and analyze the genome?  For example, the assembly of the human genome can help determine the presence of cancer cells in the body at an early stage.  Another goal - the transition to the so-called personal medicine.  If each person has a known genome, then medications and treatment can be prescribed to him not only by symptoms, but also taking into account his genetic characteristics.  Personal medicine is not yet widespread, but some clinics already provide patient sequencing services.  The price of reading the genome is constantly falling, and new assembly algorithms allow it to be faster.  However, the human genome alone is not enough for a complete diagnosis.  In each of us there are several thousand species of different bacteria that are involved in a huge number of internal processes and are necessary for life.  The assembly of the genomes of these bacteria (the total length of which exceeds the length of the human genome itself) would allow a much better study of the human body.  However, there are difficulties in their sequencing. <br><br><h4>  From person to bacteria </h4><br>  To sequence a genome, it is necessary to have a sufficiently large amount of genetic material ‚Äî identical DNA molecules.  A standard experiment requires several tens or hundreds of millions of identical cells.  In the case of multicellular organisms, including humans, this is not a problem, since a sufficient amount of genetic material is contained in a few grams of saliva.  Obtaining as many bacterial cells is more difficult - they need to be grown in the laboratory.  The problem is that not all bacteria can be cultivated.  For example, bacteria that are inside the human intestines live in large colonies and are not able to exist and multiply separately from each other.  It is impossible to cultivate a single bacteria taken from the colony. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Today, there are two fundamentally different approaches to the sequencing of uncultivated bacteria in colonies.  The first, fairly simple from a biotechnological point of view, is metagenomics.  It consists in the following: we take a colony, we select DNA molecules from all bacteria, we place them in a sequencer and we read.  As a result, we will simultaneously receive the reading of the genomes of several dozen different bacteria - from some more, from some less.  In closely related bacteria, the genomes are similar to each other, and in bacteria of different types, they are very different, but may contain identical areas.  These factors greatly complicate the assembly of such data.  Despite a number of metagenomic studies conducted and carried out today, I cannot say that there is at least one assembler that would give results on such data that would satisfy the biologists.  As a rule, collectors give short sequences that contain a large number of errors and are difficult to analyze.  Sometimes in metagenomic projects, the discovery of several genes for the entire set of different bacteria is already a success. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/322/c00/f6c/322c00f6cb52d0a9584d34cfbc703af9.png" align="right">  Another approach to sequencing uncultivated bacteria was discovered only a few years ago and consists of a rather complex biotechnological process.  We take one bacterium of interest to us from the colony, extract DNA from it, and plant several ‚Äúcopying machines‚Äù on the DNA, which begin to ‚Äúrun‚Äù along the molecule and copy its parts randomly.  When copying, new DNA strands branch off from the molecule, which in turn can also be joined by a ‚Äúcopy machine‚Äù and continue building up genetic material.  As a result, we get identical copies of the sections of DNA of interest to us, which we place in the sequencer.  This method of sequencing a single cell genome is called Multiple Displacement Amplification (MDA).  It would seem that everything is great.  But, alas, there are problems.  ‚ÄúCopy machines‚Äù work on their own and cannot be controlled.  Therefore, some sections of DNA can be copied ten thousand times, and others - never.  As a result, some regions will be missing in the assembly simply due to lack of material.  On average, as a result of this method, 95-99% of the genome can be restored, depending on its structure (some can be copied worse) and the quality of the experiment. <br><br><h4>  MDA technology issues </h4><br>  The number of times that the nucleotide in the genome is read during the sequencing process is called its coating.  In standard sequencing, the coverage of all nucleotides is approximately the same (an example of the coverage graph in the left figure below).  When using the MDA technology, we obtain data with extremely uneven coverage (the right figure below), which is a critical problem for the assembly task.  Let's try to figure out why. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/514/a35/689/514a3568957ff0c34d119486883db249.png"><img src="https://habrastorage.org/getpro/habr/post_images/f6a/954/e7d/f6a954e7d3e1dfa60a82d4710565f7fe.png"><br><br>  Let's make the following experiment.  Take the data obtained using standard sequencing, and from all readings we extract substrings of a fixed length k (next - k-measures), so that from each possible position in reading one k-measure begins (in this way the ACGTAC sequence is divided into four 3-measures : ACG, CGT, GTA and TAC).  Next, let's calculate how many times each k-mer occurs in our readings.  If some k-mer appears a sufficiently large number of times (that is, it has a high coverage), there is reason to assume that it also occurs in the original genome, which we are sequencing.  If k-mer is rare, then it most likely contains an incorrectly read nucleotide.  Sequencing errors are a random event, and the likelihood that the same error will occur immediately in a large number of k-measures is very small (the probability of incorrect reading of a single nucleotide ranges from 0.001% to 1% depending on the position of the nucleotide in the reading).  Thus, with the help of the value of the coverage of k-mers, it is possible to divide them into reliable and erroneous.  Similar methods are used in most modern assemblers and utilities for correcting errors in readings.  However, in the case of sequencing a single cell, it is impossible to divide k-measures in this way, since k-measures that are rarely encountered may correspond to a section copied a small number of times, and do not contain errors at all. <br><br>  Uneven coverage is not the only problem with the MDA approach.  When copying DNA from the main thread, a new single-stranded thread branches off.  It can randomly stick together with another spun off thread, which is why DNA sequences will be next to it, which correspond to completely different regions in the original genome.  In the place of such an erroneous gluing together, ‚Äúcopying machines‚Äù can again sit down and multiply the wrong part.  Then, as a result of sequencing, we obtain readings containing sequences from different parts of the genome (chimerical readings).  When processing such readings, the genomic collector may incorrectly combine two sequences into one.  If one wrong nucleotide is not a gross error, then two wrongly glued sections are a much more serious problem that complicates further analysis. <br><br><h4>  At last </h4><br>  It is the data obtained using the MDA technology that became the main subject of research at the Laboratory of Algorithmic Biology in St. Petersburg.  Many algorithms have been developed that have allowed to circumvent the problems of uneven coverage and chimerical readings.  These algorithms were implemented in the genome assembler SPAdes, which today is the leader in the field of bacterial genome assembly and is used by leading global laboratories (for example, in the United Genomic Institute in the USA - the world's largest center for the study of bacterial genomes). <br><br>  Then I would gladly acquaint readers with the basic principles of modern standard assemblers, talk about new approaches and algorithms that are used in the collection of data obtained in a single cell, but, unfortunately, this does not fit into one article.  For those who are interested to learn more about the assembly algorithms - below there are links to popular texts and scientific publications on this topic.  I sincerely hope that the next such article will appear faster than this time. <br><br><h5>  For those interested </h5><br><ul><li>  The function of the ‚Äúcopy machine‚Äù is performed by Phi-29 DNA polymerase, a protein capable of replicating DNA segments in random places. </li><li>  In order to identify and eliminate sequencing errors in data obtained using MDA, approaches based on, for example, clustering of k-mers by Hamming distance, topological features of the de Bruyne graph in places of errors are used. <br></li><li>  Search and removal of chimerical compounds are also mainly performed according to the topology of the graph de Bruijn. </li></ul><br><h5>  Bibliography </h5><br><ul><li>  <a href="http://bioinf.spbau.ru/sites/default/files/Pevzner_Final2.pdf">PEC Compeau and PA Pevzner.</a>  <a href="http://bioinf.spbau.ru/sites/default/files/Pevzner_Final2.pdf">Genome Reconstruction: A Puzzle with a Billion Pieces</a> (review article on genome assembly) </li><li>  <a href="http://www.pnas.org/content/98/17/9748.full.pdf">PA Pevzner, H. Tang, and M. Waterman.</a>  <a href="http://www.pnas.org/content/98/17/9748.full.pdf">An Eulerian path approach to DNA fragment assembly.</a>  <a href="http://www.pnas.org/content/98/17/9748.full.pdf">Proceedings of the United States of America, 98: 9748-9753, 2001</a> (first article on the use of Count de Bruin in the task of assembling the genome) </li><li>  <a href="http://genome.cshlp.org/content/18/5/821.full.pdf">D. Zerbino, and E. Birney.</a>  <a href="http://genome.cshlp.org/content/18/5/821.full.pdf">Velvet: Briefs of the Bruijn graphs.</a>  <a href="http://genome.cshlp.org/content/18/5/821.full.pdf">Genome Research.</a>  <a href="http://genome.cshlp.org/content/18/5/821.full.pdf">18 (2008), 821‚Äì829</a> (classic Velvet assembler) </li><li>  <a href="">A. Bankevich, S. Nurk, et al.</a>  <a href="">SPAdes: A New Genome Assembly Algorithm and Single-Cell Sequencing.</a>  <a href="">Journal of Computational Biology 19 (5) (2012), 455-477</a> (SPAdes - assembler for assembling data obtained using MDA) </li></ul></div><p>Source: <a href="https://habr.com/ru/post/186492/">https://habr.com/ru/post/186492/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../186482/index.html">Honest glow and speed</a></li>
<li><a href="../186484/index.html">jPutty for layout, jQuery plugin</a></li>
<li><a href="../186486/index.html">The simplest implementation of cross-platform multiplayer on the example of the evolution of a single .NET game</a></li>
<li><a href="../186488/index.html">MikroTik + port knocking over ICMP</a></li>
<li><a href="../186490/index.html">Roskomnadzor plans to block resources with information about blocked resources</a></li>
<li><a href="../186494/index.html">Dune 2 The Building of A Dynasty</a></li>
<li><a href="../186496/index.html">Human parser on Selenium WD</a></li>
<li><a href="../186498/index.html">cPanel :: Feature Request :: FastCGI Process Manager (FPM) SAPI</a></li>
<li><a href="../186502/index.html">Overview of the home weather station La Crosse WS9257</a></li>
<li><a href="../186504/index.html">Results of the online conference ruHaxe 2</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>