<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Systematics of prokaryotes - distant relatives</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In the summer I planned an experiment and wrote an article Using UML for an experiment on the evolutionary systematics of prokaryotes, and indirectly ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Systematics of prokaryotes - distant relatives</h1><div class="post__text post__text-html js-mediator-article">  In the summer I planned an experiment and wrote an article <a href="http://habrahabr.ru/post/147063/">Using UML for an experiment on the evolutionary systematics of prokaryotes, and indirectly about the psychology of scientists</a> .  The rough processing results were already ready by the end of the summer (thanks to <a href="https://habrahabr.ru/users/mktums/" class="user_link">mktums</a> for the help). <br><br>  Now there was a pause, and I finished off this topic, and present the results. <br><br><a name="habracut"></a>
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <h4>  Method </h4><br><br>  <i>(I will repeat something from the previous article in order not to force new readers to read it)</i> <br><br>  The main criticism of the article <a href="http://habrahabr.ru/post/146331/">Interesting results about the evolutionary systematics of prokaryotes, or ‚Äúmany-species origins‚Äù,</a> consisted in the following claim ‚Äú <i>One gene cannot be considered as a measure</i> ‚Äù.  With this, I fully agree, and this experiment fixes this fully. <br><br>  Some numbers.  Now in the NCBI there are about 2,000 genomes of bacteria (3,723 loci).  In preparation for the experiment, I isolated all the tRNAs that are labeled this way.  They turned out more than 40 thousand unique variations.  But alas, there are a lot of mistakes among them (about 50%, see previous articles, where it was discussed in detail). <br><br>  But I thought that you can skip the stage of full error correction.  How to do it?  I sorted these tRNAs by length and by the presence of the end of the CCA at the end of the sequence.  It must be said that the CCA sequence is obligatory for any tRNA, and the length can be from 74 to 96 nucleotides. <br><br>  There are many miracles in NCBI up to tRNA from one nucleotide, or more than 1300 :) (you can‚Äôt tell without a smile).  Therefore, I removed the sequences that are up to 70 and over 100 in length, as well as those that do not end in CCA. <br><br>  There are about 20,000 of them. These are the most likely tRNAs that do not contain errors from the NCBI.  With the remaining half of the tRNA - you can figure it out later. <br><br>  In fact, for a planned experiment, it makes no difference whether this particular sequence of 70‚Äì100 nucleotides in length contains errors or not.  Why?  Since I am going to double-check the genomes of 2000 bacteria, are there really such sequences - the errors will be excluded.  And tRNA is actually whether or not this is the second thing.  The main thing is that different organisms coincide significant portions of DNA.  The coincidence of the sequence length of 70-100 in the genomes is far from accidental. <br><br>  Therefore, what am I doing now?  I take these 20,000 tRNAs and find which bacteria they are present in.  If the sequence is present in only one organism, this is not interesting.  And most likely this is an erroneous sequence.  And thus a substantial percentage of errors is eliminated. <br><br>  If there is a sequence in more than one organism, this is one association (connection) between two organisms. <br><br><h4>  results </h4><br><br>  The first article made an important conclusion that <br><br><blockquote>  The multi-species origin greatly confuses the evolutionary picture, but nothing can be done about it - such is the complexity of speciation, and we only need to reflect them most accurately in conditions when not all species are known. <br>  <b>And therefore, to adequately describe, we need not phylogenetic trees.</b>  <b>At a minimum, one can speak of genital trees with two parents (for averaging), and in general, a graph.</b> <br></blockquote><br><br>  I was also advised to display the graph using Graphviz, which I did.  But Graphviz hangs when the number of links in the graph is more than 1000. And I have a total graph of 6172 links.  Therefore, here I show only a small fragment for clarity.  And I give a link to a graph of almost 1000 links. <br><br><img src="https://habrastorage.org/storage2/3dd/d72/f69/3ddd72f69c17f784b54c02653b6a35fd.png"><br><br>  <a href="">Here is the graph with the strongest bonds (links to 5 identical tRNAs are omitted, inclusive)</a> <br><br>  Each link is characterized by a minimum-maximum number of coincident (100% identical) tRNA genes.  Relationship of a genus with oneself means the number of identical tRNAs within this genus (i.e., how species differ). <br><br><h4>  Some conclusions </h4><br><br>  In fact, all this must still be visually processed, so that it would be possible to visually embrace all this set.  On the column with 1000 connections there are many genera that are not connected with anyone - but if you display weaker connections with up to 5 identical tRNAs, then you can see distant relatives.  (I am thinking of doing this as the next step, if you are willing to help, write). <br><br>  In fact, on this basis, much coincides according to the current classification.  The number of identical tRNAs is well illustrated by the range of genera from each other, the smaller the number of identical tRNAs, the more ancient the ancestor.  Those genera that have few connections are the most ancient (as they are sequenced now, and their population is currently represented by separate species).  Analyzing them you can build a fairly accurate initial evolution process. <br><br>  <b>upd.</b>  Removed from the graph two-way communication (littered image).  The total number of links was reduced to 4551. This allowed to display a larger graph: <br>  <a href="http://code.google.com/p/vlabdownload/downloads/detail%3Fname%3DProcaryota2.png">Download the image here (11.2 MB).</a>  <a href="http://code.google.com/p/vlabdownload/downloads/detail%3Fname%3DProcaryota2.png">Here is the graph with the strongest bonds (links to 3 identical tRNAs are omitted, inclusive)</a> <br>  Then connections (intermediate types) between two huge domains are visible (stand out in the image, presumably correspond to Beta and Gamma proteobacteria), and other details.  How much this corresponds to the current classification should be compared, but there is something to think about (just the detailing is such that for sure there is something that did not fall into the current scientific classification). <br><br>  <b>upd2</b> Using yEd Graph Editor it turned out to display the full graph.  Below is a mini picture. <br><br><img src="https://habrastorage.org/storage2/d56/8cd/22d/d568cd22dbe820b2a1b64fea2a761bf6.png"><br><br>  The image turns out badly because of the links are not visible details, so below the file in the format yEd Graph Editor, there at least you can enlarge, move and see.  If someone becomes interested and makes a more visible graph - I will say thanks :). <br><br>  <a href="http://code.google.com/p/vlabdownload/downloads/detail%3Fname%3DProcaryota.graphml">Graph "Systematics of prokaryotes (505 genera and 4548 connections between them)"</a> </div><p>Source: <a href="https://habr.com/ru/post/162729/">https://habr.com/ru/post/162729/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../162713/index.html">How are the variables in PHP</a></li>
<li><a href="../162717/index.html">Back to the Future - Avatars in the phone book</a></li>
<li><a href="../162723/index.html">As I did the budget amplifier on the TDA2050 for old speakers</a></li>
<li><a href="../162725/index.html">Constructing fractal shapes in Matlab</a></li>
<li><a href="../162727/index.html">Logic - weekly selection of gaming and IT industry news ‚Ññ5</a></li>
<li><a href="../162731/index.html">Developers Studyapps raised $ 200,000 on Napartner startups</a></li>
<li><a href="../162733/index.html">How to write a game in the style of tower defense</a></li>
<li><a href="../162735/index.html">gps.conf for everyone, or how to speed up GPS on Android</a></li>
<li><a href="../162737/index.html">Stellaris LaunchPad Short Review</a></li>
<li><a href="../162739/index.html">Analysis of statistics in free-to-play games: analytics tools</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>