<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Installation and initial setup of ZINBA</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="I noticed that the articles are quite large, and the questions are asked in different directions. This article was written to collect questions on ins...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Installation and initial setup of ZINBA</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/storage2/b29/dfd/c14/b29dfdc14df0b7ae656d6316ba9096a3.png" align="left"><br>  I noticed that the articles are quite large, and the questions are asked in different directions.  This article was written to collect questions on installing the ZINBA program in a separate topic.  So, to work with ZINBA, you need to know how to install it. <br><a name="habracut"></a><br><br>  Detailed <a href="http://code.google.com/p/zinba/wiki/Installation">installation painted on the developer's site in English</a> .  Therefore, I will briefly describe the necessary steps in Russian.  All the steps were performed by me in the OS OpenSUSE 12.1 x64, but must pass without incident on other Linux platforms.  We start <a href="http://www.r-project.org/">R</a> (if there are questions on setup, we will discuss in comments), we execute the following commands: <ul><li>  system ("wget <a href="">zinba.googlecode.com/files/zinba_2.01.tar.gz</a> ").  Downloads the package from the developer‚Äôs site. </li><li>  install.packages (c ('multicore', 'doMC', 'foreach', 'qvalue', 'quantreg', 'R.utils')).  Installs additional packages.  A window will appear with a list of sites to download, select the site from which to install these packages. </li><li>  install.packages ("zinba_2.01.tar.gz", repos = NULL).  The installation of the downloaded package occurs. </li></ul>  Now, after any launch of R, it is enough to run library (zinba) to activate the library.  Further, it is recommended to perform the steps for <a href="http://code.google.com/p/zinba/wiki/UsingZINBA">initial configuration</a> , namely: <br><ul><li>  We go to the ‚ÄúNecessary file downloads‚Äù section, in the ‚ÄúGenome Build‚Äù subsection we download one or several <a href="http://genome.ucsc.edu/FAQ/FAQformat.html">two-bit</a> genome indexes, it depends on which of them you are going to work with. </li><li>  In the subsection ‚ÄúMappability files‚Äù we upload a file containing the name of the genome in the title and the length of the reads, as in your FASTA / FASTQ file.  Do not forget to unzip it to any directory.  In Linux, for example, you can download and unzip it like this: ‚Äúwget <a href="http://www.bios.unc.edu/~nrashid/map">www.bios.unc.edu/~nrashid/map##.tgz;tar</a> -xzvf map ##. Tar.gz‚Äù, where ## is a number corresponding to the length reed </li><li>  Run R, execute two commands successively: 1. generateAlignability, to create an alignment index and 2. basealigncount, for a basic alignment count;  This command will create a file containing the exact binding of the peaks in several iterations. </li></ul>  Further detailed description of these two commands: <br><br><pre><code class="bash hljs">generateAlignability( mapdir=, <span class="hljs-comment"><span class="hljs-comment">#   ,     ¬´mappability¬ª outdir=, #  ,      athresh=, # ,       extension=, #       twoBitFile=, #  ,    .2bit ) basealigncount( inputfile=, #      , ,  bowtie outputfile=, #   ,    extension=, #       filetype=, #   "bed", "bowtie", or "tagAlign" twoBitFile=, #   ,    .2bit )</span></span></code> </pre> <br>  The first command is recommended to be performed once and later only if the parameters of the experiments change, such as: the number of errors and the length of the fragments.  The second command should be carried out from experiment to experiment, and then only if we need to improve the quality of finding enriched regions and peaks (as this requires additional computational load). <br>  During the execution of these actions, I received a number of errors, talked with technical support.  It turned out that I didn‚Äôt read the instructions well enough, and they didn‚Äôt have everything smoothly.  First, if the file is received using a bowtie (the usual is not .sam, without the ‚ÄìS parameter), then the last column in the file should be simply removed, there should be 7 columns in the file. It also turned out that the separator is not only tabulation, but so make sure the fields do not contain spaces.  This is true for version 2.01, promised to fix it in the future. <br>  The team itself finding the enriched regions is described below, runs for a long time and eats up all the resources.  I performed it on experimental data from 33.000.000 reads, on 4x CPUs (not to be confused with a stone, just 4 threads, one Core i7), 8Gb RAM, though in a VirtualBOX virtual machine, the calculations took 8 hours.  If there is only 8Gb of RAM, I do not recommend putting more threads; the program will work more with the paging file than with the data. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <pre> <code class="bash hljs">zinba( refinepeaks=, <span class="hljs-comment"><span class="hljs-comment">#   ( )? 1 -  , 0 -   seq=, #      input=, #     ,    ,   .    'none' filetype=, # : 'bed', 'bowtie',  'tagAlign' threshold=, #,    p value   .   ,  0.05 align=, #  ,   outdir   generateAlignability numProc=, # CPU   ,     ( Core i7  8,  7  ,   1) twoBit=, #  ,    .2bit outfile=, #      extension=, #       ###################   ################### basecountfile=, #  basecount ,   refinepeaks   1 broad=, # , TRUE  FALSE ( ) printFullOut=, #    : 1,   (  ); 0,   ( ) )</span></span></code> </pre><br>  We have prepared the package for work, now is the time to launch something to check it.  The developer offers a set of test files: <a href="">http://www.unc.edu/~nur2/zinbaweb/test_data.tgz</a> .  In the command line consistently perform the following actions. <br><br><pre> <code class="python hljs"> <span class="hljs-comment"><span class="hljs-comment"># ,       . wget http://hgdownload.cse.ucsc.edu/goldenPath/hg19/bigZips/hg19.2bit wget http://www.bios.unc.edu/~nrashid/map36.tgz wget http://www.unc.edu/~nur2/zinbaweb/test_data.tgz #   tar -xzvf map36.tgz tar -xzvf test_data.tgz #   , , align_athresh1_extension200 mkdir align_athresh1_extension200 # R. R # R     library(zinba) generateAlignability( mapdir='map36/', outdir='align_athresh1_extension200/', athresh=1, extension=200, twoBitFile='hg18.2bit' ) basealigncount( inputfile='data/ctcfGM12878rep3chr22.taf', outputfile='data/ctcfGM12878rep3chr22.basecount', extension=200, filetype='tagAlign', twoBitFile='hg18.2bit' ) zinba( align='align_athresh1_extension200/', numProc=4, seq='data/ctcfGM12878rep3chr22.taf', basecountfile='data/ctcfGM12878rep3chr22.basecount', filetype="tagAlign", outfile="data/ctcf", twoBit="hg18.2bit", extension=200, printFullOut=1, refinepeaks=1, broad=F, input='data/inputGM12878rep3chr22.taf' )</span></span></code> </pre>  In the data directory there will be files with the name ctcf and the extension .peaks, .peaks.bed, which contain the data of interest to us in the form of a <a href="http://code.google.com/p/zinba/wiki/PostAnalysis">table</a> .  The file with the .bed extension can be loaded into the genome browser and see. <br>  I worked with other data taken from the NIH website <a href="http://www.ncbi.nlm.nih.gov/geo/roadmap/epigenomics/">www.ncbi.nlm.nih.gov/geo/roadmap/epigenomics</a> , <a href="">ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByExp/sra/ SRX / SRX040 / SRX040388</a> (file SRR079566.sra).  Below is an image obtained using this data.  Similar graphs can be obtained with other source data.  With the help of the red line, I divided the picture into two parts, the experimental data are above, the standard annotation below (genes, start sites, introns, exons, etc.).  I downloaded not only the obtained bedgraph from the ZINBA program, but also the initial data (the bedgraph from the initial data was obtained using the program described in the article <a href="http://habrahabr.ru/blogs/bioinformatics/137082/">habrahabr.ru/blogs/bioinformatics/137082</a> ).  For example, I selected a few green peaks in the green squares that were in the original data, and those enrichment areas that ZINBA found.  The segments obtained from ZINBA contain thickenings.  These thickenings are counted for the given parameter refignpeaks = 1 and should correspond to the most enriched part of the region. <br><img src="https://habrastorage.org/storage2/7fc/e2a/d39/7fce2ad3967383bb9626d4caf2fc2b30.jpg"><br><br>  How correctly and reliably processed data can be understood only after a thorough study of the algorithm and comparison with the result of the work of other programs. <br><br>  Review by Andrey Kartashov, Cincinnati, OH, porter@porter.st </div><p>Source: <a href="https://habr.com/ru/post/137524/">https://habr.com/ru/post/137524/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137515/index.html">Visual Studio vNext for game developer</a></li>
<li><a href="../137516/index.html">EX.ua left alone!</a></li>
<li><a href="../137518/index.html">Thoughts while reading someone else's code</a></li>
<li><a href="../137519/index.html">Why do IT projects take 2-3 times longer than planned?</a></li>
<li><a href="../137523/index.html">Implementation diaries: the work of the Media Resources module on the example of a bank</a></li>
<li><a href="../137525/index.html">How to download Yandex.slovari only through design</a></li>
<li><a href="../137526/index.html">Consumer extremism or the desire to save when buying goods on the example of phones</a></li>
<li><a href="../137528/index.html">Chrome won first place in Russia</a></li>
<li><a href="../137529/index.html">Intel Anti-Theft Special Forces</a></li>
<li><a href="../137530/index.html">Some details about Class Based Views, part 3</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>