<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Bioinformatics: a view from the inside</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Of all the technical and natural sciences known to me, it is perhaps the worst idea about bioinformatics that people have. It is either incorrect to s...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Bioinformatics: a view from the inside</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/getpro/habr/post_images/f39/1d7/26b/f391d726b832a46f066bcd91b37e68f0.jpg" align="right">  Of all the technical and natural sciences known to me, it is perhaps the worst idea about bioinformatics that people have.  It is either incorrect to some degree or not at all.  When two years ago I started studying bionformatics, I myself didn‚Äôt have any knowledge about this science.  Over time, I began to better understand what tasks bioinformatics are facing, what they use, and what may be the result of their work. <br><br>  Bioinformatics have no test tubes, reagents, bacteria, white coats.  The main tools they have are a laptop, a pen with paper or a white board with a marker - in general, everything is like that of a programmer.  And the work itself is very similar to the work in an IT company, and the laboratory is like a small development department.  What are the differences then?  Well, I'll try to answer. <br><br><a name="habracut"></a>  Firstly, the tasks are mainly algorithmic.  That is, before you write a program, you need to read a few articles, think for yourself, discuss your ideas with colleagues and only then proceed to implement.  Secondly, we have to work with large amounts of data, and therefore the implementation should be as efficient as possible.  However, even an effective, logical and perfectly debugged program may not give the desired result.  The main reason for this is the biological origin of the data, which means a huge number of errors and a significant difference between the data from different laboratories. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Another, perhaps the most visible difference between bioinformatics and programming is research and publication.  Bioinformatics is a science, which means you just need to be aware of everything that happens in the world.  To this end, there are numerous conferences, collaborations with laboratories from other countries, and, of course, publications ‚Äî you also need to tell everyone about your achievements.  Without all this, you can diligently reinvent the wheel. <br><br>  In general, the impression of bioinformatics is exactly that, but it is best to tell it with an example, especially since there is one, and very close.  But first things first. <br><br><h4>  Laboratory of Algorithmic Biology <br></h4><br>  In 2010, the program ‚Äúmegagrants‚Äù was launched in Russia.  Under the leadership of leading Western scientists (in most cases, who had long since left Russia), new scientific laboratories were created.  The laboratory of algorithmic biology at SPbAU under the leadership of Pavel Pevzner, one of the most famous scientists in his field, became one of such.  Pavel graduated from Moscow Institute of Physics and Technology, but soon after that he left for the USA, took up Computer Science (or, to be precise, it was bioinformatics) and is now a professor at the University of California at San Diego. <br><br>  Before you tell what exactly they are doing in the laboratory, it is worthwhile to introduce readers to the current situation. <br><br><h4>  Little about genomics <br></h4><br>  I am sure that every reader has heard the word genome.  For biologists, the genome is DNA molecules - long chains consisting of four nucleotides, organized into chromosomes, folded in the nucleus of a cell.  We see the genome as a string, again consisting of four characters (A, C, G, T).  The length of the genome can reach billions or even tens of billions of characters.  Biologists do not know how to read the entire genome - only small fragments up to 150 "letters", and even then with errors.  Our task is to restore the original genome for these pieces, or as they often say, to collect. <br><br>  For clarity, you can make such a comparison: imagine a pack of identical newspapers.  Now imagine that this pack is blown up and small pieces of paper fly away, shuffle, deteriorate or even burn completely.  And then on the heap of this garbage I want to glue the original newspaper. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/fc4/316/da1/fc4316da1dc3b5a1affa5e61be50fa64.png" align="left">  The same with the genome.  The first technologies made it possible to read pieces of the genome with a length of up to several thousand characters.  These technologies were incredibly expensive ‚Äî several billion dollars were spent on assembling the first human genome and several years of hard work by hundreds of laboratory workers around the world.  Modern technologies allow you to read shorter fragments, but much cheaper and in large quantities.  Processing gigabytes of input data, of course, automatically.  For this purpose, programs are developed that are called genomic collectors, or more often assemblers (assembled from English).  Due to some features of the original genomes (for example, recurring regions), as well as a large number of errors in the input data, the result of the work of the collector is not a whole gene, but rather long enough parts of it.  The longer the resulting sites, the more they are similar to the original genome, the better the result is considered. <br><br><h4>  Genome assembly task <br></h4><br>  If we take the task of assembling a genome in the most general case, it will be nothing more than the shortest superstring problem, which is formulated as follows: find the shortest string such that each line in the given set would be its substring.  This task is NP-complete.  But if we assume that we have all the possible substrings of the source string of the same length, the problem can be solved in polynomial time.  Assembly of the genome is just such a case.  In 2001, Pavel Pevzner proposed an efficient genome assembly approach using the Count de Bruin.  The main idea of ‚Äã‚Äãthis approach is used in almost every modern genomic assembler.  However, in practice, everything is greatly complicated by the aforementioned biological errors, and therefore the main task is to develop heuristics for various subtasks arising during the assembly of genomes. <br><br>  In the laboratory of algorithmic biology, it was decided to focus on the development of the assembler.  Of course, at the time of the creation of the laboratory there was a huge number of genomic collectors.  Why was it to create another one?  In fact, the assembly task turns out to be much wider than it seems at first glance.  Biologists produce a huge number of different types of input data, each of which requires the development of new methods that take into account their specifics.  In addition, the assembly of the genome includes a large number of stages and algorithms, so even despite the fact that all modern assemblers use the same approach, their results can be very different.  The laboratory was tasked with obtaining an assembler that would surpass existing ones in many ways. <br><br><h4>  Path to bioinformatics <br></h4><br>  I got into bioinformatics, one might say, by chance.  I studied at the SPbAU Master's program and, like every student, at the beginning of the semester I had to choose a research paper.  To try myself in a new field, I chose a bioinformatics project.  At first, the scarecrow is that you may have to learn biology, instead of developing and implementing algorithms.  However, the fears, fortunately, were not justified - immersion in this subject area is exactly the same as in any other.  Gradually, you begin to understand more, learn something new, and even if biology was far from being the most favorite subject in school, interest in it appears fairly quickly.  Almost immediately, I realized that bioinformatics is exactly what I wanted to do - programming with elements of research and an interesting subject area. <br><br>  While I was working on my project, the laboratory of algorithmic biology I mentioned was organized.  In the summer of 2011, I successfully completed my internship in her and remained as a permanent research assistant.  If we talk about the laboratory as a whole - a huge number of various interesting projects that are not limited to assembling genomes, cooperation with Western laboratories, scientific conferences, a constant opportunity to learn something new and, of course, a very good team. <br><br>  One could probably talk for a long time about both work in the laboratory and bioinformatics in general, which still has a lot of open problems, and concrete approaches and algorithms for various tasks.  But it is impossible to embrace the immensity, and therefore the story will be about one thing and the next time.  And what exactly it depends on your wishes. <br><br><h4>  Links <br></h4><br><ul><li>  <a href="http://bioinf.spbau.ru/sites/default/files/Pevzner_Final2.pdf">PEC Compeau and PA Pevzner.</a>  <a href="http://bioinf.spbau.ru/sites/default/files/Pevzner_Final2.pdf">Genome Reconstruction: A Puzzle with a Billion Pieces (review article on genome assembly)</a> </li><li>  <a href="http://www.pnas.org/content/98/17/9748">PA Pevzner, H. Tang, and M. Waterman.</a>  <a href="http://www.pnas.org/content/98/17/9748">An Eulerian path approach to DNA fragment assembly.</a>  <a href="http://www.pnas.org/content/98/17/9748">Proceedings of the United States of America, 98: 9748-9753, 2001</a> </li></ul></div><p>Source: <a href="https://habr.com/ru/post/143115/">https://habr.com/ru/post/143115/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../143106/index.html">On KickStarter, online community members expose fraudsters</a></li>
<li><a href="../143108/index.html">Video review gaming laptop MSI GT60</a></li>
<li><a href="../143111/index.html">Nothing else belongs to you</a></li>
<li><a href="../143113/index.html">Installing Oracle JDK 7 in Ubuntu 12.04</a></li>
<li><a href="../143114/index.html">How Bulgakov helps to study the distribution of commas - a site for improving literacy</a></li>
<li><a href="../143116/index.html">Statistical analysis of project estimates, or why correct estimates are always exceeded</a></li>
<li><a href="../143117/index.html">Kindle Services Overview</a></li>
<li><a href="../143118/index.html">Text to Speech on Android</a></li>
<li><a href="../143119/index.html">+1 extension for Mozilla Firefox</a></li>
<li><a href="../143122/index.html">Facebook allows you to specify whether the account owner agrees to donate internal organs</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>