<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Five selfish reasons to work reproducibly</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="annotation 


 So, my fellow scientists, do not ask what you can do for reproducibility ‚Äî ask what reproducibility can do for you! 


 Here I will lis...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Five selfish reasons to work reproducibly</h1><div class="post__text post__text-html js-mediator-article"><h2 id="annotaciya">  annotation </h2><br><p>  So, my fellow scientists, do not ask what you can do for reproducibility ‚Äî ask what reproducibility can do for you! </p><br><p>  Here I will list five reasons why the reproducibility of data pays off in the long term and is of personal interest to every ambitious, career-oriented scientist. </p><br><hr><br><p>  The complex equation in the left half of the black board, an even more complex equation in the right one.  A short sentence links two equations: "A miracle happens here."  Two mathematics, in deep thought.  ‚ÄúI think you should be more specific in the second step,‚Äù says one to the other. <br><img src="https://habrastorage.org/webt/h0/ja/ga/h0jagadk0meqglw8vhy_b0m6zdq.png" alt="image"></p><br><p>  This is how the situation looks like when you are trying to understand how the author moved from a large and complex data set to a dense article with many complex graphs.  Without access to data and analytical code, such a transition can only be explained by a miracle.  And in science there should be no miracles. </p><br><p><a name="habracut"></a>  The ability to work transparently and reproducibly relies heavily on empathy ‚Äî put yourself in the shoes of one of your colleagues and ask: ‚ÄúCan this person access my data and understand the meaning of my analysis?‚Äù.  Mastering such "tools" (Box 1) requires the involvement and huge investments of your time and energy.  A priori, it is not obvious why the advantages of such a work format exceed the costs. </p><br><p>  Here are a few reasons that usually result in such cases: ‚ÄúBecause reproducibility is right,‚Äù ‚ÄúBecause it is the basis of science!‚Äù, ‚ÄúBecause the world would be a better place if everyone worked transparently and reproducibly!‚Äù.  Do you know how this reasoning sounds to me?  Like ‚Äúblah blah blah‚Äù ... </p><br><p>  Not that I consider these arguments to be untenable.  I'm just not a very idealist: I don't care what science should be.  I am a realist: I try to do the best that is in my power, based on how science really works.  And, like it or not, science is about career growth, an increase in the impact factor, the number of publications and the amount of money.  More, more, more ... So how does reproducibility help me achieve more as a scientist? </p><br><h2 id="vosproizvodimost-zachem-eto-mne">  Reproducibility: why do I need it? </h2><br><p>  In this article, I give five reasons why this approach to reproducibility pays off in the long term and is in the interests of every ambitious career-oriented scientist. </p><br><h3 id="prichina--1-vosproizvodimost-pomogaet-izbezhat-katastrofy">  Reason # 1: Reproducibility helps avoid catastrophe. </h3><br><p>  ‚ÄúAs a loud promise in cancer testing failed,‚Äù was the title of The New York Times article published in the summer of 2011 [1], which highlights the work of Keith Baggerley and Kevin Coombs, two biostatists of the Cancer Center.  M.D. Anderson.  They identified the problem of analyzing data on deaths in a series of articles by scientists from Duke University, who had a great influence on breast cancer research [2]. </p><br><p>  The problems discovered by Baggerly and Coombs could be easily noticed by any co-author of the article before it was submitted.  The data sets there are not so large, they can be easily checked on a standard laptop.  You do not have to be a statistics genius to understand that the number of patients there varies, labels are reversed, or samples are presented several times with conflicting annotations in the same data set.  Why no one noticed these problems before it was too late?  Because the data and analysis were not transparent and required knowledge of criminal bioinformatics in order to understand them [2]. </p><br><p>  This example motivates me to be more transparent and reproducible in my own work.  Even smaller incidents can embarrass you. </p><br><p>  Here is an example from my research.  Our experimental partners checked the track model we created.  However, when writing this article, we faced a serious obstacle: no matter how hard we tried, we could not reproduce our original travel model.  Perhaps the data has changed, perhaps the code was different, or maybe we just could not correctly remember the settings of the parameters of our method.  If we published this result, we would not be able to demonstrate how we arrived at the approved hypothesis from the original data.  We would have published a miracle. </p><br><p>  This experience showed me two things.  First of all, the project is more than a beautiful result.  You need to describe in detail how this result was obtained. </p><br><p>  And besides, thinking about reproducibility in the early stages, you will save time in the future.  We have spent years of our time and partners time, having failed to reproduce our own results.  All of this could have been avoided if we had been better at tracking how data and analyzes changed over time. </p><br><h3 id="prichina--2-vosproizvodimost-pozvolyaet-legche-pisat-stati">  Reason 2: Reproducibility makes it easier to write articles. </h3><br><p>  Transparency in your analysis greatly simplifies writing articles.  For example, in a dynamic document (Box 1), all results are automatically updated when data changes.  You can be sure that your numbers, graphs and tables will remain relevant.  In addition, this transparency of analysis is more attractive, more people will be able to familiarize themselves with it, and it becomes much easier to detect errors. </p><br><p>  Here is another example from my work.  In another project [3], we discussed with the doctor why some results of survival in a multicenter study did not meet our expectations.  Since all the data and analytical code were available to us in an easily readable file, we were able to study this question ourselves. </p><br><p>  Just by creating a table with a variable describing the stage of the tumor, we were able to identify the problem: we expected to see the numbers of the stages from 1 to 4, and we saw something like ‚ÄúXXX‚Äù, ‚ÄúFred‚Äù and ‚Äú999‚Äù.  The people who provided us with the data seemed to have read them badly.  Studying the data yourself turned out to be much faster and easier than going to the postdoc working on the project and saying, ‚ÄúExplain it to us.‚Äù  My coauthor and I are too busy to waste time on low-level data cleansing, and without a well-documented analysis, we would not be able to contribute.  But since we had very transparent data and code, it took us only five minutes to detect the error. </p><br><h3 id="prichina--3-vosproizvodimost-pomogaet-recenzentam-uvidet-dannye-vashimi-glazami">  Reason # 3: Reproducibility helps reviewers see data through your eyes. </h3><br><p>  Many of us love to complain about reviewing.  Most often, I hear: "The reviewers did not even read the article and have no idea what we actually investigated." </p><br><p>  This contrasts sharply with my experience of reviewing a recent article [4], for which we made the data and documented code easily accessible to reviewers.  One of them suggested making a small change in some of the analyzes, and since he had access to all the data, he was able to directly test his ideas and see how the results changed.  The reviewer was fully involved, and the only thing left to discuss was which method of data analysis would be the best.  So a constructive review should be arranged.  And this would not have been possible without the transparent and reproducible presentation of our analysis. </p><br><h3 id="prichina--4-vosproizvodimost-obespechivaet-nepreryvnost-vashey-raboty">  Reason 4: Reproducibility ensures the continuity of your work. </h3><br><p>  I would be surprised if you had not heard the following remarks before (and maybe even voiced them yourself): ‚ÄúI‚Äôm so busy that I cannot remember the details of all my projects thoroughly‚Äù or ‚ÄúI did this analysis 6 months ago.  Of course, I cannot remember all the details after such a long period ‚Äùor‚Äú My supervisor (PI) said that I should continue the project of the previous postdoc, but that postdoc was long gone and did not save any scripts or data. ‚Äù </p><br><p>  Think about it: all of these problems can be solved by documenting and making data and code available.  This is especially important for leading researchers who are working on complex long-term projects.  How can you ensure continuity of work in your laboratory if the way it is advanced is not documented in reproducible form?  In my group, I do not even discuss the results with students if they are poorly documented.  No evidence of reproducibility - no result! </p><br><h3 id="prichina--5-vosproizvodimost-pomogaet-reputacii">  Reason number 5: Reproducibility helps reputation </h3><br><p>  In several articles, we have made our data, code, and analyzes available in the form of a package for the Bioconductor [5].  When I came to work under a contract, I provided all these packages as the results of research in my laboratory. </p><br><p>  As a rule, the analysis presented in this way helps to create a reputation as an honest and thorough researcher.  If ever there is a problem with one of your articles, it will be very easy for you to protect your name and show that you have reported everything in good faith. </p><br><p>  A recent article published in the journal Science - ‚ÄúScientific Standards.  Promoting an open research culture ‚Äù[6], summarizes eight standards and three levels of recommendations for reproducibility.  Using tools such as R and knitR (Box 1) will allow you to easily follow the highest standards, which, again, is good for your reputation. </p><br>
<h2 id="chto-vas-uderzhivaet">  What is holding you back? </h2><br><p>  I convinced you?  Probably no.  Here is a selection of reactions that I often get when I insist on reproducibility (and how I respond to it): </p><br><ul><li>  ‚ÄúOnly the result matters!‚Äù.  You're wrong. </li><li>  ‚ÄúI would prefer to do real science rather than tidy up my data.‚Äù  If your results are not reproducible, you are not engaged in science at all [7]. </li><li>  ‚ÄúGo about your business!  I document my data as I want! ‚Äù  Yes please!  There are many ways to work so that it is reproducible [8] - you can choose any that you like. </li><li>  ‚ÄúExcel works fine.  I don't need any newfangled R, Python or anything else. ‚Äù  The tool you mentioned may work well if you need to make many manual edits.  But if you perform data analysis, fewer clicks and more scripts are the best solution.  Imagine that you need to do a simple analysis - for example, build a regression graph - 5 (10, 20) times.  Compare the processing of this by hand with the writing of a simple loop that does it for you.  Now imagine that you need to do it again after 3 weeks, because the data has changed a little.  In this case, it‚Äôs definitely worth using R and Python. </li><li>  ‚ÄúReproducibility sounds good, but my code and data are scattered on so many hard drives and directories that it takes too much effort to put everything in one place.‚Äù  Just think what you just said.  Lack of organization puts you and your project in mortal danger. </li><li>  "We can always sort the code and data after submitting an application for consideration."  Above, my example with a track model demonstrates the danger of such a strategy.  In addition, preparing a manuscript may take a long time, so you may not even remember all the details of your analysis when it comes time to present the results. </li><li>  "There is a lot of competition in my research, and spending too much time is too much of a risk."  And that is why you should start working with reproducibility at an early stage in order not to waste this time in the long run. </li></ul><br><h2 id="kogda-nuzhno-bespokoitsya-o-vosproizvodimosti">  When should you worry about reproducibility? </h2><br><p>  Suppose I convinced you that reproducibility and transparency - <br>  in your own interest.  When should we start to worry? <br>  Long answer: </p><br><ul><li>  before the project starts - because you may have to learn tools like R or git .; </li><li>  while you are analyzing - because if you wait too long, you may lose a lot of time trying to remember what you did two months ago; </li><li>  when you are writing an article - because you want your numbers, tables and figures to be relevant; </li><li>  when you are a co-author of the article - because you want to be sure that the analysis presented in the document with your name is correct; </li><li>  when you are viewing a document - because you will not be able to judge the results unless you know how the authors came to them. </li></ul><br><p>  The short answer is: always! </p><br><h2 id="dobitsya-kultury-vosproizvodimosti">  Achieve a culture of reproducibility </h2><br><p>  For whom reproducibility and transparency are important?  It is obvious that students and postdocs play an important role in reproducible work, because most often they are the people who actually do this work.  My advice: learn reproducibility tools as quickly as possible (Box 1) and use them in every project. </p><br><p>  With effort, you get a lot of advantages: </p><br><ul><li>  you will make fewer mistakes and it is easier to correct existing ones; </li><li>  you will be more efficient and will develop much faster in the long run; </li><li>  if you think that your supervisor is not very involved in the work, then by making the analysis more understandable, you can contribute to the fact that the mentor becomes more involved. </li></ul><br><p>  Leading researchers, leaders of groups and teams, professors ‚Äî you follow the creation of a ‚Äúculture of reproducibility‚Äù over the technical base that your students and post-docs represent.  In my laboratory, I made reproducibility a key element in documents that I distribute to beginners [9].  If you want to support your colleagues, request analysis documentation every time a team member shows you the results of your work.  You do not need to go into details - a quick glance will show how well it is made.  What really improved reproducibility in my own laboratory is the requirement that prior to the application being submitted by one member of the team, a colleague who is not participating in the project should try to independently analyze and reproduce our results. </p><br><p>  If you do not create a culture of reproducibility in your laboratory, you will miss the great scientific advantages that it has in the long term. </p><br><p>  Science is becoming more transparent and reproducible every day.  You can become leaders in this process!  Advanced trendsetters!  Come on, I know - you want it too. </p><br><hr><br><p>  <strong>Box 1</strong> </p><br><p>  At the lowest level, working reproducibly means simply avoiding the mistakes of newbies.  Keep your project in an organized state, assign informative names to files and directories, save data and code in one place with backup software.  Do not scatter data across different servers, laptops and hard drives. </p><br><p>  In order to achieve the following levels of reproducibility, you need to study some tools of computational reproducibility [8].  Overall, reproducibility is improved with fewer clicks and inserts and more scripts and coding.  For example, do your analysis in <a href="https://www.r-project.org/">R</a> or <a href="https://www.python.org/">Python</a> and document it with <a href="http://yihui.name/knitr/">knitR</a> or <a href="http://ipython.org/">IPython</a> . <br>  These tools help integrate descriptive text with analytical code into dynamic documents that can be automatically updated each time data or code changes. </p><br><p>  Next, learn how to use a version control system, such as <a href="https://git-scm.com/">git</a> , on a shared platform, such as <a href="https://github.com/">GitHub</a> .  Finally, if you want to become a professional, learn how to use <a href="http://www.docker.com/">dockers</a> who will make your analysis seamless and easily portable to different systems. </p><br><hr><br><h2 id="blagodarnosti">  Thanks </h2><br><p>  I developed a selfish approach to reproducibility for the Postdoctoral Master Class on Reproducibility, which I taught at the Gourdon Institute in Cambridge with Gordon Brown (CRUK Cambridge Institute) and Stephen J. Eglen (DAMTP Cambridge).  I thank them for their input. </p><br><p>  All materials are available on the GitHub <a href="https://github.com/bioinformatics-core-shared-training/rep-research-masterclass">link</a> , and my report is recorded on <a href="https://scientificbsides.wordpress.com/2015/07/15/five-selfish-reasons-for-working-reproducibly">my blog</a> . </p><br><div class="spoiler">  <b class="spoiler_title">List of sources</b> <div class="spoiler_text"><ol><li>  Kolata G. How bright promise in cancer testing fell apart.  The New York Times.  2011. <a href="http://www.nytimes.com/2011/07/08/health/research/08genes.html%3F_r%3D0">http://www.nytimes.com/2011/07/08/health/research/08genes.html?_r=0</a> . </li><li>  Baggerly KA, Coombes KR.  Deriving chemosensitivity from cell lines: forensic bioinformatics and reproducible research in high-throughput biology.  Ann Appl Stat.  2009; 3: 1309‚Äì34. <br>  <a href="https://projecteuclid.org/euclid.aoas/1267453942">https://projecteuclid.org/euclid.aoas/1267453942</a> . </li><li>  Martins FC, Santiago I, Trinh A, Xian J, Guo A, Sayal K, et al.  A combination of high-grade serous ovarian cancer reveals and a prognostic classifier.  Genome Biol.  2014; 15: 526. <br>  <a href="https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0526-8">https://genomebiology.biomedcentral.com/articles/10.1186/s13059-014-0526-8</a> . </li><li>  Schwarz RF, Ng CKY, Cooke SL, Newman S, Temple J, Piskorz AM, et al.  Spatial and temporal heterogeneity in high-grade serous ovarian cancer: a phylogenetic analysis.  PLoS Med.  2015; 12: 1001789. <br>  <a href="http://journals.plos.org/plosmedicine/article%3Fid%3D10.1371/journal.pmed.1001789">http://journals.plos.org/plosmedicine/article?id=10.1371/journal.pmed.1001789</a> . </li><li>  Castro MAA, Fletcher M, Markowetz F, Meyer K. FGFR2 signaling perturbation.  BioConductor Experimental Package.  <a href="http://bioconductor.org/packages/release/data/experiment/html/Fletcher2013a.html">http://bioconductor.org/packages/release/data/experiment/html/Fletcher2013a.html</a> .  Accessed 27 Nov 2015. </li><li>  Nosek BA, Alter G, Banks GC, Borsboom D, Bowman SD, Breckler SJ, et al.  Scientific standards.  Promoting an open research culture.  Science.  2015; 348: 1422‚Äì5. <br>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4550299">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4550299</a> . </li><li>  Watson M. When will'open science 'become simply'science'?  Genome Biol.  2015; 16: 101. </li><li>  Piccolo SR, Lee AB, Frampton MB.  Tools and techniques for computational reproducibility.  2015. <a href="http://biorxiv.org/content/early/2015/07/17/022707">http://biorxiv.org/content/early/2015/07/17/022707</a> .  Accessed 27 Nov 2015. </li><li>  Markowetz F. You are not working for me;  I am working with you.  PLoS Comput Biol.  2015; 11: 1004387. <br>  <a href="http://journals.plos.org/ploscompbiol/article%3Fid%3D10.1371/journal.pcbi.1004387">http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1004387</a> . <br><hr><br><p>  <strong>Twitter and blog</strong> <br>  Florian on Twitter @markowetzlab and on his blog: <a href="http://scientificbsides.wordpress.com/">http://scientificbsides.wordpress.com/</a> . </p></li></ol></div></div></div>
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <p>Source: <a href="https://habr.com/ru/post/417469/">https://habr.com/ru/post/417469/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../417457/index.html">Frontend Conf Moscow - client-side and deep and wide</a></li>
<li><a href="../417459/index.html">Know your JIT: closer to the car</a></li>
<li><a href="../417461/index.html">Ivan Tulup: asynchronism in JS under the hood</a></li>
<li><a href="../417463/index.html">Unexpected meeting. Chapter 17</a></li>
<li><a href="../417465/index.html">Overview of text entry binding methods</a></li>
<li><a href="../417471/index.html">Simple Solder MK936 SMD. DIY soldering station on SMD components</a></li>
<li><a href="../417473/index.html">Secure storage with DRBD9 and Proxmox (Part 1: NFS)</a></li>
<li><a href="../417475/index.html">Glusterfs + erasure coding: when you need a lot, cheap and reliable</a></li>
<li><a href="../417477/index.html">Hot workplace (Hot desking)</a></li>
<li><a href="../417479/index.html">Accelerate do-it-yourself string concatenation</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>