<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Choosing a population size for a genetic algorithm</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="For the soul, experimenting with genetic algorithms. 

 I wanted to share my experience, observations and reflections in a small series of essays. 

 ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Choosing a population size for a genetic algorithm</h1><div class="post__text post__text-html js-mediator-article">  For the soul, experimenting with genetic algorithms. <br><br>  I wanted to share my experience, observations and reflections in a small series of essays. <br><br>  So, the first question over which I thought: what size of the population to choose. <br>  If we formulate this question systemically, it will sound like this: 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      What is the principle, based on what criteria, to calculate the size of the population? <br><a name="habracut"></a><br><br>  To answer this question, it‚Äôs worth deciding what genetic algorithms are: <br>  In my opinion, this is such a street smart kind of brute force algorithms. <br>  (street smart, since the efficiency of using a class of genetic algorithms for finding optimal solutions in systems with an enormous amount of possible combinations of elements is empirically confirmed by billions of years of the evolution of life on Earth) <br><br>  Simplified mechanics of the genetic algorithm is as follows: <br>  - we initiate the initial population - by randomly choosing a subset of possible values ‚Äã‚Äã(chromosome population); <br>  further actually the genetic algorithm itself: <br>  - we calculate for each chromosome from the population the value of the fitness function (which is set in such a way that would characterize the parameter by which the search for the optimal solution is carried out); <br>  - choose the chromosomes that will participate in the crossing (the greater the value of the chromosome fitness function, the more likely it is to take part in the crossing) <br>  - we form the next generation with the help of a genetic crossing operator, i.e.  we carry out pairwise recombination of the original chromosomes (the influence of the mutation operator is aside for the time being) <br>  And so we continue cyclically (more or less consistently improving the population according to the optimized trait) until we find the chromosome with the maximum value. <br><br>  For simplicity, take as an example the chromosome consisting of L genes, in which the alleles of each gene are the values ‚Äã‚Äã{0, 1} <br><br>  Theoretically, chromosome populations may be within [1, 2 ^ L], <br>  where the population size 2 ^ L is the limiting case in which the genetic algorithm degenerates into a brute force algorithm (the situation when the solution is found without the use of genetic operators, since the fitness function is calculated for all possible values); <br>  and the population size 1 is a degenerate case (a random choice of any value and voluntaristly declaring it optimal). <br>  This degenerate case may well be applicable when having any answer (even not optimal) - much better than no answer at all. <br>  By the way, the scope of applicability of such an approach is well characterized by an anecdote: <br>  Chapaev Petka: devices? <br>  Petka: twenty <br>  Chapaev: what is twenty? <br>  Petka: and what devices? <br><br>  Let the size of our chromosome population be N <br>  Since we produce recombinations on a subset, but our goal is to find the optimal value for the whole set, the set of chromosomes must contain all the values ‚Äã‚Äãnecessary for recombining any of the elements of the whole set. <br>  Those.  The population size must be such that for each gene (in our case, the locus), all possible allele values ‚Äã‚Äãare represented in the entire population. <br>  Let me explain with an example: let L = 5, N = 4, and at some point the population looks like this: <br>  10110 <br>  10011 <br>  11,100 <br>  10001 <br>  So: <br>  - the first locus in all four chromosomes is represented only by allele 1; <br>  - the second locus is represented by three values ‚Äã‚Äãof 0 and one value of 1; <br>  - the third locus is represented by two values ‚Äã‚Äãof 0 and two values ‚Äã‚Äãof 1; <br>  - the fourth locus is represented by two values ‚Äã‚Äãof 0 and two values ‚Äã‚Äãof 1; <br>  - the fifth locus is represented by two values ‚Äã‚Äãof 0 and two values ‚Äã‚Äãof 1. <br><br>  Accordingly, if the optimal value is reached for chromosome 01010, then no matter how much the genetic algorithm works (without the use of the mutation operator), this value will never be found. <br><br>  Now let's calculate the probability that a random set of chromosomes will contain all the necessary elements: <br>  The probability that one of the loci will contain in all chromosomes only 1 is equal to (1/2) ^ N <br>  The probability that one of the loci in all chromosomes will contain only 0 is equal to (1/2) ^ N <br>  It is obvious that in all other cases there will be a full set of {0,1} <br>  Thus, the probability for one locus is: 1 - (1/2) ^ N - (1/2) ^ N = 1 - (1/2) ^ (N-1) <br>  Accordingly, the probability that all L loci will contain a full set of alleles each: <br>  P1 = (1 - (1/2) ^ (N-1)) ^ L <br>  Now we express N (the number of chromosomes in the population) through P (the probability that the population will contain all the elements necessary for recombination) and L (the number of loci in the chromosome) <br>  N = 1 + LOG2 (1 / (1-P1 ^ (1 / L))) <br><br>  For example, for L = 100 and P1 = 0.999 (that is, 99.9%) we get the required population size N = 18 (with the total number of possible different chromosomes ~ 10 ^ 30) <br><br>  Note that no matter how large we take the value of P1, the probability that any of the iterations will result in such a set of chromosomes from which any possible chromosome cannot be obtained by recombination is non-zero. <br>  This is where the need for a mutation operator comes into effect. <br>  The fact is that in itself it is rather harmful, since we cross breeding with an eye to the values ‚Äã‚Äãof the fitness function of the crossed chromosomes (i.e., we recombine the best-fit sets) and by this we achieve a population fitness increase, but the mutation operator is absolutely chaotic. <br>  (It is worth noting that in nature damage to chromosomes, which can lead to mutations, is a fairly frequent phenomenon, but effective repair mechanisms for the overwhelming majority of these damages are developed in cells. So, as a result, mutations are a very rare phenomenon and this is not accidental) <br>  But in the ‚Äúhomeopathic doses‚Äù the mutation operator turns out to be useful precisely because of the impossibility of maintaining a set of 100% sufficient for recombinations. <br><br>  The probability that a locus will contain either only 1 for all chromosomes or only 0 is equal to (1/2) ^ (N-1) <br>  We will try to express to some extent the necessary (to restore the lost diversity) probability of mutation P2 <br>  So, P2 is the probability of a single chromosome mutation. <br>  Then the probability of mutation of a particular chromosome locus is P2 / L. <br>  The probability that a given chromosome locus does not mutate is 1 - P2 / L <br>  The probability that this locus does not mutate on any of the chromosomes is (1 - P2 / L) ^ N <br>  Accordingly, the probability that a given locus mutates in at least one of the chromosomes is 1 - (1 - P2 / L) ^ N <br>  For reliability, it is necessary that the probability of a mutation of a given locus in at least one of the chromosomes is an order of magnitude greater than the probability that any of the values ‚Äã‚Äã(0 or 1) are absent in a given locus in all chromosomes. <br>  Moreover, the fact of mutation does not mean that we are dealing with the mutation we need. <br>  Thus, we have the inequality: <br>  1 - (1 - P2 / L) ^ N&gt; 10 * (1/2) ^ (N-1) <br>  Transforming it, we get the following value: <br>  P2&gt; L * (1- (1-10 * (1/2) ^ (N-1)) ^ (1 / N)) <br>  Focusing on the lower limit of inequality, we obtain the following formula for calculating the chromosome mutation probability: <br>  P2 = L * (1- (1-10 * (1/2) ^ (N-1)) ^ (1 / N)) <br>  For N = 18 (calculated by us, based on the values ‚Äã‚Äãof L = 100 and P1 = 99.9%), we get P2 ~ 0.05% <br><br>  Those.  for chromosomes of length L = 100 loci, we obtain the minimum required population size N = 18 and the necessary probability of mutation of one chromosome P2 = 0.04% <br>  For L = 1000, we get N = 22 and P2 = 0.03% <br><br>  <b>Update:</b> <br>  It should be noted that the above calculations of the population size and mutation frequency set only the necessary conditions for the stable performance of the genetic algorithm (that is, if these conditions are not met, the algorithm will be unstable or the algorithm will not work in principle - it will not even reach local optima). <br><br>  To assess the adequacy of conditions, it is necessary to take into account the factor of convergence / rate of convergence to local / global optima (this is the topic of the next essay - I think that in a week - two). </div><p>Source: <a href="https://habr.com/ru/post/114910/">https://habr.com/ru/post/114910/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../114904/index.html">Get feedback from Silicon Valley</a></li>
<li><a href="../114905/index.html">Empowering your google apps for domains account</a></li>
<li><a href="../114906/index.html">Happy Birthday, Pi!</a></li>
<li><a href="../114907/index.html">In Great Britain, the creators of the Gh0stMarket forum were planted</a></li>
<li><a href="../114908/index.html">ABBYY CEO Sergey Andreev answers questions from Habr's readers about the company's participation in the Skolkovo project</a></li>
<li><a href="../114911/index.html">Hash navigation in AJAX sites</a></li>
<li><a href="../114912/index.html">We manage the server via SMS</a></li>
<li><a href="../114913/index.html">Extending jQuery style functionality</a></li>
<li><a href="../114914/index.html">Age of Relevance</a></li>
<li><a href="../114915/index.html">"Beeline" fined 100,000 rubles for sending advertising messages</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>