<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Genome search with Wolfram Language (Mathematica) and HadoopLink</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Translation of the post by Paul-Jean Letourneau " Searching Genomes with Mathematica and HadoopLink ". 
 The code given in the article can be download...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Genome search with Wolfram Language (Mathematica) and HadoopLink</h1><div class="post__text post__text-html js-mediator-article"><div style="text-align:center;"><img src="https://habrastorage.org/files/932/cfb/b58/932cfbb58df74fa1ae2ecad6e2f313d1.png"></div><br>  Translation of the post by Paul-Jean Letourneau " <a href="http://blog.wolfram.com/2014/02/21/searching-genomes-with-mathematica-and-hadooplink/">Searching Genomes with Mathematica and HadoopLink</a> ". <br>  The code given in the article can be downloaded <a href="">here</a> . <hr>  <i>Note: This post is written as a continuation of the post <a href="http://blog.wolfram.com/2013/07/31/mathematica-gets-bigdata-with-hadooplink/">Big Data in Mathematica with HadoopLink</a></i> . <hr>  <b>Translator‚Äôs note</b> : the author of this article refers to the whole <i>genome</i> of a certain structural element of living matter as a gene.  This is somewhat different from standard definitions that are close in meaning, meaning either the entire set of genes of a particular species (Ridley, M. (2006). Genome. New York, NY: Harper Perennial), or the full set of genetic instructions that can be found in cell ( <a href="http://www.genome.gov/Glossary/index.cfm%3Fid%3D90">http://www.genome.gov/Glossary/index.cfm?id=90</a> ).  In this post we will use the presentation of the author. <hr>  In my previous <a href="http://blog.wolfram.com/2013/07/31/mathematica-gets-bigdata-with-hadooplink/">post,</a> I described how to write <b>MapReduce</b> algorithms ( <a href="https://ru.wikipedia.org/wiki/MapReduce">wikis</a> ) in <a href="http://www.wolfram.com/mathematica/"><em>Mathematica</em></a> using the <a href="https://github.com/shadanan/HadoopLink"><em>HadoopLink</em></a> package.  Now let's dig a little deeper and write a more serious <b>MapReduce</b> algorithm. <br><br>  I already wrote earlier about some <a href="http://blog.wolframalpha.com/2010/03/10/did-you-know-that-wolframalpha-knows-your-dna/">entertaining possibilities in the field of genomics in Wolfram | Alpha</a> .  If you're interested, you can even <a href="http://www.wolframalpha.com/examples/GenomicsAndMolecularBiology.html">search the human genome for</a> specific DNA sequences.  Biologists often need to find the location of the DNA fragment that they found in the laboratory to determine which animal the fragment belongs to or from which chromosome.  Let's use <em>HadoopLink</em> to create a genomic search engine! <br><a name="habracut"></a><br>  As before, we load the <b>HadoopLink</b> package: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/9ce/335/9aa/9ce3359aa4c33eed76b2c017f679731d.png" width="167" height="16">
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      And set the link on the main Hadoop node: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/31f/416/65a/31f41665a74f40ee9c73a2b931a46fea.png" width="453" height="119"><br><br>  To illustrate the idea, take the small-sized human mitochondrial genome from <a href="http://reference.wolfram.com/mathematica/ref/GenomeData.html"><b>GenomeData</b></a> : <br><br><img src="https://habrastorage.org/getpro/habr/post_images/aca/5ba/8c3/aca5ba8c3dac04ba08a9f4c33ea64280.png" width="472" height="141"><br><br>  First we divide the genome into separate bases (A, T, C, G): <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ea6/f0a/a30/ea6f0aa30d25932c720ff3de99374b83.png" width="482" height="304"><br><br>  They will be our key-value pairs <b>(k1, v1)</b> .  Values ‚Äã‚Äã- the initial position of each base of the genome: <br><br>  <b>{k1, v1} = {base, position}</b> <br><br>  That is, we have just created an index for the genome. <br><br>  Export this index to the <i>Hadoop Distributed File System</i> (HDFS): <br><br><img src="https://habrastorage.org/getpro/habr/post_images/9da/244/f7f/9da244f7fe9c3d9febd2b742b0bb8212.png" width="482" height="154"><br><br>  For queries, we will use a sequence of 11 bases about which we know that they are contained in the genome: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/449/ba7/032/449ba7032269c0773df8e660af3dcd99.png" width="395" height="129"><br><br>  We use a sequence with a large number of repetitions in it in order to complicate the task of our algorithms. <br><br>  Our genomic search engine for this query should return position 515: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/afe/80c/821/afe80c82196f1d248702e8d1ad28e4ab.png" width="465" height="38"><br><br>  Now we need a <b>mapper</b> and a <b>reducer</b> . <br><br>  As it was said in the <a href="http://blog.wolfram.com/2013/07/31/mathematica-gets-bigdata-with-hadooplink/">first part</a> - <b>mapper</b> takes a key-value pair (step 1), and outputs another pair (step 2): <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ef7/392/506/ef7392506b9e71d630c1e92e93b42553.png" width="455" height="245"><br><br>  <b>The mapper</b> obtains a base from the genome index as an input value and returns key-value pairs for each location in the query, where base indices are found (you will soon find out why): <br><br>  <b>(1) Input:</b> <b>{k1, v1} = {base index, position in the genome}</b> <br><br><img src="https://habrastorage.org/getpro/habr/post_images/5ab/d34/e4c/5abd34e4ce4591834c67c0abd9cab3dd.png" alt="GenomeSearchMapper" title="GenomeSearchMapper" width="460" height="171"><br><br>  The key at the exit is the position in the genome, the value at the exit is the position in the query: <br><br>  <b>(2) Conclusion:</b> <b>{k2, v2} = {position in the genome, position in the query}</b> <br><br>  What is the difference between the position in the genome and the position in the query?  The position in the request is the position of the base in the request, while the position in the genome is the position in the whole genome. <br><br>  For example, let's say <b>mapper</b> gets a key-value pair with base A at position 517: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/fed/e6c/92f/fede6c92fbf4da140a95d5d73294599e.png" width="154" height="58"><br><br>  Positions in the request for base A in the requested GCACACACACA sequence are 3, 5, 7, 9, and 11: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/03f/2c8/692/03f2c8692b5ba9690b81bd52f4138ed3.png" width="318" height="40"><br><br>  Here is a sequence with highlighted positions: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/306/a54/8da/306a548da6e60dd79af7cd4d7a95379f.png" width="445" height="107"><br><br>  <b>Mapper</b> has only one key-value pair with one index base along with the requested sequence.  He does not have the rest of the genome to compare, so he needs to find all possible ways to build a query with base A at position 517: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/30e/e74/58b/30ee7458b835e2266bdcf881efee9713.png" width="331" height="374"><br><br>  Here, the colors correspond to each A in the query (horizontally) with their resulting positions in the genome (vertically).  Take, for example, A in the third base in the query (shown in green).  If it is inserted from A to position 517, then the requested sequence will begin in the genome from position 515 (517 - 3 + 1 = 515) (also shown in green). <br><br>  In the same way, the highlighted red base (fifth position in the query) will force the requested sequence to line up, starting at position 513 in the genome (also shown in red).  And the same for the seventh position in the query with 511 position in the genome, (purple) the ninth position in the query with 509 position in the genome (orange), the eleventh position in the query with 507 position in the genome (brown). <br><br>  Only one of these arrangements is correct.  In this case, for the third position in the request (green color), which allows the requested sequence to integrate into the genome.  But the <b>mapper</b> does not know this, and therefore considers all possible matches. <br><br>  So, the <b>reducer</b> collects the keys, it will collect all the bases that correspond to the same positions in the genome: <br><br>  <b>Input</b> : <b>{k2, {v2 ...}} = {position in the genome, {position in the query ...}}</b> <br><br>  Now for a given position in the genome, the <b>reducer</b> will look for those cases where the values ‚Äã‚Äãform a complete sequence of query positions: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ccc/db1/e00/cccdb1e005a0c48870bf1a498d82d0dd.png" width="405" height="236"><br><br>  If the <b>reducer</b> finds full compliance, it gives the position of the genome: <br><br>  <b>Conclusion</b> : <b>{k3, v3} = {requested sequence, position in the genome}</b> <br><br>  Let's now request from our genomic search engine the sequence GCACACACACA: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/c08/20e/508/c0820e5084b8a0cfeff26d96949ebfaa.png" width="492" height="417"><br><br>  (See <a href="http://blog.wolfram.com/2013/07/31/mathematica-gets-bigdata-with-hadooplink/">part 1 for a</a> description of the <b>HadoopMapReduceJob</b> function). <br><br>  And importing matches to the genome from HDFS: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/cf5/4a9/add/cf54a9adde6e76fa74840a401b3b51f0.png" width="499" height="148"><br><br>  The corresponding position in the genome is 515, and this is the correct answer!  Our genomic search engine works! <br><br>  Let's now perform a search with a query to which two different positions in the genome must correspond: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/c17/442/d71/c17442d71a3e8c0da5b388a479c93ffd.png" width="475" height="58"><br><br>  This request must comply with position 10 and 2277: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/6dd/1c9/58b/6dd1c958b52162193e5cfe2846443915.png" width="501" height="621"><br><br>  Yes, he found both coincidences! <br><br>  Now let's scale it all for the entire human genome.  The first step is to create an index, this time for the entire genome, not just the mitochondrial.  To do this, I downloaded the entire human genome as text files from the <a href="">state server</a> and imported them into HDFS: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/140/686/103/140686103c33b56c5fda7c9464de5bb1.png" width="382" height="34"><br><img src="https://habrastorage.org/getpro/habr/post_images/89b/73b/cb6/89b73bcb627913fd34da47405ec0b2b8.png" width="477" height="629"><br><img src="https://habrastorage.org/getpro/habr/post_images/8bb/d3b/43e/8bbd3b43e6cdf7ba565ed3c269c9b67b.png" width="477" height="359"><br><br>  It turns out one text file per chromosome containing the raw chromosome sequences: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/fa2/b4d/b2e/fa2b4db2ef32f4df9e9d069654e2102d.png" width="504" height="499"><br><br>  Then I applied <b>MapReduce</b> to create key-value pairs in the index on the HDFS, which looks like this: <br><br>  [hs_ref_GRCh37.p13_alts.fa, 121] G <br>  [hs_ref_GRCh37.p13_alts.fa, 122] A <br>  [hs_ref_GRCh37.p13_alts.fa, 123] A <br>  [hs_ref_GRCh37.p13_alts.fa, 124] T <br>  [hs_ref_GRCh37.p13_alts.fa, 125] T <br>  [hs_ref_GRCh37.p13_alts.fa, 126] C <br>  [hs_ref_GRCh37.p13_alts.fa, 127] A <br>  [hs_ref_GRCh37.p13_alts.fa, 128] G <br>  [hs_ref_GRCh37.p13_alts.fa, 129] C <br>  [hs_ref_GRCh37.p13_alts.fa, 130] T <br><br>  One small difference from the previous example is that the key now is the <b>{chromosome, position in the genome}</b> , and the value is the basis on this position.  So now the chromosome will be in the key.  I changed the <b>mapper</b> slightly so that it could work with the new key view: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/ce2/622/8f8/ce26228f86ead80614157e468421b585.png" width="460" height="220"><br><br>  <b>Reducer</b> remained unchanged. <br><br>  Let's run a search for the same sequence: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/aa3/170/bd3/aa3170bd366fdc41198c66531954e4da.png" width="502" height="419"><br><br>  This time we get matches for the whole genome: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/401/aea/3b2/401aea3b20626194ea65575dcee0c948.png" width="503" height="345"><br><br>  And we can still improve the algorithm.  How about searching for inaccurate matches instead of exact ones?  Slightly change the <b>reducer</b> , specifying how much of the request should match: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/1c4/174/c2b/1c4174c2b8688aa330cbecbe16ff880e.png" width="467" height="372"><br><br>  This is not the most efficient way to search the genome, but it does show how easy it is to write prototypes of <b>MapReduce-</b> based algorithms and run them in <em>Mathematica.</em>  If you are interested in the details, I recommend to look at my <a href="http://cufp.org/conference/sessions/2013/paul-jean-letourneau-wolfram-programming-mapreduce">recent posts</a> .  And you will be happy to answer your questions in <a href="https://github.com/shadanan/HadoopLink"><em>HadoopLink</em> GitHub Repo</a> ! </div><p>Source: <a href="https://habr.com/ru/post/263307/">https://habr.com/ru/post/263307/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../263295/index.html">How the transition to the cloud changes the role of the internal IT service</a></li>
<li><a href="../263297/index.html">Online stores deliver correctly using DaData.ru</a></li>
<li><a href="../263301/index.html">Multi-window mode: additional competitive advantage for Android applications</a></li>
<li><a href="../263303/index.html">How to designer to prepare the transfer of Android applications to the developer</a></li>
<li><a href="../263305/index.html">NSA has posted on GitHub utility to ensure network security</a></li>
<li><a href="../263313/index.html">Introduction to the analysis of social networks on the example of VK API</a></li>
<li><a href="../263319/index.html">Turning off the strip above the tabs and dragging a new assembly for the tab bar Vivaldi 1.0.231.3</a></li>
<li><a href="../263321/index.html">EU regulators did not find a violation of the law in the activities of the Hacking Team</a></li>
<li><a href="../263323/index.html">Creating an Excel file from a select with parameters using pure PL / SQL, as an alternative to Oracle * Reports</a></li>
<li><a href="../263325/index.html">How did Pixar solve problems with the filming of "The Puzzle" (The Inside Out)</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>