<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>RNAInSpace and folding of tRNA - season closing, new season - Structural alignment</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="And in less than a year and a half how I got to assemble the tertiary structure of tRNA. Let me remind you that earlier I wrote an article on this top...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>RNAInSpace and folding of tRNA - season closing, new season - Structural alignment</h1><div class="post__text post__text-html js-mediator-article">  And in less than a year and a half how I got to assemble the tertiary structure of tRNA.  Let me remind you that earlier I wrote an article on this topic on the Habr√© <a href="http://habrahabr.ru/post/154521/">"Development of RNAInSpace, CRA algorithm, code problems on Linux and others</a> . <a href="http://habrahabr.ru/post/154521/">"</a>  I have to say that for about a year I didn‚Äôt do this, but during this time my second scientific article on this topic <a href="http://www.jitcs.ru/index.php%3Foption%3Dcom_content%26view%3Darticle%26id%3D414">‚ÄúApplication of game theory for the task of folding ribonucleic acids‚Äù was published</a> (this is for those who want to talk about it professionally).  But recently, I can say that I have received the tertiary structure of tRNA and verified it with the sample available in the database (PDB), which was obtained by biological methods (crystallography). <br><br>  Under the cut there are 3D tRNA structure drawings, explanations and future plans. <br><a name="habracut"></a><br><h4>  Tertiary structure of tRNA - results </h4><br><img src="https://habrastorage.org/getpro/habr/post_images/c83/c02/b80/c83c02b80186f3d85d247e525ddf1fca.png"><br><br><img src="https://habrastorage.org/getpro/habr/post_images/87e/692/2be/87e6922bef8f161208b72edcdb9de592.png">
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      One could make a folding roller, but I was too lazy - it shows very little, as an example of the beginning, you can watch <a href="https://www.youtube.com/watch%3Fv%3DwctCd6Q5_Qg">this</a> , then <a href="https://www.youtube.com/watch%3Fv%3D_ZQgPK5yMw0">this</a> , and then folding turns the chain into tRNA shown in the figure. <br><br>  Figures tRNA from two angles of view.  Green is the model I received, red is the model from the PDB database.  Now I can tell the specialists that RMSD = 6.71 (this is a measure of the similarity of the two models).  As we can see, the overall profile is almost the same.  Also, in my model almost all standard hydrogen bonds are formed and non-canonical hydrogen bonds are close to formation. <br><br>  It should be noted (recall, if someone has already read my articles) that I only get the tertiary model based on the primary structure (the so-called de novo), if it is possible to predict hydrogen bonding sites and find critical stacking sites.  If there is interest, I am ready to explain the details and discuss these results. <br><br><h4>  Season closing </h4><br>  Having brought this direction of my activity to some logical conclusion - in this article I would like to close the series of articles that I wrote on Habr√©.  In essence, I achieved my goals.  And here I will tell about it: <br><br>  1. <a href="http://ru.vlab.wikia.com/wiki/%25D0%25A1%25D0%25B5%25D1%2580%25D0%25B3%25D0%25B5%25D0%25B9_%25D0%25AF%25D0%25BA%25D0%25BE%25D0%25B2%25D0%25BB%25D0%25B5%25D0%25B2:%25D0%25A1%25D1%2582%25D0%25B0%25D1%2582%25D1%258C%25D1%258F:%25D0%2593%25D0%25B5%25D0%25BD%25D0%25BE%25D0%25BC%25D0%25B8%25D0%25BA%25D0%25B0_%25D0%25B1%25D1%2580%25D0%25BE%25D1%2581%25D0%25B0%25D0%25B5%25D1%2582_%25D0%25B2%25D1%258B%25D0%25B7%25D0%25BE%25D0%25B2_%25D0%25B8%25D1%2581%25D0%25BA%25D1%2583%25D1%2581%25D1%2581%25D1%2582%25D0%25B2%25D0%25B5%25D0%25BD%25D0%25BD%25D0%25BE%25D0%25BC%25D1%2583_%25D0%25B8%25D0%25BD%25D1%2582%25D0%25B5%25D0%25BB%25D0%25BB%25D0%25B5%25D0%25BA%25D1%2582%25D1%2583">The first article</a> on the Internet dates back to 2009.  In it, the task of folding is set in the spirit of cybernetic ideas. <br>  2. Next, I tried to develop an open source project <a href="https://ru.wikiversity.org/wiki/RNAInSpace">at Wikiversity</a> . <br><br>  The main thesis was the following: " <i>you can get serious results, knowing only a certain minimum and not having a specialized education either in biology, or physics, or in chemistry</i> ."  Now I have no doubt that I have obtained serious results, and the method I received surpasses all other methods that exist at the moment. <br><br>  So gentlemen, do not be afraid to start, on your way you will meet a lot of opposition and criticism of those who know little about, but are ready to show their erudition.  If results are available, they will have to retreat. <br><br>  3. I had to abandon many modern approaches in this direction, sometimes there was even a feeling that the methods were used not to solve the problem, but to show how a particular method works.  and if at first I pinned my hopes on some methods, including methods of artificial intelligence, it turned out that they are not suitable.  Only the general ideology of game theory and the agent-based approach is suitable.  And so it all comes down to certain heuristics in finding the objective function (of course, if we talk in more detail, there are fine goodies in the algorithms developed by me - but this is not for this article - not the level of immersion in the problematic). <br><br>  4. Two articles in refereed journals are enough for me personally on this topic.  Thanks for attention. <br>  5. In fact, I developed a method and approach, now it's up to technology and followers. <br><br>  6. Next, I come to the question "for what and why"?  About this in the next section. <br><br><h4>  "The difference between the living and the non-living" </h4><br>  Back in the first article, the answer to the question of why studying the three-dimensional structure of RNA was given (this is in addition to being interesting in itself, and may be useful to biologists). <br><br><blockquote>  We have a clear biological task: "To find out exactly what and how much change in the three-dimensional structure of the 50-100 nucleotide chain of RNA fundamentally affects the fact that this chain of RNA is a ribozyme."  In other words, which ribozyme mutations improve or worsen the possibility of self-replication, including their absence.  And by popularizing, this will be the detailed answer to the question of how the living differs from the non-living. </blockquote><br><br>  Of course, looking around is now somewhat naive.  But nevertheless it has a certain meaning.  I will try to explain. <br><br>  Earlier, I have repeatedly pointed out that the modern theory of sequence alignment is essentially erroneous, it allows in essence to customize the results, and not to get the true picture.  I also wrote that annotation in biological bases contains many errors <a href="http://habrahabr.ru/post/146814/">‚ÄúGenomes of sequenced organisms - errors in bases‚Äù</a> , and those who work there were forced to agree with this. <br><br>  Now, looking back, I can say that then, without knowing essentially bioinformatics, in my first article ‚ÄúI made a bet‚Äù on the so-called.  Structural alignment.  This is the finding of genes in the genome, and the subsequent comparison of genomic sequences that takes into account NOT the mutations of individual nucleotides and their statistics, but focuses on the tertiary structure of functionally similar genes. <br><br>  Indeed, now my approach to obtaining a tertiary structure makes it possible to judge whether a certain nucleotide sequence will be able to fold into one or another structure.  This means that it is possible to understand which parts of the nucleotide sequence should be conservative, and in which mutations are possible. <br><br>  All this information, which really affects the ability of the same tRNA to function, or ribozyme or any other RNA structure, is not used in a simple analysis (alignment), which means there will definitely be errors that will not even be noticeable for a researcher who does not pay attention on the functionality of the tertiary structure.  And the statistical approach, which is now commonly used for this, will further obscure this question. <br><br>  And now, when we know (approximately) the tertiary structure - we can build, let's call it - a functional profile, for example, tRNA.  And after that, and only after that - we can find with sufficient accuracy in the DNA the locations of all tRNAs. <br><br>  But to build this functional profile is not so easy.  It turns out we have a few 100% conservative areas - almost everything can change in absolute terms.  To understand this, consider the example of tRNA. <br><br>  So let's compare two tRNAs: <br><br>  gcgcggauagcucagucgguagagcaggggauugaaaauccccguguccuugguucgauuccgaguccgcgc <br>  gcggauuuagcucaguugggagagcgccagacugaagucuggagguccuguguucgauccacagaauucgca <br><br>  try to align these two tRNAs and say how are they different?  In reality, the problem is much worse - these sequences are not highlighted, as in this example - they are among millions of similar gcau signs.  And we do not know where we need tRNA. <br><br>  You can of course engage in nonsense and align these signs, making assumptions where the breaks and where the inserts occurred during the mutation. <br><br>  But you can do it easier, let's find hydrogen bonds, for a start, at least classical ones.  We get: <br><br>  (((((((.. ((((........)))). (((((((...)))) () (((.......)))))))))))) <br>  (((((((.. ((((........)))). ((((((....))))) ..... ((( ((.......)))))))))))). <br><br>  Isn‚Äôt it becoming more fun?  It turns out the difference is not so big.  It is necessary to make tolerances on plus minus 1-3 points (unpaired nucleotides) and 1-3 pairs of brackets (paired with a hydrogen bond nucleotides).  For greater accuracy, it will be possible to find a correspondence of non-canonical hydrogen bonds (which stabilize the structure at the 3D level). <br><br>  Of course, it is still difficult to find these structures among the millions of gcau signs.  But here there is a landmark.  We divide the task into parts, and look for not all the tRNAs, but those that bring phenylalanine.  And if so we know for sure that in the center is the sequence gaa.  Then we can search for all such sequences in the genome, which has gaa in the middle, and also has a corresponding profile: <br><br>  (((((((.. ((((........)))). ((((((((gaa)))))) ..... ((((( (.......)))))))))))) <br>  (((((((.. ((((........)))). ((((((gaa.))))) ..... (((((( .......)))))))))))). <br><br>  With permissible limits in the structure. <br><br>  That's what I'm going to do in the near future - reliably find all the tRNAs in the sequenced genomes of bacteria.  Maybe someone wants to participate in this - I invite. </div><p>Source: <a href="https://habr.com/ru/post/230615/">https://habr.com/ru/post/230615/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../230601/index.html">Book about venus</a></li>
<li><a href="../230603/index.html">No worries, no hassle: What you need for successful store automation</a></li>
<li><a href="../230605/index.html">Controls of CIA</a></li>
<li><a href="../230607/index.html">Modern Tornado: distributed image hosting in 30 lines of code</a></li>
<li><a href="../230613/index.html">What must a startup do to survive?</a></li>
<li><a href="../230617/index.html">Interview with the demostsener - kb ^ Farbrausch</a></li>
<li><a href="../230619/index.html">Meta-Object Protocol Common Lisp on the example of the implementation of the prototype object system</a></li>
<li><a href="../230623/index.html">7 myths about Linq to Database</a></li>
<li><a href="../230625/index.html">View DOM events in Firefox Developer Tools</a></li>
<li><a href="../230627/index.html">Google uses machine learning to improve data center efficiency</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>