<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>An introduction to genomics for programmers</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="About the author. Andy Thomason is the lead programmer for Genomics PLC . He has been working with graphic systems, games and compilers since the 70s;...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>An introduction to genomics for programmers</h1><div class="post__text post__text-html js-mediator-article">  <font color="gray">About the author.</font>  <font color="gray">Andy Thomason is the lead programmer for <a href="https://www.genomicsplc.com/">Genomics PLC</a> .</font>  <font color="gray">He has been working with graphic systems, games and compilers since the 70s;</font>  <font color="gray">specialization - code performance.</font> <br><br><h4>  Genes: a brief introduction </h4><br>  <a href="https://en.wikipedia.org/wiki/Human_genome">The human genome</a> consists of two copies of approximately 3 billion base pairs of DNA, which are encoded using the letters A, C, G and T. This is about two bits for each base pair: <br><br>  3,000,000,000 √ó 2 √ó 2/8 = 1,500,000,000 or about 1.5 GB of data. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      In fact, these copies are very similar, and the DNA of all people is almost the same: from merchants from Wall Street to Australian Aborigines. <br><br>  There are a number of ‚Äúreference genomes,‚Äù such as <a href="">Ensembl Fasta files</a> .  Reference genomes help build a map with specific characteristics that are present in human DNA, but not unique to specific people. <br><a name="habracut"></a><br>  For example, we can determine the ‚Äúlocation‚Äù of a gene that encodes a BRCA2 protein that is responsible for DNA repair in breast cancer: <a href="https://www.ensembl.org/Homo_sapiens/Gene/Summary%3Fdb%3Dcore%3Bg%3DENSG00000139618%3B%2520r%3D13:%252032315474-32400266">this gene</a> . <br><br>  It is located on chromosome 13, from position 32315474 to 32400266. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/webt/ab/nu/r-/abnur-al6qu8cteb53sienjn-d4.png" width="550"></div><br><br><h4>  Genetic variations </h4><br>  People are so similar that for the representation of a person it is usually enough to store a small set of ‚Äúvariations‚Äù. <br><br>  Over time, our DNA is damaged by cosmic rays and copying errors, so the DNA that parents transmit to children is slightly different from their own. <br><br>  <a href="https://ru.wikipedia.org/wiki/%25D0%25A0%25D0%25B5%25D0%25BA%25D0%25BE%25D0%25BC%25D0%25B1%25D0%25B8%25D0%25BD%25D0%25B0%25D1%2586%25D0%25B8%25D1%258F_(%25D0%25B1%25D0%25B8%25D0%25BE%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B8%25D1%258F)">The recombination</a> mixes the genes even more, so the child‚Äôs DNA inherits from each parent a mixture of the grandparents' DNA from this side. <br><br>  Therefore, for each change in our DNA, it is enough to keep only the differences from the reference genome.  Usually they are saved in a VCF (Variant Call Format) file. <br><br>  Like almost all files in bioinformatics, it is a file of type TSV (text format with tabs). <br><br>  You can get your own VCF file from companies like <i>23 and Me</i> and <i>Ancestry.com</i> : pay relatively little money and send a sample that is sequenced on a DNA microchip.  It highlights fragments where DNA corresponds to the expected sequences. <br><br>  An abbreviated example of <a href="http://www.internationalgenome.org/wiki/Analysis/Variant%2520Call%2520Format/vcf-variant-call-format-version-40/">the VCF specifications</a> : <br><br><pre>  ## fileDate = 20090805
 ## source = myImputationProgramV3.1
 ## reference = 1000GenomesPilot-NCBI36
 ## phasing = partial
 #CHROM POS ID REF ALT QUAL FILTER INFO FORMAT NA00001 NA00002 NA00003
 20 14370 rs6054257 GA 29 PASS NS = 3; DP = 14; AF = 0.5; DB; H2 GT: GQ: DP: HQ 0 | 0: 48: 1: 51.51 1 | 0: 48: 8: 51.51 1/1: 43: 5:.,. </pre><br>  Here we have three people named NA00001, NA00002 and NA00003 (we are very serious about the security of personal data in the world of genetics), who in position 14370 of chromosome 20 have certain differences <code>0|0</code> , <code>1|0</code> and <code>1|1</code> from G to A. <br><br>  There are two numbers per person, since we all have two copies of chromosome 20 (one from each parent; the only exception is the sex chromosome).  I was not lucky that I have only one X chromosome, so I inherited color blindness from my grandfather through my mother). <br><br>  Such options are possible: <br><br><pre>  0 | 0 both chromosomes correspond to the reference sample
 1 | 0 and 0 | 1 only one chromosome is different from the standard
 1 | 1 both chromosomes are different from the standard </pre><br>  VCF files are considered ‚Äúphased‚Äù if you can figure out which particular chromosome the variant is on or at least where it is located relative to its neighbors.  In practice, it is difficult to say which chromosome DNA came from, so you have to speculate! <br><br>  Thus, we have a bit vector <code>001011</code> , which is enough to classify three people in this variation.  These are <i>haplotypes</i> or variations of individual chromosomes. <br><br><h4>  GWAS research </h4><br>  Using this bit vector, we can try to figure out which parts of the genome affect diseases or other individual properties, such as hair color or growth.  For each option, we build a haplotype for the measured traits ( <i>phenotype</i> ). <br><br>  GWAS (Genome wide association study, polygenomic search for associations) is the basis for genetic analysis of variants.  He compares variations with observational data. <br><br>  For example: <br><br><pre>  Haplotype Height Person
 0 1.5m NA00001
 0 1.5m
 1 1.75m NA00002
 0 1.75m
 1 1.95m NA00003
 1 1.95m </pre><br>  Please note that each has two haplotypes, because we have a pair of chromosomes. <br><br>  Here we see that options 1 are associated with higher growth, and the values ‚Äã‚Äãcorrespond to linear regression: <br><br><pre>  beta Growth variation with variation variation.
 standard error Error rate. </pre><br>  In practice, the data is really a lot of noise, and the error is usually larger than the <code>beta</code> , but often we have several options, where <code>beta</code> much higher than the error.  This relationship ‚Äî the <i>Z-score</i> and its associated <i>p-value</i> ‚Äî indicates which options are most likely to affect growth. <br><br>  The easiest way to regress is to apply the <a href="https://en.wikipedia.org/wiki/Moore%25E2%2580%2593Penrose_inverse">Moore-Penrose inversion</a> . <br><br>  We compose a 2 √ó 2 covariance matrix with the scalar product of two vectors, and we solve the problem using the least squares method. <br><br>  We have trillions of data points, so it‚Äôs important to do this effectively. <br><br><h4>  Curse of disequilibrium clutch </h4><br>  Since we inherit large fragments of the genome from our parents, certain areas of DNA look very similar: they are much more similar than the case dictates. <br><br>  This is good for us, because genes continue to work in the same way as their ancestors, but badly for genomics researchers.  This means that the differences are not enough to determine the variations that caused a change in the phenotype. <br><br>  <a href="https://en.wikipedia.org/wiki/Linkage_disequilibrium">Non-equilibrium adhesion</a> (LD) determines how similar the two vectors are with variations. <br><br>  It calculates a value between -1 and 1, where <br><br><pre>  -1 Completely opposite variations.
  0 Variations are not similar.
  1 The variations are exactly the same. </pre><br>  To determine the similarity of variations, we create large square LD matrices for specific places in the genome.  In practice, many of the variations around this place are almost identical to the variant in the middle. <br><br>  The matrix looks something like this, with large squares of similarity. <br><br><pre>  v0 v2 v4 v6 v8 va vc ve vg
       v1 v3 v5 v7 v9 vb vd vf
 v0 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0 
 v1 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v2 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v3 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v4 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v5 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v6 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v7 1 1 1 1 1 1 1 1 0 0 0 0 0 0 0 0
 v8 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 v9 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 va 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 vb 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 vc 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 vd 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 ve 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 vf 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1
 vg 0 0 0 0 0 0 0 0 1 1 1 1 1 1 1 1 </pre><br>  Real values ‚Äã‚Äãare not 0 or 1, but very similar. <br><br>  A recombination occurred between v7 and v8.  Because of this, v0..v7 is different from v8..vg. <br><br>  The problem of similarity is that we know that one of the options in the group caused something, but we do not know which one. <br><br>  This limits the resolution of our <i>genomic microscope</i> , and to solve the problem will have to use additional methods such as functional genomics. <br><br><h4>  Conclusion </h4><br>  In the end, you can never be 100% sure which particular region of the genome caused a particular individual feature, this is the essence of genetics.  Biology is not an exact machine with perfect factory-made parts.  This is a boiling mass of accidents that somehow create what we call life.  That is why statistics, or ‚Äúmachine learning,‚Äù is so important, as it is now fashionable to call it. </div><p>Source: <a href="https://habr.com/ru/post/452622/">https://habr.com/ru/post/452622/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../45261/index.html">Venture Dating Service</a></li>
<li><a href="../452610/index.html">Help and request for it. Article about information security for ordinary users</a></li>
<li><a href="../452612/index.html">Machine learning of a deep neural network with reinforcement at tensorflow.js: stunts</a></li>
<li><a href="../452614/index.html">How to start programming in Adobe Illustrator. Part two</a></li>
<li><a href="../452618/index.html">What did Google I / O 2019 talk about: Android 10, AR apps and more</a></li>
<li><a href="../452624/index.html">Introduction to Spring Boot Actuator</a></li>
<li><a href="../452636/index.html">‚ÄúI am inevitability myself‚Äù: how do ecosystems appear and what can we expect from them</a></li>
<li><a href="../452638/index.html">How we launched 2GIS under CarPlay and are still unraveling</a></li>
<li><a href="../45264/index.html">SearchWiki</a></li>
<li><a href="../452644/index.html">DevConf X Web Developers Conference - June 21</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>