<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Practical bioinformatics p.4. Getting ready to work with ZINBA</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In the modern world of data analysis, using only one method or only one approach means that sooner or later you will be confronted with the fact that ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Practical bioinformatics p.4. Getting ready to work with ZINBA</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/storage2/b34/e64/eb8/b34e64eb8396c1d0527876ef846d34f3.png" alt="bioinformatic" align="left">  In the modern world of data analysis, using only one method or only one approach means that sooner or later you will be confronted with the fact that you were wrong.  To analyze the data, various methods are combined, they compare the result and, based on the comparison, they are already making more accurate predictions.  ZINBA uses exactly this approach.  The developers combined a variety of methods for analyzing DNA-seq experiments in a single package.  This package is written for the R statistical software. So what does ZINBA do?  It finds various enriched regions even in cases where some of them have been enhanced, for example, chemically or have a different degree of signal-to-noise ratio. <br><a name="habracut"></a><br>  At first I expected to make an overview article about the <a href="http://genomebiology.com/2011/12/7/R67">ZINBA</a> software product, but the more I read about the methods used in it, the deeper I dig into the algorithms and definitions.  And when I further summarized my knowledge, I realized the fact that more than one article had already accumulated data, and without introducing it, it would be difficult to touch the subject matter.  In this topic, I provide brief excerpts from the articles mentioned in the description of the ZINBA program, complementing them with my own comments.  I will wait for your comments to get to the truth together. <br>  Far-reaching plans for regulatory biology are to learn how the genome encodes a variety of gene expression mechanisms.  The link between coding and these mechanisms is found due to the possibility of finding protein binding sites throughout the genome using chromatin immunoprecipitation ( <a href="http://en.wikipedia.org/wiki/Chromatin_immunoprecipitation">ChIP</a> ) and gene expression (RNA-seq).  Going for evidence is far from necessary, there is the project Encyclopedia of DNA Elements (ENCODE [1]), within which most of the functional elements of the genome are recognized.  The first steps were made with the help of the microchip (‚Äúmicroarray‚Äù) method, but deep DNA sequencing (ChIP-seq &amp; DNA-seq, sometimes the term NGS next generation sequencing is found) caught up and overtook this technology. <br>  RNA-seq and ChIP-seq have the following positive differences: they increase the accuracy, specificity, sensitivity of measurements and allow you to view the entire genome at once.  In general, RNA-seq and ChIP-seq have some similarities with such a predecessor as a microchip, but particulars differ greatly.  But sequencing still suffers from a number of difficulties, although it is much lighter than the microchip in these matters.  Due to the fact that the cut DNA fragments are rather short, there is a possibility that the fragments may occur several times in the genome, be parts of the repeatable (repetitive) part of the DNA, or turn out to be common fragments for some start sites. <br>  It is believed that data processing should consider the ChIP reaction as enrichment.  Because approximately 60-99% of the fragments in the prepared DNA solution are usually the background, and the remaining 1-40% belong to the fragments, the protein of which could be bound ( <a href="http://en.wikipedia.org/wiki/Cross-link">cross-link</a> ) with the antibody [2].  It is this smallest part of the solution that will be sequenced.  The fact that the level of contamination is high must be taken into account when calculating the enrichment.  The number of reads per genome in some cases is incomparably small, for example, for the mammalian genome, the number of reads is less than 1% of the total genome length.  The data obtained by sequencing require, obviously, new algorithms and software. <br><br><img src="https://habrastorage.org/storage2/51a/900/a62/51a900a6282ac436478672f4f0d8cddb.png" align="left">  Fig.  one <br><br>  This slide schematically displays the hierarchy of ChIP-seq and RNA-seq analysis.  Usually the analysis goes through all the stages from the bottom up.  Different programs are applied at each level of the scheme and are sometimes divided into programs specifically for ChIP-seq and RNA-seq.  The output of one program is the input to the other (pipeline).  As can be seen from the diagram, all programs first pass through the definition of reads on the genome (Maps read), with the exception of the assembly of transcripts, the result of which are the alleged <a href="http://en.wikipedia.org/wiki/Transcriptome">transcriptomes</a> [3].  After finding the reads on the genome, the output of such a program is transferred to the input of the program for data collection and identification (Aggregate and identity).  For example, to determine enriched regions or read densities on a known annotation.  Also, these programs may include some subsequent analysis of both binding sites (for DNA-seq) and new unexplored fragments of transcription-isoformism (for ChIP-seq).  Further analysis of a higher level.  It includes the definition of a <a href="http://en.wikipedia.org/wiki/Sequence_motif">conservative motif</a> or expression level, the result of which could be the discovery of a new gene model.  The level of integration (Integrate) shows that previously obtained results can be used for current data processing (not necessarily in the framework of the current experiment, laboratory or country). <br>  So what do we get after the reads were found on the genome?  How do these data look and how do they differ from each other?  There are sites on DNA (binding sites) to which a particular protein can attach; we will call these sites binding sites or simply sites.  For example, a CTCF site means that the <a href="http://en.wikipedia.org/wiki/CTCF">CTCF</a> protein can gain a foothold in this region.  Scientists have found an antibody to CTCF protein and with the help of antibodies were able to precipitate the corresponding sections of DNA.  But, as we already know, arbitrary ones can be attached to specific antibodies along with special sections.  The rest of the methods work in a similar way: the protein is fixed on DNA, the antibody is fixed on the protein, only a sample of DNA segments is changing.  Thus, the specificity of the selected areas leaves its mark, and the design landscape of each method will have its own: somewhere narrow peaks, as in the CTCF method, the graph of a;  somewhere steep peaks with a large adjacent territory, for example, RNA polymerase II method, graph b;  during the deposition of the histone modification, a continuous ‚Äúfringe‚Äù is obtained; this is due to the peculiarity of the arrangement of the histones, figure c.  On the graphs, the coordinate on the chromosome is plotted along the axis x, and the density of reads along the axis y.  The data for the figure are taken from [4]. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <img src="https://habrastorage.org/storage2/a18/5f5/7bf/a185f57bf97a494a167b3d275c33abe0.png" align="left">  Fig.  2 <br><br>  The presence of the background forces the algorithms to try to evaluate it empirically based on the ‚Äúcontrol‚Äù.  The control is an additional experiment carried out with the initial solution with special (preimmune) antibodies, with which nothing should be connected, but still randomly connected.  As a result of this connection, some arbitrary background is obtained.  Some algorithms model a possible background, based only on the data, without control.  Whichever approach is chosen (with control and without control), the distribution of background reads cannot be unified until the end, since the background level depends on both the type of cells and the tissue from which they are obtained, and the method of sedimentation, etc. .  The technology used for amplification (PCR) and sequencing can also add artificial enrichments (artifacts) or enhance one region more than another.  There are algorithms that allow you to find enriched areas for each experiment.  Although each of the algorithms is tailored to the data corresponding to the experiment, the assumption underlying the algorithm does not always fit the possible set of enriched areas found using DNA-seq [2]. <br>  Below, for comparison of data, graphs from a more recent article on ZIMBA.  Similar graphs are blurred due to the difference in scale between the first and second pictures, but they look similar to the above explanation about the DNA-seq technology (see Figure 2). <br><br><img src="https://habrastorage.org/storage2/667/b5b/86d/667b5b86df949970a7d7025619279e39.png"><br>  Fig.  3 <br><br>  In Figure 3, in addition to the previous graphs, the landscapes of the following DNA-seq experiments are <a href="http://ru.wikipedia.org/wiki/%25D0%25A0%25D0%25B5%25D1%2581%25D1%2582%25D1%2580%25D0%25B8%25D0%25BA%25D1%2582%25D0%25B0%25D0%25B7%25D1%258B">displayed</a> : DNA-sensitive sample (DNase-seq) [5] and protein isolation / fixation with formaldehyde (FAIRE-seq) [6]. <br>  DNA is a double helix.  This phrase itself is popular, but very often it is forgotten.  What does it mean in our case?  And the fact that the program that will search for the reads on the genome will give out, along with the coordinates on the chromosome, and from which ‚Äúside‚Äù of the helix (strand) this read is located.  Spirals are divided into positive (sense) and nonpositive (nonsense / antisense).  The mechanism for obtaining reads is such that most reads will be located on two different sides with respect to the protein bound to DNA.  And in most cases, the reads will be located from the 5 'end of the corresponding helix.  Therefore, reads from both sides will tend to the center of the protein binding site.  Figure 4 depicts this schematically, red dots are reads from the 5 'end, and in the center is a protein bound to DNA. <br><br><img src="https://habrastorage.org/storage2/ce2/607/983/ce2607983a5a0d685b2e6f3797650135.png"><br>  Fig.  four <br><br>  Thus, to find the center of the protein binding site with DNA, we construct the following graph.  To do this, we postpone the difference between the number of reads from the positive and negative sides along the ‚Äúy‚Äù axis, and postpone the read coordinate corresponding to the chromosome along the ‚Äúx‚Äù axis.  In Figure 5, reads are displayed on the positive side in red, and on the negative side - in blue.  The resulting graph crosses zero very close to the center of the protein binding site (see the lower graph in Fig. 5).  There are algorithms that, up to a nucleotide, determine binding sites. <br><img src="https://habrastorage.org/storage2/d75/ba7/4da/d75ba74da05c2736259bde9ad2a8034b.png"><br>  Fig.  five <br><br>  In this topic, I tried to briefly introduce the material that is the object of study.  The differences in the landscapes of the corresponding DNA-seq experiments were shown, and the method of approximate finding the center of the protein binding site was explained.  There is a lot of work ahead: we need to understand the differences between the algorithms depending on the landscape, learn how to find peaks and try to distinguish them from artificial ones, find out which regions of the genome lying in the vicinity of the peak can be considered enriched, and all this in order to start thinking about ZINBA . <br><br>  1. Birney, E., et al., Identification and analysis of the functional elements in the 1% of the human genome by the ENCODE pilot project.  Nature, 2007. 447 (7146): p.  799-816. <br>  2. Pepke, S., B. Wold, and A. Mortazavi, Computation for ChIP-seq and RNA-seq studies.  Nat Methods, 2009. 6 (11 Suppl): p.  S22-32. <br>  3. Barski, A. and K. Zhao, Genomic location analysis by ChIP-Seq.  J Cell Biochem, 2009. 107 (1): p.  11-8. <br>  4. Barski, A., et al., High-resolution profiling of the human genome.  Cell, 2007. 129 (4): p.  823-37. <br>  5. Boyle, AP and TS Furey, High-resolution mapping studies and gene regulatory elements.  Epigenomics, 2009. 1 (2): p.  319-329. <br>  6. Giresi, PG and JD Lieb, Isolation of active regulatory elements from eukaryotic chromatin using FAIRE (Formaldehyde Assisted Isolation of Regulatory Elements).  Methods, 2009. 48 (3): p.  233-9. <br><br>  Review by Andrey Kartashov, Cincinnati, OH, porter@porter.st <br>  corrected by Ekaterina Morozova, ekaterina@porter.st. </div><p>Source: <a href="https://habr.com/ru/post/137453/">https://habr.com/ru/post/137453/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137444/index.html">Mechanical display on the Arduino - work on the bugs</a></li>
<li><a href="../137445/index.html">Yandex.Maps are now able to build routes by public transport in Kiev</a></li>
<li><a href="../137446/index.html">Metaprogramming</a></li>
<li><a href="../137449/index.html">PROhq Online Phone - New Freelance Tool</a></li>
<li><a href="../137451/index.html">Draw three-dimensional icons</a></li>
<li><a href="../137454/index.html">MySQL in NGINX: using blocking libraries in a non-blocking server</a></li>
<li><a href="../137456/index.html">FreeBSD and D-Link DI-804HV over IPSEC</a></li>
<li><a href="../137457/index.html">The gov.uk platform will be completely open. Source codes published on GitHub</a></li>
<li><a href="../137458/index.html">Why don't we make personal computers anymore?</a></li>
<li><a href="../137459/index.html">PHP deobfuscation</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>