<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>New approach to building the tree of life</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="When the British morphologist George Jackson Mivart [St. George Jackson Mivart] published in 1865 one of the first evolutionary trees, he lacked suppo...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>New approach to building the tree of life</h1><div class="post__text post__text-html js-mediator-article"><div style="text-align:center;"><img src="https://habrastorage.org/getpro/geektimes/post_images/08b/36f/728/08b36f72893eda0433440b91eed43be2.jpg" alt="image"></div><br>  When the British morphologist George Jackson Mivart [St.  George Jackson Mivart] published in 1865 one of the first evolutionary trees, he lacked supporting material.  He built a tree - a <a href="http://biodiversitylibrary.org/item/90412">branching map of</a> various species of primates - with the help of a detailed analysis of the spines of animals.  The second tree, created on the basis of comparison of limbs of animals, <a href="http://phylonetworks.blogspot.com/2012/08/who-published-first-phylogenetic-tree.html">showed other related connections</a> between primates, highlighting the problem of evolutionary biology that still exists today. <br><br>  Almost 150 years later, scientists acquired a mountain of data to build the so-called <a href="https://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B8%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B5%25D0%25BD%25D0%25B5%25D1%2582%25D0%25B8%25D1%2587%25D0%25B5%25D1%2581%25D0%25BA%25D0%25BE%25D0%25B5_%25D0%25B4%25D0%25B5%25D1%2580%25D0%25B5%25D0%25B2%25D0%25BE">phylogenetic trees</a> , a modern version of the structure created by Mivart.  Advances in DNA decoding technology and <a href="https://ru.wikipedia.org/wiki/%25D0%2591%25D0%25B8%25D0%25BE%25D0%25B8%25D0%25BD%25D1%2584%25D0%25BE%25D1%2580%25D0%25BC%25D0%25B0%25D1%2582%25D0%25B8%25D0%25BA%25D0%25B0">bioinformatics</a> make it possible to compare sequences of hundreds of genes, and sometimes whole genomes, in different species, and create the tree of life with more detail than has ever been possible. <br><a name="habracut"></a><br><img src="https://habrastorage.org/getpro/geektimes/post_images/56f/825/c4e/56f825c4e33c040a9abb10cc327d7e2f.jpg"><br>  <i>The historic tree of life from 1866 describes the kingdoms of plants, animals and single-celled</i> <br><br>  But although the abundance of data has helped to resolve some of the conflicts that have arisen over various sections of the evolutionary tree, it has brought new difficulties.  The current version of the tree of life looks more like a controversial Wikipedia page than a published book ‚Äî there are constant controversies about some of the branches.  Just as the spine and limbs have led to the emergence of contradictory primate evolution maps, scientists now know that different genes in the same organism can tell different stories. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      According to a new study, partly based on the study of yeast, the controversial picture drawn by individual genes is even more controversial than expected.  "They claim that each of the 1070 genes is involved in a conflict," says Michael <a href="http://www.yale.edu/eeb/donoghue/">Donoghue</a> , an evolutionary biologist at Yale, who is not associated with the study.  ‚ÄúWe are trying to understand the phylogenetic relationships of 1.8 million species, and we ourselves cannot sort out twenty types of yeast,‚Äù he says. <br><br>  To resolve the paradox, researchers developed an algorithm based on information theory to measure the level of confidence in the correctness of individual parts of the tree.  They hope that the new approach will help bring clarity to the periods of evolution, possessing both the most interesting and useful and the most conflicting data - for example, the <a href="https://ru.wikipedia.org/wiki/%25D0%259A%25D0%25B5%25D0%25BC%25D0%25B1%25D1%2580%25D0%25B8%25D0%25B9%25D1%2581%25D0%25BA%25D0%25B8%25D0%25B9_%25D0%25B2%25D0%25B7%25D1%2580%25D1%258B%25D0%25B2">Cambrian explosion</a> - the rapid diversification of animal life that occurred 540 million years ago. <br><br>  ‚ÄúHistorically, the most interesting episodes are connected with the areas that attracted attention and caused controversy,‚Äù such as the origin of animals, vertebrates, and flowering plants, says <a href="http://as.vanderbilt.edu/rokaslab/">Anthony Rokas</a> , a biologist at Vanderbilt University who led the new study. <br><br>  Based on the results of the new algorithm, scientists can choose only the most informative genes for the construction of phylogenetic trees.  Such an approach can make the process both more accurate and efficient.  ‚ÄúI think it will help speed up the reconstruction of the tree of life,‚Äù says Khidir Hilu, a biologist at the Virginia Institute of Technology. <br><br><h2>  Bricks of life </h2><br>  The basis of phylogenetic trees is created through the grouping of species according to their degree of kinship.  If we compare the DNA of humans, chimpanzees and fish, it becomes clear that humans and chimpanzees are closer to each other than to fish. <br><br>  Researchers once used one or more genes to compare organisms.  But in the last decade there has been an explosion of phylogenetic data that very quickly filled the bases necessary for the creation of these trees.  The analysis made it possible to fill in a few of the white spots scattered around the tree, but serious disagreements still exist. <br><br>  For example, it is still unclear whether snails are closer in relation to them - bivalve mollusks or <a href="https://ru.wikipedia.org/wiki/%25D0%259B%25D0%25BE%25D0%25BF%25D0%25B0%25D1%2582%25D0%25BE%25D0%25BD%25D0%25BE%25D0%25B3%25D0%25B8%25D0%25B5">shovel-legged mollusks</a> , Rokas says.  It is not known how exactly one of the earliest branches of animals from a tree are connected, such as jellyfish and sponges.  Scientists can demonstrate examples of conflicting trees that appear in the same scientific journals with a difference of weeks, or <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1637082/">even in the same issue</a> . <br><br>  ‚ÄúHence the question: why is it so difficult for us to come to an agreement?‚Äù Rokas says. <br><br><img src="https://habrastorage.org/getpro/geektimes/post_images/9be/085/c50/9be085c50ffa540c76d0b31b9bd33a4e.jpg"><br><br>  Rokas and his graduate student Leonidas Salichos studied this issue by <a href="http://www.ncbi.nlm.nih.gov/pubmed/23657258">evaluating the genes separately</a> , using the most useful genes ‚Äî carrying the most information related to evolutionary history ‚Äî to build their version of the tree. <br><br>  They started with 23 yeast species and selected 1070 genes.  To begin with, they created a phylogenetic tree in a standard way, by concatenation.  To do this, all the sequences from individual species come together in one megagene, and then the sequences of individual species are compared with this long sequence, on the basis of which a tree is created that best explains the differences. <br><br>  The resulting tree is accurate in terms of standard statistical analysis.  But since similar methods lead to the appearance of trees full of disagreements, Rokas and Salichos decided to delve into the subject.  They built sets of phylogenetic trees for individual yeast genes, and applied an algorithm, developed using information theory, to search for the areas of greatest fit between different trees.  The result, <a href="http://www.ncbi.nlm.nih.gov/pubmed/23657258">published in the journal Nature in May</a> , was unexpected.  It seems that each gene studied tells a slightly different evolutionary history. <br><br>  ‚ÄúVirtually all trees built for individual genes were in conflict with a tree based on a concatenation of data,‚Äù says Hilu.  ‚ÄúThis is shocking.‚Äù <br><br>  They concluded that if several genes support a particular architecture, then it must be exact.  But if different sets of genes equally support two different architectures, then the probability of their exact correspondence to reality decreases.  Rokas and Salichos used a method called <a href="https://ru.wikipedia.org/wiki/%25D0%25A1%25D1%2582%25D0%25B0%25D1%2582%25D0%25B8%25D1%2581%25D1%2582%25D0%25B8%25D1%2587%25D0%25B5%25D1%2581%25D0%25BA%25D0%25B8%25D0%25B9_%25D0%25B1%25D1%2583%25D1%2582%25D1%2581%25D1%2582%25D1%2580%25D1%258D%25D0%25BF">statistical bootstrap</a> to select the most informative genes. <br><br>  In fact, ‚Äúif you take only genes with active support, then you will get the right tree,‚Äù says Donogue. <br><br>  The revised tree coincided with a tree built on an alternative source of evolutionary information ‚Äî large-scale changes in DNA segments passed down from generation to generation ‚Äî which justified their research. <br><br>  Discoveries are not limited to yeast.  By applying the same analysis to larger and more complex forms of life, including the genetic data of vertebrates and animals, they found serious conflicts between individual genes. <br><br>  Some researchers need to get used to the idea of ‚Äã‚Äãselectively excluding data from analysis.  ‚ÄúFor many years, the overriding problem of people trying to understand the relationships of organisms was the problem of collecting enough data,‚Äù said <a href="http://www.yale.edu/townsend/jeffreytownsend.html">Jeffrey Townsend</a> , an evolutionary biologist from Yale University who is not associated with research.  ‚ÄúThe community has always been told about the need for a data set, so it‚Äôs not surprising that they approached the task in this way.‚Äù <br><br>  Although evolutionary biologists have struggled with these problems for years, the new study was the largest attempt to study the level of conflict of individual genes.  ‚ÄúPeople will have two reactions: there are more conflicts than I thought, and we need to learn how to analyze them better,‚Äù says Donague, who wants to apply the new method in his work.  However, he also points to the difficulties in confirming the accuracy of the new approach.  Although the revised tree coincides with that built on alternative genetic information, the latter may reveal its own inconsistencies.  ‚ÄúI'm not sure that we know what the relationship really is,‚Äù he says.  ‚ÄúAnd if we are not sure about the true state of things, we don‚Äôt know if we got the right tree.‚Äù <br><br><h2>  Changing picture </h2><br>  Researchers need to apply the new technique more broadly to see how it can change the concept of evolution.  However, Rokas and Salichos have already shown that it is most difficult to reconstruct short branches of the tree, or ‚Äúbushy‚Äù parts of it, representing periods of rapid speciation ‚Äî especially those closest to the base of the tree and deep in the evolutionary history. <br><br>  ‚ÄúTheoretical studies predicted this behavior, but our study demonstrates confirmation for the first time with experimental data,‚Äù Rokas said. <br><br>  Rokas argues that the new discoveries will change how researchers interpret fuzzy fragments of the tree.  ‚ÄúEvolutionary biologists usually assume that if the tree doesn‚Äôt have the necessary details, then it‚Äôs wrong.  And consequently, if we collect more data and compose better algorithms, then we will come to the correct tree, ‚Äùhe says.  But the presence of conflicting parts of the tree that persist, despite the flow of data and the use of a new type of analysis, may indicate the presence of bushy parts.  "I think in some cases the algorithm will be able to resolve this conflict, and in others it will mark the areas of conflict that we can hardly ever resolve." <br><br>  The study of these bushy parts of the tree can give a new look at particularly interesting stages of evolution, for example, the Cambrian explosion, when life has moved from the predominance of simple organisms to a diverse set of animal species. <br><br>  Other scientists agree that discoveries may affect how experts cope with conflicting ideas about evolution.  ‚ÄúI think this is a harbinger of a paradigm shift,‚Äù said Townsend.  ‚ÄúIf we use suitable methods, we have the opportunity to learn more about the issues that have tormented us for a long time.‚Äù <br><br>  Townsend, who developed his own method of selecting the most informative genes based on the <a href="http://www.ncbi.nlm.nih.gov/pubmed/17464879">speed of their evolution</a> , notes that not all members of the scientific community agree on the need for new approaches.  ‚ÄúI hope this work will help bring this problem to the forefront,‚Äù he said. <br><br>  Choosing the right number of genes to build prototypes of phylogenetic trees is not the only question that torments evolutionary biologists.  They also need to agree on how many species to include in processing - the more species in the tree, the more difficult the analysis.  Results may also differ due to differences in the quality of data collected for different species.  ‚ÄúIf we need to get a true evolutionary history of how everything is connected with each other, then what is better for this - collecting more genes or more species?  - says Donogue.  ‚ÄúI think both.‚Äù <br><br>  New approaches that allow researchers to obtain accurate results using fewer genes may allow an evolutionary tree to expand.  The ability to select only the most informative of genes can make the process more efficient, and allow scientists to create accurate trees using less data and resources.  ‚ÄúIf we could choose several genes and get the same good tree as with the whole genome,‚Äù says Hilu, ‚Äúwe could build a much more detailed tree of life ‚Äî at the genus level, or even at the species level ‚Äî instead of being content skeleton of the most important offshoots. " </div><p>Source: <a href="https://habr.com/ru/post/401343/">https://habr.com/ru/post/401343/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../401333/index.html">Goldman Sachs replaced 600 traders with 200 programmers</a></li>
<li><a href="../401335/index.html">Upward spiral. Tools to fight depression</a></li>
<li><a href="../401337/index.html">Study: 72% of anonymous traffic can be linked to real users.</a></li>
<li><a href="../401339/index.html">How I built a successful 3D printing business without knowledge and experience</a></li>
<li><a href="../401341/index.html">Mexican surgeon distracts patients from surgery using virtual reality</a></li>
<li><a href="../401345/index.html">Caltech has created a bat robot</a></li>
<li><a href="../401347/index.html">IT billionaires have turned Hawaii into a tropical paradise for technoelites</a></li>
<li><a href="../401349/index.html">Geoscan Pioneer - "school" quadrocopter</a></li>
<li><a href="../401351/index.html">Quantum magic into life</a></li>
<li><a href="../401355/index.html">And at this time</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>