<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Genome Browsers</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Not the last role in bioinformatics is visualization. Scientists in this field work with huge amounts of information that it would be nice to somehow ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Genome Browsers</h1><div class="post__text post__text-html js-mediator-article"> Not the last role in bioinformatics is visualization.  Scientists in this field work with huge amounts of information that it would be nice to somehow capture the eye and present in my head.  A vivid example of a visualization tool is the genome browser (genome browser), which I want to talk about. <br> <a href="http://genome.ucsc.edu/cgi-bin/hgTracks%3FhgHubConnect.destUrl%3D..%252Fcgi-bin%252FhgTracks%26clade%3Dmammal%26org%3DHuman%26db%3Dhg19%26position%3Dchr21%253A33%252C031%252C597-33%252C041%252C570%26hgt.positionInput%3Denter%2Bposition%252C%2Bgene%2Bsymbol%2Bor%2Bsearch%2Bterms%26hgt.suggestTrack%3DknownGene%26Submit%3Dsubmit%26hgsid%3D327287135%26pix%3D1884"><img src="https://habrastorage.org/storage2/c33/532/301/c33532301fc8cc1efec46b31a3fb74a1.png" title="UCSC genome browser"></a> <br><a name="habracut"></a><br>  As many people remember from a school biology course, a genome consists of a set of chromosomes, and a chromosome is two chains that are folded into a spiral.  Each of the chains contains a sequence of nucleotides with four types of nitrogenous bases - adenine (A), guanine (G), cytosine (C) and thymine (T).  It is easy to identify the second one by one chain, if you remember that adenine is paired up with thymine (Antoshka-Timoshka), and guanine with cytosine (goose-chicken).  Some sections of DNA are called genes, RNA is read from them, and proteins are then encoded.  Proteins are composed of <a href="http://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BC%25D0%25B8%25D0%25BD%25D0%25BE%25D0%25BA%25D0%25B8%25D1%2581%25D0%25BB%25D0%25BE%25D1%2582%25D1%258B">20 types</a> of <a href="http://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BC%25D0%25B8%25D0%25BD%25D0%25BE%25D0%25BA%25D0%25B8%25D1%2581%25D0%25BB%25D0%25BE%25D1%2582%25D1%258B">amino acids</a> (plus a pair of exotic ones), each of which is encoded in three nucleotides. <br><br>  A genome browser is a one-dimensional map that displays some nucleotide sequence (say, a chromosome or a single gene) with accompanying information.  Information is usually structured into blocks called tracks.  For example, there may be a track with genes or with separate nucleotides.  Individual entities on the tracks are often called features. <br><br>  There are genome browsers sharpened for small bacterial genomes, but the universal browser needs to show both the long chromosomes of vertebrates and individual nucleotides.  The longest human chromosome (the <a href="http://ru.wikipedia.org/wiki/1-%25D1%258F_%25D1%2585%25D1%2580%25D0%25BE%25D0%25BC%25D0%25BE%25D1%2581%25D0%25BE%25D0%25BC%25D0%25B0_%25D1%2587%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25B2%25D0%25B5%25D0%25BA%25D0%25B0">first</a> ) contains about 250 million base pairs, that is, the scale should change about a million times.  Of course, on a different scale, the information is displayed differently.  For example, in the picture above there is a track with the UCSC Genes genes, where the whole SOD1 gene and fragments of the neighboring genes fell.  At this scale, the exon-intron gene structure is displayed.  <a href="http://ru.wikipedia.org/wiki/%25D0%25AD%25D0%25BA%25D0%25B7%25D0%25BE%25D0%25BD">Exons</a> (those parts that will remain in RNA after <a href="http://ru.wikipedia.org/wiki/%25D0%25A1%25D0%25BF%25D0%25BB%25D0%25B0%25D0%25B9%25D1%2581%25D0%25B8%25D0%25BD%25D0%25B3">splicing</a> and encode protein in the long term) are marked with filled rectangles, and <a href="http://ru.wikipedia.org/wiki/%25D0%2598%25D0%25BD%25D1%2582%25D1%2580%25D0%25BE%25D0%25BD">introns</a> (gaps between exons) are indicated by arrows, which show the direction of gene reading (in this case, the SOD1 gene is located on a straight DNA strand, and BC041449 - on the back).  And this is how a piece of the SOD1 gene looks like when magnified: <br><img src="https://habrastorage.org/storage2/9d0/fdc/473/9d0fdc47352b1660837662a64aa59b5f.png" title="UCSC genome browser; SOD1 gene"><br>  Here the scale allows you to deduce the amino acid sequence of those gene fragments, which then encode the protein.  Each amino acid corresponds to a certain letter of the Latin alphabet. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      What else can be seen on the genome browser?  At the most detailed scale, you can see individual nucleotides, both on the forward and reverse DNA helix: <br> <a href="http://www.ensembl.org/Homo_sapiens/Location/View%3Fr%3D6:134489603-134489662"><img src="https://habrastorage.org/storage2/2b4/043/0cf/2b40430cf415c2746332398596ea8484.png" title="Ensembl genome browser"></a> <br>  Each nucleotide corresponds to a standard color, so you can have fun painting it, even if the letters themselves no longer fit: <br><img src="https://habrastorage.org/storage2/a0c/67f/e3b/a0c67fe3b51c139539ca08bdc6e48f09.png" title="Ensembl genome browser"><br><br>  If you roll back a little more, you can judge about the nucleotide composition by the special track GC content: <br><img src="https://habrastorage.org/storage2/844/bd4/819/844bd48190b2f5c9db7308648fdfbaf0.png" title="BioUML genome browser - GC content"><br>  Here red color means that G and C nucleotides in this place are less than 50%, and blue color is more.  One might think that A, C, G, T are just four equal states of a two-bit cell encoding genetic information, and the proportion of G and C does not mean anything interesting.  However, base pairs GC form three hydrogen bonds, and AT only two.  That is, GC is stronger, they are harder to break and the enrichment of GC or AT bonds affects the chemical processes in a given region of DNA. <br><br>  What else can you see?  Usually there are tracks with genomic variations that, for example, distinguish different people from each other.  Often, variations are expressed as point mutations, single nucleotide substitutions ( <a href="http://ru.wikipedia.org/wiki/%25D0%259E%25D0%25B4%25D0%25BD%25D0%25BE%25D0%25BD%25D1%2583%25D0%25BA%25D0%25BB%25D0%25B5%25D0%25BE%25D1%2582%25D0%25B8%25D0%25B4%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25BF%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25BE%25D1%2580%25D1%2584%25D0%25B8%25D0%25B7%25D0%25BC">Single-nucleotide polymorphism, SNP</a> ).  Many of these mutations were found when comparing the results of sequencing of the genomes of different people and placed in special databases (for example, dbSNP): <br><img src="https://habrastorage.org/storage2/1b4/bf9/e3d/1b4bf9e3d0829cae0449f258f55e3752.png"><br>  On the given fragment, there are not so few SNPs (19 by 356 nucleotides - more than 5%).  However, many of them are synonymous.  Since out of 4 <sup>3</sup> = 64 variants of three nucleotides, 20 variants of proteins are encoded, some substitutions do not affect the resulting protein.  Some substitutions fall into non-coding regions (for example, introns), so they may also not influence the result (but they can also influence). <br><br>  Another interesting thing is a comparison of the human genome with the genomes of other species.  For this, non-trivial algorithms make multiple alignment of genomes and also show it.  The very top image of the post shows a schematic alignment of a person with a <a href="http://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B0%25D0%25BA%25D0%25B0%25D0%25BA%25D0%25B0-%25D1%2580%25D0%25B5%25D0%25B7%25D1%2583%25D1%2581">rhesus monkey</a> , a mouse, a dog, an elephant, an opossum, a chicken, a frog (Xenopus tropicalis) and a <a href="http://ru.wikipedia.org/wiki/%25D0%2594%25D0%25B0%25D0%25BD%25D0%25B8%25D0%25BE-%25D1%2580%25D0%25B5%25D1%2580%25D0%25B8%25D0%25BE">zebrafish</a> .  Dark matching fragments are shown.  Notice that the darkest regions are in the coding regions of the genes.  In the same picture there is a graph of the conservatism of sites among mammals (Mammal cons), which also correlates.  But the multiple alignment in an enlarged form: <br><img src="https://habrastorage.org/storage2/75f/d8a/2e2/75fd8a2e293d0abc27cdef05812095c1.png"><br>  Minus means that a person has a nucleotide, but is absent in another species.  Orange vertical line (for example, in the line with the dog between two thymine) - on the contrary.  The number of missing nucleotides is indicated above (they are not listed).  The coding region is given in amino acid form, therefore, synonymous substitutions are not visible.  Chicken and fish, apparently, do not have a similar region.  You can see how the macaque looks like a person. <br><br>  At the very far scale, the <a href="http://ru.wikipedia.org/wiki/%25D0%259A%25D0%25B0%25D1%2580%25D0%25B8%25D0%25BE%25D1%2582%25D0%25B8%25D0%25BF">karyotype of the</a> chromosome becomes visible: <br><img src="https://habrastorage.org/storage2/fed/567/c1f/fed567c1f3f49c02e9cf2e927813455e.png"><br>  You can orient yourself by the karyotype, if you remember, for example, in which lane is your favorite gene that you are studying.  The crossing in the middle is the <a href="http://ru.wikipedia.org/wiki/%25D0%25A6%25D0%25B5%25D0%25BD%25D1%2582%25D1%2580%25D0%25BE%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0">centromere</a> . <br><br>  There are many other predefined tracks.  Some browsers allow you to load tracks from the web using a special <a href="http://en.wikipedia.org/wiki/Distributed_Annotation_System">DAS protocol</a> .  And, of course, the genome browsers allow scientists to add their own (for this there are special file formats).  User tracks can, for example, show DNA binding regions for a specific protein (for example, <a href="http://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B0%25D0%25BA%25D1%2582%25D0%25BE%25D1%2580%25D1%258B_%25D1%2582%25D1%2580%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BA%25D1%2580%25D0%25B8%25D0%25BF%25D1%2586%25D0%25B8%25D0%25B8">a transcription factor</a> ), both predicted and obtained in an experiment (for example, <a href="http://en.wikipedia.org/wiki/ChIP-sequencing">ChIP-Seq</a> ).  If you, for example, sequenced your own genome, you can download the result and compare it with the reference and known SNP. <br><br>  Genome browsers are plentiful.  Only in Wikipedia thirty pieces <a href="http://en.wikipedia.org/wiki/Genome_browser">are listed</a> , and this is definitely not all.  Many of them are specialized: sharpened for a certain organism or a certain type of data.  Of the popular desktop browsers can be noted <a href="http://bioviz.org/igb/">Integrated Genome Browser</a> and <a href="http://www.broadinstitute.org/igv/">Integrative Genomic Viewer</a> (as you can see, with the names do not bother).  Both are Java applications, there is Java Web Start. <br> <a href=""><img src="https://habrastorage.org/storage2/4da/c31/dd8/4dac31dd84e577efd262d5a900624c53.png" title="IGB (left) and IGV (right). Click to enlarge"></a> <br><br>  Of course, it is often more convenient to use the genome browser on the web.  Most of the pictures above are taken in the <a href="http://genome.ucsc.edu/cgi-bin/hgTracks">UCSC Genome Browser</a> and the <a href="http://www.ensembl.org/Homo_sapiens/Location/Genome">Ensembl Genome Browser</a> .  Both of these browsers generate images on the server.  There are more modern technical solutions.  <a href="http://neomorph.salk.edu/epigenome/epigenome.html">AnnoJ</a> , for example, renders pictures on a client on canvas, receiving JSON data from the server (a demonstration for Arabi's favorite weed of biologists - Arabidopsis).  There are more <a href="http://jbrowse.org/demos/">JBrowse</a> .  In a way, it is unique because it does not contain server code.  Data on tracks and sequences are prepared in advance on the server in the form of static files that the browser loads via AJAX.  User files are processed through the <a href="http://www.w3.org/TR/FileAPI/">File API</a> . <br><br>  The perfect genome browser doesn't exist.  In my opinion, the main problem is the speed of work.  This is especially noticeable on the web, although there are delays in the desktop.  Some tracks at certain scales are either generated very slowly, or are turned off altogether.  For visualization it is necessary to grind a lot of information, which, perhaps, is not always presented in the optimal form.  Therefore, if someone has a desire to do this, there is every chance of overcoming competitors. </div><p>Source: <a href="https://habr.com/ru/post/170429/">https://habr.com/ru/post/170429/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../170415/index.html">First look and video of the Sony Xperia Tablet Z tablet</a></li>
<li><a href="../170417/index.html">The digest of interesting news and materials from the world of ayti for the last week ‚Ññ45 (February 16 - 21, 2013)</a></li>
<li><a href="../170419/index.html">Ubuntu Touch ported to Galaxy S III + Canonical instructions for other devices.</a></li>
<li><a href="../170423/index.html">Google introduced App Launcher for windows-based Chrome browser</a></li>
<li><a href="../170425/index.html">Nokia is preparing a smartphone Lumia 520 and 720</a></li>
<li><a href="../170433/index.html">New Google Campus Project</a></li>
<li><a href="../170435/index.html">NASA will print parts of a rocket for a flight to Mars on a 3D printer</a></li>
<li><a href="../170437/index.html">Arm it!</a></li>
<li><a href="../170439/index.html">About procrastination popular, or Hollywood vs. laziness</a></li>
<li><a href="../170441/index.html">Debugging Grunt jobs in WebStorm</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>