<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Project ‚ÄúProkaryotic Genome‚Äù - a scientific startup</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="This project was conceived long ago. About 5 years ago I believed that many results in genomics can be obtained by people far from biology, which I am...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Project ‚ÄúProkaryotic Genome‚Äù - a scientific startup</h1><div class="post__text post__text-html js-mediator-article">  This project was conceived long ago.  About 5 years ago I believed that many results in genomics can be obtained by people far from biology, which I am fully.  Of course, during this time I picked up a bit of terminology and learned a little how the specialists work.  But the more I learned how the specialists work, the more it caused me to reject it.  I believe that they obviously make many unjustly complicate the result that a difficult area becomes impassable.  While everything is quite simple and quality can be done.  And yes, I'm trying to compete with them (of course, only in a certain narrow area), no matter how naive it would look. <br><br>  The whole problem of this project is that I am its only full-fledged participant.  Of course, I managed to talk with many during this time and many had a real impact on the project.  Thanks to all of them.  It is clear that a non-profit project is not much can count on success.  Yes, indeed, behind every scientific project there is a solid injection of about millions of people and a team of serious scientists.  We do not have this, but there is only humanism and enthusiasm. <br><br>  Therefore, first of all, I need advice from those who have experience in a startup of such projects on a non-commercial basis.  Secondarily, we need a team of programmers (from knowledge of biology, if necessary, I will free you :)).  And now I would like to find such enthusiasts who could provide work (say modestly) for the project‚Äôs home web page (please write to me at tac@inbox.lv or with personal messages from Habr).  And of course, any other response and suggestions are important. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      And below I will explain the idea and what the project claims for, as well as the current results, and in the worst case they are comparable to those given by the specialists.  But I am quite self-critical, therefore I am always ready to listen to criticism - preferably not in my address, but in the address of the project. <br><br><a name="habracut"></a><br><br><h4>  From idea to computer experiments </h4><br><br>  I will not explain the raw idea, a lot has already been passed and it was described by me in previous articles on Habr√©.  [However, I will insert a couple of words, because  below, many complain that I started off the bat.  The main idea / task of the project is to understand how bacteria evolved and how their DNA consistently changed.  To do this, we build a tree of divergence of species and analyze them.] I will describe a new, what is called an all-wheel drive experiment.  But first I need to introduce you to the problematic and then understand how to evaluate the results of the experiment. <br><br><h5>  Phylogenetic signal </h5><br><br>  Here we will try to discuss this term to which one biologist turned my attention. <br><br>  With the evolutionary origin of animals from a common ancestor, it is believed that it is possible to build a single tree-like hierarchical structure of the origin of species.  There is no fundamental difference what signs to take as a basis.  Simply, the more genes are included in the analysis, the less poorly grounded sections remain in the tree.  At the same time, if the objects being classified do not originate from a common ancestor, then there is no single tree-like hierarchical structure.  The classification of such objects is either fundamentally different when using different sets of characters (genes), or has a fundamentally non-woody appearance. <br><br>  But the coincidence of the resulting "trees" built on various grounds allegedly tells us about the presence of a "phylogenetic signal."  And the smaller the differences between trees built on different sets of genes, the stronger the ‚Äúphylogenetic signal‚Äù we have.  But what is important, the reverse is not true. <br><br>  It is often said that this signal is indeed present and coincides.  But this is not entirely true, since I came across one <a href="http://istina.imec.msu.ru/media/publications/articles/d25/c99/1787974/biochemistry-Rus1610.pdf">article</a> that is somewhat more critical on this subject. <br><br>  First, they indicate that: <br><br><blockquote>  It is assumed that by analyzing a multitude of genes it is possible to strengthen the phylogenetic signal before it exceeds the noise and to achieve the correct resolution of conflicts between different genes.  <b>But</b> <br><br>  [there are a number of private examples] <br><br>  All this suggests that the current methods of phylogeny reconstruction by a large number of genes do not eliminate the artifacts known for single genes.  Here, assumptions of evolution models, differences in the rates of evolution of species, alignment errors and choice of orthologs, and insufficient representativeness of taxonomic sampling can have the same effect.  To eliminate the artifacts of multigene phylogenetic analysis, data selection is proposed, which, of course, makes it less formal.  Thus, the practice of modern phylogenomics shows that the statistical support for phylogeny reconstructions increases with an increase in the number of genes being compared, but a high level of statistical support for the tree as a whole or its individual nodes cannot serve as an indicator of the correctness of phylogenetic reconstruction. <br></blockquote><br><br>  And secondly they ask: <br><blockquote>  How to find to check a gene or nucleotide worthy of unlimited trust?  The shorter the geological period of the stem group existed, the less likely it is that a randomly selected gene will carry synapomorphy, while not subject to homoplasias and reversions.  To get for sure the winning ticket in the lottery, there is a way - to buy the whole edition.  Given the speed of development of sequencing technology and computer processing, with respect to the genomes this may in a few years seem like such a stupid idea.  On the other hand, if the related resemblance in the species is large, then it will be found in many genes from among those chosen at random and even, probably, in one sufficiently extended gene, like 18S or 28S rRNA. <br></blockquote><br><br>  <u>This is what is called a classic of biology.</u>  <u>And now let's try to think about it.</u> <br><br>  In previous articles on the role of such ‚Äútrustworthy‚Äù genes, I suggested and showed what happens if it is a tRNA gene.  This gene is no worse than rRNA, which now enjoys "unlimited confidence."  But in this article [in continuation of it] I will show further what will happen if you ‚Äúbuy the whole circulation‚Äù.  But before that, it‚Äôs necessary to figure out what the bad option is when rRNA enjoys ‚Äúinfinite trust‚Äù. <br><br>  And it turns out that it‚Äôs not at all the choice of one or another gene or nucleotide sequence.  And it is right that they dream (but strangely they don‚Äôt make it) about comparing over a large variety of genes.  The point is in the method.  And it has a statistical nature, and those who look at it a little more soberly admit that the above article has problems: ‚ÄúHere, assumptions of evolution models, differences in the rates of evolution of species, <i>alignment errors</i> and choice of orthologs, lack of representativeness of taxonomic sampling can affect it‚Äù . <br><br>  All of this separately worsens the <i>phylogenetic signal</i> in one way or another.  Most of all claims to <a href="http://ru.wikipedia.org/wiki/%25D0%2592%25D1%258B%25D1%2580%25D0%25B0%25D0%25B2%25D0%25BD%25D0%25B8%25D0%25B2%25D0%25B0%25D0%25BD%25D0%25B8%25D0%25B5_%25D0%25BF%25D0%25BE%25D1%2581%25D0%25BB%25D0%25B5%25D0%25B4%25D0%25BE%25D0%25B2%25D0%25B0%25D1%2582%25D0%25B5%25D0%25BB%25D1%258C%25D0%25BD%25D0%25BE%25D1%2581%25D1%2582%25D0%25B5%25D0%25B9">alignment</a> errors (I will not explain that it is Wikipedia read this by reference).  It is because of this that one has to deal with statistics and the errors associated with it.  Correctly do the alignment, especially for small sequences now do not know how - it really does not take into account the conservatism of some fragments.  To do this, hydrogen bonds in the tertiary structure must be taken into account - but this is usually not done during alignment. <br><br>  But rRNA is, firstly, long, and secondly, separately, there are a lot of errors, but statistically they still give some kind of signal.  But what quality is it below and we will look at the example of comparing trees constructed with 16S rRNA and 23S rRNA (these are the longest RNA sequences of which the ribosome consists).  These trees were obtained in the project <a href="http://www.arb-silva.de/projects/living-tree/">The All-Species Living Tree</a> .  But, third, they are now writing a sufficient number of articles on the construction of phylogenetic trees, but for some reason such a question as ‚Äúanalysis of the prevalence of phylogenetic signal over noise‚Äù is not discussed. <br><br><h5>  And what about the alternative? </h5><br><br>  The only option to object to criticism like the one above (‚Äúa high level of <u>statistical</u> support for the tree as a whole or its individual nodes cannot serve as an indicator of the correctness of phylogenetic reconstruction‚Äù) is to move from statistical reasoning in which common sense does not allow 100% certainty, it is to proceed to the conclusions of a deterministic nature.  And for this you need to get rid of alignment in the analysis and choose those nucleotide sequences that can be analyzed without alignment. <br><br>  I am surprised, but experts of this alternative do not offer and do not see.  Although it at least shows more stable results.  Why?  Let's deal with this. <br><br>  After all, no matter what tree I would not conclude, the confidence in it will be no more / no less than other trees.  But there experts built (as for example, in the project The All-Species Living Tree), and here you say you built a "charlatan."  And there will always be objections. <br><br>  Likewise, any method is vulnerable to criticism, as long as there is no confidence in the results.  Therefore, we need a criterion for the correctness of the results.  The stability of the ‚Äúphylogenetic signal‚Äù claims such a criterion. <br><br>  But before choosing him for this - I would like the reader to understand why this signal may be unstable at all.  There can be 3 reasons: <br><br>  1. Evolution does not follow Darwin, i.e.  organisms simply have no common ancestor and have never had it.  Considering, firstly, that now there is a phenomenon of horizontal transfer, and secondly, that the hypothesis of the RNA world has already been practically proven, and then individual organisms could arise independently of each other - Darwinian evolution is in fact a big question.  Therefore, here we simply agree that it is simply more convenient for the human mind to hierarchically consider the origin of species and the Darwinian evolution for us is just a convenient way of presenting information, similar to drawing charts instead of textual information. <br>  2. Method errors.  For example, alignment, to which I expressed a great deal of distrust.  It is because of incorrect alignment that the signal is deviated to a large extent. <br>  3. Different number of examples in the sample. <br><br>  When we have the influence of all three causes, we cannot with certainty distinguish the resulting noise ‚Äî it is an objective reason or a subjective one.  Those.  we cannot say either the problem in our method, the problem in our representativeness of the sample or, nevertheless, evolution does not go entirely according to Darwin. <br><br>  Researchers can very easily say, ‚Äúbut you know, our method works perfectly, the sample is wonderful, and the small errors that you see are just the way it is in nature‚Äù.  But first, let's quantify the errors.  Secondly, let's replace the statistical approach with the deterministic one.  Third, we will analyze the total available for the deterministic approach. <br><br><h5>  The advantage of the deterministic approach </h5><br><br>  To demonstrate the advantage of the deterministic approach, I will propose a thought experiment.  It can be done experimentally in reality, but the public will simply get tired of the dryness of presentation, and most importantly since Aristotle‚Äôs time we know that the experiment does not prove anything in absolute categories, but only allows us to say ‚Äúwe see this in this data, but this does not mean that It cannot be otherwise. ‚Äù  And we need to judge it in absolute categories. <br><br>  So a mental experiment.  Compare the statistical and deterministic approach.  In statistical terms, we analyze 1000 organisms on a single 16S rRNA gene, which has a great length of about 1600 characters (and this is done in the overwhelming number of cases in the study).  Suppose we have a reliable set of rRNA for all 1000 organisms.  But to build a phylogenetic tree, we need to make an alignment.  But before alignment, divide the rRNA into two equal parts and make the alignment and the subsequent construction of the tree according to the first and second parts separately. <br>  Since the sample is the same, 3rd reason has no effect.  On the 1st reason we agreed not to refer.  But it is obvious that the alignment will affect the appearance of the tree, at least to a small extent.  a certain evolutionary distance is calculated there, but for different parts it will be a little different.  As a result, the first and second tree will be different and this will be the 2nd reason - a method error. <br><br>  What we have for a deterministic approach.  Here we focus on such genes that are completely identical in different organisms, but they cannot be long, because  all long is more likely to mutate.  But instead of one gene of 1600 characters, we have a set of 10-20 genes with 70-150 characters.  Such characteristics, for example, correspond to the tRNA genes.  Again, suppose that we have a reliable set of these genes.  Then the question is: if the tRNA sequences are divided into two parts and the construction of two different trees - will they coincide or not?  Answer: they will match by 100%.  This is due to the fact that when building a tree, in fact, the sequences are replaced with identifiers, and then all the manipulations take place on the basis of only combinations of genes.  Therefore, if the genes were correctly identified on the basis of half the sequence, then there will be no further distortion. <br><br>  That is, under ideal conditions and the same sampling rate, the deterministic approach has a distinct advantage, and has no errors of the 2nd kind. <br><br>  And then we can talk about the errors of the 3rd kind and how they affect the phylogenetic signal.  But we must understand that in the deterministic approach we only have errors of the 3rd kind, and in the statistical one, which is now accepted everywhere, we cannot separate the influence of the errors - the ‚Äúnoise‚Äù of the 2nd and 3rd kinds. <br><br><h4>  Actually experiment </h4><br><br><h5>  ‚Ññ1.  Comparing 16S and 23S trees </h5><br><br>  So, we need to compare two trees <a href="http://www.arb-silva.de/fileadmin/silva_databases/living_tree/LTP_release_102/LSU/LTPs102_LSU_tree.pdf">built on the 23S rRNA gene</a> and <a href="http://www.arb-silva.de/fileadmin/silva_databases/living_tree/LTP_release_108/LTPs108_SSU_tree.pdf">built on the 16S rRNA gene,</a> which are the last result of the project <i>The All-Species Living Tree</i> . <br><br>  But you can only compare comparable things.  And then it's time to talk about how to measure the magnitude of the error of the 3rd kind, i.e.  how the sample size and its composition affect the result.  Specialists here would suggest us to do statistical surveys of any probability distributions, estimates of displacement, variance, etc.  turbid indices and indifferent coefficients.  In contrast, we must compare in such a way that each digit would allow us to understand what this means. <br><br>  First, the format of phylogenetic trees hides one important thing - they obviously do not display the parent, although it is there as the intersection of lines on the same level.  In fact, here we need to solve the issue of converting the .newick format, for example, into the .gml format, i.e.  get a full tree, where all the ancestors will have a name. <br><br>  Secondly, the fact is that the data on the 16S gene is almost 10 times more.  And we need to remove such leaves of trees that are in the tree 16S, but they are not in the tree 23S, and vice versa.  Only then will we get what can be compared with each other.  But after such removal (cutting) of ‚Äúleaves‚Äù on the tree, which we cannot compare, their supposed ancestors can remain and if they no longer have other ‚Äúleaves‚Äù, they should also be removed so that they do not litter the tree. <br><br>  Third, and most importantly, the above-described circumcision does not solve all the problems of bringing the tree to the same denominator.  It may be a situation that an ancestor has only one leaf, and this ancestor has in turn again only one ancestor, and so on several times.  Those.  as a result, we have ‚Äúlong threads‚Äù on the tree.  All these ‚Äúsingle‚Äù ancestors do not allow us to compare with another tree (23S) in which these ancestors do not exist, because  it was built on another smaller sample, and it is natural that a large sample suggests a large number of ancestors in order to more accurately reflect the divergence of the species.  But in order for it to be comparable it is necessary to exclude such ‚Äúsingle‚Äù ancestors, and raise the leaves from them to a level where there is an ancestor with more than one leaf (that is, where there is real divergence). <br><br>  This process of ‚Äúlifting leaves to divergence sites‚Äù again leaves ancestors, which can be eliminated and steps 2 and 3 should be repeated until all unnecessary ancestors are excluded. <br><br>  Little sketching for understanding: <br><br><img src="https://habrastorage.org/storage2/e98/a6d/e31/e98a6de31dd168c08382aa9ad36cea62.png"><br><br>  Right option to all manipulations.  In the center there is an option where the sheet ‚ÄúEscherichia_albertii‚Äù is cut, which is not in the compared tree.  On the left is the option where the superfluous ancestor "n23" is removed.  In reality, only 3000 remain more and more seriously from 18,000 nodes. It may also give the impression that important ancestors have been removed, but if they are not removed the comparison result will only get worse, since in the smaller tree the ‚Äúremoved‚Äù ancestors cannot appear, and compare all we need comparable things, not a ‚Äúkettle with a pan‚Äù. <br><br>  Now, if we strictly approach the comparison, then the coincidence of the trees is when the leaves having one parent in one tree also have one parent in the compared tree.  And we can count the number of such cases.  But in order to assess proximity, one must also have some distribution of errors.  The magnitude of the error can be calculated as follows.  If a pair of ‚Äúleaves‚Äù in one tree has one parent, then in the compared tree we find their <a href="http://ru.wikipedia.org/wiki/%25D0%259D%25D0%25B0%25D0%25B8%25D0%25BC%25D0%25B5%25D0%25BD%25D1%258C%25D1%2588%25D0%25B8%25D0%25B9_%25D0%25BE%25D0%25B1%25D1%2589%25D0%25B8%25D0%25B9_%25D0%25BF%25D1%2580%25D0%25B5%25D0%25B4%25D0%25BE%25D0%25BA">smallest common ancestor LCA</a> and count the number of intermediate ancestors from one leaf to LCA and from second to LCA ‚Äî we add the resulting numbers and plot as a dot on the error distribution. <br><br>  As a result, we have such a schedule, about 50% of the correct cases, and the others are somewhat erroneous, the error really fades. <br><br><img src="https://habrastorage.org/storage2/dd7/8f6/a6f/dd78f6a6f9fe82cc7416b2fbd73b1ef0.png"><br><br>  As we can see, the specialists are far from ideal, the signal is obtained somewhere 50% noisy and then, although some regularity breaks through, it is unstable.  Therefore, there is something to improve. <br><br><h5>  To be continued‚Ä¶ </h5><br><br>  It turns out somehow long, so the results of the deterministic approach, I will put in a separate article.  There we will look at how much it will be possible to improve the quality of the evolutionary tree (phylogenetic signal).  The experiment is not fully completed, but I hope for the best :) <br><br>  PS <b>upd.</b>  There is a high probability that the issue with the site will be resolved.  Thanks to the good people :) Now we need the chief editor of the site / image-maker - who knows how to correct both grammar and semantic text correction so that my ‚Äúcheeky style‚Äù does not distort specialists, and at the same time is understandable to ordinary people. </div><p>Source: <a href="https://habr.com/ru/post/166361/">https://habr.com/ru/post/166361/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../166339/index.html">CRUX 3.0 has been released</a></li>
<li><a href="../166341/index.html">How bad should the code be?</a></li>
<li><a href="../166351/index.html">Marked text in android.widget.TextView</a></li>
<li><a href="../166353/index.html">Haskell Polymorphia and Classes</a></li>
<li><a href="../166357/index.html">Little Brave Arkanoid (Part 2 - YAML)</a></li>
<li><a href="../166363/index.html">A few words about productivity</a></li>
<li><a href="../166367/index.html">Features of using roles with authentication in Oracle since 10.2.0.5</a></li>
<li><a href="../166369/index.html">Living computer museum</a></li>
<li><a href="../166373/index.html">Sending temperature data from the TL-MR3020 router and the Raspberry Pi to People's Monitoring</a></li>
<li><a href="../166375/index.html">And again about ... LAMP and basic secure mini-hosting with your own hands</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>