<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Comparison of biological sequences</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Tetrapeptide HABR is incomparable 

 Comparison of sequences of characters is a seemingly simple matter, but, applied to biology, it encounters a lot ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Comparison of biological sequences</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/getpro/habr/post_images/f03/205/d96/f03205d9613c56ae187461bb8b75192e.png" alt="image"><br>  Tetrapeptide HABR is <i>incomparable</i> <br><br>  Comparison of sequences of characters is a seemingly simple matter, but, applied to biology, it encounters a lot of problems almost out of the blue.  Some problems of modern biology cannot be solved at all at the present stage of development of computer technology.  In this article, I will show you on fingers what is so special about biological sequences and why they need special algorithms.  Biologists and especially bioinformatics are not recommended to read - there is a risk of dying of boredom. <br><br>  Biological sequences are the primary structure of biological macromolecules.  Namely proteins and DNA / RNA.  (There are still carbohydrates, for example, starch, but they consist of identical monomers and therefore uninteresting.) The DNA sequence determines the protein sequence, the protein sequence determines its spatial structure, the structure determines the protein functions, and the set of functions of different proteins is called life.  It is the differences in the functioning of different proteins that we, in essence, differ from each other.  It is necessary to compare molecules, roughly speaking, for two reasons: 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      1) comparing in pairs the proteins of different organisms, we can say which organisms are more similar to each other, and which are less; <br>  2) comparing simultaneously a dozen or two proteins, we can find structurally and functionally important areas (they are usually identical in related organisms), which is useful when creating artificial proteins (drug design, nanotechnology, in a good sense of the word). <br><br><a name="habracut"></a><br>  It is better to compare proteins, not DNA, if only because proteins have more ‚Äúalphabet‚Äù (20 amino acids versus 4 nucleotides), and the probability of a coincidence is lower, therefore, we‚Äôll deal with proteins later.  Generally speaking, it would be necessary to compare structures, not sequences (since we still do not quite know how the second defines the first);  but it so happened that we learned well how to figure out the sequences, and obtaining a spatial structure is a very labor-intensive matter and, in general, many people think art is more likely.  Therefore, we have to compare the sequence.  Comparison of sequences of biological macromolecules is called alignment - the lines are written one under the other in such a way as to achieve coincidence or similarity of characters in the greatest number of positions.  One amino acid residue in a protein corresponds to <a href="http://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BC%25D0%25B8%25D0%25BD%25D0%25BE%25D0%25BA%25D0%25B8%25D1%2581%25D0%25BB%25D0%25BE%25D1%2582%25D1%258B" title="Single letter designations of amino acids">one letter of the Latin alphabet</a> in the sequence. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/44d/58e/967/44d58e9678e05b71e27e90a3619fc9eb.png"><br><br><h5>  Accounting for the degree of similarity of monomers </h5><br>  We are accustomed to consider two sequences all the more similar, the more characters in them coincide.  In biology, this approach is no good.  Consider two pairs of sequences: <br><br> <code> 1          2</code> <br> <code>ASDLV          ASDLV</code> <br> <code>ATEIV          AWDKV</code> <br> <br>  The old <a href="http://ru.wikipedia.org/wiki/%25D0%25A0%25D0%25B0%25D1%2581%25D1%2581%25D1%2582%25D0%25BE%25D1%258F%25D0%25BD%25D0%25B8%25D0%25B5_%25D0%25A5%25D1%258D%25D0%25BC%25D0%25BC%25D0%25B8%25D0%25BD%25D0%25B3%25D0%25B0">Hamming</a> clearly says that the second alignment is better: in the first pair only two of the five characters match, and in the second - three.  But let's look at their structures: <br><br><img src="https://habrastorage.org/getpro/habr/post_images/d16/d39/a11/d16d39a11378c24347cb6c928b0cca41.png"><br><br>  S and T, D and E - those residues that distinguish the sequences of the first pair - differ in structure by exactly one carbon atom, L and I - are isomers in general.  In most cases, such a difference does not affect the structure of the protein.  But replacing a small polar serine with a large non-polar tryptophan or hydrophobic leucine with a charged lysine, as in the second pair, will greatly change the structure of the molecule.  Thus, in the first alignment, the sequences are almost identical, and in the second, they are very different. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/e81/1c0/ad0/e811c0ad0899f2419d8652ee6043f19c.png"><br><br>  Accounting for the structural features of the compared characters is quite simple: the so-called ones are used.  replacement matrix, as the one in the figure.  The greater the number at the intersection of matched letters - the greater the quantitative measure of similarity (called "Score", or "Score"). <br><br> <code>ASDL V                  ASDLV</code> <br> <code>ATEI V                  AWDKV</code> <br> <code>Score = 4+1+2+2+4 = 13     Score = 4-3+6-2+4 = 9</code> <br> <br>  Matrices are compiled on the basis of the statistics of amino acid substitutions in proteins with already known structure: the more often there is a substitution of one letter for another, the greater the number at their intersection.  The problem is that periodically new matrices appear, better than the previous ones;  accordingly, they are all bad and do not take into account something. <br><br><h5>  Accounting for insertions and deletions </h5><br>  I do not know how others, but I was once struck by how similar the proteins of different organisms are.  Where is a person - and where is a rabbit, and serum albumin in both of them differ by only a few amino acids.  But they differ by a few: mutational variability is an inherent property of the living.  Essentially, assessing the similarity of proteins, we estimate the number of mutations that occurred on the path of the evolutionary transformation of one protein into another.  Accounting for mutations of the first type - substitutions - we have discussed above.  Two more types - insertions and deletions (deletes) - we will consider now. <br><br>  Insertions and deletions in bioinformatics are considered as one entity for the simple reason - having two sequences, one of which has an extra letter, we cannot determine which of the sequences is ‚Äúoriginal‚Äù.  The hypothetical common ancestor of both organisms died out millions of years ago, so we do not know for sure if there is an insertion or deletion in this case.  For this reason, the term "indel" (insertion + deletion) has appeared;  In Russian bioinformatics, the term ‚Äúdeletion‚Äù or tracing from English ‚Äúgap‚Äù is more common. <br><br><img src="https://habrastorage.org/getpro/habr/post_images/0f8/59f/a13/0f859fa13edb2f9ec4bc1497c061ce1f.png"><br>  <i>Two possible variants of the origin of related sequences.</i>  <i>A couple of sequences from the bottom line are real-life ‚Äúmolecules,‚Äù at the top, a hypothetical ancestor that could be either on the left or on the right (or not at all).</i> <br><br>  If there are deletions in alignment, they are penalized.  The more deletions - the more points are removed from the account.  Again, consider two alignments: <br><br> <code>ASGHDLV          AMSDCLV</code> <br> <code>AT--EIV          A-TD-VV</code> <br> <br>  In both alignments "dropped out" in two amino acids.  In both cases, formally, two mutations occurred.  In fact, in the first case, <i>one</i> mutation occurred.  Two times long.  But one.  One evolutionary event.  And the probability of it is more than two separate, as in the second example.  Therefore, the alignment score on the left should be higher, and the first two sequences should be considered more similar. <br><br>  To account for deletions is used so-called.  an affine penalty consisting of a penalty ‚Äúfor the discovery‚Äù and ‚Äúfor the continuation‚Äù of the deletion.  The first is usually an order of magnitude more than the second (about 10 and 1, respectively, or 10 and 0.1).  In the example on the left, we have one discovery and one continuation, on the right, two discoveries. <br><br>  The problem of accounting for deletions is not limited to this.  Depending on the location of the deletion in the molecule, it may, to a greater or lesser extent, affect the structure.  Some algorithms, for example, Clustal, take into account "residual-specific fines", some - not.  This usually comes down to the fact that deletions in hydrophilic regions ‚Äî potential loops dangling in the aquatic environment ‚Äî are fined less than in hydrophobic (potential tightly packed globule nuclei).  The question of how these fines should differ, and whether hydropathicity should serve as a criterion for differences, apparently, everyone decides for himself ... <br><br><img src="https://habrastorage.org/getpro/habr/post_images/573/cef/616/573cef616b1eaaf2e23aa596e5d3a261.png"><br>  <i>Mutation in the area number 1 - tightly packed and strictly structured (beta strand) globule nucleus - can greatly disrupt the structure of the protein.</i>  <i>The mutation in the plot number 2 (unstructured relatively labile loop), most likely, does not affect it.</i> <br><br><h5>  Alignment Calculation </h5><br>  So far, we have considered the assessment of the finished alignment, without wondering how it turns out.  The average protein consists of several hundreds of amino acids, aligning such long lines "by eye" obviously will not work.  Although there are different ways of obtaining alignments, <a href="http://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BB%25D0%25B3%25D0%25BE%25D1%2580%25D0%25B8%25D1%2582%25D0%25BC_%25D0%259D%25D0%25B8%25D0%25B4%25D0%25BB%25D0%25BC%25D0%25B0%25D0%25BD%25D0%25B0_%25E2%2580%2594_%25D0%2592%25D1%2583%25D0%25BD%25D1%2588%25D0%25B0">the Needleman ‚Äì Wunsch algorithm</a> and its modifications are now considered optimal.  I will omit the essence of the algorithm: the mathematical audience can easily figure it out on its own; the rest <s>should not even try,</s> I cannot explain it in the article on Habr√©.  Suffice it to say that to align two sequences with lengths m and n, an array of the order of size m * n is used;  since m and n are numbers of the same order, the order of the array is n <sup>2</sup> .  Modern consumer computers cope with this calmly.  While sequences - two. <br><br>  In the beginning, I said that by aligning a dozen or two proteins at the same time, we can find structurally and functionally important areas in them.  For example, the amino acids in the figure that make up the active center of trypsin-like proteases clearly express themselves in that they are identical in 20 of these proteins (the corresponding positions are indicated by asterisks). <br><br><img src="https://habrastorage.org/getpro/habr/post_images/1df/a09/73e/1dfa0973e897a6a3e240a31209169b94.png"><br>  <i>The amino acids that determine the catalytic properties of the enzyme are conservative, i.e.</i>  <i>do not change in the process of evolution.</i>  <i>The active center of the enzyme must comply with the geometry and chemistry of the substrate up to the atom, so the mutation in it means the ‚Äúend of career‚Äù of the enzyme as a catalyst.</i>  <i>In fact, of course, mutations occur anywhere in the molecule;</i>  <i>but proteins with a mutation in the functional region make their host unviable, therefore it seems to us that they do not exist.</i> <br><br>  In conditions when obtaining a complete three-dimensional structure of a protein is still in a certain sense a luxury, the ability to find such patterns is very valuable.  The ambush here is that we cannot get a mathematically optimal alignment of even a dozen proteins.  If a two-dimensional array is used to align two proteins, then a ten-dimensional, respectively, is needed to align 10 proteins.  Even if each sequence has a length of 100 characters (which corresponds to a very small squirrel), the dimension of the array is so huge (100 <sup>10</sup> ) that a modern computer will count it for thousands of years.  About how all the same biologists get out in this situation - I will tell, probably, another time. </div><p>Source: <a href="https://habr.com/ru/post/221813/">https://habr.com/ru/post/221813/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../221803/index.html">VPS Optimization Checklist for PHP / Mysql / Nginx</a></li>
<li><a href="../221805/index.html">Fast web application - network trepanation</a></li>
<li><a href="../221807/index.html">Benchmark 14 sorting algorithms on arrays with different dimensions and content</a></li>
<li><a href="../221809/index.html">Space free launch</a></li>
<li><a href="../221811/index.html">The story of parsing one aspx site</a></li>
<li><a href="../221815/index.html">We create a website using Laravel and Recurly. Part 2</a></li>
<li><a href="../221817/index.html">PocketBook 626 vs Sony PRS-T3: tested in humans</a></li>
<li><a href="../221819/index.html">Using Git hooks to block edits of published commits</a></li>
<li><a href="../221821/index.html">Macrorails on MSP430</a></li>
<li><a href="../221823/index.html">Crazy Cube World Game. Creature. Part 1</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>