<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Finishing the genome: fast, high quality, inexpensive</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="I think that many Habr readers have already heard about bioinformatics, perhaps even directly about the task of assembling the genome . Many people ar...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Finishing the genome: fast, high quality, inexpensive</h1><div class="post__text post__text-html js-mediator-article">  I think that many Habr readers have already heard about bioinformatics, perhaps even directly about the task of <a href="https://ru.wikipedia.org/wiki/%25D0%25A1%25D0%25B1%25D0%25BE%25D1%2580%25D0%25BA%25D0%25B0_%25D0%25B3%25D0%25B5%25D0%25BD%25D0%25BE%25D0%25BC%25D0%25B0">assembling the genome</a> .  Many people around the world are busy writing genomic assemblers - programs that interpret the raw data of machines for sequencing and produce the resulting DNA sequence of the studied organism.  However, in most cases, the whole genome "out of the box" can not be obtained.  In this article I will try to explain why the genome cannot be assembled with one click of the mouse and describe the process of its ‚Äúfinishing‚Äù - perhaps the most time-consuming step in the entire assembly, sometimes lasting several years. <br><br>  Also, I will tell you how we can sometimes significantly facilitate this process, using the already collected genomes of closely related organisms.  I was engaged in this task within the framework of writing my master's thesis at the <a href="http://mit.spbau.ru/">St. Petersburg Academic University</a> , and I studied in conjunction with the <a href="http://bioinformaticsinstitute.ru/">Institute of Bioinformatics</a> .  Since the resulting algorithm is quite specific, I will begin by describing the problem as a whole, I will give an overview of some of the "hardwired" methods for solving it, and then talk a little about what happened to me. <br><br><a name="habracut"></a><br><h4>  Introduction </h4><br>  So we want to assemble the genome.  On Habr√© there are already a couple of good publications on the topic of genomic assemblers: <a href="http://habrahabr.ru/company/spbau/blog/143115/">one</a> , <a href="http://habrahabr.ru/company/spbau/blog/186492/">two</a> . <br>  If bioinformatics is something new for you, I highly recommend starting over with them, since this article implies knowledge of some basic concepts. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Just in case, let me remind you that a genomic assembler (collector) is a program that accepts short (several hundred nucleotides) overlapping pieces of the genome, called reads, and assembles them into a single sequence.  More precisely, it tries to assemble - in most cases, even for relatively small bacterial genomes (several million nucleotides), the whole sequence does not work.  This happens for a number of reasons, described in the articles above.  I will briefly say that this happens because of the presence of repetitive pieces in the genome, areas that are physically difficult to read, as well as errors in the sequencing process. <br><br><img src="https://habrastorage.org/files/5ee/55e/e34/5ee55ee34b9545f7bd62df33fe0d9c76.png"><br><br>  As a result, instead of giving us the whole genome, we get its pieces, called contigs.  They are already much longer (tens and hundreds of thousands of nucleotides), and in their totality they make up the original sequence.  However, their correct order is unknown to us.  What to do next? <br><br>  To begin with, let us ask ourselves the question: why do we need a whole genome?  After all, the length of contigs is enough, for example, to search for genes inside them.  On the other hand, some genes will still be damaged due to fragmentation.  But, what is more important, for some studies it is the structure of the genome that is necessary  the order of the studied genes in it. <br><br>  For example, this is important when studying evolution.  As we well know, DNA mutates over time.  Most often, small changes occur: one nucleotide is replaced with another, or several consecutive nucleotides are cut / inserted into another place.  But there are much more rare, but serious changes.  For example, a large piece of the genome can roll over or "move" to another position.  It happens that whole chromosomes merge with each other, or vice versa, they break up into two parts.  In their <a href="http://dl.acm.org/citation.cfm%3Fid%3D796277">work,</a> Hannehalli and Pevzner showed that the human and mouse genomes are separating a total of 131 such rearrangements.  Agree, not so much. <br><br><h4>  Genome draft </h4><br>  Now that I am interested in you, let's get this genome together!  We have a set of contigs - its pieces, the order of which is unknown to us.  What will we do next?  We can be helped by the technology of targeted sequencing, which allows one to read a specific and relatively short (up to several thousand nucleotides) region of the genome containing a specific pattern at the ends.  For this, polymerase chain reaction ( <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0%25D0%25B7%25D0%25BD%25D0%25B0%25D1%258F_%25D1%2586%25D0%25B5%25D0%25BF%25D0%25BD%25D0%25B0%25D1%258F_%25D1%2580%25D0%25B5%25D0%25B0%25D0%25BA%25D1%2586%25D0%25B8%25D1%258F">PCR</a> ) is used, and then the resulting product is read by a much more expensive, but reliable, sequencing <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B5%25D1%2582%25D0%25BE%25D0%25B4_%25D0%25A1%25D1%258D%25D0%25BD%25D0%25B3%25D0%25B5%25D1%2580%25D0%25B0">method, the Sanger method</a> .  By running such a reaction and selecting the ends of two contigs of interest as a pattern, we can find out if they are adjacent in the genome, and also read the missing sequence between them.  If our assumption is not true, and the contigs are too far apart, the reaction simply will not work. <br><br>  But after all, we have hundreds (if we collect a bacterium) and thousands (mammals) of Contigs, respectively tens of thousands and millions of their possible "compounds."  There is not enough time or money to check each of them.  What do we do in this situation?  Maybe we can somehow assess which compounds are most likely to exist, and which ones should not even be checked? <br><br>  Of course we can!  There are various methods (some of which I will now discuss) that allow you to combine contigs into the so-called scaffold, which is their ordered set.  There are gaps of a certain length between contigs, symbolizing an unknown sequence: <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/ee1/691/300/ee16913001e54fb7af8a55e923446727.png"></div><br><br>  It is clear that having such a ‚Äúdraft‚Äù of the genome, it will be much easier for us to fill in the gaps in it.  Let's consider what are the ways of combining contigs into scaffolds (this process is called scaffolding). <br><br><h4>  Scaffolding technology </h4><br><h5>  Paired reading libraries </h5><br>  What could be better than a short read?  Yes, unless two are the same!  Let's cut the DNA that we have to sequence into pieces of some known length, and then read each one from the beginning and end.  Of course, most of the sequence between the ends will remain unread, but now we have a couple of readings, the distance between which we know. <br><br>  Such pairs of reads give us new additional information.  Let's "align" (align) them (that is, find an entry, possibly inaccurate) on our contigs.  If at the same time two reads are aligned to different contigs, then the latter can be combined into a single scaffold: <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/04f/92f/589/04f92f589ca54f85a63c8ded77bf4038.png"></div><br><br>  An important criterion of this technology is the distance between pairs of reads - the so-called insertion length.  The larger it is, the longer we can pass over the longer genome intervals.  However, with an increase in the length of the insert, the cost of the experiment increases significantly, as well as the number of errors. <br><br><h5>  Long reeds </h5><br>  Recently, technologies have begun to emerge that make it possible to produce much longer reads (up to several tens or even hundreds of thousands of base pairs) in relatively large quantities.  For example, this opportunity provides technology <a href="http://www.pacificbiosciences.com/">PacBio</a> .  However, in all of them so far there are two significant drawbacks: firstly, the high cost of the process, and secondly, the presence of a large number of errors. <br><br>  Collecting a genome with so many inaccuracies is another task.  Some went the other way - the creation of hybrid assemblies.  They combine the usual short reads, as well as long, but less accurate.  The idea is intuitively clear: at first, as usual, we perform the assembly using only short reads, and then align the resulting contigs with long reads.  This, as in the case of pair reads, gives us information about the location of contigs relative to each other.  Also, the part of the genome between them becomes available: <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/f4f/a8a/41f/f4fa8a41fdbe45a488fb9ba246bdf722.png"></div><br><br>  Using such a technique, not so long ago a group of scientists was able to fully automatically assemble the <a href="http://www.ncbi.nlm.nih.gov/pubmed/22750883">bacterial genome</a> .  Also, there is a noticeable success in the assembly using <a href="http://biorxiv.org/content/early/2014/08/14/008003">long reads</a> alone.  However, the whole technology, unfortunately, remains the lot of the elect because of its high cost and low prevalence. <br><br><h5>  Hi-c </h5><br>  Hi-C is a very fresh and promising technology that allows you to measure the interaction of pieces of the genome in space.  As you probably know, DNA is not just a long linear molecule, it is also packed in a complex way into a spatial structure, which is called a chromosome.  Without going into details, I will say that Hi-C allows you to get pairs of reads for which the corresponding pieces of the genome are close in space, but they are not necessarily close if we stretch the entire molecule in one line.  The picture below shows the intensity of the interaction of chromosome segments with each other (along both axes - linear coordinates on the chromosome): <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/e2c/dfa/c15/e2cdfac157d64059a5cd4bf8dfce9708.png"></div><br><br>  So we can get quite interesting information reflecting the form of DNA packaging in the cell.  How can this be used for scaffolding?  It's very simple: pieces of DNA that are linearly close to each other will, on average, more often interact with each other (which we see in the picture above).  Again, I will not indulge in long explanations and refer the reader to the <a href="http://www.ncbi.nlm.nih.gov/pubmed/24185095">original article</a> on this topic. <br><br><h4>  Reference Assembly </h4><br>  Finally, we very smoothly approached the topic of my work.  Since the advent of the first sequencing technologies, mankind has managed to accumulate quite a lot of data.  There are already a large number of fully (or almost completely) assembled genomes of various organisms.  Why not take advantage of this information, instead of carrying out expensive experiments? <br><br>  And you can use!  There are many purely computational methods under the common name ‚Äúreference-assisted assembly‚Äù, which involve the use of sequences of already assembled related organisms to improve the quality of the assembly of the new sample.  The basic idea here is similar to the one used for assembly with long reads: we find the alignment of the contigs now to the reference genome, and combine them in the appropriate order: <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/c48/b55/6cc/c48b556cc6f34bce9799bb9e05b1bcc4.png"></div><br><br>  But, as we know, evolution does not stand still, and the differences in the collected and reference genome can be significant.  First, it can lead to difficulties when searching for alignment.  But it's not so bad.  At the very beginning of the article, I mentioned large-scale genomic rearrangements.  It is clear that if such rearrangements between the collected and reference genome are present, then we risk getting a completely wrong result.  So, we need an algorithm that analyzes such restructuring and tries to "put everything in its place." <br><br>  I was writing this algorithm as my master's thesis.  Its main difference from existing solutions is the use of several reference genomes simultaneously.  This gives us additional information about how the genomes have changed in the course of evolution, and therefore allows us to more reliably assess the order of the Contigs in our genome. <br><br>  Without going into details, I will try to explain why several references are better than one.  Let's start with the fact that between all the genomes that are given to us at the entrance, there is some kind of evolutionary relationship.  Simply put, we can build a <a href="https://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B8%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B5%25D0%25BD%25D0%25B5%25D1%2582%25D0%25B8%25D1%2587%25D0%25B5%25D1%2581%25D0%25BA%25D0%25BE%25D0%25B5_%25D0%25B4%25D0%25B5%25D1%2580%25D0%25B5%25D0%25B2%25D0%25BE">phylogenetic tree</a> , in the leaves of which our genomes will be located, both reference - R and target - T. The tree shows their directed evolution from some common ancestor, and the length of the branches corresponds to the duration of the processes: <br><br><div style="text-align:center;"><img src="https://habrastorage.org/files/161/871/dfd/161871dfd88d48b180e34db3b21593a7.png"></div><br><br>  Now, knowing the structure of the reference genomes, we need to restore the target.  To do this, it is necessary to understand how the evolution of the whole family proceeded.  It is important for us to predict in which places of the phylogenetic tree genomic rearrangements took place, the consequences of which we observe in the leaves.  Knowing this, we will be able to determine which reorganizations occurred before, and which after the separation of the genome we are collecting into a separate branch.  So, we can and say which of them is most likely present in our assembly. <br><br>  According to the results of the <a href="http://bioinformatics.oxfordjournals.org/content/30/12/i302.abstract">article</a> was written, which became my master's thesis.  The program itself can be found on the githaba - <a href="http://fenderglass.github.io/Ragout/">fenderglass.github.io/Ragout</a> .  A detailed description of the algorithm, unfortunately, goes beyond the publication of the Habr format, and it would not be too clear without an additional background.  But if this article arouses the interest of the community, then in the future I will continue to write about the bioinformatic algorithms associated with this topic, and I will gradually delve into the field. <br><br>  <s>PS I also wanted to write in Bioinformatics, but I lacked karma.</s>  - already wrote </div><p>Source: <a href="https://habr.com/ru/post/238759/">https://habr.com/ru/post/238759/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../238749/index.html">Bioluminescent lamp with luminous bacteria inside</a></li>
<li><a href="../238751/index.html">Ten best cities in the world for remote teams and freelancers</a></li>
<li><a href="../238753/index.html">Box2D explosions</a></li>
<li><a href="../238755/index.html">Some interesting and useful things for web developer # 30</a></li>
<li><a href="../238757/index.html">Market Watch for Apple Watch - Pebble Update, Pebble Alliance and Jawbone</a></li>
<li><a href="../238761/index.html">Lumia SensorCore SDK: new opportunities for developing mobile applications. Part 1: Overview</a></li>
<li><a href="../238763/index.html">Microsoft introduced Windows 10</a></li>
<li><a href="../238765/index.html">The best publications of social networks</a></li>
<li><a href="../238771/index.html">‚ÄúBleed the mind. Inexpensive. Is there any sense of a mask for lucid dreams?</a></li>
<li><a href="../238775/index.html">Windows 10 Technical Preview is available for download.</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>