<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Protein Structure: An Introduction for IT Professionals</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="It is nice to see that habravchane regularly interested in other subject areas - for example, biology (more specifically, the structure and function o...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Protein Structure: An Introduction for IT Professionals</h1><div class="post__text post__text-html js-mediator-article">  It is nice to see that habravchane regularly interested in other subject areas - for example, biology (more specifically, the structure and function of biological macromolecules).  However, some posts (for example, <a href="http://habrahabr.ru/post/181055/">this one</a> ) cause a specialist simply physical pain due to the abundance of completely wild factual errors.  In this post I want to talk about the structure and function of the protein.  About what we know and about what we don‚Äôt know, as well as about the computational tasks in this area that require solutions and are interesting for IT specialists.  I will try to talk concisely and tezisno, to have more information, and less water.  Anyone interested in the structure of proteins, please under the cat, there are a lot of letters. <br><a name="habracut"></a><br><h4>  1. Why are proteins important? </h4><br>  As Friedrich Engels said, ‚ÄúLife is a way of existence of protein bodies‚Äù.  In the 19th century, they still did not know about the role of DNA in the inheritance of genetic information, but Uncle Frederick‚Äôs statement is still largely true - proteins do most of the work in our cells.  This includes maintaining the structure (cell shape), chemical catalysis, and motor function (muscle contraction, for example), and transport (say, hemoglobin protein transports oxygen from the lungs to the tissue and carbon dioxide in the opposite direction) and complex regulatory functions to maintain consistency internal environment (for example, protein hormones and all intracellular regulatory systems) and many others.  In a word, if something happens in our body, proteins are necessarily involved (though not only them). <br><br><h4>  2. What is protein? </h4><br>  From a chemical point of view, a protein is a linear (unbranched) polymer consisting of monotonously repeating identical blocks of the ‚Äúmain chain‚Äù, to which various ‚Äúside groups‚Äù are attached.  Since the main chain blocks are asymmetric, the entire polypeptide chain of the protein has a direction, the N- and C-termini of the polypeptide chain are distinguished. <br><img src="https://habrastorage.org/storage2/66f/525/70c/66f52570c22c785ce0efb1311c83582e.jpg"><br>  The chain length is from 70 to more than 1000 monomers (amino acid residues), the average length for higher organisms is about 500-600 amino acid residues, for bacteria this value will be less, most likely 300-400 residues.  In total, there are 20 standard amino acids in nature, the same for both bacteria and humans, that is, 20 different side groups can stick out from the main chain. <br><img src="https://habrastorage.org/storage2/fce/3a0/950/fce3a095025d7a3fab60aa27c2db6db3.jpg"><br>  (A correction is possible here - some chemical groups can be modified after protein synthesis, for example, phosphorylated. However, this is not considered as another amino acid, but is considered as a product of the original modification. Also, higher non-canonical amino acids are possible in higher organisms, but this is a rare event. That is, strictly speaking, 22 different amino acids, 20 of them are basic and 2 rare, plus some side groups can be rarely chemically modified). <br><br>  From generation to generation, genetic information is transmitted in the form of DNA, there are so-called ‚Äúprotein-coding regions‚Äù in it.  In these places, DNA unambiguously (for botans - up to alternative splicing and RNA editing) encoded information about the linear amino acid sequence for the synthesis of this protein, plus there are corresponding machines in the cell that can synthesize a protein according to information originally encoded in DNA. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Since protein is a linear polymer assembled from 20 standard monomers, its so-called ‚Äúprimary structure‚Äù can be easily represented as a string, for example: <br><br><pre> 
 &gt; small ubiquitin-related modifier 3 precursor [Homo sapiens]
 MSEEKPKEGVKTENDHINLKVAGQDGSVVQFKIKRHTPLSKLMKAYCERQG
 LSMRQIRFRFDGQPINETDTPAQLEMEDEDTIDVFQQQTGGVPESSLAGHSF
</pre><br><br>  This is the amino acid sequence of a small human protein in the FASTA format, the first line starting with ‚Äú&gt;‚Äù describes its name, followed by the amino acid sequence in accordance with the standard coding (for example, M is methionium, S is serine, etc., a total of 20 letters of standard single-letter code), on the left is the N-terminus of the protein, on the right is its C-terminus.  For different proteins, the length of the string will be obviously different, since proteins have different lengths.  The sequences of all known proteins can be found in the public domain here: <a href="http://www.ncbi.nlm.nih.gov/">www.ncbi.nlm.nih.gov</a> <br><br><h4>  3. Protein Structure </h4><br>  Well, the primary structure is sorted out, but does the protein work in an expanded linear form?  Of course not.  Here it should be noted that from a structural point of view there are different classes of proteins: globular, membrane and fibrillar.  Membrane proteins, as the name implies, live only in cell membranes, to stabilize their structure, you need a special membrane environment, we will not consider them in this review.  Fibrillar proteins have a simple regular structure, look like elongated fibers, they are insoluble in water and perform structural functions (for example, hair consists of keratin, fibrillar proteins include protein from natural silk).  Recently, they began to distinguish a class of disordered proteins ‚Äî proteins that do not have a constant three-dimensional structure, or that acquire it only for a short time when interacting with other proteins.  The most interesting from a practical point of view, the class of proteins, which we will consider - globular water-soluble proteins, most proteins belong to this class. <br><br>  A linear polypeptide chain in water is capable of spontaneously folding into a complex three-dimensional structure (globule) and only in such a folded form can proteins perform chemical catalysis and other interesting work.  Therefore, it is fundamentally important for us to know exactly the three-dimensional folding of the protein, since it is only at this level that it becomes clear how the protein works. <br><br>  <i>Question</i> : how many three-dimensional structures correspond to a specific protein? <br>  <i>Answer</i> : One, up to a small mobility of small ‚Äúdisordered‚Äù loops.  Only one exception is known, when two sufficiently different structures correspond to one sequence, these are <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D1%2580%25D0%25B8%25D0%25BE%25D0%25BD%25D1%258B">prions</a> . <br><br>  <i>Question</i> : Why does the protein have only one three-dimensional structure? <br>  <i>Answer</i> : for chemical catalysis, we need to arrange the corresponding chemical groups in a strictly defined way in space.  This requires a rigid structure.  That is, the whole protein must be rigid in order to maintain the chemical groups of amino acids of the active center in the right places (in reality, many proteins consist of two or more rigid parts that can move relative to each other, this is necessary for regulating the activity of the protein ( <a href="http://biohimija.ru/46/">allosteric regulation</a> ) so that a certain signal could turn on and off the chemical activity of the protein-enzyme).  To make the structure rigid and stable, nature made sure that the structure of each protein corresponded to the energy minimum of this system of atoms and this minimum was so deep that the protein did not ‚Äújump out‚Äù from it.  All other parasitic structures have more energy and the protein still falls into the energy minimum corresponding to the native structure. <br><img src="https://habrastorage.org/storage2/5f9/f50/0c2/5f9f500c26781bc39bb39821c2c256c2.jpg"><br><br>  <i>Question</i> : what keeps the three-dimensional structure of the protein? <br>  <i>Answer</i> : In short, mainly on a large number of non-covalent interactions.  In principle, chemical groups of a protein can form: (1) a hydrogen bond, these groups exist in the main chain and in some side groups, (2) an ionic bond is an electrostatic interaction between oppositely charged side groups, (3) Van der Waalsovo interaction and (4) the hydrophobic effect, which keeps the overall structure of the protein.  The point is that there are always hydrophobic aromatic residues in a protein, it is energetically unprofitable for them to contact with polar water molecules, and it is advantageous to ‚Äústick together‚Äù with each other.  Thus, when the protein is folded, the hydrophobic groups are pushed out of the water environment, ‚Äústicking together‚Äù with each other and forming a ‚Äúhydrophobic core‚Äù, while the polar and charged groups, on the contrary, tend to the water environment, forming the surface of the protein globule.  Also (5), the side groups of two cysteine ‚Äã‚Äãresidues can form a disulfide bridge between them ‚Äî a full-fledged covalent bond that rigidly fixes the protein. <br><br>  Accordingly, all amino acids are divided into hydrophobic, polar (hydrophilic), positively and negatively charged.  Plus cysteines capable of forming a covalent bond between each other.  Glycine has special properties - it does not have a side group that severely restricts the conformational mobility of other residues, so it can ‚Äúbend‚Äù very much and is located in places where the protein chain must be unfolded.  In Proline, on the contrary, the side group forms a ring covalently linked to the main chain, rigidly fixing its conformation.  Prolines are found where it is necessary to make the protein chain rigid and stiff.  Many diseases are associated with the mutation of proline to glycine, due to which the structure of the protein slightly ‚Äúfloats‚Äù. <br><br>  <i>Question</i> : how do we even know about the three-dimensional structures of the protein? <br>  <i>Answer</i> : from the experiment, this is absolutely reliable data. <br>  Now there are 3 methods for experimentally determining the structure of a protein: nuclear magnetic resonance (NMR), cryo-EM (electron microscopy), and x-ray diffraction analysis of protein crystals. <br><br>  NMR allows you to determine the structure of the protein in solution, but it works only for very small proteins (for large ones it is impossible to make deconvolution). <br><img src="https://habrastorage.org/storage2/d95/980/a61/d95980a612de68b2ed82112f59961235.jpg"><br>  This method was important for general evidence that a protein has only one three-dimensional structure and that the structure of a protein in a crystal is identical to the structure in solution.  This is a very expensive method, since it is required to obtain a protein with isotopic tags. <br><br>  Cryo-EM is a simple freezing of protein solution and microscopy.  The minus of the method is low resolution (only the general shape of the molecule is visible, but how it is arranged inside is not visible), plus the protein density is close to the density of water / solvent, therefore the signal sinks in a high noise level.  In this method, computer technologies for working with pictures and statistics are actively used to pull the signal out of noise. <br><img src="https://habrastorage.org/storage2/a40/d85/253/a40d852531dbc5a18906f889c1b9087d.jpg"><br>  Millions of protein molecules are selected, classes are divided according to the orientation of the molecule relative to the substrate, class averaging, eigenimages generation, a new round of averaging, and so on until it converges.  Then from the information from different classes you can restore the three-dimensional view of the molecule with a low resolution.  If there is an internal symmetry of the particles (for example, during cryo-EM analysis of viruses), then each particle can be averaged according to the symmetry operators - then the resolution will be even better, but worse than in the case of X-ray analysis. <br><br>  X-ray diffraction analysis is the main method for determining protein structures.  The main advantage is that it is potentially possible to obtain crystals of even very large complexes from many dozens of proteins (for example, this is how the structure of the ribosome was determined - the Nobel Prize of 2009).  The minus of the method - you first need to get a protein crystal, but not every protein wants to crystallize. <br><img src="https://habrastorage.org/storage2/1dd/cd6/7eb/1ddcd67eb1464ed2c1f6c23865948a53.jpg"><br>  But after the crystal is obtained, the x-ray diffraction can unambiguously determine the positions of all (ordered) atoms in the protein molecule, this method gives the highest resolution and allows you to see the positions of individual atoms in the best cases.  It was proved that the structure of the protein in the crystal is uniquely consistent with the structure in solution. <br><br>  Now the convention is in effect - if you determined the structure of the protein by any of the experimental physical methods, the structure should be placed in open access to the protein data bank (Protein Data Bank - PDB, <a href="http://www.pdb.org/">www.pdb.org</a> ), currently there are more than 90,000 structures (however, many of them are repetitive, for example, complexes of the same protein with different small molecules, such as drugs).  In PDB, all structures are in a standard format called, suddenly, pdb.  This is a text format in which each atom of the structure corresponds to one line, in which the number of the atom in the structure, the name of the atom (carbon, nitrogen, etc.), the name of the amino acid that includes the atom, the name of the protein chain (A, B, C, etc.) , if it is a crystal of a complex of several proteins), the number of amino acids in the chains and the three-dimensional coordinates of the atom in angstroms relative to the origin, plus the so-called temperature factor and population (these are purely crystallographic parameters). <br><br><pre> ATOM 1 N HIS A 17-12.690 8.753 5.446 1.00 29.32 N  
 ATOM 2 CA HIS A 17 -11.570 8.953 6.350 1.00 21.61 C  
 ATOM 3 C HIS A 17-10.274 8.970 5.544 1.00 22.01 C  
 ATOM 4 O HIS A 17-10.193 8.315 4.491 1.00 29.95 O  
 ATOM 5 CB HIS A 17-11.462 7.820 7.380 1.00 23.64 C  
 ATOM 6 CG HIS A 17-12.551 7.811 8.421 1.00 21.18 C  
 ATOM 7 ND1 HIS A 17-13.731 7.137 8.194 1.00 28.94 N  
 ATOM 8 CD2 HIS A 17-12.634 8.384 9.644 1.00 21.69 C  
 ATOM 9 CE1 HIS A 17-14.492 7.301 9.267 1.00 27.01 C  
 ATOM 10 NE2 HIS A 17-13.869 8.058 10.168 1.00 22.66 N  
 ATOM 11 N ILE A 18 -9.269 9.660 6.089 1.00 19.45 N  
 ATOM 12 CA ILE A 18-7.710 9.377 5.605 1.00 18.67 C  
 ATOM 13 C ILE A 18-7.122 8.759 6.749 1.00 16.24 C  
 ATOM 14 O ILE A 18-7.425 8.919 7.929 1.00 18.80 O  
 ATOM 15 CB ILE A 18-7.228 10.640 5.088 1.00 20.22 C  
 ATOM 16 CG1 ILE A 18-7.062 11.686 6.183 1.00 18.52 C  
 ATOM 17 CG2 ILE A 18-7.781 11.176 3.889 1.00 24.61 C  
 ATOM 18 CD1 ILE A 18-6.161 12.824 5.749 1.00 28.21 C  
 ATOM 19 N ASN A 19-6.121 8.023 6.349 1.00 15.46 N  
 ATOM 20 CA ASN A 19 -5.239 7.306 7.243 1.00 14.34 C  
 ATOM 21 C ASN A 19-4.012 8.178 7.507 1.00 14.83 C  
 ATOM 22 O ASN A 19-3.431 8.715 6.575 1.00 18.03 O  
 ATOM 23 CB ASN A 19-4.825 6.003 6.573 1.00 17.71 C  
 ATOM 24 CG ASN A 19-6.062 5.099 6.413 1.00 21.26 C  
 ATOM 25 OD1 ASN A 19,66.606 4.651 7.400 1.00 26.18 O  
 ATOM 26 ND2 ASN A 19-6.320 4.899 5.151 1.00 31.73 N  
</pre><br><br>  Then there are special programs that, according to the data from this text file, can graphically display the beautiful three-dimensional structure of the protein molecule, which can be twisted on the monitor screen and, as <a href="http://www.york.ac.uk/chemistry/staff/academic/emeritus/gdodson/obituary/">Guy Dodson</a> said, ‚Äútouch the molecule with the mouse‚Äù (for example, <a href="http://www.pymol.org/">PyMol</a> , <a href="http://www.ccp4.ac.uk/MG/">CCP4mg</a> , old <a href="http://www.umass.edu/microbio/rasmol/">RasMol</a> ) .  That is, it is easy to look at protein structures - you put the program, load the necessary structure from the PDB and enjoy the beauty of nature. <br><br><h4>  4. Analyzing the structure </h4><br>  So, we understand the basic idea: protein is a linear polymer that collapses in an aqueous solution under the action of many weak interactions into a stable and unique three-dimensional structure for this protein, and is capable of performing its function in this form.  There are several levels of organization of protein structures.  Above, we have already become acquainted with the primary structure - a linear sequence of amino acids, which can be written out in a line. <br><img src="https://habrastorage.org/storage2/582/338/5c9/5823385c919d97857d8224d8a0e23d2d.jpg"><br><br>  The secondary structure of the protein is determined by the interaction of the atoms of the main protein chain.  As mentioned above, the main chain of the protein includes donors and hydrogen bond acceptors, thus, the main chain can acquire some structure.  More precisely, several different structures (details nevertheless depend on differing side groups), since the formation of different alternative hydrogen bonds between the main chain groups is possible.  Structures are such: alpha-helix, beta sheets (consisting of several beta strands), which are parallel and anti-parallel, beta turn.  Plus, part of the chain may not have a pronounced structure, for example, in the region of the rotation of the protein loop.  These types of structures have their well-established schematics - an alpha-helix in the form of a spiral or a cylinder, beta-strands in the form of wide arrows.  The secondary structure can be reliably predicted by the primary (the standard is <a href="http://www.compbio.dundee.ac.uk/www-jpred/">JPred</a> ), the alpha-helix is ‚Äã‚Äãpredicted most accurately, with beta strands are lining. <br><br>  The tertiary structure of the protein is determined by the interaction of the side groups of amino acid residues, this is the three-dimensional structure of the protein.  One can imagine that the secondary structure is formed and now these spirals and beta strands want to fit together in a compact three-dimensional structure so that all hydrophobic side groups calmly ‚Äústick together‚Äù together in the depths of the protein globule, forming a hydrophobic core, and the polar and charged residues stuck out out into the water, forming the surface of the protein and stabilizing the contacts between the elements of the secondary structure.  The tertiary structure is schematically depicted in several ways.  If you just draw all the atoms, you get porridge (although when we analyze the active center of the protein, we want to look at all the atoms of the active residues). <br><img src="https://habrastorage.org/storage2/6f5/9ec/791/6f59ec7915fd4d5d09e37bd09a12e68a.jpg"><br>  If we want to see how the whole protein works in general, we can display only some atoms of the main chain to see its progress.  Alternatively, you can draw a beautiful scheme, where elements of the secondary structure are schematically drawn over the real arrangement of atoms - this is how the protein is seen at a glance.  After studying the entire structure in a general, schematic form, you can display the chemical groups of the active center and focus on them.  The task of predicting the tertiary structure of a protein is nontrivial and, in general, cannot be solved, although it can be solved in particular cases.  More details below. <br><br>  The quaternary structure of the protein - yes, there is one, though not all proteins.  Many proteins work by themselves (monomers, in this case, the monomer means a single folded polypeptide chain, that is, the whole protein), then their quaternary structure is equal to the tertiary one.  However, a lot of proteins work only in a complex consisting of several polypeptide chains (subunits or monomers - dimers, trimers, tetramers, multimers), then this assembly of several separate chains is called a quaternary structure.  The most commonplace example is <a href="">hemoglobin consisting of 4 subunits</a> , the most beautiful example in my opinion is the <a href="">bacterial protein TRAP</a> consisting of 11 identical subunits. <br><br><h4>  5. Computational problems </h4><br>  Protein is a complex system of thousands of atoms, so without using computers in the structure of the protein can not figure out.  There are many tasks, both solved at an acceptable level and completely unsolved.  I will list the most relevant: <br><br>  <u>At the level of the primary structure</u> - the search for proteins with a similar amino acid sequence, the construction of evolutionary trees on them, and so on - the classic tasks of bioinformatics.  The main hub is NCBI - The National Center for Biotechnology Information, <a href="http://www.ncbi.nlm.nih.gov/">www.ncbi.nlm.nih.gov</a> .  BLAST is standardly used to search for proteins with a similar sequence: <a href="http://blast.ncbi.nlm.nih.gov/Blast.cgi">blast.ncbi.nlm.nih.gov/Blast.cgi</a> <br><br>  Prediction of protein solubility.  The point is that if we read the genome of an animal, determine protein sequences using it, deflect these genes into E. coli or the baculovirus expression system, then it turns out that when expressed in these systems, about a third of the proteins will not fold into the correct structure. and, as a result, will be insoluble.  It turns out that large proteins actually consist of separate ‚Äúdomains‚Äù, each of which represents an autonomous, functional part of a protein (carrying one of its functions) and often ‚Äúcutting out‚Äù a separate domain from a gene, you can get a soluble protein, determine its structure and experiment with it.  People try to use machine learning (neural networks, SVM and other classifiers) to predict protein solubility, but it works quite badly (Google will show a lot of things on request to ‚Äúprotein solubility prediction‚Äù - there are many servers, but in my experience they all work disgustingly on my squirrels).  Ideally, I would like to see a service that would reliably say where those soluble domains are in the protein so that they can be cut out and work with them - there is no such service. <br><br>  <u>At the level of the secondary structure</u> - the prediction of the same secondary structure by the primary ( <a href="http://www.compbio.dundee.ac.uk/www-jpred/">JPred</a> ) <br><br>  <u>At the level of the tertiary structure</u> - the search for proteins with similar three-dimensional structures ( <a href="http://ekhidna.biocenter.helsinki.fi/dali_server/start">DALI</a> , <a href="http://en.wikipedia.org/wiki/Structural_alignment">en.wikipedia.org/wiki/Structural_alignment</a> ), <br>  Search for structures in a given sub-structure.  For example, I have an arrangement of three amino acids of the active center in space.  I want to find structures that contain the same three amino acids in the same relative arrangement, or find structures of proteins, mutation of which will make it possible to arrange the necessary amino acids in the right way.  (google "protein substructure search") <br>  Prediction of potential mobility of a three-dimensional structure, possible conformational changes - normal mode analysis, <a href="http://www.igs.cnrs-mrs.fr/elnemo/">ElNemo</a> . <br><br>  <u>At the level of the quaternary structure</u> - suppose the structures of the two proteins are known.  It is known that they form a complex.  Predict the structure of the complex (determine how these two proteins will interact through shape matching, for example).  Google "protein-protein docking" <br><br><h4>  6. Prediction of protein structure </h4><br>  He singled out this computational problem into a separate section, because it is large, fundamental and not solved in the general case. <br><br>  Experimentally, we know that if you take a protein, fully deploy it and throw it into water, it will roll back to its original state within a time from milliseconds to seconds (this statement is true at least for small globular proteins without any pathologies).  This means that all the information needed to determine the three-dimensional structure of a protein is implicitly contained in its primary sequence, so you want to learn how to predict the three-dimensional structure of a protein by the amino acid sequence <i>in silico</i> !  However, this problem is generally not solved so far.  What is the matter?  The fact is that in the primary sequence there is no explicit information necessary for the construction of the structure.  Firstly, there is no information about the conformation of the main chain - and it has significant mobility, although it is somewhat limited for steric reasons.  Plus, each side chain of each amino acid may be in different conformations, for long side groups such as arginine, it may be more than a dozen conformations. <br><br>  What to do?  There is a fairly well-known habravchanam the most common approach, called "molecular dynamics" and suitable for all molecules and systems.   ,      ,    ,    ,       ,   .    ?                  ,   ,   .     ‚Äì   ,      ,  ‚Äì   . ,       .    ,                            ,     ( <a href="http://en.wikipedia.org/wiki/Anton_(computer)"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">http://en.wikipedia.org/wiki/Anton_(computer)</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> )! However, it is too early to celebrate victory. They took a very small (its size is 5-10 times less than the average protein) and one of the fastest-folding proteins, the classic model protein, on which folding was studied. For large proteins, the calculation time increases nonlinearly and will take years, that is, there is still something to work on. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Another approach is implemented in </font></font><a href="https://www.rosettacommons.org/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Rosetta</font></font></a> .        (3-9 )   ,        PDB,    -     ,  .   - ,            ,    : ¬´    ,  -     ?¬ª. <br><br>       ‚Äì         ,   .       <a href="http://fold.it/portal/">FoldIt</a> ,       , -   (   ‚Äì !).        ,  <a href="http://predictioncenter.org/index.cgi">CASP</a> .   ,       ,     PDB,        PDB,         <a href="http://predictioncenter.org/index.cgi">CASP</a> .   ,      ,         ,    .  ,   <a href="http://fold.it/portal/">FoldIt</a> ,   , -  <a href="http://predictioncenter.org/index.cgi">CASP</a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">among professionals modeling protein structures and predicted protein structure more accurately. However, even these successes do not suggest that the problem of predicting the structure of the protein is solved - very often the model is very far from the real structure. </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">All this concerned </font><i><font style="vertical-align: inherit;">ab initio</font></i><font style="vertical-align: inherit;"> protein modeling.</font></font><i><font style="vertical-align: inherit;"></font></i> ,       .     ,      PDB        .        . ,          30%     (      ,         ).     ( )   ,   ¬´ ¬ª,    ¬´¬ª       ,     ,  -    .          ,    ‚Äì   .     ‚Äì <a href="http://salilab.org/modeller/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">Modeller</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> , </font></font><a href="http://swissmodel.expasy.org/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">SwissModel</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> . </font></font><br><br><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">You can solve other problems, for example, try to model what will happen if you bring this or that mutation into the protein. For example, if you replace the hydrophilic amino acid on the surface of a protein with another hydrophilic one, then most likely the structure of the protein will not change at all. If you replace the amino acid from the hydrophobic nucleus with another hydrophobic one, but of a different size, then most likely the protein placement will remain the same, but will slightly ‚Äúeat up‚Äù by an angstrom. If we replace the amino acid from the hydrophobic nucleus with the charged one, then most likely the protein simply ‚Äúexplodes‚Äù and cannot coagulate.</font></font><br><br>  ,              . ,   -,         ,       ‚Äì      <a href="http://phys.protres.ru/lectures/protein_physics/index.html">   ¬´ ¬ª</a> .           ¬´      ?¬ª, ¬´     ?¬ª, ¬´     ?¬ª. <br><br>    :        ,   .    ,    ,       .                  2-3 ,           .             ‚Äì             3 . ,    .   ,      ,          ¬´/¬ª   ,        ,         .      ,         .    , ¬´   ,   ¬ª. <br><br>  ,  ,   ,      . <br><br>      ,  ,    . <br><br>         : <br> 1) ¬´ ¬ª   .         ,      : <a href="http://phys.protres.ru/lectures/protein_physics/index.html">phys.protres.ru/lectures/protein_physics/index.html</a> (     ) <br> 2) , ¬´  ¬ª,   II ¬´ ¬ª.      Djvu <br> 3)  ,     ,    PubMed ( <a href="http://www.pubmed.org/"><font style="vertical-align: inherit;"><font style="vertical-align: inherit;">www.pubmed.org</font></font></a><font style="vertical-align: inherit;"><font style="vertical-align: inherit;"> ) - he should be asked to read about "protein engineering" and the like.</font></font></div><p>Source: <a href="https://habr.com/ru/post/181850/">https://habr.com/ru/post/181850/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../181838/index.html">Using CMake with Qt 5</a></li>
<li><a href="../181840/index.html">A module that simplifies working with OpenLayers and monitoring systems</a></li>
<li><a href="../181842/index.html">UN expert urged the world community to slow down the creation of combat robots with artificial intelligence</a></li>
<li><a href="../181844/index.html">Creating multiboot HDD in linux (GRUB2 + memdisk + grub4dos)</a></li>
<li><a href="../181848/index.html">Configuring the working environment for Backend PHP development</a></li>
<li><a href="../181852/index.html">We use Twitter for its intended purpose</a></li>
<li><a href="../181854/index.html">How to stop noticing the limited traffic with free WiFi</a></li>
<li><a href="../181858/index.html">Course lectures "Startup". Peter Thiel. Stanford 2012. Session 14</a></li>
<li><a href="../181860/index.html">6 hours with MegaFon Login or looking for a budget smartphone with android for a vacation trip</a></li>
<li><a href="../181864/index.html">Graphene photosensor is 1000 times more sensitive to light than CMOS and CCD</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>