<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Bioinformatic Pipeline with Docker</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In this article, I want to share the experience of developing the pipeline using Docker to analyze biomedical data. Probably, one reader will be inter...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Bioinformatic Pipeline with Docker</h1><div class="post__text post__text-html js-mediator-article">  In this article, I want to share the experience of developing the <a href="https://en.wikipedia.org/wiki/Pipeline_(computing)">pipeline</a> using Docker to analyze biomedical data.  Probably, one reader will be interested in the bioinformatical pipeline itself, and on others - the use of the Docker, so we divide the article into two parts. <a name="habracut"></a><br><br><h2>  Part 1. Bioinformatical pipeline, or what we did and why. </h2><br>  Technologies for reading the DNA sequence of living organisms are gradually developing.  Scientists are learning more and more what parts of DNA are responsible for what and how they affect the body in general.  In all these works there is a great medical potential.  Aging, cancer, genetic diseases - DNA sequence analysis can be a powerful tool in combating them.  We built a pipeline that analyzes the sequence of some parts of a person‚Äôs DNA and tries to predict whether it can have cardiomyopathy, a genetically determined heart disease. <br><br>  Why are only some parts of DNA taken?  Reading all the DNA (which is 3.2 billion nucleotides or "letters") will cost much more.  And in order to understand whether a particular person has ‚Äúerrors‚Äù that lead to one or another ‚Äúgenetically determined‚Äù disease, it is enough to read only those areas that, so to speak, affect the development of these diseases.  And it is much cheaper. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      How do scientists know which parts of the DNA to read?  The answer is: comparing the genomes of healthy and sick people.  These data are quite reliable, because today mankind knows the complete <a href="http://www.internationalgenome.org/">genomes of more than a thousand people</a> around the world. <br><br>  When readings on the necessary DNA segments are obtained, it is required to make a forecast whether the owner is waiting for a disease or not.  That is, to understand the sequence in these areas such as in a healthy person, or in them there are deviations that occur in sick people. <br><br>  Quite a lot of research is being done on such studies, therefore there are <a href="https://software.broadinstitute.org/gatk/best-practices/">best practices</a> on how to do this.  They describe in detail how the cleared data align to the reference genome (that is, the genome of some abstract healthy person), how to find the ‚Äúsingle letter‚Äù differences between them, and then analyze these differences: weed out the insignificant, and others look in the biomedical databases.  For all these actions, the bioinformatics community has developed many programs: bwa, gatk, annovar, etc.  The pipeline is designed so that the output of the desired program is fed to the input of the next one, and so on, until the desired result is obtained.  There are many ways to implement pipeline, in our work, inspired by the excellent course ‚Äú <a href="https://stepik.org/course/1612/syllabus">Management of Computing</a> ‚Äù, we used snakemake. <br><br><img src="https://pp.userapi.com/c824201/v824201992/7956d/EQIvyCWV7hE.jpg" alt="biopipeline"><br><br>  With the help of the pipeline, we analyzed data for one family, some of whose members were diagnosed with cardiomyopathy (in the figure they are in the red box).  Variations were found (that is, deviations from the reference genome), which, according to medical bases, are found in people with this disease (in the figure they are indicated by blue and green colors). <br><br><img src="https://pp.userapi.com/c824201/v824201992/7955a/Nl1iWJHHGWM.jpg" alt="image"><br><br>  What conclusions can be made from all this?  As expected, the pipeline itself cannot deliver a diagnosis.  He can only give information according to which the doctor decides whether the person is at risk for sickness or not.  This ambiguity is associated with the fact that cardiomyopathy is a complex disease, depending on both genetic and external factors.  The biochemical mechanisms of its occurrence are not known (all this is difficult), so it is impossible to say exactly which sets of variations will lead to the disease.  All that is is statistics on the sick and healthy, which allows the doctor to assess the probability of the disease and, if necessary, start treatment in time. <br><br>  We also attempted to assess the quality of the work of the pipeline.  As mentioned above, the pipeline finds variations - ‚Äúsingle-letter‚Äù deviations of the DNA sequence of a person from the reference genome.  Then he analyzes them and searches for information on them in biomedical databases.  The most ambiguous stage that requires fine-tuning is finding these variations.  Here we need to find a balance between redundancy - when there are too many variations, most of which are garbage, and insufficiency - when variations are chosen so strictly that we lose the necessary information.  Therefore, quality testing has come down to checking how the pipeline finds variations in the data about which we know the ‚Äúcorrect answer‚Äù.  As this data, <a href="https://github.com/genome-in-a-bottle/about_GIAB">Genome in a Bottle</a> was taken - a certain as accurately as possible read by the human genome, for which there is reliable data on variations.  The result of the quality check gave an 85% match, which is pretty good. <br><br><h2>  Part 2. Using Docker </h2><br>  If you express the main idea of ‚Äã‚Äãthis article in one sentence, it would be like this: ‚ÄúUse Docker in your pipelines, it is much more convenient with it‚Äù.  Indeed, what problems do people usually think of using something with pipelines?  If the pipeline is on your work computer, you can inadvertently change the environment or the dependencies of the programs used, automatic updates are possible - all this can lead to what the pipeline will consider a little differently from earlier or start producing errors.  Also, the deployment of the pipeline on a new computer can be problematic: you need to install all the programs, again, keep track of versions and dependencies, take into account the operating system.  With Docker using all these problems will not, and to run the pipeline on a new computer, you will not have to install anything at all (except Docker). <br><br>  The idea behind Docker is that each program used by the pipeline will run in an isolated container, in which the developer builds the necessary dependencies and environments.  He describes everything he needs in the corresponding Dockerfile, then the docker build command builds an image of the container that can be downloaded to dockerhub.  When someone (pipeline or another user) wants to use this program with these dependencies, he simply downloads the desired image with dockerhub and the docker create command creates the necessary container on his computer. <br><br><img src="https://pp.userapi.com/c824201/v824201992/79563/qVI00Zf_OkE.jpg" alt="docker"><br><br>  Our docker pipeline is available on <a href="https://github.com/phedo/biopipeline">github</a> .  Each time, calling any program, the pipeline runs the corresponding container, passes the necessary parameters to it and the calculation goes.  In fact, all the work of the programmer was to write a Dockerfile for each container.  It indicates the base image (FROM), which commands to execute in the specified image (RUN), or which files to add (ADD), you can specify the working directory (WORKDIR), into which, when the container is started, the folder with the data necessary for calculations is mounted.  An image is created based on the Dockerfile: <br><br><pre><code class="bash hljs">$ docker build -t imagename .</code> </pre> <br>  And it is loaded into any repository, for example, <a href="https://hub.docker.com/">dockerhub.com</a> . <br><br>  We describe some typical cases for our pipeline.  Read more about the Dockerfile on the <a href="https://docs.docker.com/engine/reference/builder/">official website</a> : <br><br>  You need to run the standard program installed from the repositories, for example, picard-tools.  The dockerfile will be: <br><br><pre> <code class="hljs sql">FROM ubuntu:14.04 RUN apt-get <span class="hljs-keyword"><span class="hljs-keyword">update</span></span> &amp;&amp; apt-<span class="hljs-keyword"><span class="hljs-keyword">get</span></span> <span class="hljs-keyword"><span class="hljs-keyword">install</span></span> -y picard-tools \ &amp;&amp; mkdir /home/<span class="hljs-keyword"><span class="hljs-keyword">source</span></span> WORKDIR /root/<span class="hljs-keyword"><span class="hljs-keyword">source</span></span></code> </pre> <br><pre> <code class="bash hljs">$ docker run -it --rm -v $(<span class="hljs-built_in"><span class="hljs-built_in">pwd</span></span>):/root/<span class="hljs-built_in"><span class="hljs-built_in">source</span></span> picard-tools picard-tools MarkDuplicates INPUT={input} OUTPUT={output[0]} METRICS_FILE={output[1]}</code> </pre> <br>  You need to run your shell script, annotation_parser.sh, which parses the file.  To do this, you can use the standard docker-style ubuntu: <br><br><pre> <code class="bash hljs">$ docker run -it --rm -v $(<span class="hljs-built_in"><span class="hljs-built_in">pwd</span></span>):/root/<span class="hljs-built_in"><span class="hljs-built_in">source</span></span> -w=<span class="hljs-string"><span class="hljs-string">"/root/source"</span></span> ubuntu:16.04 /bin/sh scripts/annotation_parser.sh {input} {output}</code> </pre> <br>  You need to run a script that is not in the standard repositories.  Dockerfile: <br><br><pre> <code class="hljs sql">FROM ubuntu:16.04 RUN apt-get <span class="hljs-keyword"><span class="hljs-keyword">update</span></span> &amp;&amp; apt-<span class="hljs-keyword"><span class="hljs-keyword">get</span></span> <span class="hljs-keyword"><span class="hljs-keyword">install</span></span> -y perl &amp;&amp; apt-<span class="hljs-keyword"><span class="hljs-keyword">get</span></span> <span class="hljs-keyword"><span class="hljs-keyword">install</span></span> -y wget \ &amp;&amp; mkdir /root/<span class="hljs-keyword"><span class="hljs-keyword">source</span></span> <span class="hljs-keyword"><span class="hljs-keyword">ADD</span></span> annovar /root/annovar ENV <span class="hljs-keyword"><span class="hljs-keyword">PATH</span></span>=<span class="hljs-string"><span class="hljs-string">"/root/annovar:${PATH}"</span></span> WORKDIR <span class="hljs-string"><span class="hljs-string">"/root/source"</span></span></code> </pre> <br><pre> <code class="bash hljs">$ docker run -it --rm -v $(<span class="hljs-built_in"><span class="hljs-built_in">pwd</span></span>):/root/<span class="hljs-built_in"><span class="hljs-built_in">source</span></span> annovar table_annovar.pl {input} reference/humandb/ -buildver hg38 -out {params.name} -remove -protocol refGene,cytoBand,exac03,avsnp147,dbnsfp30a -operation gx,r,f,f,f -nastring . -vcfinput</code> </pre> <br>  You need to run a java-application.  Here we use <a href="https://docs.docker.com/engine/reference/builder/">ENTRYPOINT</a> (https://docs.docker.com/engine/reference/builder/#entrypoint), which allows us to run the container as an executable file. <br><br><pre> <code class="hljs pgsql"><span class="hljs-keyword"><span class="hljs-keyword">FROM</span></span> ubuntu:<span class="hljs-number"><span class="hljs-number">16.04</span></span> RUN apt-<span class="hljs-keyword"><span class="hljs-keyword">get</span></span> <span class="hljs-keyword"><span class="hljs-keyword">update</span></span> &amp;&amp; apt-<span class="hljs-keyword"><span class="hljs-keyword">get</span></span> install -y <span class="hljs-keyword"><span class="hljs-keyword">default</span></span>-jre \ &amp;&amp; mkdir /home/source <span class="hljs-keyword"><span class="hljs-keyword">ADD</span></span> GenomeAnalysisTK.jar /root/GenomeAnalysisTK.jar WORKDIR "/root/source" ENTRYPOINT ["/usr/bin/java", "-jar", "/root/GenomeAnalysisTK.jar"]</code> </pre><br><pre> <code class="bash hljs">$ docker run -it --rm -v $(<span class="hljs-built_in"><span class="hljs-built_in">pwd</span></span>):/root/<span class="hljs-built_in"><span class="hljs-built_in">source</span></span> gatk -R {input[0]} -T HaplotypeCaller -I {input[1]} -o {output}</code> </pre> </div><p>Source: <a href="https://habr.com/ru/post/346184/">https://habr.com/ru/post/346184/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../346172/index.html">Moving to Japan</a></li>
<li><a href="../346174/index.html">Analyzing local functions in C # 7</a></li>
<li><a href="../346176/index.html">Defer: from Go to PHP</a></li>
<li><a href="../346178/index.html">640 KB is really enough for everyone</a></li>
<li><a href="../346180/index.html">Algorithm Paccos. Understandable article on consensus in a distributed system</a></li>
<li><a href="../346186/index.html">Practice writing tests. Yandex lecture</a></li>
<li><a href="../346188/index.html">Web messengers and the event 'beforeunload': how to save a million messages when you close the page</a></li>
<li><a href="../346190/index.html">Choosing a data center in St. Petersburg, practical experience of choice</a></li>
<li><a href="../346194/index.html">How to identify and develop talents in IT</a></li>
<li><a href="../346196/index.html">Vim: Search Javascript documentation</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>