<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Darwinian Evolution of Bacteria - The Complete Picture</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="I will start with a provocative statement - "biologists do not publish the details of their research." It would seem that there are so many articles, ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Darwinian Evolution of Bacteria - The Complete Picture</h1><div class="post__text post__text-html js-mediator-article">  I will start with a provocative statement - "biologists do not publish the details of their research."  It would seem that there are so many articles, so many studies ... but where is the description and detail of the information that is obtained?  It is not in principle.  And articles without such information are empty and controversial.  Everyone praises his method, but how many people have bothered to verify the data of others, and most importantly, could he do it? <br><br>  We can only welcome the appearance of such bio-information bases as the <a href="http://www.ncbi.nlm.nih.gov/genome">NCBI genomes</a> and <a href="http://www.rcsb.org/pdb/home/home.do">PDB</a> , in which researchers place data on the sequenced genomes and structures of RNA and proteins.  And most importantly, some scientists, before publishing an article, first put the data into bioinformatics databases. <br><br>  You tell me there are many other bases - but I‚Äôll tell you they are less serious, and as a rule these are two posts with some adaptation.  But the main thing is that all other bio-informational information, it may be said secondary, does not fit into the bases.  And in the articles, however, there are various speculations. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Of course, it looks like this only for amateurs like me.  Real professionals have everything in a pharmacy.  Therefore, you can not bother to answer these pretentious statements.  We'll just talk about what bioinformatics looks like in its private areas through the eyes of an amateur.  But maybe this story will lead you to something. <br><br>  We will discuss below the construction of an evolutionary tree according to Darwin, let's look at how much this is true and I will eventually give a complete tree (within the available information) of the evolution of bacteria based on the most conservative tRNA genes.  And I will give an explanation about the method of constructing such a tree. <br><br>  Experts in bioinformatics recommend reading from section number 5, skipping all my pathos. <br><br><a name="habracut"></a><br><br><h4>  ‚Ññ1.  Taxonomy </h4><br><br>  Being an amateur, I always wondered about one thing - how can organisms be classified and systematized when there is no information about DNA, when strains of organisms are not yet sequenced?  And please, the <a href="http://ru.wikipedia.org/wiki/Bergey%27s_Manual_of_Systematic_Bacteriology">Burgey Handbook</a> only in the last edition began to take into account information about genes.  And before that, it took into account only the structural and functional characteristics of bacteria. <br><br>  I'm not talking about those conservative biologists who say in all seriousness that taxonomy should be built not only on the basis of a genomic comparison, but also on the basis of morphological and physiological data.  And this is the gene century, we must return to the time of C. Linnaeus? <br><br>  But in the absence of a more authoritative publication than the Burgey Handbook, bio-information bases on taxonomy, such as <a href="http://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi%3Fmode%3DUndef%26name%3DBacteria%26lvl%3D3%26srchmode%3D1%26keep%3D1%26unlock">in the NCBI</a> , are more complete and sometimes have links to sequenced strains ‚Äî the principle of building such a tree is just a repost of the Burgey reference book. <br><br>  Say no ... ok, you can easily find the difference.  But you will never understand why a tree is exactly what it is.  It is certainly attributed to this or that species, who gave such a name to the taxon, and if the article is lucky, and if it‚Äôs very lucky, the article will briefly describe why this taxon was placed one way or another in systematics. <br><br>  Further, if we take individual articles on the construction of phylogenetic trees - in them, at best, a very small number of species are considered, and the trees are built using completely transparent methods and not large enough. <br><br><h4>  ‚Ññ2.  The problem of amateur </h4><br><br>  There are many professionals who try to make it so that the problem of an amateur is his lack of education and under-awareness. <br><br>  This is partly true, but only partly.  Amateurs are not doing their job, because having their profession - they are also interested in things from others and think in what other area they can apply their knowledge.  And when they see something like this as I described above for taxonomy, they are confused. <br><br>  They take the most naive method, since they need a result, not a reason to write an article, and build an evolution tree.  Then the professionals begin to resent how so - they are doing it professionally, but there are no results ... the grants are not all used.  Although it is possible to take one person to build it all without any particular difficulties and without scoring his head with methods in which complexity is introduced for the sake of complexity.  And so it turns out the result of the amateur. <br><br>  It can be discussed, but it can be discussed seriously only when professionals have at least something comparable and equally transparent.  And now we come to this. <br><br><h4>  Number 3.  Multicultural origin and other nonsense </h4><br><br>  Who has read my previous articles knows that I have already written on this topic since the article <a href="http://habrahabr.ru/post/146331/">Interesting results about the evolutionary systematics of prokaryotes or ‚Äúmultiple species‚Äù</a> , and not so long ago gave more complete results in the article <a href="http://habrahabr.ru/post/162729/">Systematics of prokaryotes - distant relatives</a> .  Here I would like to tell you how my worldview has changed as this study progressed. <br><br>  Initially, the article showed that on the basis of one type of tRNA that transferred alanine, one can find a stable connection between different species, genera, etc.  I interpreted this connection as sexual inheritance, since  it was possible to find organisms in which tRNA_Ala existed from two other species.  There were some exceptions, but they were relatively few.  ‚ÄúWhat could be simpler,‚Äù the amateur exclaimed, ‚Äúit's the same genes from mom and dad, and biologists fool us with asexual reproduction.‚Äù <br><br>  My critics hardly noticed this idea at that time (apparently writing off the horizontal transfer - although the mum-dad communication was very constant), but noted that it was not serious to draw conclusions based on one gene. <br><br>  I readily agreed, but I thought to myself - how many genes do you analyze yourself?  Correctly, as a rule, one 16S, only it will be longer, but it will be rugged with mutations.  But what can we compare with others ... go ahead. <br><br>  Then I took all the sequenced genomes and all the tRNAs available in the NCBI and clarified the information (see. <a href="http://habrahabr.ru/post/162729/">Systematics of prokaryotes - distant relatives</a> ). <br><br>  Critics did not become less, but she became more emotional.  Yeah, I thought it becomes harder to object, and the arguments of opponents are far from being considered and indirect. <br><br>  But I saw that the whole picture became very confusing, there was a feeling that the genera interact somewhere weaker, somewhere stronger - but almost like everyone with each.  One or another type of gene they had in common. <br><br>  Imagine that this could really be evolution.  as if all the genes were thrown into one pot, and then they would scoop a random set from this pot and create a look ‚Äî it was somehow difficult.  But the results talked about it inexorably. <br>  Different genera, although slightly distinguished in groups, looked as though tRNA were transferred horizontally in a random way. <br><br><h4>  ‚Ññ4.  Darwinian evolution - as a way of thinking </h4><br><br>  In fact, there is no 100% reason not to believe in ‚Äúmultiple origin‚Äù.  This is exactly the same speculation as Darwinian evolution.  These speculations should be explicitly called methods of interpreting experimental data.  ‚ÄúMulticultural origin‚Äù is a display of a graph on which there are exact links on genes between genera. <br><br>  But in this graph there is no direction of evolution, this graph makes no assumptions about the past.  It simply shows the facts of kinship of modern organisms.  At the same time the relationship of these organisms can be distant and on the basis of this graph it is not possible to tell when the divergence of species occurred. <br><br>  Darwinian evolution is another way of interpretation, which makes it possible to imagine the most detailed course of evolution. <br><br>  But here the amateur faced again with bewilderment from classical ideas, or rather just from the lack of results.  I was told by an opponent that such a concept as ‚Äúancient‚Äù is bad for biology, since  based on the available methods, it is impossible to estimate the relative time of occurrence of the species.  But after all, after clarifying a number of points, we all agreed on the following: <br><br><blockquote>  I: We can speak of the degree of conservatism of the species as a combination of the presence of conservative molecules closer to luca.  That seems to be the difference with us. <br>  Opponent: Yes, I agree with that.  that is, if we can restore the ‚Äúancestral state‚Äù by a large number of genes (which in itself is a rather difficult task), then for each specific species we will be able to determine how close it is to this ancestral state.  I think that there will not be much difference between different species, but definitely there is, for whom evolution went a little faster, there are those for whom a little slower.  Intuitively, I suspect that the value obtained will correlate very well with the length of each specific branch (from the root) for each specific species. <br></blockquote><br><br>  This is what I called the interpretation of Darwinian evolution.  But let me specially note that although all Darwinists (that is, classical taxonomists and phylogenetics) should deal with this, they build trees using measures that are more similar to interpreting "many-species origin", and of course it is difficult for them to talk about "antiquity of the species" ¬ªBy definition, such an interpretation - as mentioned above, there is no direction of evolution and there can be no. <br><br>  But the opponent turned out to be wrong in his assessment ‚ÄúI think that there will not be much difference between the different types‚Äù - it is and significant, and this will be demonstrated further - just look at the resulting tree of evolution. <br><br><h4>  ‚Ññ5.  Method of restoring the direction of evolution </h4><br><br>  <b>From here can read those who are squeamish about the pathetic text of the amateur that is above.</b> <br><br>  To understand it is necessary to read the article <a href="http://habrahabr.ru/post/162729/">Systematics of prokaryotes - distant relatives</a> , it describes the basics, which are the input data.  Explaining further, I assume that you understand what this graph means, for example, and how it was constructed: <br><img src="https://habrastorage.org/storage2/bb7/a7e/aee/bb7a7eaee5649540ead9b58249662823.png"><br>  Now we need to figure out how to transform it into a tree with directional evolution, for example: <br><br><img src="https://habrastorage.org/storage2/75f/a18/91b/75fa1891b1c3d62da933f4c88bd7bff4.png"><br><br>  In this tree we restore the ancestors of the modern genera of bacteria.  Modern sorts of bacteria have names and are represented as leaves of a tree, while their ancestors are indicated by a set of numbers. <br><br>  Each digit is the identifier of the tRNA group that an ancestor must possess in order to pass on to its descendants in the next generation.  If he did not possess such a group of tRNA, then we definitely could not get the current state of the relationship (coincidence of identical tRNA), which is available in the column of "multiple origin" above. <br><br>  Thus, the algorithm for constructing such a tree consists of two parts: <br>  1. Distribution of tRNA into groups, so that the entire analyzed set can be appealed only in groups without switching to single tRNA ‚Äî this is necessary for two purposes (1) it is an order of magnitude more convenient to deal with groups than with a large set of tRNA.  Duplicate information is eliminated, and the group is the minimum unit of divergence.  (2) Groups can be sorted by the number of tRNAs entering there.  The probability of divergence (separation) of a larger group in different genera is higher with a smaller number of ancestral divergences (branch length). <br><br>  2. Actually building ancestral tree. <br><br>  Further, I will describe only the general principle of the implementation of these two parts. <br><br>  Division into groups: <br>  1. At the entrance there is information of the form: <br><br>  1 10 000913,003420,00686818,011215,013800,016316,017374, <br>  2 9 000434,000487,005891,005892,011142,01111163, <br>  2 10 000913,003420,006868,007509,011215,013800,016316,017374, <br>  2 8 000487,003420,005891,006678,011163,013218,007509, <br><br>  it describes a graph of ‚Äúmulti-species origin‚Äù, namely a set of links, where ‚Äú1‚Äù is the identification of one kind, ‚Äú10‚Äù is the identification of the second kind, ‚Äú000913,003420,00686818.011215.013800,016316.017374,‚Äù are those tRNAs that identical in both the first and second kind. <br>  2. The first group is created, as a set of all generally different tRNAs. <br>  3. Distribution into groups takes place, if tRNA on relationships between genera belongs to a group, this set is replaced with a group identification, but if the occurrence is partial, then what tRNA is labeled is missing, or, conversely, which tRNA, are only from this group. <br>  4. The division of the group into two.  The above made distribution into groups is analyzed, the first partial entry is taken - a new group is created, and the missing part remains with the previous group. <br>  5. Repeating paragraph 3. So gradually, there will be a division into groups without partial occurrences. <br>  6. Groups are sorted by value 1 - a group is, say, a set of 20 tRNAs, and after 300 groups there is an entry of 1-2 tRNA <br><br>  Ancestral tree construction: <br>  1. Based on the division of the genus links into groups of tRNA, it is possible to recover which groups of tRNA are in each genus.  So if between births there is such a connection <br>  1 10 307 | 864 | 867 | 897 | 909 | 911 | <br>  6 10 307 | 862 | 864 | 867 | 897 | 909 | 911 | <br>  This means that groups 307 | 864 | 867 | 897 | 909 | 911 |  There is also a 1st kind and a 10th.  But the 862 group, for example, has only the 10th and 6th, but not the 1st. <br>  2. All genera do leaves of the tree <br>  3. We take the 1st group (remember that it is the largest, which means it is less fragmented and is younger).  <a href="http://ru.wikipedia.org/wiki/%25D0%259D%25D0%25B0%25D0%25B8%25D0%25BC%25D0%25B5%25D0%25BD%25D1%258C%25D1%2588%25D0%25B8%25D0%25B9_%25D0%25BE%25D0%25B1%25D1%2589%25D0%25B8%25D0%25B9_%25D0%25BF%25D1%2580%25D0%25B5%25D0%25B4%25D0%25BE%25D0%25BA">We find the smallest common ancestor</a> for all genera that possess this group of tRNA.  If there is no such ancestor, create it.  If there is, but the smallest common ancestor is not marked with the corresponding identifier of the tRNA group - we label. <br>  4. Repeat p.3.  for all groups <br><br>  Well, the actual <b>result of the tree of the evolution of bacteria can be seen in the picture</b> : <br><br>  <a href="">We look in high resolution</a> <br><br>  PS I understand that I did not provide the actual results of the division into groups, that this tRNA cannot be concretely understood, and the method is described only from a bird's eye view.  Really interested, I can provide all the information, but I expect from them that they will try to double-check me at least in something and do not hesitate to make it publicly. </div><p>Source: <a href="https://habr.com/ru/post/164763/">https://habr.com/ru/post/164763/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../164753/index.html">Make your own photos in the style of Lytro</a></li>
<li><a href="../164755/index.html">Node.JS for startups or production problems</a></li>
<li><a href="../164757/index.html">'Elite: Dangerous' Sets a record as the most expensive project on Kickstarter, which successfully raised money for implementation</a></li>
<li><a href="../164759/index.html">He left his grandmother</a></li>
<li><a href="../164761/index.html">Sandbox review from 12/26/12 to 1/5/13</a></li>
<li><a href="../164767/index.html">Google closed Maps for Windows Phone users</a></li>
<li><a href="../164769/index.html">Internal device llst, part 1. Introduction to Smalltalk</a></li>
<li><a href="../164771/index.html">-90% piracy on iOS or Hackulos closes</a></li>
<li><a href="../164773/index.html">Android GameStick: The World's Smallest Game Console</a></li>
<li><a href="../164775/index.html">Tracking directory changes: how it is done in different operating systems</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>