<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Tasks of the domestic sequencer (June, 2019)</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Before reading this upgrade, it is advisable to familiarize yourself with the previously published source . 

 The very first and most popular Minion ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Tasks of the domestic sequencer (June, 2019)</h1><div class="post__text post__text-html js-mediator-article">  <i>Before reading this upgrade, it is advisable to familiarize yourself with the <a href="https://habr.com/ru/post/408139/">previously published source</a> .</i> <br><br>  The very first and most popular Minion nanoporous sequencer (the minimum set in the UK is $ 1000, in Russia is 150 thousand rubles), developed by Oxford Nanopore Technologies (ONT), works on disposable cells, each of which costs $ 900 ( Russia - 135 thousand rubles.).  Such cells allow for 2 ... 3 days to digitize DNA with a length of 10 ... 20 Gb.  This is too little for sequencing the human genome (need&gt; 100 Gb), but too much for all other clinical diagnostic purposes (&lt;1 Gb is enough). <br>  Recently, sales of more productive (and more expensive) models started - PromethION 24/48 ($ 165,000 / $ 285,000), aimed mainly at continuous genome-wide sequencing.  More expensive are their more efficient consumable cells ($ 2,000 / piece,&gt; 100 Gb).  True, they can cost in bulk cheaper ($ 1,800,000 for 720 pieces, $ 625 / piece), but for the Russian budget such acquisitions (200 ... 250 million rubles per batch) are impermissible.  And not only for the Russian.  Therefore, the attention of visitors to the recent ONT conference (May 22-24) attracted a message about the start of sales of Flongle - an adapter insert to the MinION sequencer, which allows working with less efficient (~ 1 Gb) but relatively cheap ($ 90) disposable cells. <br><br><img src="https://habrastorage.org/webt/ik/ul/db/ikuldb6gzu6oq4pxn6gbhtuys9e.png">
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      <a href="https://nanoporetech.com/products">https://nanoporetech.com/products</a> <br><br>  The appearance of Flongle can cause an explosive growth in the use of NGS (next generation sequencing) in clinical practice and press off conventional PCR diagnostics, the market size of which in Russia exceeds one billion rubles, and is measured in the world by billions of dollars.  But for the majority of Russian patients, NGS diagnostics will still remain an unaffordable luxury, since the cost of cells to Flongle in the domestic market will be above 10 thousand rubles.  And if you add to it the cost of consumable reagents, transport costs and overhead costs, then the estimated cost of sequencing a single DNA / RNA sample will not fall below 20 thousand rubles. <br>  Hence the need for the development of domestic nanoporous sequencers and their provision of cheap consumables and reagents. <br><a name="habracut"></a><br><h4>  Nanorider </h4><br>  It is possible to significantly reduce the cost of nanopor sequencing by reusing the Flongle cells and importing the consumables necessary for working with them.  But first you need to get a test device that allows you to monitor the formation of bilayer lipid membranes (BLM) on 126 wells of the working cell, and assess the quantity and quality of nanopores embedded in these membranes.  Such a device (‚ÄúNanorider‚Äù?) Should track changes in electrical conductivity (and / or impedance) of individual wells when BLM is formed on them, and record pico-ampere currents flowing through single ion channels. <br>  The reader can be connected to a computer via USB 2.0 from standard elements: FPGA ($ 100 ... $ 200), current amplifiers (MAX9923FEUB - 200 ... 300 rubles), ADC (1 ... 2 thousand rubles), etc. ... Only a board with a non-standard contact pad (10x13) of type LGA (Land Grid Array) for docking with a cell from Flongle.  Or with its simplified domestic analogue, containing a smaller number (16 ... 32) of sensory wells. <br><br><img src="https://habrastorage.org/webt/qd/z6/nt/qdz6ntxrcsuboqswbjm3q-ee6fu.gif"><br><br>  <a href="https://nanoporetech.com/resource-centre/videos/sub1000">Flongle cell touch pad</a> <br><br>  Reducing the calculated cell performance by 4 ... 8 times (up to 10 ... 20 Mb) is not critical for solving many tasks (determining pathogens, HLA typing, paternity determination, personal identification, etc.), but it allows you to assemble a nanoporous sequencer on conventional amplifiers picoampere currents.  And do without the use of special 512-channel chips, in the development of which the Oxfords invested tens of millions of dollars.  True, each multichannel nanorider should contain more than a dozen current amplifiers, but the ADC can still be several times less than the proprietary combination of MinION + Flongle (&gt; 200 thousand rubles).  As for the poor performance of the nanorider, then for Russian users who do not have decent computers (doctors, biologists, biohackers, etc.), it may even be useful. <br><br><h4>  Kilorider </h4><br>  The Oxford-based 512-channel pico-amp current amplifiers are contained in each disposable cell to the MinION.  Spent cells can be thrown away, but it is better to use the chips contained in them for the production of Flongle analogs characterized by increased productivity.  If these chips are equipped with surface contacts, then they can work with contact cells containing 2048 (512x4) sensory wells each. <br>  Developing cells with such LGAs will not be easy, but it must be borne in mind that the requirements for contacts are less stringent than those for processors and motherboards, for which one poor contact can destroy the entire system.  For a sequencer, the inoperability of even half the contacts of a cell can be considered acceptable.  The remaining half is enough to ensure performance at 5 ... 10 Gb. <br>  The logical development of this idea will be the unification of several similar readers in one device.  This will allow you to get an analogue of the GridION X5 sequencer, designed for simultaneous operation with five (or more) reusable cells of domestic production.  Such a sequencer can be in demand in many clinical diagnostic laboratories. <br><br><img src="https://habrastorage.org/webt/bl/pv/cr/blpvcrxtirbdhrsukdnipnrc8nc.png"><br><br>  <a href="https://store.nanoporetech.com/devices">https://store.nanoporetech.com/devices</a> <br><br>  PromethION disposable sequencer cells contain 9,000 sensor wells on chips.  And the chips themselves are 3000-channel pico-amp current amplifiers.  The flow of information cells issued by them from 24 (for PromethION 24) or from 48 (for PromethION 48) is so large that in its ‚Äúraw‚Äù form it cannot be transferred to a regular external computer.  Therefore, sequencers of this type are necessarily equipped with their own supercomputer.  But if the sequencer will work with only one cell, then it can be connected to a regular gaming laptop with a Thunderbolt 3.0 port. <br>  Alteration of free (ejected) chips from ONT to work with contact cells is not an insoluble problem.  But sequencers based on the use of such chips can only work with ONT software.  And the efforts that will have to be spent on reworking these programs can negate the benefits of using ready-made chips.  And to develop such chips from scratch is too expensive. <br><br><h4>  Megaarder </h4><br>  This year, beta testing of the AXBIO (USA) nanoporous sequencer should begin.  It uses chips with a million channels of current amplification, produced in Japan by the Israeli company Tower Semiconductor Ltd ... Expendable reagents and sales of these sequencers will be handled by the Chinese branch of the company AXBIO. <br><br><img src="https://habrastorage.org/webt/cw/uq/oy/cwuqoydkjxb_lmd_0g3kmruuijs.png"><br><br>  In the United States and the European Union, more than 7,000 patents are devoted to nanoporus sequencing, which creates unpredictable financial risks for local companies trying to engage in similar technologies.  Apparently, this is the main reason for the high internationalization of this development.  And allows Russian companies and laboratories to participate in the testing and improvement of the AXBIO technology without any restrictions, and also to produce (or assemble) such sequencers on its territory. <br>  It is advisable to filter terabytes of raw data received by a nanoporous sequencer with megapixel chips (megarider).  And transfer to the computer only the longest and high-quality reads.  This will not only improve the quality of sequencing, but also reduce computer requirements, which is also important in mass production and widespread use of genomic sequencers.  In addition, the excessive performance of mega-readers can reduce the duration of sequencing of the human genome from 2 ... 3 days (on ONT) to several hours. <br><br><h4>  The amount of technology </h4><br>  The success of ONT was facilitated by the organization of parallel conveyor optimization of the five basic elements of nanopore sequencing technology - a membrane, a nanopore, a motor (helicase), sequencing conditions, and decoding algorithms of read signals. <br><br><img src="https://habrastorage.org/webt/_t/nw/e1/_tnwe1gumo1n5qn5ecrsb0iy6qq.png"><br><br>  <a href="https://nanoporetech.com/resource-centre/videos/sub1000">https://nanoporetech.com/resource-centre/videos/sub1000</a> <br><br>  The guaranteed shelf life of cells for nanoporous sequencing depends mainly on the stability of the bilayer lipid membrane.  Now it is 6 weeks, although usually such membranes are destroyed within a few days.  To increase their shelf life, hydrophobic polymer additives, surface-active block copolymers, cholesterol, antioxidants, cross-linking of fatty acids, etc. are used, and such modifications can interfere with the formation of ion channels.  Therefore, to achieve stability BLM is very difficult. <br>  This problem can be fundamentally solved by the formation of membranes in the working cells of the sequencer and the introduction of nanopores into them immediately before use.  Independent preparation of cells for work will provide the possibility of their regeneration and repeated use.  As for the shelf life, for dry cells it is practically unlimited, and the reagents necessary for their activation and regeneration can be stored for months. <br>  The bottleneck in the technology of nanoprovod sequencing is at the same time the bottleneck of the nanopore, which determines the quality of the read information.  Therefore, one of the main tasks of ONT was the search for membrane protein, which allows to obtain results of acceptable quality. <br>  The first such protein was CsgG, an outer membrane protein found in many gram-negative bacteria.  Its nine subunits form an ion channel in the membrane with wide entrance and exit vestibules, and one bottleneck.  But it was possible to obtain sufficiently high-quality nanopores (R9, R9.4.1, R9.5.1) only after searching and analyzing hundreds of modified CsgG variants.  This summer, cells with the next generation pores (R10 or R10b), characterized by increased accuracy of reading homopolymer repeats, should appear on sale. <br><br><img src="https://habrastorage.org/webt/87/zq/2g/87zq2golwasbtfguqdkgkbfz_fo.png"><br>  <a href="https://nanoporetech.com/about-us/news/london-calling-clive-brown-and-team-plenary">https://nanoporetech.com/about-us/news/london-calling-clive-brown-and-team-plenary</a> <br><br>  The obligatory component of ONT's nanoporos sequencing technology is helicase, a molecular motor that unwinds double-stranded DNA and inhibits the promotion of single-stranded DNA through a nanopore.  The maximum speed of DNA unwinding by such a motor now reaches 450 base pairs per second, although the speed of the electronic components of the MinION makes it possible to increase it to 1000 nucleotides per second.  Theoretically, the acceleration of the work of helicase can double the performance of the MinION, but in practice this may lead to a deterioration in the quality of sequencing.  Therefore, faster and more efficient helicases are unlikely to improve the performance of nanoporous sequencers such as MinION and PromethION. <br>  The quality of sequencing depends on a variety of conditions - the magnitude of the voltage applied to the membrane, the electrical conductivity and the ionic composition of the reaction mixture, the amount and nature of impurities plugging the pores, the ability to clean the blocked pores by changing the polarity of the potential, the structure of the adapters used, etc., etc. Therefore, nanoporous sequencing requires careful optimization of all its parameters and compositions of working solutions (Run Conditions) that can adversely affect the quality of digitization of nucleotide followers  awns.  For example, it has recently been shown that reducing the concentration of ATP cleaved by helicase gradually reduces the accuracy of sequencing results, and now Oxfords are trying to eliminate this problem by introducing ATP regeneration system into the reaction mixture. <br>  The fifth element of nanopor sequencing technology is artificial intelligence, built on recurrent neural networks (RNN) and on the use of some special algorithms and programs that improve the recognition of homogeneous repetitions (Flip-flop, Medaka).  In July, the next release of the Guppi and MinKNOW programs should appear, in which the Oxfords promise to add the possibility of identifying methylated bases (5mC and 6mA).  This is important not only for epigenomic studies, but also to improve the recognition of common bases. <br>  The large volumes of information read by nanopor sequencer and the complexity of its processing impose increased requirements on the hardware of such software and hardware complexes.  Therefore, not every computer is suitable for connecting MinION, GridION and PromethION put the latest generation of graphics accelerators, and Oxford Nanopore Technologies became one of the first buyers of a large batch of Jetson AGX Xavier (NVIDIA) modules designed for devices with artificial intelligence. <br><br>  More recently, nanoporous technology was considered nothing more than a useful addition to more accurate fluorescent and semiconductor sequencing technologies.  But a gradual increase in the accuracy of DNA reading, combined with a large length of readable sequences, allowed her, if not to get ahead of all competitors, then at least determine the main vector of the development of genomic sequencing technologies. <br><br><img src="https://habrastorage.org/webt/ps/nz/ax/psnzaxcgow2g3teqlz0wtjx8zny.png"><br><br>  A lot of time may be needed to master all the elements of nanopor sequencing technology, but this is not the biggest problem.  The main problem is the lack of targeted financing of such complex developments in Russia.  True, the recently approved Federal Scientific and Technical Program for the Development of Genetic Technologies for 2019‚Äì2027 (section ‚ÄúProgram Implementation Directions‚Äù) says that in the short term (3 ... 6 years) a prototype of the device will be developed for high-performance genomic sequencing.  But to find this single line on 30 pages of text is not so easy.  It is easier to agree with the Chinese on the purchase of AXBIO sequencers. <br>  It is even easier to find craftsmen who can assemble the simplest nanorider ‚Äúon their knees‚Äù.  And then walk with an outstretched hand in search of investors who will agree to finance the development of domestic technology of nanoporous sequencing. </div><p>Source: <a href="https://habr.com/ru/post/455156/">https://habr.com/ru/post/455156/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../455132/index.html">Antiquities: Philips Velo 1, tech enthusiast nightmare</a></li>
<li><a href="../455138/index.html">JetBrains Night Moscow: how it was</a></li>
<li><a href="../455148/index.html">Solve decision-making tasks based on data.</a></li>
<li><a href="../45515/index.html">Would you like to be able to buy any gadget in Russia without hemorrhoids?</a></li>
<li><a href="../455154/index.html">‚ÄúMyself Chubais‚Äù or alternative energy in the village</a></li>
<li><a href="../455158/index.html">Wonders of accounting methodology: depreciation fund</a></li>
<li><a href="../455160/index.html">Entity Framework 6 with Full-Text Search via LINQ</a></li>
<li><a href="../455166/index.html">VPN on Beeline router to bypass blocking</a></li>
<li><a href="../45517/index.html">Winners of the Runet Prize</a></li>
<li><a href="../455176/index.html">Is it possible without Redux?</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>