<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>The first bacterial genome, designed using a computer</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Caulobacter crescentus is a safe bacterium that lives in fresh water throughout the world. 

 All known genomes of organisms are stored in a database ...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>The first bacterial genome, designed using a computer</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/webt/rc/q0/vo/rcq0vomwtu_tt9pnfslgxlpbbs8.jpeg"><br><br>  <i>Caulobacter crescentus is a safe bacterium that lives in fresh water throughout the world.</i> <br><br><blockquote>  All known genomes of organisms are stored in a database owned by the <a href="https://www.ncbi.nlm.nih.gov/">National Center for Biotechnology Information</a> in the United States.  Now in the database there is a record: <i>Caulobacter ethensis-2.0</i> .  This is the world's first fully computerized genome of a living organism, developed by scientists from <a href="https://ethz.ch/">ETH Zurich</a> .  It must be emphasized that, although the <i>Caulobacter ethensis-2.0</i> genome was physically obtained in the form of a very large DNA molecule, the corresponding organism does not yet exist. </blockquote>  <i>Caulobacter ethensis-2.0 is</i> based on the genome of the well-studied and safe freshwater bacterium <i>Caulobacter crescentus</i> , which is found in nature in spring water, rivers and lakes around the globe.  Does not cause any diseases.  <i>Caulobacter crescentus is</i> also a model organism commonly used in research laboratories to study the life of bacteria.  The genome of this bacterium contains 4000 genes.  Scientists have previously shown that only about 680 of these genes are crucial in the survival of bacteria in the laboratory. <br><a name="habracut"></a><br>  Beat Kristen, a professor of experimental systems biology at ETH Zurich, and his brother Matthias Kristen, a chemist at ETH Zurich, took the minimal genome of <i>Caulobacter crescentus</i> as a base.  They intended to chemically synthesize this genome from scratch as a continuous circular chromosome.  This task is considered really challenging: a chemically synthesized bacterial genome, presented 11 years ago by the American pioneer of genetics <a href="https://en.wikipedia.org/wiki/Craig_Venter">Craig Venter</a> , was the result of 10 years of work by 20 scientists.  It is said that the project cost was 40 million dollars. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <h2>  Rationalization of the assembly process </h2><br>  While Venter's group made an exact copy of the natural genome, scientists from ETH Zurich radically altered the genome using a computer algorithm.  Their motivation was twofold: one - that it was much easier to synthesize the genomes, and the second - the solution of fundamental problems of biology. <br><br>  To create a DNA molecule the size of a bacterial genome, scientists must act step by step.  In the case of the <i>Caulobacter</i> genome, scientists from ETH Zurich synthesized 236 genome fragments, which they subsequently sewed.  ‚ÄúThe synthesis of these fragments is not always simple,‚Äù explains Mathias Kristen.  ‚ÄúDNA molecules not only have the ability to adhere to other DNA molecules, but depending on the sequence, they can also twist into loops and knots, which can complicate the synthesis process or make it impossible,‚Äù explains Mathias Kristen. <br><br><h2>  Simplified genomes </h2><br>  To synthesize genome fragments in the simplest way, and then put together all the fragments in the most correct way, scientists radically simplified the genome sequence without changing the actual genetic information (at the protein level).  There are many opportunities to simplify the genomes, because biology has built-in reserves for storing genetic information.  For example, many amino acids have two, four or more possibilities to write their information in DNA. <br><br>  The algorithm developed by scientists from ETH Zurich makes the best use of this redundancy of the genetic code.  Using this algorithm, they calculated the most economical DNA sequence in order to synthesize and construct the genome, which they used in their work. <br><br><img src="https://habrastorage.org/webt/kt/cr/qv/ktcrqvq9cp5xmdyaa0n-zu6fqt0.jpeg"><br><br>  <i>Caulobacter ethensis-2.0 genome in vitro</i> <br><br>  As a result, scientists made many small changes to the minimal genome, which, however, is impressive: more than one-sixth of the 800,000 letters of DNA in the artificial genome were replaced compared to the ‚Äúnatural‚Äù minimal genome.  ‚ÄúThanks to our algorithm, we have completely rewritten our genome into a new sequence of DNA letters, which is no longer similar to the original sequence.  However, the biological function at the protein level has been preserved, ‚Äùsays Bit Kristen. <br><br><h2>  Litmus test in genetics </h2><br>  The rewritten genome is also interesting from a biological point of view.  ‚ÄúOur method is a litmus test to see if we, biologists, correctly understand genetics, and it allows us to highlight possible gaps in our knowledge,‚Äù Beat Kristen explains.  Naturally, the rewritten genome can contain only information that researchers actually understood.  A possible ‚Äúhidden‚Äù superfluous information that is in the DNA sequence and is not yet understood by scientists would be lost in the process of synthesizing a new code. <br><br>  Scientists have grown bacteria strains that contain both the natural <i>Caulobacter</i> genome and fragments of a new artificial genome.  By turning off some of the natural genes in these bacteria, scientists were able to test the function of artificial genes.  They tested each of the artificial genes in a multi-step process. <br><blockquote>  In these experiments, scientists found that only about 580 of 680 artificial genes were functional.  ‚ÄúWith the knowledge, we can improve our algorithm and develop a fully functional version of genome 3.0,‚Äù says Beat Kristen. </blockquote><br><h2>  Giant potential in biotechnology </h2><br>  ‚ÄúAlthough the current version of the genome is not yet perfect, our work nevertheless shows that biological systems are built in such a simple way that in the future we will be able to develop project specifications on a computer in accordance with our goals, and then build them‚Äù, - says Matthias Kristen.  And this can be obtained in a relatively simple way, as Bit Kristen stresses: ‚ÄúWhat it took Craig Venter for ten years, our small group performed using our new technology during the year at a cost of only 120,000 Swiss francs.‚Äù <br><blockquote>  ‚ÄúWe believe that it will also soon be possible to produce functional bacterial cells with such a genome,‚Äù says Beat Kristen.  Such a development will have great potential.  Among possible future applications are synthetic microorganisms that can be used in biotechnology, for example, in the synthesis of complex pharmaceutically active molecules or vitamins.  The technology can be applied universally in all microorganisms, and not only in <i>Caulobacter</i> .  Another possibility would be the production of DNA vaccines. </blockquote>  ‚ÄúNo matter how promising the research results and their possible applications are, they require a deep understanding of the purposes for which this technology can be used, and at the same time, how to prevent abuse,‚Äù says Beat Kristen.  It is not yet clear when the first bacterium with an artificial genome will be obtained, but now it is clear that it can be obtained and will develop.  ‚ÄúWe need to use the time that we have for the purpose of intensive discussions between scientists, as well as in society as a whole.  We are ready to contribute to this discussion with all the know-how we have. ‚Äù </div><p>Source: <a href="https://habr.com/ru/post/449372/">https://habr.com/ru/post/449372/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../449362/index.html">UPS for medical facilities: Delta Electronics health care experience</a></li>
<li><a href="../449364/index.html">Background: what is Continuous Delivery</a></li>
<li><a href="../449368/index.html">Do I need to clean the string in javascript?</a></li>
<li><a href="../44937/index.html">The main indicator of the effectiveness of a startup</a></li>
<li><a href="../449370/index.html">Big Data analytics - realities and prospects in Russia and the world</a></li>
<li><a href="../449374/index.html">Overview of English-language sites to search for permanent remote IT work</a></li>
<li><a href="../449376/index.html">History of one NOT vulnerability</a></li>
<li><a href="../449380/index.html">Is the counselor competent? Problems recommending "don't reinvent the wheel"</a></li>
<li><a href="../449384/index.html">Quiz: Is the remote control suitable for you (not freelance!)?</a></li>
<li><a href="../449386/index.html">Internet for summer resident. We get the maximum speed in 4G networks. Part 1. Choosing the right router.</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>