<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Practical bioinformatics, part 2</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="This article will tell you how to process the data obtained after the pipeline, the output of which will be the sam / bam file [1], create a simple be...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Practical bioinformatics, part 2</h1><div class="post__text post__text-html js-mediator-article">  This article will tell you how to process the data obtained after the pipeline, the output of which will be the sam / bam file [1], create a simple bed graph file (http://genome.ucsc.edu/FAQ/FAQformat.html) and view it using the UCSC genome browser [2].  It is very difficult to decide on what to write programs for, because there are already a huge number of foreign developments and I don‚Äôt want to write a wheel at all where this stage has been passed.  Long suffering, I decided to stop at C ++, although Python and R were treated as equals.  There is also an idea that graphics may be needed, and even under Linux, so Qt has been added to C ++.  I hope that in this article I will tell you in sufficient detail about all of the above to answer the question asked at the beginning of my journey and voiced in the first part of the story. <br><a name="habracut"></a><br>  To work with sam / bam files, we need the samtools [1] package compiled.  Download the package from the site <a href="http://samtools.sourceforge.net/">samtools.sourceforge.net</a> , expand to a directory, for example, / usr / src go to the created directory and type make.  I did not install XCurses on my system and I replaced the line ‚ÄúLIBCURSES = -lXCurses‚Äù with ‚ÄúLIBCURSES = -lncurses‚Äù and everything was assembled.  The result of the make program is the compiled program samtools and the library libbam.a. <br>  We need C ++ classes that will store information about the reads, genes, introns, exons, etc.  For the organization of such classes, I used boost.intervals, although not quite in the form that I wanted.  Neither boost.intervals, nor boost.icl allow storing complete information about segments.  I need the following information about the set of segments: <br><ul><li>  any pair of start / end, start / length; </li><li>  all segments are closed, i.e.  include ends; </li><li>  how many segments begin at a given point; </li><li>  how many segments intersect at a point; </li><li>  how many segments start in a given segment or intersect with a given segment. </li></ul><br>  In particular, this information will answer the question what is the height of the coverage with segments of the current segment of the coordinate axis. <br>  With each new article I will replenish and change these classes.  Now they are enough to build a program and estimate what is missing for the future.  Here is an example of the resulting classes. <br><br>  <u>Reads.hpp</u> <br><br><pre><code class="cpp hljs"><span class="hljs-meta"><span class="hljs-meta">#</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">ifndef</span></span></span><span class="hljs-meta"> _READS_29122011_HPP_ #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">define</span></span></span><span class="hljs-meta"> _READS_29122011_HPP_ #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">include</span></span></span><span class="hljs-meta"> </span><span class="hljs-meta-string"><span class="hljs-meta"><span class="hljs-meta-string">&lt;boost/numeric/interval.hpp&gt; #include &lt;QString&gt; #include &lt;QMap&gt; #include &lt;QtDebug&gt; namespace genome { namespace bni = boost::numeric; typedef bni::interval&lt;int&gt; read_position; /********************************************************************************** **********************************************************************************/ class Read { public: Read(): multiplying(1), length(0){}; Read(Read const &amp;r): multiplying(r.multiplying), length(r.length), position(r.position){ sentenceRepresentation=r.sentenceRepresentation; qualityRepresentation=r.qualityRepresentation;}; Read(int start,int len,QString sr="",QString qr=""): multiplying(1), length(len), position(start,start+len-1), sentenceRepresentation(sr), qualityRepresentation(qr){}; int getLevel() {return multiplying;}; void plusLevel() {++multiplying;}; int getStart() {return position.lower();}; int getLength() {return length;}; void operator+= (const int&amp; c) {this-&gt;multiplying+=c;}; bool operator== (const Read&amp; r) const {return this-&gt;position==r.position;}; bool operator!= (const Read&amp; r) const {return this-&gt;position!=r.position;}; void operator++ (int) {this-&gt;multiplying++;}; private: int multiplying; int length; read_position position; QString sentenceRepresentation; QString qualityRepresentation; }; /********************************************************************************** **********************************************************************************/ typedef QMap&lt;int,Read&gt; cover_map; class Cover { public: Cover():max_len(0){}; void add(Read&amp;); int getHeight(int); int getHeight(int,int); int getStarts(int); int getStarts(int,int); QList&lt;int&gt; getStarts(); cover_map::iterator getBeginIterator(){return covering.begin();}; cover_map::iterator getEndIterator(){return covering.end();}; bool operator== (const Cover&amp; c) const {return this==&amp;c;}; bool isEmpty(){return covering.size()==0;}; // static Cover empty(){ return Cover();}; private: cover_map covering; int max_len; }; /********************************************************************************** **********************************************************************************/ class Lines { public: Lines(){}; Lines(Lines&amp;){}; void addLine(QString, Read&amp;); Cover&amp; getLineCover(QString); QList&lt;QString&gt; getLines(void) { return lines.keys(); }; /* */ private: QMap&lt;QString,Cover&gt; lines; }; /********************************************************************************** **********************************************************************************/ class GenomeDescription: public Lines { public: quint64 notAligned; // number of reads (ussualy form sam/bam file) that are not aligned quint64 total; /* */ void setGene(const QChar &amp;sense,const QString &amp;chrName,const qint32 &amp;pos,const qint32 &amp;,const qint32 &amp;len) { Read r(pos,len); addLine(chrName+sense,r); }; /* */ GenomeDescription():Lines(), notAligned(0), total(0) {}; }; } #endif</span></span></span></span></code> </pre> <br>  Now let's get down to wrapping functions for working with sam files.  The difference between sam and bam is that the first is a plain text file, albeit formatted, and the second is compressed and binary, i.e.  structured.  I prefer to work with bam with a large number of reads, the sam file can reach gigabytes. <br>  <u>BEDHandler.hpp</u> 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <pre> <code class="cpp hljs"><span class="hljs-meta"><span class="hljs-meta">#</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">ifndef</span></span></span><span class="hljs-meta"> BEDHandler_H #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">define</span></span></span><span class="hljs-meta"> BEDHandler_H #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">include</span></span></span><span class="hljs-meta"> </span><span class="hljs-meta-string"><span class="hljs-meta"><span class="hljs-meta-string">&lt;config.hpp&gt; #include &lt;Reads.hpp&gt; template &lt;class Storage, class Result&gt; class BEDHandler : public QState { private: Storage *sam_input; Result *output; QSettings setup; QFile _outFile; public: BEDHandler(Storage &amp;sam,Result &amp;output,QState *parent=0); ~BEDHandler(); protected: virtual void onEntry(QEvent* event); }; #include &lt;BEDHandler.cpp&gt; #endif</span></span></span></span></code> </pre><br>  <u>BEDHandler.cpp</u> <br><pre> <code class="cpp hljs"><span class="hljs-comment"><span class="hljs-comment">//------------------------------------------------------------- //------------------------------------------------------------- template &lt;class Storage,class Result&gt; BEDHandler&lt;Storage,Result&gt;::~BEDHandler() { } //------------------------------------------------------------- //------------------------------------------------------------- template &lt;class Storage,class Result&gt; BEDHandler&lt;Storage,Result&gt;::BEDHandler(Storage&amp; sam,Result &amp;_output,QState * parent): QState(parent), sam_input(&amp;sam), output(&amp;_output) { _outFile.setFileName(setup.value("outFileName").toString()); _outFile.open(QIODevice::WriteOnly|QIODevice::Truncate); } //------------------------------------------------------------- //------------------------------------------------------------- template &lt;class Storage,class Result&gt; void BEDHandler&lt;Storage,Result&gt;::onEntry(QEvent*) { if(!setup.contains("graphWindow")) setup.setValue("graphWindow",200); if(!setup.contains("siteShift")) setup.setValue("siteShift",75); if(!setup.contains("separateStrand")) setup.setValue("separateStrand",false); if(!setup.contains("HeaderString")) setup.setValue("HeaderString","track type=bedGraph name=%1"); if(setup.value("HeaderString").toString()!="") { _outFile.write((setup.value("HeaderString").toString().arg(_outFile.fileName())+"\n").toLocal8Bit()); _outFile.flush(); } quint32 window=setup.value("graphWindow").toUInt(); quint32 shift= setup.value("siteShift").toUInt(); QString line; QString chrome; foreach(line,sam_input-&gt;getLines()) { if(line.endsWith("-")) continue; chrome=line; chrome.truncate(line.length()-1); QMap &lt;int,int&gt; bed; { genome::cover_map::iterator i=sam_input-&gt;getLineCover(chrome+QChar('+')).getBeginIterator(); genome::cover_map::iterator e=sam_input-&gt;getLineCover(chrome+QChar('+')).getEndIterator(); while(i!=e) { int val=i.key()+shift; bed[val-val%window]+=i.value().getLevel(); ++i; } } { genome::cover_map::iterator i=sam_input-&gt;getLineCover(chrome+QChar('-')).getBeginIterator(); genome::cover_map::iterator e=sam_input-&gt;getLineCover(chrome+QChar('-')).getEndIterator(); while(i!=e) { int val=i.key()-shift; if(val&lt;0) val=0; bed[val-val%window]+=i.value().getLevel(); ++i; } } QMap&lt;int,int&gt;::iterator i = bed.begin(); for(;i!=bed.end();i++) { _outFile.write(QString(chrome+"\t%1\t%2\t%3\n"). arg(i.key()).arg(i.key()+window).arg(i.value()).toLocal8Bit()); _outFile.flush(); } } }//end of function</span></span></code> </pre><br><br>  I will not give the rest of the code in the article; <a href="">here is the link to the archive</a> , the structure is quite simple: at the root are two directories, thirdparty and src, in the first one there is samtools, in the second one - sam2bedgraph and global.  In order to build a project, you need to run qmake and then make in the sam2bedgraph directory.  Checked under openSUSE 12.1 x64 with native Qt (4.7.4) and boost (1.46.1). <br><br>  The link to the description of the structure of the bedgraph file is given in the first paragraph, but I will briefly mention it.  The first line specifies the characteristics of the file, if present.  The file body contains at least 4 columns.  The first column indicates the name of the chromosome, in the second and third - the coordinates of the beginning and end of the segment ("window"), respectively.  In the last column - ‚Äúheight‚Äù, in our case, this number began reading on this window.  The size of the "window" for each file is best chosen experimentally.  I use 200 bp for ChIP-seq, 20 bp for RNA-seq. <br>  A program can have two parameters: an input .bam file and an output file where the bedgraph data will be written.  Parameters can be not set, then the program will try to open the default file input.bam and create output output.data.  In the user's home directory, a directory will be created with the configuration file .config / CCHMC / sam2bedgraph.ini, in which you can change the default file name values, change the length of the ‚Äúwindow‚Äù for which we count heights and set the appearance of the first information line. <br>  The resulting file can be downloaded to the genome browser.  Go to the site <a href="http://genome.ucsc.edu/">genome.ucsc.edu</a> [2] click on the link Genomes. <br><img src="https://habrastorage.org/storage2/290/c25/3bc/290c253bccbaad08b44ee02a6bcb1eef.jpg"><br><br>  We get to the next page, where it is necessary to set the conditions under which our bam file was received: Clade (Evolutionary Line), genome (Genome), assembly (Abstract, Assembly).  And then click the ‚Äúadd custom tracks‚Äù button. <br><img src="https://habrastorage.org/storage2/6cd/086/c03/6cd086c03b8b8676a47be473f764ae7d.jpg"><br><br>  Now you can click the ‚ÄúChoose file‚Äù button and upload our file by clicking the ‚ÄúSubmit‚Äù button.  As a result, we get to the page with the annotation, which will be displayed information from the database and from our bedgraph.  If you install the genome browser locally, you can add a bedgraph to the site not temporarily during the session, but constantly.  You can organize special links to the site, which will indicate where to get the bedgraph file.  The site contains instructions on how to copy the genome browser. <br>  You can upload multiple files, which I did.  You can see the result in the following screenshot. <br><br><img src="https://habrastorage.org/storage2/2bf/221/4e0/2bf2214e0da455d5caf65f071c666bb8.jpg"><br>  As can be seen in this screenshot, there are areas with a zero level, and there are areas with enrichment, where there are peaks.  Sometimes there is a background or noise.  You can also clearly see the difference between ChIP-seq and RNA-seq.  The definition of these sections from the program is a separate question, the unequivocal answer to which is not.  The fact is that there can be as many peaks as there are genes in the genome or, worse, as exons.  And to separate the enriched areas in the experiment is very time consuming. <br><br>  1. Li, H., et al., The Sequence Alignment / Map format and SAMtools.  Bioinformatics, 2009. 25 (16): p.  2078-9. <br>  2. Kent, WJ, et al., The human genome browser at UCSC.  Genome Res, 2002. 12 (6): p.  996-1006. <br>  3. Dreszer, TR, et al., The UCSC Genome Browser database: extens and updates 2011. Nucleic Acids Res, 2012. 40 (Database issue): p.  D918-23. </div><p>Source: <a href="https://habr.com/ru/post/137082/">https://habr.com/ru/post/137082/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137076/index.html">Historical essay on the protection of information</a></li>
<li><a href="../137077/index.html">Postindustrial or Production 2.0</a></li>
<li><a href="../137078/index.html">Video review of Android applications and games - kedDroid</a></li>
<li><a href="../137079/index.html">Interview with Herman Klimenko</a></li>
<li><a href="../137080/index.html">Meet PayPal Standard Checkout</a></li>
<li><a href="../137084/index.html">Not a computer mouse and high technology</a></li>
<li><a href="../137085/index.html">BlackBerry PlayBook in police service</a></li>
<li><a href="../137086/index.html">How to model business processes in eEPC notation?</a></li>
<li><a href="../137087/index.html">PayPal and US shipping address</a></li>
<li><a href="../137088/index.html">Beta-test Bitcasa do it yourself</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>