<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Analysis of the genome of bacteria. Continuation</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="In the previous article , the discussion turned out to be too flashy. But we opened our site and there will be more extensive information (where? - wr...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Analysis of the genome of bacteria. Continuation</h1><div class="post__text post__text-html js-mediator-article">  In the <a href="http://habrahabr.ru/post/166361/">previous article</a> , the discussion turned out to be too flashy.  But we opened our site and there will be more extensive information (where? - write letters).  I promised to write a sequel about my experiment, so those who are interested in the problems of building evolutionary trees - please under the cat. <a name="habracut"></a><br><br><h5>  ‚Ññ1.  Selection of all homologous sequences (paralogs) </h5><br><br>  In the last article, we compared evolutionary trees built on the 16S and 23S genes.  My method differs in that it offers to compare what is not mutated in organisms.  In early articles on Habr√©, I suggested using tRNA, because  These are the most conservative sequences.  But it gave not much information.  Therefore, I wondered - how do you find all those sequences that are not mutated in organisms?  To do this in real time, I went for a little trick.  The fact is that before any DNA sequence is inherited, it will most certainly (if it is useful) be represented in the genome by several copies.  Those.  we are talking about paralogs. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      <i>If within the same organism a gene doubled as a result of a chromosomal mutation, then its <u>copies</u> are called paralogs.</i> <br><br>  So if you find all the paralogs in one organism, then if inheritance occurred, they were transferred to other organisms.  We only need then to select those who did not have time to mutate. <br><br>  Those.  we do the following: <br>  1. We are looking for in each DNA (the genome of the organism) something that generally has duplicates from 50 to 150 characters. <br>  2. For each duplicate found, we are looking for its occurrences in all organisms, i.e.  learn and make up the base as ALL of the paralogs are included in the set of the genomes <br><br>  (in order not to be distracted from the point of how to do this, I will tell you either as a separate article or rather, if you are interested, I will write an article on our website with time) <br><br><h5>  ‚Ññ2.  Actually building an evolutionary tree </h5><br><br>  I <a href="http://habrahabr.ru/post/164763/">have already told</a> how to build an evolutionary tree according to my method.  Therefore, focus on the results of cross-checking.  Let me remind you that the cross-check of two trees <a href="http://www.arb-silva.de/fileadmin/silva_databases/living_tree/LTP_release_102/LSU/LTPs102_LSU_tree.pdf">built on the 23S rRNA gene</a> and <a href="http://www.arb-silva.de/fileadmin/silva_databases/living_tree/LTP_release_108/LTPs108_SSU_tree.pdf">built on the 16S rRNA gene,</a> which is the last result of the project <i>The All-Species Living Tree</i> , gave the following error distribution (compared to the previous article, converted to percentages of the total number of ): <br><br><img src="https://habrastorage.org/storage2/851/de9/84d/851de984d572ed7d099b71a777a35ecb.png"><br><br>  I was hoping that my approach would give better results, but alas, he gave about the same in quality - but others in essence.  First, about the quality, then the cross-check was done like this.  Since about a million occurrences of paralogs were found in the organism‚Äôs genome, i.e.  there are a million entries like ‚ÄúDNA sequence ID such-and-such enters into the organism of such-and-such‚Äù, then for cross-checking I divided this set randomly into two samples.  He built trees on them and compared the constructed trees in the same way.  It turned out the following: <br><br><img src="https://habrastorage.org/storage2/891/ff3/f8b/891ff3f8b712191d0770fec977cda215.png"><br><br>  Thus, in essence, the confidence in these trees is about the same.  Both are correct by about 50%. <br><br>  Of course, it seems that the information in the genomes is not so much that only half of the sample could be obtained similarity.  Therefore, I thought that I would use the available information as economically as possible.  And I thought that you can do this cross-analysis.  Take all the available information to build a complete tree, and compare it with half-trees.  Those.  take the entire million records and compare them first with one half a million, and then with the second.  In the figure below, tree images (and by reference in full resolution) are plotted using the full sample, and those nodes that are fairly stable ‚Äî that is, are shown in red.  cross-analysis did not give more than one error. <br><br>  As you can see, everything is not so bad, some of the branches are completely red, but the closer to the root, the less information and the position of the species in the tree does not pass the test. <br><br>  But what is interesting, I then checked the tree I received and the tree of the project The All-Species Living Tree (after reduction to one composition).  It turned out that they coincide by only 25%. <br><br>  And I had an <b>important question of interpretation</b> , can someone tell me what it could mean.  It turns out that my method of building trees can be trusted and also apparently can be trusted and the classical method used in the project The All-Species Living Tree.  They do not differ significantly in the level of coincidences.  But why do they not coincide with each other?  They are shown to show two variants of the same.  But how can there be at the same time two half-truths that coincide only by 25%? <br><br><img src="https://habrastorage.org/storage2/88f/a34/bc1/88fa34bc1a44cbb1a19ea91fd8222192.png"><br>  <a href="">Full-size format can be found here</a> and <a href="">here</a> . <br><br>  I also thought that inconsistencies appear non-random, and somewhere at the level of families of organisms.  In the second version of the tree image, it can be seen that the species are clustered in groups, and within the group there are many coincidences, while the position of the groups themselves is inaccurate. <br><br>  There are two options - or really little data yet, few sequenced intermediate species.  Or, nevertheless, really, at the level above the families, they have no common ancestor, and evolution does not follow Darwin?  At least for the time being we do not have reliable data that a common ancestor existed at all. </div><p>Source: <a href="https://habr.com/ru/post/168593/">https://habr.com/ru/post/168593/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../168583/index.html">Gossip. A new commercial of the new Chromebook Pixel with a touch screen and a resolution of 2560 x 1700 px</a></li>
<li><a href="../168585/index.html">Yandex overtook Microsoft in the world search</a></li>
<li><a href="../168587/index.html">Now you can ask the company to fill in your description.</a></li>
<li><a href="../168589/index.html">The camera module for the Raspberry Pi will cost $ 25 and shoot FullHD video</a></li>
<li><a href="../168591/index.html">Regular Expression Crossword</a></li>
<li><a href="../168595/index.html">Step-by-step guide on "How to take the Cisco exam?"</a></li>
<li><a href="../168597/index.html">Undocumented Microsoft SQL Server features: STATISTICS_ONLY, DBCC AUTOPILOT and SET AUTOPILOT</a></li>
<li><a href="../168599/index.html">GetUpps! - application store for MegaFon subscribers</a></li>
<li><a href="../168601/index.html">PostgreSQL 9.2 Start!</a></li>
<li><a href="../168603/index.html">Microsoft accuses Google of reading user letters</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>