<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Genetic engineering of bacteria: how it all began and how it works in the laboratory today</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="Despite the existence of obvious obstacles and difficulties that sometimes stand in the way of the development and introduction of products of genetic...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Genetic engineering of bacteria: how it all began and how it works in the laboratory today</h1><div class="post__text post__text-html js-mediator-article"><img src="https://habrastorage.org/files/205/c3c/017/205c3c017a5d4117acbfa050382bae16.jpg" align="left">  Despite the existence of obvious obstacles and difficulties that sometimes stand in the way of the development and introduction of products of genetic engineering (GI), the XXI century can no longer be imagined without the fruits of this important and diverse technology in the arsenal of a modern biologist.  The most commonly used organism in GI are bacteria. <br><br>  What is GI and why do we need it?  Why are bacteria so popular with genetic engineers?  What is the easiest way to insert the gene into the bacterium?  What difficulties can you encounter when working with these organisms?  What happened before: the creation of the first genetically engineered bacteria or the discovery of the structure of DNA and the genome?  Read about this and many other things under the cut. <br><a name="habracut"></a><br><h3>  0. Brief educational program in biology </h3><br>  This paragraph provides a brief description of the so-called <a href="https://ru.wikipedia.org/wiki/%25D0%25A6%25D0%25B5%25D0%25BD%25D1%2582%25D1%2580%25D0%25B0%25D0%25BB%25D1%258C%25D0%25BD%25D0%25B0%25D1%258F_%25D0%25B4%25D0%25BE%25D0%25B3%25D0%25BC%25D0%25B0_%25D0%25BC%25D0%25BE%25D0%25BB%25D0%25B5%25D0%25BA%25D1%2583%25D0%25BB%25D1%258F%25D1%2580%25D0%25BD%25D0%25BE%25D0%25B9_%25D0%25B1%25D0%25B8%25D0%25BE%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B8%25D0%25B8">Central dogma of molecular biology</a> .  If you have basic knowledge of molecular biology, feel free to skip to step 1. <br><br><img src="https://habrastorage.org/files/25c/0ba/e8a/25c0bae8a83e40be9dbe8f25aba70df3.jpg"><br>  <i>The central dogma of molecular biology in one picture</i> 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      So, let's begin.  All information on all stages of development and properties of any organism, whether <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D1%2580%25D0%25BE%25D0%25BA%25D0%25B0%25D1%2580%25D0%25B8%25D0%25BE%25D1%2582%25D1%258B">prokaryotes</a> (bacteria), <a href="https://ru.wikipedia.org/wiki/%25D0%2590%25D1%2580%25D1%2585%25D0%25B5%25D0%25B8">archaea</a> or <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D1%2583%25D0%25BA%25D0%25B0%25D1%2580%25D0%25B8%25D0%25BE%25D1%2582%25D1%258B">eukaryotes</a> (all other single and multicellular), is encoded in genomic DNA, which is a complex of two polynucleotide chains complementary to each other ( complementary DNA nucleotides: AT and GC).  Eukaryotic chromosomes are linear double-stranded DNA molecules, and prokaryotic chromosomes are looped.  Often genes make up only a small part of the entire genome (about 1.5% in humans). <br><br><img src="https://habrastorage.org/files/ee5/dc5/6e4/ee5dc56e44f041ea8bd08053971e8e63.JPG"><br>  <i>Examples of DNA and RNA monomers.</i>  <i>"Deoxy" in the name of DNA means the absence of an oxygen atom in position 2 '(in the figure position 2' is circled in red).</i> <br><br><img src="https://habrastorage.org/files/a23/182/f84/a23182f849e94ff2969f480a3ed798db.png"><br>  <i>Two complementary strands of DNA.</i>  <i>Dotted lines indicate hydrogen bonds between the bases.</i>  <i>As can be seen, adenine and thymine form between themselves two hydrogen bonds, and guanine and cytosine - three.</i>  <i>Therefore, the GC bond is stronger and the GC-rich regions of double-stranded DNA are more difficult to divide into two chains.</i> <br><br>  Please note that each of the chains has a 5'-end and 3'-end.  It can be seen that about the 5'-end of the left chain is the 3'-end of the right and vice versa, therefore the chains are called "anti-parallel".  Also, the 5'- and 3'-end is in RNA.  The positions 5 'and 3' are chosen to designate the beginning and the end because it is through them that covalent bonds are formed in the DNA and RNA chains. <br><br><img src="https://habrastorage.org/files/6aa/1b9/1b7/6aa1b91b7b1046c39d4fd928c0264ac9.GIF"><br>  <i>DNA and RNA chains.</i> <br><br>  DNA and RNA sequences are always recorded from the 5'-end to the 3'-end.  There are several reasons for this: <br><br><ul><li>  The synthesis of new DNA and RNA chains starts from the 5'-end ( <a href="https://ru.wikipedia.org/wiki/%25D0%2594%25D0%259D%25D0%259A-%25D0%25BF%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0%25D0%25B7%25D0%25B0">DNA polymerases</a> (enzymes that synthesize the complementary DNA strand on a DNA or RNA template) and <a href="https://ru.wikipedia.org/wiki/%25D0%25A0%25D0%259D%25D0%259A-%25D0%25BF%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0%25D0%25B7%25D0%25B0">RNA polymerases</a> (enzymes that synthesize a complementary RNA strand on the DNA or RNA template) go through the template in direction 3 '-&gt; 5', therefore, the new chain is synthesized in the direction 5 '-&gt; 3'); </li><li>  The ribosome reads codons, moving along the mRNA in the direction of 5 '-&gt; 3'; </li><li>  The amino acid sequence is written in the coding chain of DNA in the direction of 5 '-&gt; 3' (a significant part of the mRNA is an exact copy of the region of the coding DNA chain with the replacement of thymine by uracil and with a hydroxyl group (-OH) instead of hydrogen in position 2 ', of course); <br><br><img src="https://habrastorage.org/files/adb/553/83d/adb55383d088486ebda7f29a71bea37a.jpg"><br></li><li>  Finally, it‚Äôs just convenient to have a generally accepted write rule. </li></ul><br>  A gene is a region of genomic DNA that defines the sequence of nucleotides of an RNA molecule: <br><br><ul><li>  Encoding RNA: messenger RNA (mRNA), in which the amino acid sequences of the corresponding protein are encoded as codons.  You can also find the name "informational RNA", then the abbreviation looks like "mRNA"; </li><li>  Non-coding RNA: transport RNA, ribosomal RNA, and others. </li></ul> <br>  The role of tRNA is to deliver an amino acid mRNA-ribosome to the complex.  In addition, it is tRNA that is responsible for the recognition of mRNA codons; for this, each tRNA contains a so-called ‚Äúanticodon‚Äù - a triplet complementary to the mRNA codon. <br><br><img src="https://habrastorage.org/files/44d/4a8/b8a/44d4a8b8a6e745a3999c437c40191e22.png"><br>  <i>The process of translation, catalyzed by the ribosome.</i>  <i>In the figure, the codons UUU and UCG contained in the mRNA are recognized by the AAA and AGC anticodons contained in the tRNA molecules.</i>  <i>Transport RNA with the CCC anticodon has already given its amino acid to the growing protein chain, and tRNA with the CAG anticodon is waiting for its turn.</i>  <i>The section of an mRNA molecule shown in the figure consists of four codons: GGGUUUUCGGUC.</i>  <i>The codon GGG corresponds to the amino acid glycine, UUU to phenylalanine, UCG to serine, GUC to valine.</i>  <i>So this region of mRNA encodes a fragment of a protein with the amino acid sequence glycine-phenylalanine-serine-valine.</i> <br><br>  Ribosomal RNAs are indispensable components of the ribosome.  The main function of rRNA is to provide the translation process: it participates in reading information from mRNA using tRNA adapter molecules and catalyzing the formation of peptide bonds between the amino acids attached to tRNA and the growing protein chain. <br><br><img src="https://habrastorage.org/files/32c/b09/58c/32cb0958c015406f9d45040b5c5f3cae.jpg"><br>  <i>The main types of RNA molecules (in fact, they are much larger).</i> <br><br>  The protein is a chain of amino acids covalently linked to each other through a peptide bond (the fact that it is this can be seen in the spoiler a little further).  After synthesis, a chain of amino acids must adopt a certain spatial structure - the " <a href="https://ru.wikipedia.org/wiki/%25D0%259A%25D0%25BE%25D0%25BD%25D1%2584%25D0%25BE%25D1%2580%25D0%25BC%25D0%25B0%25D1%2586%25D0%25B8%25D1%258F">conformation</a> " (about the spatial structure of proteins <a href="https://geektimes.ru/post/181850/">on Geektimes already told before me</a> ).  In addition, many large proteins actually consist of several proteins combined by hydrophobic interactions and hydrogen bonds into a single stable structure.  In this case, each of the ‚Äúbuilding proteins‚Äù is called a ‚Äúsubunit‚Äù, and the resulting large protein is called a ‚Äúmulti-subunit‚Äù. <br><br><div class="spoiler">  <b class="spoiler_title">20 amino acids that make up proteins</b> <div class="spoiler_text"><img src="https://habrastorage.org/files/3c7/602/57c/3c760257c8b94ce088af2e95a75211cd.png"><br></div></div><br><div class="spoiler">  <b class="spoiler_title">Peptide bond</b> <div class="spoiler_text"><img src="https://habrastorage.org/files/4c7/dad/6f0/4c7dad6f018442918430e77695161fc3.png"></div></div><br><br><img src="https://habrastorage.org/files/64c/b0d/140/64cb0d1404824fba84cd5a27b8001aeb.jpg"><br>  <i>Ribosome complex.</i>  <i>The picture is taken from the <a href="https://geektimes.ru/users/olegkovalevskiy/" class="user_link">OlegKovalevskiy</a> publication <a href="https://geektimes.ru/post/238451/">‚Äú3D printing of protein molecule models‚Äù</a> .</i> <br><br>  In the case of genes encoding a protein, the process of decoding genetic information looks like this: <br><br><ol><li>  RNA polymerase recognizes the <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D1%2580%25D0%25BE%25D0%25BC%25D0%25BE%25D1%2582%25D0%25BE%25D1%2580">promoter</a> and binds to it (if it is ‚Äúopen,‚Äù we will go further on the regulation of promoter activity); </li><li>  On the DNA template, the enzyme RNA polymerase synthesizes according to the principle of complementarity and synthesizes a ‚Äúpreform‚Äù of messenger RNA (pre-mRNA, in eukaryotes) or ready-made functional mRNA (in prokaryotes).  This process is called <a href="https://ru.wikipedia.org/wiki/%25D0%25A2%25D1%2580%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BA%25D1%2580%25D0%25B8%25D0%25BF%25D1%2586%25D0%25B8%25D1%258F_(%25D0%25B1%25D0%25B8%25D0%25BE%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B8%25D1%258F)">‚Äútranscription‚Äù</a> ; </li><li>  (only in eukaryotes) The pre-mRNA molecule undergoes modifications ("ripens") and becomes functional mRNA; </li><li>  mRNA is recognized by the <a href="https://ru.wikipedia.org/wiki/%25D0%25A0%25D0%25B8%25D0%25B1%25D0%25BE%25D1%2581%25D0%25BE%25D0%25BC%25D0%25B0">ribosome</a> , an enzyme that decodes the triplet code of mRNA and synthesizes peptide / protein based on it.  The amino acids from which the ribosome builds a protein are delivered in combination with transport RNA ( <a href="https://ru.wikipedia.org/wiki/%25D0%25A2%25D0%25A0%25D0%259D%25D0%259A">tRNA</a> ).  This process is called <a href="https://ru.wikipedia.org/wiki/%25D0%25A2%25D1%2580%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BB%25D1%258F%25D1%2586%25D0%25B8%25D1%258F_(%25D0%25B1%25D0%25B8%25D0%25BE%25D0%25BB%25D0%25BE%25D0%25B3%25D0%25B8%25D1%258F)">‚Äútranslation‚Äù</a> ; </li><li>  The peptide / protein can undergo posttranslational modifications (‚Äúmaturation‚Äù by analogy with mRNA) and become functional.  An important factor is that the system of posttranslational modification of eukaryotes is <b>much</b> more complicated and diverse than in prokaryotes, therefore not every eukaryotic protein can be correctly synthesized by bacteria. </li></ol><br>  In addition to the coding regions in the genome, there are numerous fragments that are also somehow involved in transcription.  The sites located close to the gene and called promoters are recognized by RNA polymerases (they say that the gene is under the control of this promoter).  Different promoters are recognized by different RNA polymerases.  For example, a gene under the control of a <a href="https://ru.wikipedia.org/wiki/%25D0%2591%25D0%25B0%25D0%25BA%25D1%2582%25D0%25B5%25D1%2580%25D0%25B8%25D0%25BE%25D1%2584%25D0%25B0%25D0%25B3%25D0%25B8">bacteriophage</a> promoter will not be transcribed into bacteria unless RNA polymerase of the corresponding bacteriophage is synthesized in it.  <u>Generally</u> . <br><br>  Also, each gene can have several regulatory sequences, which can be located either directly near the promoter (or even overlap with it), or at a distance of tens of thousands of base pairs from it.  Elements that enhance transcription are called <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D0%25BD%25D1%2585%25D0%25B0%25D0%25BD%25D1%2581%25D0%25B5%25D1%2580">‚Äúenhancers‚Äù</a> , suppressing transcription are called ‚Äúsilencers‚Äù, and the proteins interacting with them are called <a href="https://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B0%25D0%25BA%25D1%2582%25D0%25BE%25D1%2580%25D1%258B_%25D1%2582%25D1%2580%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BA%25D1%2580%25D0%25B8%25D0%25BF%25D1%2586%25D0%25B8%25D0%25B8">‚Äútranscription factors‚Äù</a> .  Although it is also accepted to call transcription factors necessary components of the transcription initiation complex, without which transcription is impossible in principle.  The fact is that only to start the synthesis of an RNA molecule on the DNA template in eukaryotes and archaea, an assembly of the whole supramolecular complex is necessary.  The simplest such complex includes a holoenzyme RNA polymerase and six so-called <a href="https://ru.wikipedia.org/wiki/%25D0%259E%25D0%25B1%25D1%2589%25D0%25B8%25D0%25B5_%25D1%2584%25D0%25B0%25D0%25BA%25D1%2582%25D0%25BE%25D1%2580%25D1%258B_%25D1%2582%25D1%2580%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BA%25D1%2580%25D0%25B8%25D0%25BF%25D1%2586%25D0%25B8%25D0%25B8">‚Äúcommon transcription factors‚Äù</a> (TFIIA, TFIIB, TFIID, TFIIE, TFIIF, and TFIIH).  The complex itself is called the <a href="https://en.wikipedia.org/wiki/Transcription_preinitiation_complex">‚ÄúTranscription preinitiation complex‚Äù</a> ( <a href="https://www.youtube.com/watch%3Fv%3DicZjgZozkB8">video</a> , each component of the complex is highlighted in one color or another). <br><br>  The prokaryotic transcriptional complex is completely different, so there is no point in inserting a eukaryotic gene into a bacterium together with a eukaryotic promoter.  A prokaryotic analogue of common transcription factors of eukaryotes and archaea can be called a protein, called the <a href="https://en.wikipedia.org/wiki/Sigma_factor">‚Äúsigma factor‚Äù</a> . <br><br><img src="https://habrastorage.org/files/cec/d46/30d/cecd4630d8594701a802fd386c998102.jpg"><br>  <i>Prokaryotic transcriptive complex.</i>  <i>The letters indicated in the figure are the generally accepted designations of the corresponding subunits.</i>  <i>œÉ70 - sigma factor <a href="https://ru.wikipedia.org/wiki/%25D0%2593%25D0%25B5%25D0%25BD%25D1%258B_%25D0%25B4%25D0%25BE%25D0%25BC%25D0%25B0%25D1%2588%25D0%25BD%25D0%25B5%25D0%25B3%25D0%25BE_%25D1%2585%25D0%25BE%25D0%25B7%25D1%258F%25D0%25B9%25D1%2581%25D1%2582%25D0%25B2%25D0%25B0">genes household</a> E. coli</i> <br><br>  The genomes of prokaryotes and eukaryotes have many similarities, and the previously mentioned Central Dogma of Molecular Biology is fair for both kingdoms.  However, there are also many significant differences.  For example, for bacteria, a system of operons is typical - genes grouped together that participate in the same process and are not transcribed individually, but as part of one long mRNA.  In eukaryotes, everything is completely different: the genes involved in the same process are scattered across different chromosomes, and the genes themselves are divided into coding fragments <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D0%25BA%25D0%25B7%25D0%25BE%25D0%25BD">"exons" by</a> non-coding regions <a href="https://ru.wikipedia.org/wiki/%25D0%2598%25D0%25BD%25D1%2582%25D1%2580%25D0%25BE%25D0%25BD">"introns"</a> .  In this case, first, the gene is transcribed completely, and then, at the RNA stage, the introns are cut out, and the exons are stitched to form the coding mRNA.  This process is called <a href="https://ru.wikipedia.org/wiki/%25D0%25A1%25D0%25BF%25D0%25BB%25D0%25B0%25D0%25B9%25D1%2581%25D0%25B8%25D0%25BD%25D0%25B3">"splicing</a> . <a href="https://ru.wikipedia.org/wiki/%25D0%25A1%25D0%25BF%25D0%25BB%25D0%25B0%25D0%25B9%25D1%2581%25D0%25B8%25D0%25BD%25D0%25B3">"</a>  At the same time, not all of the available exons can be inserted into the finished mRNA, but only some of them, in this case they speak of <a href="https://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BB%25D1%258C%25D1%2582%25D0%25B5%25D1%2580%25D0%25BD%25D0%25B0%25D1%2582%25D0%25B8%25D0%25B2%25D0%25BD%25D1%258B%25D0%25B9_%25D1%2581%25D0%25BF%25D0%25BB%25D0%25B0%25D0%25B9%25D1%2581%25D0%25B8%25D0%25BD%25D0%25B3">‚Äúalternative splicing‚Äù</a> .  Thus, a eukaryotic cell can synthesize several proteins, transcribing one and the same gene.  Among other things, this gives a very important consequence: inserting the eukaryote gene into the bacterium ‚Äúas it is in the chromosome‚Äù often does not make sense, since the bacterium is simply not able to produce splicing. <br><br>  There is another important difference.  For prokaryotes, the presence of DNA-genetic material outside the circular "chromosome", the so-called <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D0%25BB%25D0%25B0%25D0%25B7%25D0%25BC%25D0%25B8%25D0%25B4%25D1%258B">"plasmid"</a> - small circular double-stranded DNA molecules is characteristic.  In addition, prokaryotes do not have organoids, including the nucleus: all components of the bacterial cell are free to travel throughout the entire intracellular space.  In eukaryotes, there are no plasmids, but there are <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D0%25BB%25D0%25B0%25D1%2581%25D1%2582%25D0%25B8%25D0%25B4%25D1%258B">plastids</a> and <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B8%25D1%2582%25D0%25BE%25D1%2585%25D0%25BE%25D0%25BD%25D0%25B4%25D1%2580%25D0%25B8%25D1%258F">mitochondria</a> , whose genome includes plasmids (according to the most reasonable hypothesis, plastids and mitochondria are ‚Äúdescendants‚Äù of the prokaryotic architecture of the genome of cyanobacteria and bacteria captured by ancient single-cell proto-eukaryotes inside).  In addition, for eukaryotes, the presence of the nucleus and other intracellular compartments surrounded by its own membrane is typical.  Therefore, genetic engineering of a eukaryotic cell requires different approaches than genetic engineering of bacteria. <br><br>  The genetic code itself is organized as follows.  Each gene / exon consists of a set of triplets / codons - sequences of three nucleotides, between which there are no gaps.  Triplet organization is valid for both genes in the composition of DNA, and for the coding part of mRNA.  In the process of translation, transport RNA (tRNA) carrying a certain amino acid will ‚Äúrecognize‚Äù the corresponding three-letter triplets.  The ribosome detaches the amino acid from tRNA and attaches it to the growing amino acid chain, which, at the end of the broadcast, will either immediately become a mature, full-featured protein, or before that it will additionally undergo a number of modifications.  In this case, only one amino acid corresponds to each triplet, but several different codons can correspond to one amino acid.  This is understandable, because the standard genetic code has 61 coding codons, and <a href="https://ru.wikipedia.org/wiki/%25D0%2590%25D0%25BC%25D0%25B8%25D0%25BD%25D0%25BE%25D0%25BA%25D0%25B8%25D1%2581%25D0%25BB%25D0%25BE%25D1%2582%25D1%258B">only 20 proteinogenic amino acids</a> (of all codons, of course, 4 * 4 * 4 = 64, but three of them are non-coding, instead they serve as a signal to stop translation and are called " stop codons "). <br><br><img src="https://habrastorage.org/files/33c/9f5/8d1/33c9f58d19994145a7697ad86975bc32.png"><br>  <i>Codons in the standard genetic code.</i>  <i>Thank you for <a href="https://ru.wikipedia.org/wiki/%25D0%2593%25D0%25B5%25D0%25BD%25D0%25B5%25D1%2582%25D0%25B8%25D1%2587%25D0%25B5%25D1%2581%25D0%25BA%25D0%25B8%25D0%25B9_%25D0%25BA%25D0%25BE%25D0%25B4">wikipedia</a> picture.</i> <br><br>  So proteins - these are the very elements that are the last link in the chain between genomic DNA and the properties of the organism, the so-called <a href="https://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B5%25D0%25BD%25D0%25BE%25D1%2582%25D0%25B8%25D0%25BF">‚Äúphenotype‚Äù</a> .  Therefore, in order to somehow change the characteristic we need for an organism important to us, we need to change its DNA in such a way so that certain proteins appear in its cells, which will provide us with a target result.  This is the main idea of ‚Äã‚Äãall genetic engineering. <br><br><h3>  1) For what purposes are bacteria used in genetic engineering and why they </h3><br>  So, we figured out how and why the sequence of genomic DNA affects the properties and characteristics of the organism.  Of course, it will be very good if the trait is completely determined by just one gene - the insertion of a small fragment is no longer a serious problem.  For example, often the resistance of a plant to a herbicide or pest is determined by a single gene, therefore creating varieties with the desired resistance in such cases is not difficult (as opposed to bringing such a plant to the market).  The same is true for many bacterial resistances to antibiotics (in fact, bacteria have many defense mechanisms against antibiotics, but they work in an independent manner).  The opposite example is, for example, the attempt of scientists to teach plants to assimilate nitrogen from the atmosphere.  The fact is that the only source of nitrogen for plants is soil, in which nitrogen-containing compounds suitable for plant absorption are synthesized by microorganisms (either introduced as fertilizer by a careful gardener or a passing dog).  Obviously, the creation of plants with an alternative mechanism of nutrition would be very useful for agriculture.  But, unfortunately, this process is so complicated that the problem of its ‚Äútransfer‚Äù from the microorganism to the plant could not be solved until now. <br><br>  Finally, if our goal is to obtain a protein for some specific purposes (the study of the structure and functions of the protein, the creation of medical drugs or laboratory reagents based on it, etc.), then we are obviously also quite happy with the insertion of a single gene into the cell, which in this case, it is customary to call the ‚Äúorganism-producer‚Äù. <br><br>  Genetic engineering bacterium is a potential source material for creating: <br><br><ol><li>  producer of the protein we need on a laboratory or industrial scale; </li><li>  an active agent in some kind of chemical transformation of one compound into another, be it a fermentation process in the food industry, the creation of more favorable conditions for plant growth due to the introduction of a ‚Äúbacterial fertilizer manufacturer‚Äù into the soil or the utilization of steel scrap; </li><li>  <a href="http://humbio.ru/humbio/genexp/00083307.htm">clone libraries of genes</a> (a topic whose good description will increase the size of the article to indecent); </li><li>  medically significant drug, for example, to restore the microflora of the gastrointestinal tract; <br></li><li>  <i>Agrobacterium tumefaciens</i> bacteria strains for subsequent genetic modification of plants. <br></li></ol><br>  * could forget something, so additions in the comments are welcome. <br><br>  An interesting fact is that the first successful experiments in the field of genetic engineering of bacteria occurred long before the epochal work of Watson and Crick.  Moreover, on the basis of these experiments, the very fact that the information is contained in DNA was proved, after which scientists could not waste their time on hypotheses about RNA and protein. <br><br>  This work, carried out in 1944, is known as the <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D0%25BA%25D1%2581%25D0%25BF%25D0%25B5%25D1%2580%25D0%25B8%25D0%25BC%25D0%25B5%25D0%25BD%25D1%2582_%25D0%25AD%25D0%25B2%25D0%25B5%25D1%2580%25D0%25B8,_%25D0%259C%25D0%25B0%25D0%25BA%25D0%25BB%25D0%25B5%25D0%25BE%25D0%25B4%25D0%25B0_%25D0%25B8_%25D0%259C%25D0%25B0%25D0%25BA%25D0%25BA%25D0%25B0%25D1%2580%25D1%2582%25D0%25B8">Experiment of Avery, MacLeod and McCarthy</a> , the basis for which was the <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D0%25BA%25D1%2581%25D0%25BF%25D0%25B5%25D1%2580%25D0%25B8%25D0%25BC%25D0%25B5%25D0%25BD%25D1%2582_%25D0%2593%25D1%2580%25D0%25B8%25D1%2584%25D1%2584%25D0%25B8%25D1%2582%25D0%25B0">work of Frederick Griffith</a> , in the course of which it was found that infection with killed pathogenic and live non-pathogenic pneumococcal strains causes the development of the disease, while separately, they do not cause significant symptoms.  From this experiment it was concluded that the killed bacteria are capable of transmitting something to a non-pathogenic "colleague", as a result of which it becomes dangerous.  But what do they pass on to each other?  By 1944, there were three main candidates: DNA, RNA, and protein.  In order to establish a carrier, an elegant experiment was carried out: at that time, enzymes were available that could destroy DNA separately (DNase), separate RNA (RNAse) and proteins (proteinase) separately.  It was shown that the transfer of pathogenic properties did not occur only in those cases when the preparation of a dead pathogenic strain was treated with DNAse and was not dependent on the treatment of the preparation with RNase and proteinase. <br><br>  Thus, it was proved that the carrier of information about the trait is DNA.  In addition, it was clearly shown that spontaneous penetration of a foreign DNA molecule into a bacterial cell is possible. <br><br>  Why are bacteria so popular with obvious flaws (for example, the absence of eukaryotic post-translational modifications)?  It's simple.  They are unpretentious in work, easy to use and do not require expensive nutrient media. <br><br><h3>  2) How to create a genetic construct that is introduced into the bacterium </h3><br>  Modern genetic engineering of bacteria is mainly the introduction of a <a href="http://humbio.ru/humbio/tarantul_sl/0000107b.htm">plasmid vector</a> (modified bacterial plasmid containing the target gene and a set of other necessary elements, which are discussed below).  The change in the chromosome of the bacterium is less typical, but this procedure is also not something outlandish: for example, the bacteriophage T7 RNA polymerase gene was introduced into the chromosome of E. coli using a vector based on the <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D1%2580%25D0%25BE%25D1%2584%25D0%25B0%25D0%25B3">prophage</a> Œª in the process of creating one of the most popular strains in the laboratory.  There are three reasons why a researcher will often choose to insert a gene into a plasmid vector: <br><br><ul><li>  first, it is cheaper to insert a plasmid vector into a bacterium than to embed something into the chromosome; </li><li>  secondly, the methodology for ensuring the stability and inheritance of a plasmid vector in a bacterial cell, as well as the procedure for creating the desired genetic structure is well developed and simple to implement; </li><li>  thirdly, everything works fine with plasmid vectors. </li></ul><br>  A typical plasmid vector for working with bacteria is a small circular double-stranded DNA molecule carrying the target protein gene under the control of a specific promoter and a number of necessary genes and regulatory elements, the presence of which provides a constant amount of plasmid in the cell (‚Äúcopy number control‚Äù).  Obviously, even in the case of ultra-efficient synthesis of mRNA from a vector, there is little sense if it exists in the bacteria in the number of pairs: during the division process, the probability of forming a daughter cell without the desired plasmid is high. <br><br>  In addition to the gene and promoter, the main elements of the plasmid vector are: <br><br><ol><li>  <a href="https://ru.wikipedia.org/wiki/%25D0%25A2%25D0%25BE%25D1%2587%25D0%25BA%25D0%25B0_%25D0%25BD%25D0%25B0%25D1%2587%25D0%25B0%25D0%25BB%25D0%25B0_%25D1%2580%25D0%25B5%25D0%25BF%25D0%25BB%25D0%25B8%25D0%25BA%25D0%25B0%25D1%2586%25D0%25B8%25D0%25B8">ori</a> - plasmid replication start region.  Need to maintain a constant amount of the plasmid and its inheritance by daughter cells; <br><br></li><li>  marker gene - a gene that gives bacteria carrying a plasmid vector a certain property that allows them to be distinguished from unmodified bacteria.  Most often, antibiotic resistance is used as a marker, then the transformed cell is able to live and divide into a medium containing an antibiotic (‚Äúselective medium‚Äù).  In addition, the presence of an antibiotic in a nutrient medium helps to protect the cell culture from infection by unwanted organisms that may accidentally enter the nutrient medium.  Often, there is one resistance gene in the work, but it is quite typical to use two resistance genes at once. <br><br>  Another example is the use of the Œ≤-glucuronidase gene (GUS).  This enzyme turns certain compounds into colored or fluorescent, which can be detected visually by colony coloration.  Naturally, these compounds need to be added to a selective nutrient medium.  Another example is the use of the <a href="https://ru.wikipedia.org/wiki/%25D0%2597%25D0%25B5%25D0%25BB%25D1%2591%25D0%25BD%25D1%258B%25D0%25B9_%25D1%2584%25D0%25BB%25D1%2583%25D0%25BE%25D1%2580%25D0%25B5%25D1%2581%25D1%2586%25D0%25B5%25D0%25BD%25D1%2582%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25B1%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25BA">green fluorescent protein (GFP)</a> gene (although the use of GUS and GFP is more typical for working with plant and animal cells); <br><br></li><li>  the site responsible for the copy number control (there are few plasmids in the cell - it‚Äôs bad, a lot - it‚Äôs too bad); <br><br></li><li>  <a href="http://humbio.ru/humbio/tarantul_sl/000010d4.htm">polylinker</a> is a nucleotide sequence containing several unique overlapping restriction endonuclease recognition sites (cutting double-stranded DNA at a specific location).  The polylinker is needed in order to embed the target gene in an ‚Äúempty‚Äù vector. </li></ol><br><img src="https://habrastorage.org/files/9d6/e4e/591/9d6e4e5912ef4423b95fbbef6b09526f.gif"><br>  <i>The figure shows the restriction sites of the endonuclease <a href="https://en.wikipedia.org/wiki/BamHI"><i>BamH1</i></a> and <a href="https://en.wikipedia.org/wiki/BamHI"><i>EcoR1</i></a> .</i>  <i>Both enzymes recognize a certain portion of six base pairs and introduce single-stranded breaks in different places (in the figure they are indicated by arrow-triangles).</i> <i>       ,   ¬´ ¬ª ( ,   ¬´ ¬ª).</i> <br><br><img src="https://habrastorage.org/files/18a/35c/6f9/18a35c6f9ee8426c9a00960a640dbca1.gif"><br> <i>   .    ori,      ,  10   .</i> <br><br>  ,     .     ?  ,    ? <br> ,        .    : <br><br><ol><li>       ,    ; </li><li>      <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D0%25BE%25D0%25BB%25D0%25B8%25D0%25BC%25D0%25B5%25D1%2580%25D0%25B0%25D0%25B7%25D0%25BD%25D0%25B0%25D1%258F_%25D1%2586%25D0%25B5%25D0%25BF%25D0%25BD%25D0%25B0%25D1%258F_%25D1%2580%25D0%25B5%25D0%25B0%25D0%25BA%25D1%2586%25D0%25B8%25D1%258F"></a> . </li></ol><br><br><img src="https://habrastorage.org/files/8ac/209/dd2/8ac209dd279543e48b9b46944082042b.gif"><br> <i>        (      ).       .</i> <br><br><div class="spoiler"> <b class="spoiler_title">     F1  R1</b> <div class="spoiler_text"><img src="https://habrastorage.org/files/994/b96/e11/994b96e114a64371b0ac1caa846fed62.png"><br></div></div><br>     ,      ,         ,      .  ,     (     <i>BamH1</i>  <i>EcoR1</i> )    ,      ,   ¬´  ¬ª       .    ,                .            ,     <a href="https://ru.wikipedia.org/wiki/%25D0%259B%25D0%25B8%25D0%25B3%25D0%25B0%25D0%25B7%25D1%258B">-</a> ,        . <br><br>         ,          ,         ,    ¬´¬ª .      ,    ,                . ,         ,     ¬´¬ª      . <br><br><img src="https://habrastorage.org/files/827/834/e6f/827834e6ff2b44ad8500c6619fc4610a.jpeg"><br> <i>        , ,  , ,  ,   .         <i>E. coli</i> .</i> <br><br>     .     ,         .     ,    . ¬´¬ª   ,          :   ,   . ,       ,      .       (,   )   ,         .            ,   . <br><br>  .           .  ,           (          10%     ),         -  <a href="https://ru.wikipedia.org/wiki/%25D0%25A6%25D0%25B8%25D1%2582%25D0%25BE%25D0%25BF%25D0%25BB%25D0%25B0%25D0%25B7%25D0%25BC%25D0%25B0"></a> !             .        ,  ¬´¬ª     ¬´ ¬ª.  : <br><br><ol><li>      <a href="https://ru.wikipedia.org/wiki/%25D0%259B%25D0%25B0%25D0%25BA%25D1%2582%25D0%25BE%25D0%25B7%25D0%25BD%25D1%258B%25D0%25B9_%25D0%25BE%25D0%25BF%25D0%25B5%25D1%2580%25D0%25BE%25D0%25BD">  <i>E. coli</i></a> ( <i>lac</i> -)   . <br><br>   ,        . -,     <i>lac</i> -,    ,      .  :      ,   ?              . <br><br> ,      <i>lac</i> -.     ,     ,         <i>lac</i> -.  , <i>lac</i> -   ,      <u>  </u> .       ,        .      .              ‚Äî <a href="https://ru.wikipedia.org/wiki/%25D0%2591%25D0%25B0%25D0%25BA%25D1%2582%25D0%25B5%25D1%2580%25D0%25B8%25D0%25BE%25D1%2584%25D0%25B0%25D0%25B3%25D0%25B8"></a> . ,     7. <br><br>  ,           , ,      . <br><br><img src="https://habrastorage.org/files/9b4/7c4/026/9b47c4026dec43d981aa3c8587b75d46.png"><br><br>    ,  <i>E. coli</i>  -,     ,       -  . <br><br>       <i>E. coli</i>   -  7     -,    <i>lac</i> -.     ,       ¬´  7 +     <i>lac</i> -¬ª,        . <br><br>            .           ,       .   ,               ,         .    ¬´¬ª. <br><br>    -:         ,  ,      ,  ,     ,      .       <a href="https://ru.wikipedia.org/wiki/%25D0%2598%25D0%259F%25D0%25A2%25D0%2593"></a> . <br><br></li><li>       pL <a href="https://ru.wikipedia.org/wiki/%25D0%25A4%25D0%25B0%25D0%25B3_%25D0%25BB%25D1%258F%25D0%25BC%25D0%25B1%25D0%25B4%25D0%25B0"> Œª</a> . <br><br>    - cI.        ,  cI857:         30‚Å∞     42‚Å∞.              30‚Å∞,      42‚Å∞,      . </li></ol><br>   ,  .     ‚Äî         .  But that's another story. </div><p>Source: <a href="https://habr.com/ru/post/402025/">https://habr.com/ru/post/402025/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../402013/index.html">The evolution of chat bots: from darling Eliza to aggressive paranoiac Taya</a></li>
<li><a href="../402015/index.html">So expensive: on Thursday, the Bitcoin exchange rate for the first time exceeded the ounce of gold</a></li>
<li><a href="../402017/index.html">Programming the human nervous system</a></li>
<li><a href="../402019/index.html">Spreadtrum SC9861G-IA - Atom cores for export</a></li>
<li><a href="../402023/index.html">The growing threat of video game hackers</a></li>
<li><a href="../402027/index.html">NASA has opened free access to its software.</a></li>
<li><a href="../402029/index.html">SmartPoket keys organizer, or how we reinvent the wheel</a></li>
<li><a href="../402031/index.html">The system for working with a hotbed for role-playing games</a></li>
<li><a href="../402033/index.html">What is the strength of the brand Bluedio (response to hipsters)</a></li>
<li><a href="../402039/index.html">Google eliminates the production of Chromebooks Pixel</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>