<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Using UML for experiment on the evolutionary systematics of prokaryotes, and indirectly about the psychology of scientists</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="This article is a continuation of two others. Interesting results about the evolutionary systematics of prokaryotes, or ‚Äúmany species‚Äù , Genomes of se...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Using UML for experiment on the evolutionary systematics of prokaryotes, and indirectly about the psychology of scientists</h1><div class="post__text post__text-html js-mediator-article"> This article is a continuation of two others. <a href="http://habrahabr.ru/post/146331/">Interesting results about the evolutionary systematics of prokaryotes, or ‚Äúmany species‚Äù</a> , <a href="http://habrahabr.ru/post/146814/">Genomes of sequenced organisms are errors in the bases</a> . <br><br>  After them, I had the honor to receive some feedback from both interested and professionals in this matter.  Also, as you could see, there was a rather lively discussion.  On the one hand, I would like to respond to the comments received. <br><br>  On the other, put a new experiment.  And it would be desirable to attract to this those who are interested in such things.  If you do not have time - maybe you have free CPU time :)? 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
    <a name="habracut"></a><br><br>  <b>NCBI Position</b> <br><br>  Thanks to <a href="http://habrahabr.ru/users/kalobok/" class="user_link">Kalobok</a> , who works at NCBI, we managed to find out why ‚Äúpeople from NCBI do not do such simple cross-analysis‚Äù (see <a href="http://habrahabr.ru/post/146814/">Genomes of sequenced organisms - errors in the bases</a> ).  I must say the conversation with <a href="http://habrahabr.ru/users/kalobok/" class="user_link">Kalobok</a> was not very pleasant.  At the beginning, like many commentators, he tried to teach me, in every possible way to hint that I didn‚Äôt understand anything, I‚Äôm doing everything wrong, etc. <br><br>  Here are a few characteristic quotes from the correspondence: " <i>The essence of your claims ... is related to the incorrect use of data. ... there is nothing to correct. You just need to use the data correctly.</i> ", " <i>As for the anticodons - I already said, I do not remember the details, so I can not comment. Judging by your level of knowledge, you also don‚Äôt have much reason to talk about this topic. First, figure out whether it is possible to compare the genomes of different organisms at all.</i> <i>will answer that this is not</i>  <i>there is nothing else to discuss further. Try again to understand that NCBI doesn‚Äôt have dropout students, but professionals. And I‚Äôm not inclined to suspect that they are driving more than half of the defective data. Rather, I will assume that a layperson is mistaken , which got into someone else's field.</i> "," <i>Consider that NCBI is focused primarily on biologists. They are completely satisfied with the available data and tools. And a single non-biologist programmer with incomprehensible ideas of the weather does not.</i>  "," <i>Here is the opinion of the head of the group of bacterial genomes after reading the article: Yes, quite naive ... Such work has been going on for the past 20 years.</i>  <i>And this is some kind of lone craftsman</i> " <br><br>  Well, that sort of thing.  Probably, few people would restrain themselves in such a tone to lead the discussion.  But alas, such modern morals of people who have received a diploma of biology (biophysics, biochemistry ...) and started to understand programming a bit and now work in a respectable place. <br><br>  <b>How to survive in this evil world :)</b> <br><br>  What to do to a person who does not have a corresponding diploma, but has knowledge in a narrow, but not his own field?  Alas, it will always be related to it, starting with ‚Äú <i>your post is teeming with the self-confidence of a gifted student</i> ‚Äù, to containing an instructive, patronizing tone. <i>But in general it‚Äôs very interesting. Correct the language, add a literature review, describe methods and results, remove speculations from the discussions and maybe something will turn out.</i> "," <i>&gt; and there they will twist $ only at the temple - They won't do it, normal work. If you add links, you can get a diploma; But the discussion with the author shows that it would not have gone further</i> ". <br><br>  But the main thing is to understand a few psychological moments.  A person with a diploma and with a warm place - is girdling.  Even without having relevant knowledge (not in a broad aspect, but in this particular task), he allows himself to speak in the spirit of "superiority over the interlocutor."  The discussion, as a rule, is not carried out in essence, but the weakest argument of the interlocutor is sought, or deliberate speculation, then ‚Äúmy son, read, teach this‚Äù is advised, and as a rule, irrelevant to the question, strong arguments are ignored, and then it turns out you are a diploma, who is your father and mother, etc. <br><br>  With this, I met, well, over the years 10 repeatedly.  My advice is simple - ignore.  Do not be fooled by provocations, and do not teach what you are told - you do not need it.  The first such case was in my school when I spoke on literary characters, for example, about ‚ÄúThe Master and Margarita‚Äù or about Natasha Rostova from ‚ÄúWar and Peace‚Äù.  Then they told me how to speak negatively about what I did not read.  Then I was seduced and read "The Master ..." and then I was able to speak with all the scrupulousness on this subject.  It was easier with Natasha (the novel read only diagonally), they wrote an essay - they wrote there convincingly that this girl should not be given such important attention - it is not worth it.  The rating was excellent, with a comment in the form ‚Äúeverything is very well justified with quotations, but it may be worth looking at it from the other side, as a manifestation of the Russian soul ...‚Äù.  Not worth it, I said then, and went into adulthood :) <br><br>  Over time, I felt sorry for the time when you spend it on the beaverd.  Why do you ask literature? Yes, everything is one to one, since then everything is laid down - either you will be led or you will decide for yourself. <br><br>  And yet - all this does not mean that such discussions should be avoided.  Never allow yourself to behave like your opponent (although it is often not easy) - look for the truth in his words, they are really small, but if the opponent leads a discussion with you - he is already interested, and from time to time he gives out something Useful for you - learn how to filter. <br><br>  However, sorry for this lyrical digression.  Further in essence. <br><br>  <b>What ended the deal with NCBI?</b> <br><br>  As expected, they admitted their mistakes, but did so with a good face on their face :) <br><br>  " <i>The data that you took from FTP is the original sequences and annotations sent by the researchers. They are not yet verified by the NCBI, and there may be quite a few errors there [they are marked as] NCBI review ... Even in verified data, such errors can occur, both because of not 100% reliability of checks, and for historical reasons (in many of the old records they relied on the reliability of data submitters and did not make additional checks - so far they are). such data are periodically reviewed</i>  <i>are being corrected. One of the biologists clearly said that if for some reason he used raw data from the genome, he would simply correct them manually, but most likely would use tRNAdb [this is another database where there is less data, but they corrected].</i> " <br><br>  ‚Äú <i>Here, by the way, another comrade responded. He says that now our standard program for checking data simply does not check the correctness of tRNA. Because it turns out to be very expensive in computing power. They plan to write a separate program for this, but for now there are more priority tasks. So wait.</i> " <br><br>  Therefore, the lyrics are lyrics, but the fact turned out to be a fact.  It can take a long time to ‚Äúpounce‚Äù on a non-biologist programmer, but the fact is that more than 50% of the data from NCBI is not verified - there is a reliable and recognized fact.  It should not be taken as a criticism of NCBI - they do and contain a lot of good information, which is valuable even with errors.  This is just for information to biologists who told fairy tales in comments in past articles. <br><br>  It seems they are going to correct this data, but this is not a priority for them, since  Many of these errors are not noticed, if they themselves notice the correct.  But if they correct it themselves, because  they do not trust lists of errors from others. <br><br>  <b>We do not wait for bug fixes.</b>  <b>But what can be done without it?</b> <br><br>  The main criticism of the article <a href="http://habrahabr.ru/post/146331/">Interesting results about the evolutionary systematics of prokaryotes, or ‚Äúmany-species origins‚Äù,</a> consisted in the following claim ‚Äú <i>One gene cannot be considered as a measure</i> ‚Äù.  With this, I fully agree, and new experiments should fix this. <br><br>  Some numbers.  Now in NCBI there are about 2000 genomes of bacteria.  In preparation for the experiment, I isolated all the tRNAs that are labeled this way.  They turned out more than 40 thousand unique variations.  But alas, there are many mistakes among them. <br><br>  But I thought that you can skip the stage of full error correction.  How to do it?  I sorted these tRNcs by length and presence of the end of the CCA at the end of the sequence.  It must be said that the CCA sequence is obligatory for any tRNA, and the length can be from 74 to 96 nucleotides. <br><br>  There are many miracles in NCBI up to tRNA from one nucleotide, or more than 1300 :) (you can‚Äôt tell without a smile).  Therefore, I removed the sequences that are up to 70 and over 100 in length, as well as those that do not end in CCA. <br><br>  There are about 20,000 of them. These are the most likely tRNAs that do not contain errors from the NCBI.  With the remaining half of the tRNA - you can figure it out later. <br><br>  In fact, for a planned experiment, it makes no difference whether this particular sequence of 70‚Äì100 nucleotides in length contains errors or not.  Why?  Since I am going to double-check the genomes of 2000 bacteria, are there really such sequences - the errors will be excluded.  And tRNA is actually whether or not this is the second thing.  The main thing is that different organisms coincide significant portions of DNA.  The coincidence of the sequence length of 70-100 in the genomes is far from accidental.  Already after a length of&gt; 10, the coincidence coincidence approaches zero, and at 70-100 this is already that important part of the genome that cannot simply coincide in different organisms. <br><br>  Therefore, what am I doing now?  I take these 20,000 tRNAs and find which bacteria they are present in.  If the sequence is present in only one organism, this is not interesting.  And most likely this is an erroneous sequence.  And thus a substantial percentage of errors is eliminated. <br><br>  If there is a sequence in more than one organism, this is one association (connection) between two organisms. <br><br>  Then the question arose how to visualize it well.  The idea is that the organism is a class.  The current phylogenetic taxonomy in the form of a tree is the inheritance between classes. <br><br>  tRNA is a class property, and the aggregation of these properties in different organisms is a horizontal gene transfer (the same association). <br><br>  Having generated the corresponding backbone of the code, you can display it automatically using UML and visually see all these links in the class diagram. <br><br>  <b>What is the problem?</b> <br><br>  Now the problem is in CPU time.  I make up the base of having 20,000 tRNA in 2,000 genomes of bacteria.  Only about 100 tRNAs are processed per day.  Therefore, I would be grateful to those who are interested in and help with processor time - well, such as an undeveloped project :) <br><br>  If anyone is interested, write in private messages - you need a place on hard about 50GB, some time for me to explain what's what, and then I can send packets of 100 tRNA for processing, and you send the results after processing. </div><p>Source: <a href="https://habr.com/ru/post/147063/">https://habr.com/ru/post/147063/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../147055/index.html">Continuous Testing in .NET</a></li>
<li><a href="../147056/index.html">#SumIT #Habr #Beer (07/06/2012)</a></li>
<li><a href="../147058/index.html">Chasing the idea</a></li>
<li><a href="../147059/index.html">Samsung Galaxy S3 - disassembly</a></li>
<li><a href="../147060/index.html">Software licenses in a nutshell</a></li>
<li><a href="../147064/index.html">From engineer to manager. Part 3: Reports</a></li>
<li><a href="../147068/index.html">Another closure of unnecessary Google services</a></li>
<li><a href="../147071/index.html">Amazon let down diesel generators</a></li>
<li><a href="../147072/index.html">Walmart and the Soviet Union: who is who in the cloud market</a></li>
<li><a href="../147075/index.html">Infinitely profitable program</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>