<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Epigenetic Aging Biomarkers</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="The history of epigenetic biomarkers began in 2013. Then the pioneer in this direction, an expert in the field of genetics and biostatistics, an emplo...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Epigenetic Aging Biomarkers</h1><div class="post__text post__text-html js-mediator-article">  The history of epigenetic biomarkers began in 2013.  Then the pioneer in this direction, an expert in the field of genetics and biostatistics, an employee of the University of California at Los Angeles, Steve Horvath, presented his new revolutionary method for determining the biological age, called the "epigenetic clock."  As the name implies, this method is based on changes in the epigenome, namely DNA methylation. <br><br>  DNA methylation is one of the epigenetic mechanisms of gene expression regulation.  In the course of methylation, the methyl group CH3-with special enzymes is attached to one of the bases of DNA, cytosine.  As a result, 5-methylcytosine is formed and gene expression is inactivated - the transcription process is blocked.  As is known today, DNA methylation is a dynamic process.  It can change under the influence of external factors, is associated with the development of a number of pathologies and can be inherited by the next generations.  Methylation plays a key role in the deactivation of foreign DNA, as well as in the processes of development and aging.  Age-related changes in methylation, known as "epigenetic drift", are described.  Thus, hypomethylation (demethylation) and the associated chromosomal instability are observed with age.  In addition, the reverse process occurs during aging - hypermethylation of certain promoter regions, including certain tumor suppressor genes, which is associated with the development of pathologies [1].  In general, today it is believed that changing methylation plays a key role in aging. <br><br><a name="habracut"></a>  Based on the fact that the chronological age is associated with predictable epigenome changes, hypo- and hypermethylation in many regions throughout the genome, the first generation of epigenetic biomarkers of aging based on DNA methylation was developed.  In 2011, the first article by scientists at the University of California at Los Angeles, entitled ‚ÄúEpigenetic Age Predictor‚Äù, showed that DNA methylation has a clear correlation with age.  In 2103, Steve Horvath identified 353 methylation sites using computer-aided learning techniques that were highly correlated with the chronological age of a person, which formed the basis of the first ‚Äúepigenetic hours‚Äù or ‚ÄúHorvat hours‚Äù [2]. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      The importance of this discovery is difficult to overestimate.  I must say that the success story of S. Horvat is full of not invented drama.  According to the memoirs of Horvat himself, his article on the new epigenetic method of age measurement at first did not want to be accepted not in one of the journals.  He constantly had to hear from the reviewers the same phrase, that "this is too good to be true."  But Croat did not give up and devoted all his time to finalizing the new method: ‚ÄúI wrote in every free time, as if it was the last time to complete my article.‚Äù  Just before the end of the work in the family of Horvat, a misfortune happened - his recently born daughter died.  But this is not knocked out of the rut stubborn scientist.  The last edition of his article was again not accepted in Genome Biology.  And having received another critical comment from reviewers, Horvath, in his recollection, did three things that should not be done: ‚ÄúAfter reading the reviewers' comments, I spent the next 10 minutes doing three things that should never be done.  First, I went to the fridge and drank three bottles of beer as fast as I could.  Secondly, I went back to the computer and composed a letter to the editor.  Third, I sent it. ‚Äù  And this time fortune smiled at him, his article was accepted. <br><br><div style="text-align:center;"><img src="https://habrastorage.org/webt/oc/tw/3a/octw3ajfvutra7f-fgbowokqexo.jpeg"></div> <br>  <i>Steve Horvath.</i> <br><br>  How can an epigenetic clock be useful?  According to the author of the discovery, this method can play a large role in evaluating the effectiveness of anti-aging interventions, as it allows to significantly shorten the waiting time for results.  In addition, it is assumed that the sites of methylation identified by Horvath are not just markers, but also active participants in aging (at least some).  What gives researchers a potential therapeutic target in the fight against aging and age-related pathologies.  Also, this new method can be useful in criminal proceedings when it is necessary to establish an age by samples of fabric.  But, of course, the most important "epigenetic clock" later acquired in predicting diseases and increasing the risk of mortality. <br><br>  As it turned out, ‚Äúepigenetic clocks‚Äù showed the value ‚Äúzero‚Äù in embryonic stem and induced pluripotent cells, spermatozoa, egg cells and placental cells.  And during the first 5 years of life - an accelerated course associated with the development of the body.  By the year 21, the ‚Äúepigenetic clock‚Äù gradually slowed down its course, and then went at a more or less uniform speed, changing its indicators under the influence of external factors.  Moreover, it turned out that different tissues age at different speeds.  The most susceptible to the aging process was the breast tissue in women.  From the point of view of epigenetics, the brain is aging most slowly, and blood cells and bone tissue show slightly accelerated aging [3]. <br><br>  In the same 2013, an article by Chinese and American scientists was published, representing another version of the ‚Äúepigenetic clock‚Äù - ‚ÄúHannum clock‚Äù.  Bioengineer of the University of California at San Diego Gregory Hannum and colleagues studied DNA methylation profiles of 450,000 CpG dinucleotides from human blood cells between the ages of 19 and 101, highlighting the 71 methylation sites most correlated with age [4].  The main difference between the ‚ÄúHannum clock‚Äù and the ‚ÄúHorvath clock‚Äù is in their specificity: DNA samples are taken from blood cells, and not from any tissue, as in the Croat method. <br><br>  <i>The main value of the ‚Äúepigenetic clock‚Äù became clear soon enough.</i>  <i>Comparing their progress with chronological age has great predictive value for estimating the risk of all-cause mortality and the development of many pathologies.</i>  <i>If the ‚Äúepigenetic clock‚Äù is in a hurry, diseases arise, accelerated aging and shortening of the life span.</i>  <i>If they are slower than chronological age, there are good chances for longevity.</i>  <i>Studies have shown that the "epigenetic hours" have a high correlation with all-cause mortality and age-related diseases.</i> <br><br>  In 2015, Horvath and his colleagues conducted research to determine the relationship between the occurrence of lung cancer and indicators of epigenetic age.  After analyzing the data of 2,029 people, it was found that the acceleration of epigenetic age was associated with an increased risk of lung cancer, and this relationship was stronger in smokers and those over 70 years of age: ‚ÄúThe results showed that standardized indicators of acceleration of epigenetic age (IEAA) were significantly associated with the incidence of lung cancer (HR: 1.50, P = 3.4 √ó 10 -3).  In addition, we have shown that the association can be even stronger among older people (70 years or older) or those who are currently smokers.  Overall, our results indicate that IEAA may be a useful biomarker for assessing susceptibility to lung cancer from the point of view of biological aging ‚Äù[5]. <br><br>  In the same year, another group of researchers established a relationship between the ‚Äúepigenetic age‚Äù and the risk of all-cause mortality in people over 60 years of age.  For the analysis, data were taken from four studies (4,658 people in total), whose average age of participants was 79.1, 69.5, 66.3 and 72.9 years, respectively.  The epigenetic age was determined by two methods: ‚ÄúHorvat clock‚Äù (based on 353 CpG methylation sites) and ‚ÄúHannum clock‚Äù (based on 71 CpG methylation site).  Both methods showed a strong correlation with each other, although the basic sets had matches only for 6 CpGs sites.  The results of the study showed that the acceleration of the ‚Äúepigenetic age‚Äù compared to the chronological one by 5 years increased the risk of mortality in humans by 16%.  The authors concluded: ‚ÄúAccelerated aging indicators derived from DNA methylation are inherited factors that predict mortality regardless of health status, lifestyle factors, and known genetic factors.  Therefore, it can be assumed that the predicted age of DNA methylation is an ‚Äúepigenetic clock,‚Äù which measures the biological age, going together, but not always in parallel with the chronological age, and can give predictions of life expectancy ‚Äù[6]. <br><br>  In 2016, German oncologists conducted a study that described the relationship between accelerated epigenetic aging and mortality from cancer and cardiovascular diseases and all other causes.  For their work, they used the ‚Äúepigenetic clock‚Äù of Horvath and Hannum.  The age of DNA methylation was assessed in a cohort of 1,863 elderly people in the ESTHER study, whose average age was 62.5 years.  The results showed that the epigenetic age, exceeding the chronological age, was associated with a higher mortality.  Acceleration by 5 years of ‚Äúepigenetic age‚Äù, determined by the method of Horvat, gave an increase in mortality by 22%, and by 16% using the method of Hannum [7]. <br><br>  In 2016, a large international team of researchers led by Steve Horvath conducted a large-scale meta-analysis in which data from 13,089 people from three racial / ethnic groups were studied: whites, Latin Americans and African Americans.  The epigenetic age here was also determined by two methods: ‚Äúthe clock of Horvath‚Äù and ‚Äúthe clock of Hannum‚Äù.  This work showed that the acceleration of ‚Äúepigenetic hours‚Äù for 1 year (compared to chronological age) increased the risk of all-cause mortality to 4%.  Moreover, the opposite effect was observed: the slowing down of the ‚Äúepigenetic hours‚Äù led to a decrease in the risk of mortality.  In addition, the researchers recorded an interesting phenomenon: ‚ÄúWe found that 5 percent of people have a faster biological age, which leads to a shorter lifespan.  Accelerated aging increases the risk of death for these people by 50 percent in any adult age. ‚Äù[8, 9]. <br><br>  A number of external factors are also known that influence the course of the ‚Äúepigenetic clock‚Äù.  In 2017, Horvath and his colleagues described the relationship of diet, alcohol use, education and exercise with the course of epigenetic time.  According to the authors themselves, their work confirmed a long-known truth: a diet with vegetables, fish and lean meat, moderate alcohol consumption, physical activity and education slow down the epigenetic time and contribute to the prolongation of life.  Epigenome aging increased levels of insulin and glucose, C-reactive protein and triglycerides, as well as overweight and increased blood pressure.  In the same study, the scientists did not find a positive effect of the anti-diabetic drug metformin on the course of epigenetic time [10].  In the same year, Finnish researchers once again demonstrated the relationship of obesity with accelerated epigenetic aging [11] <br><br><img src="https://habrastorage.org/webt/wz/i8/ev/wzi8evgti8afw35pdls3ggak6ei.jpeg"><br><br>  Fig.  1 <b>Factors affecting the epigenetic age</b> (from [10]). <br>  <i>Blue and red arrows show the factors that slow down and accelerate aging, respectively.</i>  <i>The clock symbolizes the outer epigenetic clock (extrinsic epigenetic clock, the extended version of the Hannum method) and the inner epigenetic clock (intrinsic epigenetic clock, Horvath clock).</i>  <i>Positively affect the course of epigenetic time (slow): fish, poultry, vegetables and fruits, high density lipoproteins, moderate alcohol, education and physical activity.</i>  <i>Negatively affect the course of epigenetic time (accelerate): elevated levels of insulin, glucose, C-reactive protein, triglycerides, increased blood pressure, increased body weight and the wrong ratio of waist and hips.</i> <br><br>  Different teams of researchers conducted work that described the relationship of the epigenetic time with physical and cognitive functions, Down and Werner syndromes, HIV infection, Alzheimer's disease and menopause [12-17].  A clear relationship was also demonstrated between longevity and the slowing down of the ‚Äúepigenetic hours‚Äù on the example of the Italian 100-year-old centenarians [18].  For a long time, Steve Croat did not succeed in predicting the accelerated course of biological age with the Hutchinson-Gilford progeria using his method.  But this year this problem was solved: he and his colleagues created special ‚Äúepigenetic clocks‚Äù based on fibroblasts, consisting of 391 CpG sites called ‚Äúskin-blood clock‚Äù [19]. <br><br>  The next stage of S. Horvat‚Äôs work was the development of a more efficient biomarker of aging, with improved diagnostic capabilities that predict the risk of mortality from various causes and the development of age-related diseases.  The disadvantage of the previous ‚Äúepigenetic hours‚Äù, according to Horvat, was that the use of chronological age as a reference for determining age-related deviations may exclude CpG sites, whose methylation profiles do not exhibit strong, time-dependent changes.  Instead, they show only a discrepancy between biological and chronological age.  Thus, it is important not only to capture CpG, which reflect the difference with chronological time, but also those that demonstrate differences in risk and physiological status among persons of the same chronological age. In 2018, a new biomarker, called DNAm PhenoAge, was created. <br><br>  At the first stage, the team of S. Horvat analyzed on a large sample of people, 9926 people from the NHANES III study, the relationship of 42 clinical biomarkers and chronological age with the risk of mortality.  As a result, 9 biomarkers were identified to predict phenotypic age: <br><br>  <b>1. Albumin</b> <b><br></b>  <b>2. Creatinine</b> <b><br></b>  <b>3. Glucose</b> <b><br></b>  <b>4. C-reactive protein</b> <b><br></b>  <b>5. The percentage of lymphocytes</b> <b><br></b>  <b>6. The average volume of red blood cells</b> <b><br></b>  <b>7. The width of the distribution of red blood cells</b> <b><br></b>  <b>8. The number of leukocytes</b> <b><br></b>  <b>9. Alkaline phosphatase.</b> <br><br>  Then they validated selected biomarkers ‚Äî they tested their model of phenotypic age on a different sample of people (6209 people).  The audit showed an increase in phenotypic age with an increase in the risk of all-cause mortality: a one-year increase in the phenotypic age was associated with a 9% increase in the risk of all-cause mortality, a 9% increase in the risk of death from aging-related diseases, a 10% increase in the risk of death from heart disease. vascular diseases, a 7% increase in the risk of death from cancer, a 20% increase in the risk of death from diabetes and a 9% increase in the risk of death from respiratory diseases. <br><br>  At the second stage of the study, the authors associated DNA methylation with a phenotypic age.  Scientists have determined exactly which methylated GpC sites are associated with these 9 clinical biomarkers predicting a phenotypic age.  They identified 513 CpG sites that predicted phenotypic age.  A new epigenetic marker was named DNAm PhenoAge.  The test showed a fairly high accuracy of the new biomarker: between 1998 and 2007, the mean change in DNAm PhenoAge was 8.51 years, while the mean change in clinical phenotypic age was 8.88 years. <br><br>  After that, the researchers conducted a run-in of their new biomarker marker using data from 4 large studies, the Women's Health Initiative (n = 4207), the Framingham Heart Study (n = 2553), the Normative Aging Study (n = 657) and the Jackson Heart Study (n = 1747)  The results showed that DNAm PhenoAge was significantly associated with the subsequent risk of mortality in all studies (regardless of chronological age): the one-year increase in DNAm PhenoAge is associated with an increased risk of all-cause mortality of 4.5%. <br><br>  Genetic analysis of DNAm PhenoAge showed that the 513 CpG sites used in it have the same 41 CpGs with the Horvat clock and the same 6 CpGs with the Hannum clock.  Five identical CpG were detected in all three epigenetic methods.  149 CpG from DNAm PhenoAge are located in areas of CpG dinucleotide clusters (CpG islands). <br><br>  All the data obtained during the study by Steve Horvath and his team showed that the new biomarker has a great predictive value in determining the risks of the occurrence of age-related diseases and the risk of mortality.  ‚Äú <i>Using an innovative two-step process, we developed a new epigenetic biomarker of aging, DNAm PhenoAge, which greatly exceeded previous biomarkers in terms of predictions for various aging factors, including all-cause mortality, cancer, general health, physical functioning, and Alzheimer's disease.</i>  <i>Although this biomarker was developed using data from whole blood, it strongly correlates with age in each tissue and cell under test.</i>  <i>Based on in-depth transcriptional analysis in sorted cells, we found that an increased epigenetic age relative to chronological age is associated with an increase in the activation of pro-inflammatory and interferon pathways and a decrease in activation of transcriptional / translational mechanisms, response to DNA damage and mitochondrial signatures.</i>  <i>In general, this single epigenetic biomarker of aging can encompass risks for a variety of different results in different tissues and cells and provide an insight into important pathways of aging</i> [20]. <br><br>  Author: Alexey Rzheshevsky. <br><br><div class="spoiler">  <b class="spoiler_title">Bibliography</b> <div class="spoiler_text"><ol><li>  Vayserman A.M., Voitenko V.P., Mekhova L.V.  Epigenetic epidemiology of age-related diseases.  Ontogenesis.  2011. 42, 1‚Äì21; </li><li>  Horvath S. DNA methylation of human tissues and cell types.  Genome Biol.  2013. 14, R115. </li><li>  Josh Mitteldorf.  Methylation Aging Clock: An Update.  February 14, 2018. </li><li>  Hannum, G;  Guinney, J;  Zhao, L;  Zhang, L;  Hughes, G;  Sadda, S;  Klotzle, B;  Bibikova, M;  Fan, JB;  Gao, Y;  Deconde, R;  Chen, M;  Rajapakse, I;  Friend, S;  Ideker, T;  Zhang, K (2013).  Genome-wide methylation profiles reveal human aging rates.  Mol cell.  49: 359-367. </li><li>  Morgan E. Levine, H. Dean Hosgood, Brian Chen, Devin Absher, Themistocles Assimes and Steve Horvath.  DNA methylation age of blood choices.  Aging (Albany NY).  2015 Sep;  7 (9): 690‚Äì700. </li><li>  Riccardo E Marioni, Sonia Shah, et al.  DNA methylation age of blood prediction all-cause mortality in later life.  Genome Biol.  2015;  16 (1): 25. </li><li>  Laura Perna, Yan Zhang, Ute Mons, Bernd Holleczek, Kai-Uwe Saum, and Hermann Brenner.  Epigenetic acceleration of cancer, cardiovascular, and all-cause mortality.  Clin Epigenetics.  2016;  8: 64. </li><li>  Brian H. Chen, Riccardo E. Marioni et al.  DNA methylation-based measures of biological age: meta-analysis of prediction time to death (Albany NY).  2016 Sep;  8 (9): 1844-1859. </li><li>  Epigenetic clock predicts life expectancy.  ScienceDaily.  28 September 2016. </li><li>  Quach A1, Levine ME1 et al.  Epigenetic clock analysis of diet, exercise, education, and lifestyle factors.  Aging (Albany NY).  2017 Feb 14; 9 (2): 419-446. </li><li>  Nevalainen T, Kananen L, Marttila S, Jylh√§v√§ J, Mononen N, K√§h√∂nen M, Raitakari OT, Hervonen A, Jylh√§ M, Lehtim√§ki T, Hurme M.  Clin Epigenetics.  2017 Feb 14; 9:20. </li><li>  Marioni RE, Shah S, McRae AF, Ritchie SJ, Muniz-Terrera G, Harris SE, Gibson J, Redmond P, Cox SR, Pattie A, Corley J, Taylor A, Murphy L, et al .. The epigenetic clock is correlated with the Lothian Birth Cohort 1936. Int J Epidemiol.  2015;  44: 1388‚Äì96. </li><li>  Horvath S, Garagnani P, Bacalini MG, Pirazzini C, Salvioli S, Gentilini D, Di Blasio AM, Giuliani C, Tung S, Vinters HV, Franceschi C. Accelerated epigenetic aging in Down syndrome.  Aging Cell.  2015;  14: 491‚Äì95. </li><li>  Horvath S, Levine AJ.  HIV-1 Infection Accelerates  J Infect Dis.  2015;  212: 1563‚Äì73. </li><li>  Maierhofer A, Flunkert J, Oshima J, Martin GM, Haaf T, Horvath S. Accelerated epigenetic aging in Werner syndrome.  Aging (Albany NY).  2017;  9: 1143‚Äì52. </li><li>  Levine ME, Lucky AT, Bennett DA, Hormones, and Alzheimer's disease related cognitive functioning.  Aging (Albany NY).  2015;  7: 1198‚Äì211 </li><li>  Levine ME, Lu AT, Chen BH, Hernandez DG, Singleton AB, Ferrucci L, Bandinelli S, Salfati E, Manson JE, Quach A, Kusters CD, Kuh D, Wong A, et al. Menopause accelerates biological aging.  Proc Natl Acad Sci USA.  2016;  113: 9327‚Äì32. </li><li>  Horvath S, Pirazzini C, Bacalini MG, Gentilini D, Di Blasio AM, Delledonne M, Mari D, Arosio B, Monti D, Passarino G, De Rango F, D'Aquila P, Giuliani C, et al. Decreased epigenetic age of PBMCs from the Italian semi-supercentenarians and their offspring.  Aging (Albany NY).  2015;  7: 1159‚Äì70. </li><li>  Steve Horvath, Junko Oshima et al.  Hildeson Gilford Progeria Syndrome and ex vivo studies.  Aging.  Volume 10, Issue 7, pp 1758-75. </li><li>  20. Morgan E. Levine, Ake T. Lu, et al.  An epigenetic biomarker of aging for lifespan and healthspan.  Aging (Albany NY).  2018 Apr;  10 (4): 573‚Äì591. </li></ol></div></div></div><p>Source: <a href="https://habr.com/ru/post/428377/">https://habr.com/ru/post/428377/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../428367/index.html">We made the world's smallest sonar modem</a></li>
<li><a href="../428369/index.html">Getting out of the wheel of Samsara, extremism and some greens - analysis of tasks from the GridGain booklet at the Joker 2018 conference</a></li>
<li><a href="../428371/index.html">Old play IBM</a></li>
<li><a href="../428373/index.html">iPhone inconvenient to use</a></li>
<li><a href="../428375/index.html">The book "Working with BigData in the clouds. Processing and storing data with examples from Microsoft Azure ¬ª</a></li>
<li><a href="../428379/index.html">Real design process. Step-by-step story on how to create a business-oriented site</a></li>
<li><a href="../428381/index.html">$ 10 million investment and Wozniak's praise is a long way to create a computer designer for children</a></li>
<li><a href="../428383/index.html">Sony has published a full list of games for the PlayStation Classic</a></li>
<li><a href="../428385/index.html">More coffee, less caffeine: Intel 9th ‚Äã‚ÄãGen (Part 1)</a></li>
<li><a href="../428387/index.html">How to prolong life and remove pain from terminally ill patients: palliative medicine in Russia</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>