<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>What can bacteria tell about human evolution?</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="To discover the deep history and shape the future health of our species, we need to learn from the microbes that accompanied us on our evolutionary jo...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>What can bacteria tell about human evolution?</h1><div class="post__text post__text-html js-mediator-article"><h2>  To discover the deep history and shape the future health of our species, we need to learn from the microbes that accompanied us on our evolutionary journey. </h2><br><img src="https://habrastorage.org/getpro/geektimes/post_images/fa3/c71/5b6/fa3c715b604b0fdb07755c214bdadd15.jpg"><br><br>  Human nature has a desire to know its origin.  One by one we explore our family trees to find ancestors lost in history.  Together, scientists study data from a huge number of sources, from ancient fossils to modern genomes, and determine where humanity came from and how we became the species that exists today. <br><br>  Over the past decade, research in this area has undergone revolutionary changes due to a sharp drop in the cost <a href="https://ru.wikipedia.org/wiki/%25D0%25A1%25D0%25B5%25D0%25BA%25D0%25B2%25D0%25B5%25D0%25BD%25D0%25B8%25D1%2580%25D0%25BE%25D0%25B2%25D0%25B0%25D0%25BD%25D0%25B8%25D0%25B5">of genome sequencing</a> .  The human genome project was launched in 1990 and <a href="https://www.genome.gov/11006943/human-genome-project-completion-frequently-asked-questions/">cost $ 2.7 billion</a> - about $ 100 for each sequenced genome.  Today, the genome can be sequenced <a href="https://www.cnbc.com/2015/12/10/unlocking-my-genome-was-it-worth-it.html">for $ 1000- $ 2000</a> , and we have already come close to the long-standing goal <a href="https://techcrunch.com/2017/01/10/illumina-wants-to-sequence-your-whole-genome-for-100/">of $ 100</a> . <br><a name="habracut"></a><br>  Although most of the work with the genome has so far focused on the study of genetic health risks and diseases, we can explore the history of our species through genetic reconstruction.  But our own genes do not necessarily tell us the whole history of our travels and migrations as a species or about all the risks to our health. 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      Therefore, in recent years, researchers <a href="http://www.nature.com/nature/focus/humanmicrobiota/index.html">have</a> begun to look <a href="http://www.nature.com/nature/focus/humanmicrobiota/index.html">more closely</a> at our ‚Äúsecond genome‚Äù - the <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B8%25D0%25BA%25D1%2580%25D0%25BE%25D1%2584%25D0%25BB%25D0%25BE%25D1%2580%25D0%25B0_%25D1%2587%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25B2%25D0%25B5%25D0%25BA%25D0%25B0">microflora</a> genes.  Our <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC5433529/">microflora</a> is all microscopic organisms that live inside and outside of us, play a role in our <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3601187/">digestion</a> , <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4056765/">train our immune system</a> to respond correctly to <a href="https://ru.wikipedia.org/wiki/%25D0%259F%25D0%25B0%25D1%2582%25D0%25BE%25D0%25B3%25D0%25B5%25D0%25BD">pathogens</a> that make <a href="https://www.ncbi.nlm.nih.gov/pubmed/22940212">key vitamins</a> , and take a place that <a href="https://www.ncbi.nlm.nih.gov/pubmed/21270894">pathogens</a> could <a href="https://www.ncbi.nlm.nih.gov/pubmed/21270894">take</a> instead.  Intestinal microbes are the ‚Äúworld within the world‚Äù that evolved with us, their masters, while the early ancestors of people moved from place to place, ate new food and met new animals and environmental conditions.  Our today's <a href="http://learn.genetics.utah.edu/content/microbiome/">microflora</a> (the collective genetic material of the microflora) reflects part of that deep story. <br><br><h2>  Extreme symbionts in our cells </h2><br>  There are several ways to extract information about human history from these organisms within us.  One of them is the use of parts of our own cells, which are essentially microbial: our <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B8%25D1%2582%25D0%25BE%25D1%2585%25D0%25BE%25D0%25BD%25D0%25B4%25D1%2580%25D0%25B8%25D1%258F">mitochondria</a> .  These <a href="https://ru.wikipedia.org/wiki/%25D0%259E%25D1%2580%25D0%25B3%25D0%25B0%25D0%25BD%25D0%25B5%25D0%25BB%25D0%25BB%25D1%258B">organelles</a> can be considered "extreme symbionts": these are remnants of microorganisms that once lived free, but now are part of all <a href="https://ru.wikipedia.org/wiki/%25D0%25AD%25D1%2583%25D0%25BA%25D0%25B0%25D1%2580%25D0%25B8%25D0%25BE%25D1%2582%25D1%258B">eukaryotes</a> (complex cells) that produce energy and regulate metabolism. <br><br>  Mitochondria retain <a href="https://ghr.nlm.nih.gov/primer/basics/mtdna">their own DNA</a> , separate from the cell nucleus.  For many types of studies, <a href="https://ru.wikipedia.org/wiki/%25D0%259C%25D0%25B8%25D1%2582%25D0%25BE%25D1%2585%25D0%25BE%25D0%25BD%25D0%25B4%25D1%2580%25D0%25B8%25D0%25B0%25D0%25BB%25D1%258C%25D0%25BD%25D0%25B0%25D1%258F_%25D0%2594%25D0%259D%25D0%259A">mitochondrial DNA</a> (mtDNA) is preferred over nuclear DNA.  Unlike nuclear DNA, it is not a mixture of our parents' genetic material.  Since mtDNA is inherited exclusively from the egg and is passed on through the generations along the maternal line, it looks more like a clone of your mother (and her mother, and her mother, etc.).  And although eukaryotes have only one copy of nuclear DNA in one nucleus, they have many mitochondria, and therefore many copies of each mtDNA gene.  Since the mtDNA genome is much smaller than nuclear DNA (it contains 37 genes instead of 20,000), it is easier to analyze. <br><br>  An analysis of mtDNA, conducted in the 1980s, led to the <a href="https://www.nature.com/nature/journal/v325/n6099/abs/325031a0.html">conclusion</a> that <a href="https://ru.wikipedia.org/wiki/%25D0%2593%25D0%25B8%25D0%25BF%25D0%25BE%25D1%2582%25D0%25B5%25D0%25B7%25D0%25B0_%25D0%25B0%25D1%2584%25D1%2580%25D0%25B8%25D0%25BA%25D0%25B0%25D0%25BD%25D1%2581%25D0%25BA%25D0%25BE%25D0%25B3%25D0%25BE_%25D0%25BF%25D1%2580%25D0%25BE%25D0%25B8%25D1%2581%25D1%2585%25D0%25BE%25D0%25B6%25D0%25B4%25D0%25B5%25D0%25BD%25D0%25B8%25D1%258F_%25D1%2587%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25B2%25D0%25B5%25D0%25BA%25D0%25B0">mankind originated from Africa</a> , and the date of birth of the common maternal ancestor was determined 100,000‚Äì200,000 years ago.  And although this statement is considered generally accepted today, at the time it was <a href="https://www.nature.com/nature/journal/v329/n6135/abs/329111b0.html">controversial</a> , as some biologists and anthropologists <a href="http://anthrojournal.com/issue/october-2011/article/analysis-of-two-competing-theories-on-the-origin-of-homo-sapiens-sapiens-multiregional-theory-vs-the-out-of-africa-2-model">believed</a> that modern people emerged as a group derived from a diverse but interbreeding population of archaic people scattered throughout the Old World ( <a href="https://ru.wikipedia.org/wiki/%25D0%2593%25D0%25B8%25D0%25BF%25D0%25BE%25D1%2582%25D0%25B5%25D0%25B7%25D0%25B0_%25D0%25BC%25D1%2583%25D0%25BB%25D1%258C%25D1%2582%25D0%25B8%25D1%2580%25D0%25B5%25D0%25B3%25D0%25B8%25D0%25BE%25D0%25BD%25D0%25B0%25D0%25BB%25D1%258C%25D0%25BD%25D0%25BE%25D0%25B3%25D0%25BE_%25D0%25BF%25D1%2580%25D0%25BE%25D0%25B8%25D1%2581%25D1%2585%25D0%25BE%25D0%25B6%25D0%25B4%25D0%25B5%25D0%25BD%25D0%25B8%25D1%258F_%25D1%2587%25D0%25B5%25D0%25BB%25D0%25BE%25D0%25B2%25D0%25B5%25D0%25BA%25D0%25B0">hypothesis of human multiregional origin</a> ). <br><br>  Microbes within us can also help shed light on the travels of our ancestors, since they are also inherited in families and have long been associated with the human population.  One example is <a href="https://ru.wikipedia.org/wiki/Helicobacter_pylori">Helicobacter pylori</a> , a gastric bacterium that can cause ulcers and stomach cancer, but which many individuals can tolerate without any symptoms.  <a href="http://www.mayoclinic.org/diseases-conditions/h-pylori/symptoms-causes/syc-20356171">H. pylori is</a> transmitted <a href="https://www.ncbi.nlm.nih.gov/pubmed/11218379">from person to person</a> , probably through saliva (orally), or through contact with feces, and possibly through contaminated food and water.  Other Helicobacter species colonize the intestines of mammals, which indicates a long co-evolution of these types of bacteria, humans and our relatives.  In the past, H. pylori probably colonized a <a href="https://www.ncbi.nlm.nih.gov/pubmed/28456631">large percentage of people</a> , but this prevalence has decreased in many countries over the past hundred years due to improvements in sanitation and hygiene. <br><br>  Studies over the past 15 years have studied the evolution of H. pylori, collecting and sequencing bacterial strains from different people from all over the world.  Researchers have found that H. pylori collected in Africa <a href="https://www.nature.com/nature/journal/v445/n7130/full/nature05562.html">have the greatest genetic diversity</a> (just like human populations in East Africa), and that, in principle, one can track the main human migrations from this continent and across the globe by studying the genetic structure of this bacterium.  Genetic analysis also indicated that the bacterium had evolved together with people of about 60,000 years old - starting not far from the moment when modern people began to migrate from Africa, and carried along with them H. pylori and other bacteria.  Consequently, the H. pylori genome can be used to determine the evolutionary history of some populations of humans. <br><br><h2>  Restoring our past by their genes </h2><br>  But why bother doing this if you can study human bones or the genome to get this information?  For example, the same story told by the genetic data of two different organisms can be a serious confirmation of the correctness of the hypothesis, especially when these organisms differ as much as humans and bacteria.  In addition, sometimes data on a single genome can fill gaps in data that are not capable of filling other data.  Data on the H. pylori genomes were able to help <a href="http://www.pnas.org/content/101/14/4746.short">divide two ethnic communities</a> in the city of Ladakh in India, despite the fact that the genetic markers of people available at that time could not. <br><br>  Today, instead of studying the only microbial variant, studying a large collection of all of them can better inform us about the state of the human race and where we can go.  The idea of ‚Äã‚Äãthe <a href="https://en.wikipedia.org/wiki/Holobiont">golobiont</a> - the <a href="http://journals.plos.org/plosbiology/article%3Fid%3D10.1371/journal.pbio.1002226">host and all the microbes associated with it</a> , analyzed as the only hologen - is already acquiring a certain form, while we begin to understand the thousands of microflora that can live inside and outside our bodies. <br><br>  Our microflora not only reflects the evolution of man - it affects her.  Thanks to microbes associated with us, we can acquire abilities that have a beneficial effect on the population.  In a <a href="https://www.nature.com/nature/journal/v464/n7290/full/nature08937.html">2010 study,</a> it was found that in many people from Japan, their intestinal microbes possess a gene that allows them to produce an enzyme that helps them more effectively break down carbohydrates derived from algae.  This gene is not in the intestines of people from North America, where algae are not among the main products. <br><br>  This gene could be obtained by the intestinal bacterium Bacteroides plebeius, probably from the marine bacterium Zobellia galactanivorans.  The Japanese could eat Zobellia for a long time, and it could get into their intestines either in whole or in parts, including free DNA.  Since bacteria can get genes through <a href="http://www.sciencedirect.com/topics/biochemistry-genetics-and-molecular-biology/horizontal-gene-transfer">horizontal genetic transfer</a> , Bacteriodes could pick up this gene in the intestinal environment.  Then this gene could begin to benefit both bacteria and their host by opening additional food sources, and therefore remained in the population due to natural selection. <br><br><h2>  Microbial inconsistency </h2><br>  Starting to understand the relationship between our microbes and our ancestors from ancient times, we can use this deep symbiosis not only to interpret history, but also to shape our future health.  H. pylori can cause stomach cancer, but its ability to contribute to the development of cancer seems to be a function of how well the bacterial strain fits the host.  In a <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC3910595/">study</a> on stomach cancer and H. pylori in Colombia, researchers found that African strains of H. pylori were most likely to cause cancer in the Colombian population ‚Äî but the same strains were not so carcinogenic in Africans.  This observation indicates the possibility of preventing gastric cancer on an individual basis through minimizing the inconsistency of the host and its bacteria. <br><br>  Now, moving on to a deeper understanding of the presence and functioning of our microbes, we begin to understand how these long-term symbiosis could affect what we are today.  Recent studies have confirmed that the microflora in general is <a href="http://journals.plos.org/plosbiology/article%3Fid%3D10.1371/journal.pbio.2000225">much more similar</a> in related organisms than in less closely related organisms.  Microflora as a whole once can help us understand the evolutionary connections of different species. <br><br>  And although the auxiliary capabilities of the microflora in <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4290017/">understanding diseases are</a> sometimes excessively exaggerated, the most interesting may be that our microflora tells us about our ancestors lost in history. </div><p>Source: <a href="https://habr.com/ru/post/374121/">https://habr.com/ru/post/374121/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../374111/index.html">The results of the contest "It is time for awesome stories"</a></li>
<li><a href="../374113/index.html">Autonomous car and motorcycle collide in San Francisco</a></li>
<li><a href="../374115/index.html">Bitcoin, FinTech, machine learning and astrology</a></li>
<li><a href="../374117/index.html">"Retail" Internet of Things</a></li>
<li><a href="../374119/index.html">The colony. Chapter 18: An Important Find</a></li>
<li><a href="../374123/index.html">Who chooses music for movies and TV shows? Music Supervisor</a></li>
<li><a href="../374125/index.html">Facebook helps Americans who are hit by trolls</a></li>
<li><a href="../374127/index.html">What spontaneous commerce teaches online retailers and payment providers</a></li>
<li><a href="../374129/index.html">The Pentagon has declassified the archives of the program for the study of UFOs, which spent about $ 20 million</a></li>
<li><a href="../374131/index.html">Mining browser cryptocurrency unexpectedly rose from the dead</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>