<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Practical bioinformatics, part 3. Assessment of the significance of experimental data</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="At each stage of the experiment, starting from the preparation of the material, continuing with PCR and ending with sequencing, an accumulation of err...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Practical bioinformatics, part 3. Assessment of the significance of experimental data</h1><div class="post__text post__text-html js-mediator-article">  At each stage of the experiment, starting from the preparation of the material, continuing with PCR and ending with sequencing, an accumulation of error occurs.  We need a mechanism to assess the significance of the result.  What is the probability that the reads, which turned out to be at a certain part of the genome, turned out to be there by chance?  The approach presented in this article is applicable to data obtained using DNA-seq, and talks about the possibility of applying the Poisson distribution to assess significance. <br><a name="habracut"></a><br>  This illustration schematically depicts the process of preparing the material for sequencing.  At each stage, an error is possible, resulting in a read on an arbitrary region of the genome. <br><br><img src="https://habrastorage.org/storage2/91c/b1c/50b/91cb1c50bad066104af038e54d983761.jpg"><br><ul><li>  A.1 A cell with DNA is treated with formaldehyde for better protein fixation.  Then DNA is cut into fragments in various ways, including <a href="http://en.wikipedia.org/wiki/Chromatin_immunoprecipitation">ultrasound</a> . </li><li>  A.2-3 An antibody is added that is sensitive to proteins.  Fragments marked with protein are attached to the antibody.  Some fragments are connected randomly. </li><li>  A.4. Deposition of antibodies with DNA fragments occurs.  Purification of DNA fragments from antibodies and proteins. </li><li>  B.1-4 These items show the steps for preparing the obtained DNA fragments for sequencing.  Adding special adapters.  Multiplication with PCR.  Placement on the slide. </li></ul>  The resulting slide is placed in a sequencing machine, for example <a href="">Illumina</a> .  The result of the operation of the equipment and the programs accompanying it is the FASTA / FASTQ file.  Here is an example of the contents of a FASTQ file: <br><pre><code class="bash hljs">@HWI-XXXXX_0470:1:1:1761:946<span class="hljs-comment"><span class="hljs-comment">#0/1 NCCCTTTGGATTTGTCCTTTCAGATGCATTAGCCAT +HWI-XXXXX_0470:1:1:1761:946#0/1 BKNNJMHHEJRTRRR_______________Y[[[Y_</span></span></code> </pre>  The first line, starting with ‚Äú@‚Äù, and the third, starting with ‚Äú+‚Äù, are descriptive, contain the name, slide number, etc.  The second line is the read sequence of nucleotides, in our case there are 36 of them. The fourth line is the probabilities of reading the nucleotides.  Below is a link to an exact description of the file format and how to get numbers from the letter representation <a href="http://en.wikipedia.org/wiki/FASTQ_format">http://en.wikipedia.org/wiki/FASTQ_format</a> . <br>  The file with all the reads, of which millions in one sequencing, is transferred to the <a href="http://bowtie-bio.sourceforge.net/">bowtie</a> program, which is currently the fastest.  The execution line for the bowtie program can be: <br><pre> <code class="bash hljs">/usr/src/bowtie-0.12.7/bowtie -q -v 2 -m 1 --best --strata -p 8 -S \ /usr/src/bowtie-0.12.7/indexes/hg19 - | samtools view -Sb - &gt;<span class="hljs-variable"><span class="hljs-variable">${DIRNAME}</span></span>/<span class="hljs-variable"><span class="hljs-variable">${SEQNAME}</span></span>.bam</code> </pre><br>  The parameters specified in the example characterize the following program behavior: <br>  -q - processes the input file as FASTQ, not FASTA <br>  -v - the number of errors in the read, in the above example, 2 <br>  -m - how many entries (reads) in the genome to search for, in this case, 1 <br>  -p - how many threads to use, I proceeded from the number of processor cores - 8 <br>  -S - means that the output file will be recorded in the .sam format <br>  /usr/src/bowtie-0.12.7/indexes/hg19 - the path to the previously indexed genome, performed by the bowtie-build program.  Also ready path can be downloaded from the Internet. <br>  - (a dash at the end of the second line) means output to standard output for transfer to the samtools program, which converts it to .bam on the fly. <br>  In the description, I missed the parameters --best --strata, which set a special bowtie mode: if the program finds more than one occurrence of a read in the genome with a different number of errors, then it displays the result with the least error.  If the results with the least error are more than one, they are not displayed.  This increases the time of the program, but at the same time allows to increase the sensitivity of the reads in the genome.  Here is an example of the output of the .sam / .bam file: <br><br><pre> <code class="bash hljs">@HD VN:1.0 SO:unsorted @SQ SN:chr10 LN:135374737 @SQ SN:chr11 LN:134452384 ‚Ä¶ @PG ID:Bowtie VN:0.12.7 CL:<span class="hljs-string"><span class="hljs-string">"/usr/src/bowtie-0.12.7/bowtie -q -v 1 -m 1 --best --strata -p 8 -S"</span></span> USI-EAS21_8_3445_7_1_452_540 0 chr5 39195556 255 23M * 0 0 CTCCAGATTCTCTGGAAAGATGG SSSSSSPSSSSSSSSGNNSJSSL XA:i:0 MD:Z:23 NM:i:0 USI-EAS21_8_3445_7_1_170_242 0 chr14 73509925 255 23M * 0 0 CTGCATTAGACCTAGGCTTAGAA SSSSSSSSSSSSSSSSSSSGSNN XA:i:0 MD:Z:23 NM:i:0 USI-EAS21_8_3445_7_1_290_985 0 chr7 142613233 255 23M * 0 0 AGCTGACTGGCAAGCAACAGAGT JSSSSSSSSSSSPSSGSSLSNSL XA:i:0 MD:Z:23 NM:i:0 USI-EAS21_8_3445_7_1_881_711 4 * 0 0 * * 0 0 ATCGGAAGAGCTCGTATGCCGTC SSSSPSSSSSSSSSSSSSSSSNS XM:i:0 USI-EAS21_8_3445_7_1_897_318 0 chr3 50101427 255 23M * 0 0 GGGCGCAGAACCGCTGCTGCTGC SPSSSSSSSPPSSSSNSPPPLSL XA:i:1 MD:Z:15A7NM:i:1</code> </pre>
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      The header (lines starting with @SQ, @HD, @PG) contains descriptions: the names of chromosomes and their lengths, whether the file is sorted, the command with which the file was received.  The body consists of lines, each line is read, but with coordinates in the genome.  More details can be found here <a href="http://genome.sph.umich.edu/wiki/SAM">http://genome.sph.umich.edu/wiki/SAM</a> . <br><br><img src="https://habrastorage.org/getpro/geektimes/post_images/e64/7a9/05c/e647a905caccded1c9ad1cc52acd0b25.jpg"><br><br>  Genome distribution is different, but in enriched areas, it is considered that the experiment is more likely to succeed if the distribution has the following properties: even coverage (high coverage), high complexity (high complexity), continuity of coverage (continuous coverage).  Examples, as reads can cover the genome, are depicted in the figures above. <br><br>  To determine the enriched areas, we use the following reasoning.  We add all the lengths of the chromosomes and subtract the length of the repeating sections (approximate length 2282603114 for hg19), we take the resulting length in G. We divide the length G into windows of width W (for example, 500).  We take for N the number of the resulting windows (G / W approximately 4565206.228).  Let's count the number of reads in the experiment that were successfully found in the genome (let it be 4E6 out of 4.5E6), we take their number as S. Then the average number of reads per window, we denote <img src="https://habrastorage.org/getpro/geektimes/post_images/0e3/867/b37/0e3867b370e22592116c362cd2e0084b.png" alt="lamda">  , will be <img src="https://habrastorage.org/getpro/geektimes/post_images/0e3/867/b37/0e3867b370e22592116c362cd2e0084b.png" alt="lamda">  = S * W / G = S / N. <br>  Poisson distribution <img src="https://habrastorage.org/getpro/geektimes/post_images/76b/c12/f8b/76bc12f8bdcf3387919a341dc33bdf8b.png" alt="poisson ditribution">  expresses the probability of the origin of a given number k of events over a period of time with a known average origin of events.  The cumulative Poisson distribution will express the probability of occurrence of events of up to k.  We will use the cumulative distribution, since we are interested in the probability that the number of reads in the window is less than k.  Null hypothesis: k reads in a window by accident.  Alternative hypothesis: k reads in the window is not accidental.  The probability that our null hypothesis is true is even <img src="https://habrastorage.org/getpro/geektimes/post_images/1ff/f59/a60/1fff59a6069ca4bc85771446c03731e6.png" alt="Cumulative Poisson Distribution">  .  The closer P (k, <img src="https://habrastorage.org/getpro/geektimes/post_images/0e3/867/b37/0e3867b370e22592116c362cd2e0084b.png" alt="lamda">  ) to 0, the less likely that the reads hit the specified window by accident.  In our example, we obtain that if the number of reads in the window is up to 15, then 1-P (15.0.876193) = 2.55351E-15, i.e.  The number of reads up to 15 can enter a window 500 wide by chance with a probability of 2.55351E-15.  This number is also called p value.  Consequently, if there are more than 15 reads, then they are even less likely to enter the window by accident.  Thus, in the program, you can set the condition: if p value &lt;= 1.e16, then the data in the window should be left for consideration, the rest should be discarded. <br><br>  Below is an example of a program that uses part of the classes described in the <a href="http://habrahabr.ru/blogs/bioinformatics/137082/">previous article</a> , and is complemented by a new class of PoissonHandler.  Additional files to the <a href="">first</a> archive can be downloaded <a href="">here</a> .  In the new poisson directory, type qmake, then make.  The assembled poisson program builds a significance table for 500bp windows located behind the transcriptional start site of each gene, the annotation is selected from the browser's genes.  A more detailed work with the genome browser will consider in another article.  Next, reads from the .bam file that fall into our windows are selected, a Poisson distribution is built for them.  The result is written to the file. <br><br>  <u>PoissonHandler.hpp</u> <br><pre> <code class="cpp hljs"><span class="hljs-meta"><span class="hljs-meta">#</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">ifndef</span></span></span><span class="hljs-meta"> PoissonHandler_H #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">define</span></span></span><span class="hljs-meta"> PoissonHandler_H #</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">include</span></span></span><span class="hljs-meta"> </span><span class="hljs-meta-string"><span class="hljs-meta"><span class="hljs-meta-string">&lt;config.hpp&gt; class PoissonHandler : public QState { Q_OBJECT private: GenomeDescription *sql_input; GenomeDescription *sam_input; QSettings setup; public: PoissonHandler(GenomeDescription *sql,GenomeDescription *sam,HandledData &amp;output,QState *parent=0); ~PoissonHandler(); protected: virtual void onEntry(QEvent* event); }; #endif //</span></span></span></span></code> </pre><br>  <u>PoissonHandler.cpp</u> <br><pre> <code class="cpp hljs"><span class="hljs-meta"><span class="hljs-meta">#</span><span class="hljs-meta-keyword"><span class="hljs-meta"><span class="hljs-meta-keyword">include</span></span></span><span class="hljs-meta"> </span><span class="hljs-meta-string"><span class="hljs-meta"><span class="hljs-meta-string">"PoissonHandler.hpp"</span></span></span><span class="hljs-meta"> </span><span class="hljs-comment"><span class="hljs-meta"><span class="hljs-comment">//------------------------------------------------------------- //------------------------------------------------------------- PoissonHandler::~PoissonHandler() { } //------------------------------------------------------------- //------------------------------------------------------------- PoissonHandler::PoissonHandler(GenomeDescription *sql,GenomeDescription *sam,HandledData &amp;,QState * parent): QState(parent), sql_input(sql), sam_input(sam) { } //------------------------------------------------------------- //------------------------------------------------------------- void PoissonHandler::onEntry(QEvent*) { if(!setup.contains("AvgTagWindow")) setup.setValue("AvgTagWindow",500); if(!setup.contains("siteShift")) setup.setValue("siteShift",75); qint32 window=setup.value("AvgTagWindow").toInt(); qint32 half_window=0;//window/2; qint32 shift=setup.value("siteShift").toInt(); int c=0;//sense count, even/odd c sense/nonsense QMap&lt;QString,int&gt; poisson; QMapIterator&lt;QChar,QMap&lt;QString,QMap&lt;qint32,qint32&gt; &gt; &gt; sense(sql_input-&gt;genome); while(sense.hasNext()) { sense.next(); c++; QMapIterator&lt;QString,QMap&lt;qint32,qint32&gt; &gt; dictionary(sense.value()); while(dictionary.hasNext()) { dictionary.next(); QMapIterator&lt;qint32,qint32&gt; s(dictionary.value()); //sql result for strand+chromosome if((sam_input-&gt;genome['+']).contains(dictionary.key())) //chromosome chr1,chr2... { QMapIterator&lt;qint32,qint32&gt; k( (sam_input-&gt;genome['+'])[dictionary.key()] );//+ strand,sam file, iteration over strand+chromosome s.next(); //c++; k.next(); while (k.hasNext()) // + strand, iteration over strand+chromosome+each position { //if align is inside of the segment of the start site qint32 x1=k.key()- s.key() +half_window +shift; qint32 x2=x1-window; if( (x1^x2) &lt; 0 ) {//inside segment poisson[((sql_input-&gt;genome_h[sense.key()])[dictionary.key()])[s.key()]]+=k.value(); k.next(); } else { if((k.key() +shift)&gt;(s.key()+half_window)) { if(!s.hasNext()) break; s.next();// c++; } else { k.next(); } } }//while k.hasNext }//if + strand sam input if((sam_input-&gt;genome['-']).contains(dictionary.key())) { QMapIterator&lt;qint32,qint32&gt; j( (sam_input-&gt;genome['-'])[dictionary.key()] );//- strand,sam file s.toFront(); s.next(); j.next(); while (j.hasNext()) // - strand { //if align is inside of the segment of the start site qint32 x1=j.key()- s.key() +half_window -shift; qint32 x2=x1-window; if( (x1^x2) &lt; 0 ) {//inside segment poisson[((sql_input-&gt;genome_h[sense.key()])[dictionary.key()])[s.key()]]+=j.value(); j.next(); } else { if((j.key() -shift)&gt;(s.key()+half_window)) { if(!s.hasNext()) break; s.next(); } else { j.next(); } } } }//if - strand sam file }//while dictionary }//while sense //Poisson { qreal lamda=(qreal)(window*(sam_input-&gt;total-sam_input-&gt;notAligned))/(sam_input-&gt;tot_len*0.741); QFile _outFile; _outFile.setFileName(setup.value("outFileName").toString()); _outFile.open(QIODevice::WriteOnly|QIODevice::Truncate); qreal exp_l=qExp(-lamda); QString str; foreach(str,poisson.keys()) { //P(x,L)=(e**-1)*(L**x)/x! // qreal pois=exp_l; //qExp(-lamda)*qPow(lamda,poisson[str]); // for(int i=2; i&lt;= poisson[str];i++) // pois=pois*lamda/i; //P(x,L)=(e**-1)*E(1-k)(L**x(i))/x(i)! qreal pois=exp_l; qreal delim=1.0; qreal sum=1.0; for(int i=1; i&lt;= poisson[str];i++) { delim*=lamda/i; sum+=delim; } pois*=sum; _outFile.write( (QString("%1\t%2\t%3\n").arg(str).arg(poisson[str]).arg(pois,0,'e',20)).toLocal8Bit() ); } _outFile.flush(); _outFile.close(); }//if(1) }//end of function //------------------------------------------------------------- //-------------------------------------------------------------</span></span></span></span></code> </pre><br>  These small algorithms will later be useful for implementing more complex ones, such as <a href="http://code.google.com/p/zinba/">ZINBA</a> or <a href="http://home.gwu.edu/~wpeng/Software.htm">SICER</a> , for example.  We will talk about them in another article.  Often the task is to find enriched areas without using annotations.  And then step by step, you need to select data that fell into the window and passed through the filter. <br><br>  <b>Comprehensive comparative analysis of strand-specific RNA sequencing methods</b> <br>  Nat Methods.  2010 September;  7 (9): 709-715. <br>  Joshua Z. Levin, 1,6 Moran Yassour, 1,2,3,6 Xian Adiconis, 1 Chad Nusbaum, 1 Dawn Anne Thompson, 1 Nir Friedman, 3,4 Andreas Gnirke, 1 and Aviv Regev1,2,5 <br>  <a href="http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310/">www.ncbi.nlm.nih.gov/pmc/articles/PMC3005310</a> <br><br>  <b>Genomic Location Analysis by ChIP-Seq</b> <br>  J Cell Biochem.  2009 May 1; 107 (1): 11-8. <br>  Artem Barski * and Keji Zhao * <br>  Laboratory of Molecular Immunology, National Heart, Lung and Blood Institute, NIH, Bethesda, MD 20892 <br>  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19173299">www.ncbi.nlm.nih.gov/pubmed/19173299</a> <br><br>  <b>An integrated strategy for the next-generation sequencing data.</b> <br>  Genome Biol.  2011 Jul 25; 12 (7): 120. <br>  Peng W, Zhao K. <br>  Department of Physics, The George Washington University, 725 21st Street NW, Washington, DC 20052, USA.  wpeng@gwu.edu <br>  <a href="http://www.ncbi.nlm.nih.gov/pubmed/21787381">www.ncbi.nlm.nih.gov/pubmed/21787381</a> <br><br>  <b>A clustering approach for identification of enriched domains from histone modification ChIP-Seq data.</b> <br>  Bioinformatics.  2009 Aug 1; 25 (15): 1952-8.  Epub 2009 Jun 8. <br>  Zang C, Schones DE, Zeng C, Cui K, Zhao K, Peng W. <br>  Department of Physics, The George Washington University, Washington, DC 20052, USA. <br>  <a href="http://www.ncbi.nlm.nih.gov/pubmed/19505939">www.ncbi.nlm.nih.gov/pubmed/19505939</a> <br><br>  <b>ZINBA integrates local covariates with DNA-seq data, even within amplified genomic regions.</b> <br>  Genome Biol.  2011 Jul 25; 12 (7): R67. <br>  Rashid NU, Giresi PG, Ibrahim JG, Sun W, Lieb JD. <br>  Department of Biostatistics, Gillings School of Global Public Health, The University of North Carolina at Chapel Hill, Chapel Hill, NC 27599, USA. <br>  <a href="http://www.ncbi.nlm.nih.gov/pubmed/21787385">www.ncbi.nlm.nih.gov/pubmed/21787385</a> <br><br>  Review by Andrey Kartashov, Cincinnati OH, mail: porter@porter.st </div><p>Source: <a href="https://habr.com/ru/post/137267/">https://habr.com/ru/post/137267/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../137255/index.html">Developer menu</a></li>
<li><a href="../137256/index.html">Practical advice for those who decide to sell through couponers</a></li>
<li><a href="../137258/index.html">Meet text-align-last</a></li>
<li><a href="../137261/index.html">Why does Google use the Duck.com domain?</a></li>
<li><a href="../137266/index.html">Start! First post</a></li>
<li><a href="../137270/index.html">Arduino + Portal</a></li>
<li><a href="../137271/index.html">Robots from iRobot "train" to work on Fukushima</a></li>
<li><a href="../137272/index.html">How to make ‚Äútwo in one‚Äù out of a home computer - home theater + personal computer</a></li>
<li><a href="../137273/index.html">Game trailer (or all sites, yes sites)</a></li>
<li><a href="../137274/index.html">Old Japanese laptop FMV-BIBLO NU13D</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>