<!doctype html>
<html class="no-js" lang="en">

<head>
  <!-- Global site tag (gtag.js) - Google Analytics -->
  <script async src="https://www.googletagmanager.com/gtag/js?id=UA-134931760-1"></script>
  <script>
    window.dataLayer = window.dataLayer || [];
    function gtag(){dataLayer.push(arguments);}
    gtag('js', new Date());

    gtag('config', 'UA-134931760-1');
  </script>

  <meta charset="utf-8">
  <meta http-equiv="x-ua-compatible" content="ie=edge">
  <title>Citizenship by DNA, or whether Jews buy genetic tests</title>
  <link rel="icon" type="image/x-icon" href="/favicon.ico" />
  <meta name="description" content="According to our experience, the analysis of the human population by DNA, raises three big questions to the public: can genes and ethnic groups be int...">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">

  <link rel="stylesheet" href="../../css/main.css">

  <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.3.1/jquery.min.js"></script>
  <script>window.jQuery || document.write('<script src="../../js/vendors/jquery-3.3.1.min.js"><\/script>')</script>

  <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
  <script>
       (adsbygoogle = window.adsbygoogle || []).push({
            google_ad_client: "ca-pub-6974184241884155",
            enable_page_level_ads: true
       });
  </script>
</head>

<body>
  <!--[if lte IE 9]>
    <p class="browserupgrade">You are using an <strong>outdated</strong> browser. Please <a href="https://browsehappy.com/">upgrade your browser</a> to improve your experience and security.</p>
  <![endif]-->
  <header class="page-header js-page-header">
    <a class="page-header-logo-container" href="https://weekly-geekly.github.io/index.html"></a>
    <div class="page-header-text">Geekly Articles each Day</div>
  </header>
  <nav class="page-headings-container js-page-headings-container"></nav>
  <div class="tools-bar js-tools-bar">
    <!-- <a href="../../search.html" title="Search">üîé</a> -->
    <a class="js-list-of-headings-button" data-state="closed" href="#" title="Headings">üìú</a>
    <a class="js-go-to-top-button" href="#" title="Go to Top">‚¨ÜÔ∏è</a>
    <a class="js-go-to-bottom-button" href="#" title="Go to Bottom">‚¨áÔ∏è</a>
  </div>
  <a href="http://bit.ly/donateToWeeklyGeekly" class="donate-btn">DONATE</a>
  <section class="page js-page"><h1>Citizenship by DNA, or whether Jews buy genetic tests</h1><div class="post__text post__text-html js-mediator-article">  According to our experience, the analysis of the human population by DNA, raises three big questions to the public: can genes and ethnic groups be interconnected, how does origin analysis come from a technical point of view, and whether genetic tests are able to ‚Äúidentify Jews‚Äù.  For some reason, the question of Jewish identity in DNA is of great concern both to those who have indisputable evidences of belonging to the B-chosen people, and those who don‚Äôt eat matzo and do not read Torah. <br><br>  In the new material Genotek on Geektimes we will try to answer everything in order.  And yes, the Jews also define. <br><br><img src="https://habrastorage.org/files/379/071/624/3790716240844fbca684ae8ee769c077.jpg"><br><a name="habracut"></a><br><h3>  Race aka population groups in biology, medicine, and genetics </h3><br>  Mankind has a bad habit of justifying violence by the ‚Äúinnate‚Äù superiority of one race over another - that‚Äôs why modern biologists approach the question of genetic differences between populations with the sapper‚Äôs caution.  (Not) the existence of biological boundaries between racial and ethnic groups was fiercely discussed throughout the 20th century, but no final consensus on this issue has been reached so far ( <a href="">1</a> ). 
      <br>
        <script async src="//pagead2.googlesyndication.com/pagead/js/adsbygoogle.js"></script>
        <ins class="adsbygoogle"
          style="display:block; text-align:center;"
          data-ad-layout="in-article"
          data-ad-format="fluid"
          data-ad-client="ca-pub-6974184241884155"
          data-ad-slot="8945601208"></ins>
        <script>
          (adsbygoogle = window.adsbygoogle || []).push({});
        </script>
      <br>
      There were hopes that the sequencing of the human genome would reconcile all.  The genome, read ‚Äúfrom‚Äù and ‚Äúto‚Äù, will show that the boundaries between groups are of a social nature, and the genes are the same for all.  It turned out differently: a careful study of the human nucleotide code revived and heightened interest in biological differences between racial and ethnic populations.  Similar, in general, genes found slightly different allelic variants associated with the risk of disease ( <a href="http://jamanetwork.com/journals/jama/fullarticle/193524">2</a> ), drug metabolism ( <a href="http://search.proquest.com/openview/32ae3a753954821e8cd59b86a0c01a4b/1%3Fpq-origsite%3Dgscholar%26cbl%3D27990">3</a> ), the body‚Äôs response to environmental conditions ( <a href="http://www.sciencedirect.com/science/article/pii/S0027510700001615">4</a> ), and these variants were encountered in different populations with different frequencies. <br><br>  The search for non-existent "Indian" or "African" genes was stopped, but research in the field of medical and population genetics continues to draw parallels between the biological characteristics and the ethnicity of the participants.  The use of the terms "race" and "ethnicity" in such works is actively discussed (and often condemned).  There have been attempts to introduce rules that make researchers justify the need to use ‚Äúslippery‚Äù categories and clarify what exactly is meant by specific terms.  In February last year, in Science, one of the most authoritative natural science journals, an ambiguous article ( <a href="http://science.sciencemag.org/content/351/6273/564">5</a> ) was published, suggesting that we completely abandon the use of the term ‚Äúrace‚Äù in genetic research, replacing it with a more correct and neutral ‚Äúancestry‚Äù - ‚Äúorigin‚Äù . <br><br>  But even in the face of uncertainty with terms, the division of humankind into population groups is still necessary: ‚Äã‚Äãin particular, for the correct conduct of clinical trials of drugs and the assessment of the risk of diseases.  For example, the three allelic variants of the NOD2 gene - R702W, G908R and 1007fs - are associated with an increased risk of Crohn's disease in European Americans ( <a href="http://www.nature.com/nature/journal/v411/n6837/full/411603a0.html">6</a> , <a href="http://www.nature.com/nature/journal/v411/n6837/full/411599a0.html">7</a> ), however, none of these variants is associated with Crohn's disease in Japanese ( <a href="http://www.gastrojournal.org/article/S0016-5085(02)00066-5/abstract">8</a> ).  Known alleles of the CCR5 gene affecting the rate of development of immunodeficiency in HIV-infected patients ( <a href="">9</a> ): among them was found an option that slows the progression of the disease in European Americans, but accelerates its development in African Americans ( <a href="">10</a> ).  In Asians, a correlation was found between the polymorphisms of the p53 protein gene, which regulates the stress response and suppresses the development of tumors, and the average winter temperatures in the populations' habitats - genetic adaptation to frost ( <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667983/">11</a> ).  And if in the past, only information provided by the participants themselves was used to divide the sample into ethnic groups, in the post-genomic era they are increasingly complemented and specified by genetic evaluation of the subject's origin. <br><br><h3>  Genetic variations between populations </h3><br>  In everyday life, we divide people into groups by appearance or language of communication.  Most Danes resemble each other more than each of them - like an Italian ( <a href="https://pmsol3.wordpress.com/2011/04/07/world-of-averages-europeave/">here is a</a> cool visualization with averaged portraits of different nationalities).  Danes and Italians are much closer to each other than each of them to the inhabitants of sub-Saharan Africa: human phenotypes are clustered according to a geographical pattern.  The distribution of genotypes has a similar structure: members of a local group, as a rule, have closer family ties than residents of remote areas, and populations inhabiting one region are closer than those whose habitats are separated by geographic barriers (for example, mountain ranges or water array). <br><br>  At the same time, the genetic diversity of the human population is lower than in many species.  This is explained by the fact that humanity is a kind of young: some groups had relatively little time to accumulate differences.  Two randomly selected people differ from each other in each of ~ 1000 nucleotides, whereas two chimpanzees do not coincide every ~ 500 "letters".  And yet, in total, there are about 3 million potential ‚Äúpoints of divergence‚Äù in the human genome.  Most of these discrepancies, called single nucleotide polymorphisms (SNPs), are neutral or almost neutral, but some of them are responsible for the phenotypic differences between people. <br><br>  The distribution of neutral polymorphisms (since they do not have a biological meaning, are not subject to directed evolutionary selection, carried by the wind of migrations) in the world population reflects the demographic history of our species.  Genetic and archaeological evidence indicates that the size of the human population has increased significantly over the past 100,000 years.  People settled outside Africa, colonizing the rest of the world.  The resettlement process influenced the geographical distribution of alleles in two ways: first, the ‚Äúfounder effect‚Äù had an effect - in the population of immigrants, as a rule, only part of the genetic variants from the entire pool of their diversity in the ancestral population was represented;  secondly, the so-called ‚Äúassortative crossing‚Äù took place, i.e.  the pairs were formed mainly within their group, which limited the spread of existing and de novo emerging polymorphisms among individuals inhabiting different geographic areas.  These processes led to a gradual accumulation of genetic differences. <br><br>  In the context of population groups, genomic markers began to be studied in the 70s ‚Äì 80s, in the 90s they were used to identify the population of a particular person.  Researchers have again and again demonstrated that genetic polymorphisms make it possible to successfully isolate population groups and determine the group identity of an individual.  At the same time, it was shown that people living on the same continent are, as a rule, closer to each other genetically than people from different continents.  Initially, in such studies information about the place of birth, race, ethnic group was known from the very beginning and was used in conjunction with genetic data;  if the subjects were distributed blindly to the clusters, solely on the basis of genetic traits, the correspondence between geographical origin, ethnicity and population structure was less pronounced.  As further studies have shown, success depended on the genetic markers used and their number (more - better), the correct choice of reference populations and other factors ( <a href="http://www.nature.com/nrg/journal/v5/n8/full/nrg1401.html">12</a> ). <br><br>  By 2004, in the US, the genetic definition of population ownership was used not only in biomedical research, but also in investigating crimes: <a href="http://www.nature.com/nrg/journal/v5/n8/abs/nrg1405.html">this</a> article from Nature contains a fascinating story of how the police, desperate to find a criminal, ordered a DNA test in a commercial company, decided on skin color of the suspect and solved the case.  Proposals for the analysis of genetic origin have successfully fallen into a wave of people's general interest in their own past.  ‚ÄúRoots mania‚Äù was the name given to this hobby in an article in Time devoted to ‚ÄúAmerica's latest obsession‚Äù - genealogical research. <br><br>  Experts studying the origin and evolution of peoples actively use genomic methods.  For example, in 2013, an international team of researchers used genetic analysis to refute the hypothesis of the origin of Ashkenazi Jews from the Khazars ( <a href="">13</a> ).  The set of genomic data used by the authors is publicly available: it contains more than 100 world populations.  We propose to model a small study with us: to determine the place of Genotek clients in this sample, and at the same time to understand the technical details of the definition of population affiliation. <br><br><h3>  Purpose of the study </h3><br>  Determine the place of Genotek clients among reference populations.  Find out if there are representatives of Ashkenazi Jews in our sample.  Demonstrate the principles and methods of analyzing the population of the individual person. <br><br><h3>  Research objectives </h3><br>  To process the genotyping data of 722 subjects using the ADMIXTURE program, using the data set from Behar et al., 2013 as a training sample. <br><br><h3>  Materials and methods </h3><br>  The original work of Behar et al., 2013, used data from 1,774 people: among them were representatives of 88 non-Jewish populations (from Arabia, Central Asia, East Asia, Europe, the Middle East, North Africa, Siberia, South Asia, and sub- Saharan Africa) and 18 Jewish populations.  The authors needed an extensive data set to accurately determine the location of Ashkenazi in the context of world populations: the task was to present all three geographical regions from which this group could hypothetically occur - Europe, Middle East and the Khazar Khaganate.  The authors emphasized the difference between the approach to sampling, representing modern European, Middle Eastern and Jewish populations - direct descendants of ancestral populations, and samples corresponding to the Khazar kaganate, which ceased to exist about 1000 years ago.  The hitch is that none of the currently existing populations is a direct heir of the Kaganate.  Authors of the South Caucasus (Abkhazians, Armenians, Azeris, Georgians), North Caucasians (Adygs, Balkarians, Chechens, Kabardians, Ossetians and several other nationalities), Chuvash and Tatars, were chosen by the authors as possible contemporary representatives of the Khazars. <br>  We added samples of 722 people from various regions of Russia to the data set. <br><br>  For the statistical analysis, the ADMIXTURE program was used, which allows one to estimate the most probable origin of an individual on the basis of data on genotypes.  In addition, the authors of the article under discussion used other statistical methods that gave a similar answer to the question posed.  We will focus on ADMIXTURE, since it is this algorithm that allows us to estimate the percentage contribution of ancestral populations to the studied genomes. <br><br>  ADMIXTURE uses Monte Carlo methods in Markov chains (Markov chain Monte Carlo, MCMC).  Here is a <a href="http://dalexander.bol.ucla.edu/preprints/admixture-preprint.pdf">link</a> to the article of the authors of the algorithm for those who want to understand in more detail the mathematical side of the process. <br><br><h4>  Consider how ADMIXTURE works on the example of samples and populations from our set </h4><br>  We have a total of 2,496 specimens / individuals, each of which belongs to one of 106 modern populations.  We assume that modern populations are likely to have evolved from a relatively small number of ancestral populations.  ‚ÄúAncestral populations‚Äù in such an analysis are some ancient genomic clusters united according to the principle of genetic similarity.  ADMIXTURE allows both arbitrarily putting forward assumptions about the number of such clusters in a sample, as well as selecting the optimal number that most correctly describes the actual distribution of genomic data. <br><br>  After receiving information about the genotypes and the estimated number of ‚Äúancestral‚Äù populations (K), ADMIXTURE builds a model that evaluates the contribution of each of the ‚Äúancestral‚Äù populations to each sample.  When interpreting data, both the quantitative composition of the genome (the percentage ratio of clusters) and the qualitative one ‚Äî their presence or absence in specific genomes ‚Äî is important.  Based on these data, it is possible to make assumptions about the evolutionary processes in a population, in particular, about the presence or absence of common ‚Äúroots‚Äù among population groups.  However, the conclusions will be legitimate if the model we have built is good: the optimal value of K is chosen. <br><br><h4>  Choose the optimal value of K </h4><br>  How to determine how many ‚Äúancestral‚Äù populations most closely match the true for this sample?  In an empirical way! <br><br>  ADMIXTURE is a smart program: building a model of the genetic structure of populations based on data on individuals' genotypes (evaluating the contribution of each of the ancient genomic clusters to each of the genomes of the sample) for a given number K, it does not forget to compare it with reality at the end.  Check how well the input data is described by the constructed model.  The measure of comparison is ‚Äúerror‚Äù - a value that describes the discrepancy between the model and real data.  The greater the error, the worse the assumption about the number of ancestral populations corresponds to reality. <br><br>  How to choose the optimal value of K?  We run the ADMIXTURE algorithm on this sample, substituting different values ‚Äã‚Äãof K, and we obtain for each K its own error value.  We build a graph of the dependence of the magnitude of the error on K. This is the graph that the authors of the article turned out to be: <br><br><img src="https://habrastorage.org/files/04b/0d7/098/04b0d70981c54e9ea0ceb8cd7b188225.jpg"><br><br>  The optimal value of K is at the minimum point of the function.  If the minimum on the graph is not detected (the function is constantly increasing or decreasing), you will have to build models, choosing new K, until you find the right one. <br><br>  Even with optimally selected K, the reliability of the analysis results depends on the correctness of the sample: <br><br>  1. Individuals should not be related to each other. <br>  2. Single nucleotide polymorphisms (SNP), which are used for genotyping, should be evenly distributed across the genome with a sufficiently high density. <br>  3. SNP alleles must be in equilibrium adhesion, that is, the probability of the presence of a given allele in a particular individual should depend only on the frequency of this allele in the population, but not on other alleles in the genome. <br><br>  As can be seen from the graph, the optimal K for this sample was 10 ‚Äúancestral‚Äù populations. <br><br><h3>  results </h3><br>  The results of the analysis are presented by ADMIXTURE like this (only part of the data is visible in the figure): <br><br><img src="https://habrastorage.org/files/412/64f/9fb/41264f9fb89b4ba0b70dca27b21ecd46.jpg"><br><br>  Each cluster has its own color, and populations differ (or do not differ) in the shares of clusters in the genome.  <a href="http://admixture.genotek.ru:4000/">Here</a> is an interactive version of the picture for detailed study: move your mouse and scroll to see all populations or to consider some of the groups in more detail. <br><br>  In general, within the ‚Äúpopulation‚Äù of Genotek, the ratio of clusters expectedly corresponds to a pattern characteristic of populations of Eastern European origin.  Interesting starts at the level of individual samples: <br><br><img src="https://habrastorage.org/files/397/425/c43/397425c432d14dbe841591e3310a4197.jpg"><br><br>  Although exactly the population closest to a given sample is determined by numerical values, a lot of information can be obtained by visual comparison of patterns.  We suggest that you independently identify the closest populations for samples of four Genotek clients from the image. <br><br><div class="spoiler">  <b class="spoiler_title">Answer</b> <div class="spoiler_text">  In this picture, samples 1 and 2 are of Asian origin: the predominance of the pink cluster is typical for the Japanese and the khan people in our sample, blue is for the Yakuts, the third sample shows the ratio of components characteristic of Russians, Belarusians, Ukrainians and Poles, and the fourth is typical Ashkenazi Jew. <br></div></div><br>  A total of 722 samples, we found 9 Ashkenazi Jews. <br><br><h3>  Conclusion </h3><br>  Population - is not the only factor determining the ethnic identity of a person.  However, it is still possible to identify a correlation between ethnic groups and the structure of the genome of their representatives.  Such an analysis is used both for scientific and medical purposes, as well as for the study of one's own roots by everyone.  It is important to understand that the models are constantly being improved, and the results for greater accuracy should be considered together with other data, for example, the family genealogical tree. <br><br>  No evidence of Khazar origin of Ashkenazi was found by the authors of the original article.  Genetic tests, of course, "can" identify the Jews - but do not forget that "Jewry" is, above all, a state of mind. <br><br>  In the near future, the updated Genealogy DNA test with advanced results will be launched in Genotek: we will increase the number of populations to hundreds, add Jewish populations.  We will update the information in your account for everyone who has ever handed us your genetic material.  If you are still not genotyped, we invite you to <a href="https://www.genotek.ru/">join</a> . <br><br><h3>  Bibliography </h3><br><ol><li>  <a href="">Foster M., Sharp R. (2002).</a>  <a href="">Race, Ethnicity, and Genomics: Social Classifications as Proxies of Biological Heterogeneity.</a>  <a href="">Genome Res.</a> </li><li>  <a href="http://jamanetwork.com/journals/jama/fullarticle/193524">Collins FS, McKusick VA (2001).</a>  <a href="http://jamanetwork.com/journals/jama/fullarticle/193524">Implications of the Human Genome Project for medical science.</a>  <a href="http://jamanetwork.com/journals/jama/fullarticle/193524">Jama.</a> </li><li>  <a href="http://search.proquest.com/openview/32ae3a753954821e8cd59b86a0c01a4b/1%3Fpq-origsite%3Dgscholar%26cbl%3D27990">Nebert DW, Menon AG (2001) Pharmacogenomics, ethnicity, and susceptibility genes.</a>  <a href="http://search.proquest.com/openview/32ae3a753954821e8cd59b86a0c01a4b/1%3Fpq-origsite%3Dgscholar%26cbl%3D27990">Pharmacogenomics J.</a> </li><li>  <a href="http://www.sciencedirect.com/science/article/pii/S0027510700001615">Olden K., Guthrie J. (2001).</a>  <a href="http://www.sciencedirect.com/science/article/pii/S0027510700001615">Genomics: Implications for toxicology.</a>  <a href="http://www.sciencedirect.com/science/article/pii/S0027510700001615">Mutat.</a>  <a href="http://www.sciencedirect.com/science/article/pii/S0027510700001615">Res.</a> </li><li>  <a href="http://science.sciencemag.org/content/351/6273/564">Yudell M., Roberts D., DeSalle R., Tishkoff S. (2016).</a>  <a href="http://science.sciencemag.org/content/351/6273/564">Taking race out of human genetics.</a>  <a href="http://science.sciencemag.org/content/351/6273/564">Science.</a> </li><li>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411603a0.html">Ogura, Y. et al.</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411603a0.html">(2001).</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411603a0.html">Ampshift mutation in NOD2 associated with susceptibility to Crohn's disease.</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411603a0.html">Nature.</a> </li><li>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411599a0.html">Hugot, JP et al.</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411599a0.html">(2001).</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411599a0.html">Association of NOD2 leucine-rich sequences with susceptibility to Crohn's disease.</a>  <a href="http://www.nature.com/nature/journal/v411/n6837/full/411599a0.html">Nature.</a> </li><li>  <a href="http://www.gastrojournal.org/article/S0016-5085(02)00066-5/abstract">Inoue, N. (2002).</a>  <a href="http://www.gastrojournal.org/article/S0016-5085(02)00066-5/abstract">Lack of common NOD2 variants in Japanese patients with Crohn's disease.</a>  <a href="http://www.gastrojournal.org/article/S0016-5085(02)00066-5/abstract">Gastroenterology.</a> </li><li>  <a href="">Martin, MP et al. (1998).</a>  <a href="">Genetic acceleration of AIDS progression by a promoter variant of CCR5.</a>  <a href="">Science.</a> </li><li>  <a href="">Gonzalez, E. et al. (1999).</a>  <a href="">Race-specific HIV-1 disease-modifying effects associated with CCR5 haplotypes.</a>  <a href="">Proc.</a>  <a href="">Natl Acad.</a>  <a href="">Sci.</a>  <a href="">USA.</a> </li><li>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667983/">Shi, Hong et al.</a>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667983/">(2009).</a>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667983/">Tumor Suppressor Pathway in Eastern Asia.</a>  <a href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2667983/">American Journal of Human Genetics.</a> </li><li>  <a href="http://www.nature.com/nrg/journal/v5/n8/full/nrg1401.html">Bamshad M., Wooding S., Salisbury B. et al.</a>  <a href="http://www.nature.com/nrg/journal/v5/n8/full/nrg1401.html">(2004).</a>  <a href="http://www.nature.com/nrg/journal/v5/n8/full/nrg1401.html">Deconstructing the relationship between genetics and race.</a>  <a href="http://www.nature.com/nrg/journal/v5/n8/full/nrg1401.html">Nat Rev Genet.</a> </li><li>  <a href="">Behar DM et al.</a>  <a href="">(2013).</a>  <a href="">No Evidence from Khazar Origin for the Ashkenazi Jews.</a>  <a href="">Human Biology.</a> </li></ol></div><p>Source: <a href="https://habr.com/ru/post/402923/">https://habr.com/ru/post/402923/</a></p>
<section class="more-articles-navigation-panel js-more-articles-navigation-panel">
<h4>More articles:</h4>
<nav class="list-of-articles-container js-list-of-articles-container"><ul class="list-of-pages js-list-of-pages">
<li><a href="../402909/index.html">For those who love it: advanced lenses for Canon, Nikon and Sony cameras</a></li>
<li><a href="../402913/index.html">3D printer Raise3D N2 Dual</a></li>
<li><a href="../402915/index.html">We test the modular printer "3D START" - part 3. Program settings for the display</a></li>
<li><a href="../402919/index.html">Frequency response graphs for headphones Bluedio and other brands: understand, expose, choose</a></li>
<li><a href="../402921/index.html">Catch us if you can</a></li>
<li><a href="../402925/index.html">Israeli experts found 40 zero-day vulnerabilities in Samsung's Tizen OS</a></li>
<li><a href="../402927/index.html">Completion of the Chinese ST-Link v2: add the SWO debug information output interface and foot reset</a></li>
<li><a href="../402929/index.html">MPTU-2-48</a></li>
<li><a href="../402931/index.html">What happened before the Big Bang? [Fraser Cain]</a></li>
<li><a href="../402937/index.html">The practice of choosing speakers: an interview with an expert Pult.ru</a></li>
</ul></nav>
</section><br />
<a href="../../allArticles.html"><strong>All Articles</strong></a>
<script src="../../js/main.js"></script>

<!-- Yandex.Metrika counter -->
<script type="text/javascript" >
  (function (d, w, c) {
      (w[c] = w[c] || []).push(function() {
          try {
              w.yaCounter52496797 = new Ya.Metrika({
                  id:52496797,
                  clickmap:true,
                  trackLinks:true,
                  accurateTrackBounce:true,
                  webvisor:true
              });
          } catch(e) { }
      });

      var n = d.getElementsByTagName("script")[0],
          s = d.createElement("script"),
          f = function () { n.parentNode.insertBefore(s, n); };
      s.type = "text/javascript";
      s.async = true;
      s.src = "https://mc.yandex.ru/metrika/watch.js";

      if (w.opera == "[object Opera]") {
          d.addEventListener("DOMContentLoaded", f, false);
      } else { f(); }
  })(document, window, "yandex_metrika_callbacks");
</script>
<noscript><div><img src="https://mc.yandex.ru/watch/52496797" style="position:absolute; left:-9999px;" alt="" /></div></noscript>

<!-- Google Analytics -->
  <script>
    window.ga = function () { ga.q.push(arguments) }; ga.q = []; ga.l = +new Date;
    ga('create', 'UA-134931760-1', 'auto'); ga('send', 'pageview')
  </script>
  <script src="https://www.google-analytics.com/analytics.js" async defer></script>

</section>

  <footer class="page-footer">
    <div class="page-footer-legal-info-container page-footer-element">
      <p>
        Weekly-Geekly | <span class="page-footer-legal-info-year js-page-footer-legal-info-year">2019</span>
      </p>
    </div>
    <div class="page-footer-counters-container page-footer-element">
      <a class="page-footer-counter-clustrmap" href='#'  title='Visit tracker'><img src='https://clustrmaps.com/map_v2.png?cl=698e5a&w=271&t=t&d=6iCFw7uJz0zcOaoxz5k5PcLCJUzv2WG8G5V8M3U6Rc4&co=3a3a3a&ct=ffffff'/></a>
    </div>
  </footer>
</body>

</html>